% Generated by Sphinx.
\def\sphinxdocclass{report}
\newif\ifsphinxKeepOldNames \sphinxKeepOldNamestrue
\documentclass[letterpaper,10pt,english]{sphinxmanual}
\usepackage{iftex}

\ifPDFTeX
  \usepackage[utf8]{inputenc}
\fi
\ifdefined\DeclareUnicodeCharacter
  \DeclareUnicodeCharacter{00A0}{\nobreakspace}
\fi
\usepackage{cmap}
\usepackage[T1]{fontenc}
\usepackage{amsmath,amssymb,amstext}
\usepackage{babel}
\usepackage{times}
\usepackage[Bjarne]{fncychap}
\usepackage{longtable}
\usepackage{sphinx}
\usepackage{multirow}
\usepackage{eqparbox}


\addto\captionsenglish{\renewcommand{\figurename}{Fig.\@ }}
\addto\captionsenglish{\renewcommand{\tablename}{Table }}
\SetupFloatingEnvironment{literal-block}{name=Listing }

\addto\extrasenglish{\def\pageautorefname{page}}

\setcounter{tocdepth}{2}


\title{PyBioMed Documentation}
\date{\begin{figure}[h]
		\includegraphics[width=10cm]{logocbdd.png}
		\centering
	\end{figure}
}
\release{1}
\author{CBDD Group}
\newcommand{\sphinxlogo}{}

\makeatletter
\def\PYG@reset{\let\PYG@it=\relax \let\PYG@bf=\relax%
    \let\PYG@ul=\relax \let\PYG@tc=\relax%
    \let\PYG@bc=\relax \let\PYG@ff=\relax}
\def\PYG@tok#1{\csname PYG@tok@#1\endcsname}
\def\PYG@toks#1+{\ifx\relax#1\empty\else%
    \PYG@tok{#1}\expandafter\PYG@toks\fi}
\def\PYG@do#1{\PYG@bc{\PYG@tc{\PYG@ul{%
    \PYG@it{\PYG@bf{\PYG@ff{#1}}}}}}}
\def\PYG#1#2{\PYG@reset\PYG@toks#1+\relax+\PYG@do{#2}}

\expandafter\def\csname PYG@tok@gd\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.63,0.00,0.00}{##1}}}
\expandafter\def\csname PYG@tok@gu\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.50,0.00,0.50}{##1}}}
\expandafter\def\csname PYG@tok@gt\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.27,0.87}{##1}}}
\expandafter\def\csname PYG@tok@gs\endcsname{\let\PYG@bf=\textbf}
\expandafter\def\csname PYG@tok@gr\endcsname{\def\PYG@tc##1{\textcolor[rgb]{1.00,0.00,0.00}{##1}}}
\expandafter\def\csname PYG@tok@cm\endcsname{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.25,0.50,0.56}{##1}}}
\expandafter\def\csname PYG@tok@vg\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.73,0.38,0.84}{##1}}}
\expandafter\def\csname PYG@tok@vi\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.73,0.38,0.84}{##1}}}
\expandafter\def\csname PYG@tok@mh\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\expandafter\def\csname PYG@tok@cs\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.50,0.56}{##1}}\def\PYG@bc##1{\setlength{\fboxsep}{0pt}\colorbox[rgb]{1.00,0.94,0.94}{\strut ##1}}}
\expandafter\def\csname PYG@tok@ge\endcsname{\let\PYG@it=\textit}
\expandafter\def\csname PYG@tok@vc\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.73,0.38,0.84}{##1}}}
\expandafter\def\csname PYG@tok@il\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\expandafter\def\csname PYG@tok@go\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.20,0.20,0.20}{##1}}}
\expandafter\def\csname PYG@tok@cp\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@gi\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.63,0.00}{##1}}}
\expandafter\def\csname PYG@tok@gh\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.00,0.50}{##1}}}
\expandafter\def\csname PYG@tok@ni\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.84,0.33,0.22}{##1}}}
\expandafter\def\csname PYG@tok@nl\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.13,0.44}{##1}}}
\expandafter\def\csname PYG@tok@nn\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.05,0.52,0.71}{##1}}}
\expandafter\def\csname PYG@tok@no\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.38,0.68,0.84}{##1}}}
\expandafter\def\csname PYG@tok@na\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@nb\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@nc\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.05,0.52,0.71}{##1}}}
\expandafter\def\csname PYG@tok@nd\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.33,0.33,0.33}{##1}}}
\expandafter\def\csname PYG@tok@ne\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@nf\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.02,0.16,0.49}{##1}}}
\expandafter\def\csname PYG@tok@si\endcsname{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.44,0.63,0.82}{##1}}}
\expandafter\def\csname PYG@tok@s2\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@nt\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.02,0.16,0.45}{##1}}}
\expandafter\def\csname PYG@tok@nv\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.73,0.38,0.84}{##1}}}
\expandafter\def\csname PYG@tok@s1\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@ch\endcsname{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.25,0.50,0.56}{##1}}}
\expandafter\def\csname PYG@tok@m\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\expandafter\def\csname PYG@tok@gp\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.78,0.36,0.04}{##1}}}
\expandafter\def\csname PYG@tok@sh\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@ow\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@sx\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.78,0.36,0.04}{##1}}}
\expandafter\def\csname PYG@tok@bp\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@c1\endcsname{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.25,0.50,0.56}{##1}}}
\expandafter\def\csname PYG@tok@o\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.40,0.40,0.40}{##1}}}
\expandafter\def\csname PYG@tok@kc\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@c\endcsname{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.25,0.50,0.56}{##1}}}
\expandafter\def\csname PYG@tok@mf\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\expandafter\def\csname PYG@tok@err\endcsname{\def\PYG@bc##1{\setlength{\fboxsep}{0pt}\fcolorbox[rgb]{1.00,0.00,0.00}{1,1,1}{\strut ##1}}}
\expandafter\def\csname PYG@tok@mb\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\expandafter\def\csname PYG@tok@ss\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.32,0.47,0.09}{##1}}}
\expandafter\def\csname PYG@tok@sr\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.14,0.33,0.53}{##1}}}
\expandafter\def\csname PYG@tok@mo\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\expandafter\def\csname PYG@tok@kd\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@mi\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\expandafter\def\csname PYG@tok@kn\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@cpf\endcsname{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.25,0.50,0.56}{##1}}}
\expandafter\def\csname PYG@tok@kr\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@s\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@kp\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@w\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.73,0.73,0.73}{##1}}}
\expandafter\def\csname PYG@tok@kt\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.56,0.13,0.00}{##1}}}
\expandafter\def\csname PYG@tok@sc\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@sb\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@k\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@se\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}


\def\PYGZbs{\char`\\}
\def\PYGZus{\char`\_}
\def\PYGZob{\char`\{}
\def\PYGZcb{\char`\}}
\def\PYGZca{\char`\^}
\def\PYGZam{\char`\&}
\def\PYGZlt{\char`\<}
\def\PYGZgt{\char`\>}
\def\PYGZsh{\char`\#}
\def\PYGZpc{\char`\%}
\def\PYGZdl{\char`\$}
\def\PYGZhy{\char`\-}
\def\PYGZsq{\char`\'}
\def\PYGZdq{\char`\"}
\def\PYGZti{\char`\~}
% for compatibility with earlier versions
\def\PYGZat{@}
\def\PYGZlb{[}
\def\PYGZrb{]}
\makeatother

\renewcommand\PYGZsq{\textquotesingle}

\begin{document}

\maketitle
\tableofcontents
\phantomsection\label{index::doc}


The python package PyBioMed is designed by \href{http://home.scbdd.com/index.php?s=/Home/Index.html\&t=english}{CBDD Group} (Computational Biology \& Drug Design Group), Xiangya School of Pharmaceutical Sciences, Central South University. To develop a powerful model for prediction tasks by machine learning algorithms such as sckit-learn, one of the most important things to consider is how to effectively represent the molecules under investigation such as small molecules, proteins, DNA and even complex interactions, by a descriptor. PyBioMed is a feature-rich package used for the characterization of various complex biological molecules and interaction samples, such as chemicals, proteins, DNA, and their interactions. PyBioMed calculates nine types of features including chemical descriptors or molecular fingerprints, structural and physicochemical features of proteins and peptides from amino acid sequence, composition and physicochemical features of DNA from their primary sequences, chemical-chemical interaction features, chemical-protein interaction features, chemical-DNA interaction features, protein-protein interaction features, protein-DNA interaction features, and DNA-DNA interaction features. We hope that the package can be used for exploring questions concerning structures, functions and interactions of various molecular data in the context of chemoinformatics, bioinformatics, and systems biology.


\chapter{Overview}
\label{overview::doc}\label{overview:the-pybiomed-documentation}\label{overview:overview}
To develop a powerful model for prediction tasks by machine learning algorithms such as sckit-learn, one of the most important things to consider is how to effectively represent the molecules under investigation such as small molecules, proteins, DNA and even complex interactions, by a descriptor. PyBioMed is a feature-rich package used for the characterization of various complex biological molecules and interaction samples, such as chemicals, proteins, DNA, and their interactions. PyBioMed calculates nine types of features including chemical descriptors or molecular fingerprints, structural and physicochemical features of proteins and peptides from amino acid sequence, composition and physicochemical features of DNA from their primary sequences, chemical-chemical interaction features, chemical-protein interaction features, chemical-DNA interaction features, protein-protein interaction features, protein-DNA interaction features, and DNA-DNA interaction features. We hope that the package can be used for exploring questions concerning structures, functions and interactions of various molecular data in the context of chemoinformatics, bioinformatics, and systems biology.  The python package PyBioMed is designed by CBDD Group (Computational Biology \& Drug Design Group), Xiangya School of Pharmaceutical Sciences, Central South University.


\section{Who uses PyBioMed?}
\label{overview:who-uses-pybiomed}
For those researchers from different biomedical fields, the PyBioMed package can be used to analyze and represent various complex molecular data under investigation. PyBioMed will be helpful when exploring questions concerning structures, functions and interactions of various molecular data in the context of chemoinformatics, bioinformatics, and systems biology.


\section{Goals}
\label{overview:goals}
PyBioMed is intended to provide
\begin{itemize}
\item {} 
Tools for pretreating molecules, proteins sequence and DNA sequence

\item {} 
Calculating chemical descriptors or molecular fingerprints from
molecules' structures

\item {} 
Calculating structural and physicochemical features of proteins and peptides
from amino acid sequence

\item {} 
Calculating composition and physicochemical features of DNA
from their primary sequences

\item {} 
Calculating interaction features including chemical-chemical interaction features,
chemical-protein interaction features, chemical-DNA interaction features,
protein-protein interaction features, protein-DNA interaction features
and DNA-DNA interaction features.

\item {} 
Getting molecular structures, protein sequence and DNA sequence from Internet through
the molecular ID, protein ID and DNA ID.

\end{itemize}


\section{Feature overview}
\label{overview:feature-overview}
The table below shows the descriptors and the number of the descriptor that PyBioMed can calculate in four modules including PyMolecule, PyProtein, PyDNA and PyInteraction. PyMolecule module can calculate 14 different types of molecular descriptors and 18 different types of molecular fingerprints. PyProtein module can calculate 14 types of protein descriptors. PyDNA module can calculate 14 types of DNA descriptors and the number in the table appears when parameters are `all\_property = True, lamada=2, w=0.05'. PyInteraction module can calculate three types of descriptors.

\begin{longtable}{| p{2.5cm}<{\centering} | p{10cm} |}
	\caption{The descriptors in the PyBioMed package }
	\hline
	\textbf{\large{Types}} & \centerline{\textbf{\large{Features}}} \\
	\hline
	& Constitution(30) \\
	& Connectivity descriptors (44) \\
	& Topology descriptors (35) \\
	& Basak descriptors (21) \\
	& Burden descriptors (64) \\
	& Kappa descriptors (7) \\
	& E-state descriptors (237) \\
	& Moran autocorrelation descriptors (32) \\
	& Geary autocorrelation descriptors (32) \\
	& Molecular property descriptors (6) \\
	& Moreau-Broto autocorrelation descriptors (32) \\
	& Charge descriptors (25) \\
	& MOE-type descriptors (60) \\
	& CATS2D descriptors (150) \\
	& Daylight-type fingerprints (2048)\\
	& MACCS fingerprints (166)\\
	\textbf{PyMolecule} & Atom pairs fingerprints \\
	& TopologicalTorsion fingerprints \\
	& E-state fingerprints (79)\\
	& FP2 fingerprints (1024)\\
	& FP3 fingerprints (210)\\
	& FP4 fingerprints (307)\\
	& ECFP2 fingerprints (1024)\\
	& ECFP4 fingerprints (1024)\\
	& ECFP6 fingerprints (1024)\\
	& Morgan fingerprints (1024)\\
	& Ghosecrippen fingerprints (110)\\
	& FCFP2 fingerprints (1024)\\
	& FCFP4 fingerprints (1024)\\
	& FCFP6 fingerprints (1024)\\
	& Pharm2D2point fingerprints (135)\\
	& Pharm2D3point fingerprints (2135)\\\hline
	
	
	& Amino acid composition (20) \\
	& Dipeptide composition (400) \\
	& Tripeptide composition (8000) \\
	& CTD composition (21) \\
	& CTD transition (21) \\
	& CTD distribution (105) \\
	\textbf{PyProtein} & M-B autocorrelation (240) \\
	& Moran autocorrelation (240) \\\hline
	\multicolumn{2}{|r|}{\small\sl continued on next page}\\\hline
	\multicolumn{2}{|l|}{\small\sl continued from previous page}\\\hline
	& Geary autocorrelation (240) \\
	& Conjoint triad features (343) \\
	& Quasi-sequence order descriptors (100) \\
	& Sequence order coupling number (60) \\
	& Pseudo amino acid composition 1 (50) \\
	& Pseudo amino acid composition 2 (50) \\\hline
	
	
	& Basic kmer (16)\\
	& Reverse compliment kmer (12)\\
	& DAC (76)\\
	& DCC (2812)\\
	& DACC (2888)\\
	& TAC (24)\\
	\textbf{PyDNA} & TCC (264)\\
	& TACC (288)\\
	& PseDNC (18)\\
	& PseKNC (66)\\
	& PC-PseDNC (18)\\
	& PC-PseTNC (66)\\
	& SC-PseDNC (92)\\
	& SC-PseTNC (88)\\
	\hline
	
	
	& Feature type 1 \\
	\textbf{PyInteraction} & Feature type 2 \\
	& Feature type 3 \\\hline
	
\end{longtable}



\section{The Python programming language}
\label{overview:the-python-programming-language}
Python is a powerful programming language that allows simple and flexible representations of biochemical molecules, and clear and concise expressions of bioinformatics algorithms. Python has a vibrant and growing ecosystem of packages that PyBioMed uses to provide more features such as RDkit and Pybel. In addition, Python is also an excellent “glue” language for putting together pieces of software from other languages which allows reuse of legacy code and engineering of high-performance algorithms. Equally important, Python is free, well-supported, and a joy to use. In order to make full use of PyBioMed, you will want to know how to write basic programs in Python. Among the many guides to Python, we recommend the documentation at \url{http://www.python.org}.


\chapter{Getting Started with PyBioMed}
\label{User_guide:getting-started-with-pybiomed}\label{User_guide::doc}\label{User_guide:pyinter}
This document is intended to provide an overview of how one can use the PyBioMed functionality from Python. If you find mistakes, or have suggestions for improvements, please either fix them yourselves in the source document (the .py file) or send them to the mailing list: \href{mailto:oriental-cds@163.com}{oriental-cds@163.com} and \href{mailto:gadsby@163.com}{gadsby@163.com}.

This document is intended to provide an overview of how one can use the PyBioMed functionality from Python. If you find mistakes, or have suggestions for improvements, please either fix them yourselves in the source document (the .py file) or send them to the mailing list: \href{mailto:oriental-cds@163.com}{oriental-cds@163.com} and \href{mailto:gadsby@163.com}{gadsby@163.com}


\section{Installing the PyBioMed package}
\label{User_guide:installing-the-pybiomed-package}
PyBioMed has been successfully tested on Linux and Windows systems. The user could download the
PyBioMed package via: \url{http://pybiomed.scbdd.com/download/PyBioMed-1.0.zip} or  \url{https://github.com/gadsbyfly/PyBioMed/blob/master/doc/download/PyBioMed-1.0.zip}. The installation process of PyBioMed is very easy:

\begin{notice}{note}{Note:}
You first need to install RDKit and pybel successfully.
\end{notice}

On Windows:

(1): download the PyBioMed-1.0.zip

(2): extract or uncompress the PyBioMed-1.0.zip file

(3): cd PyBioMed-1.0

(4): python setup.py install

On Linux:

(1): download the PyBioMed package (.tar.gz)

(2): tar -zxvf PyBioMed-1.0.tar.gz

(3): cd PyBioMed-1.0

(4): python setup.py install


\section{Getting molecules}
\label{User_guide:getting-molecules}
The \sphinxcode{PyGetMol} provide different formats to get molecular structures, protein sequence and DNA sequence.


\subsection{Getting molecular structure}
\label{User_guide:getting-molecular-structure}
In order to be convenient to users, the {\hyperref[reference/Getmol:module\string-Getmol]{\sphinxcrossref{\sphinxcode{Getmol}}}} module provides the tool to get molecular structures by the molecular ID from website including NCBI, EBI, CAS, Kegg and Drugbank.

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed}\PYG{n+nn}{.}\PYG{n+nn}{PyGetMol} \PYG{k}{import} \PYG{n}{Getmol}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{DrugBankID} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{DB01014}\PYG{l+s+s1}{\PYGZsq{}}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{smi} \PYG{o}{=} \PYG{n}{Getmol}\PYG{o}{.}\PYG{n}{GetMolFromDrugbank}\PYG{p}{(}\PYG{n}{DrugBankID}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n}{smi}
\PYG{g+go}{N[C@@H](CO)C(=O)O}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{smi}\PYG{o}{=}\PYG{n}{Getmol}\PYG{o}{.}\PYG{n}{GetMolFromCAS}\PYG{p}{(}\PYG{n}{casid}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{50\PYGZhy{}12\PYGZhy{}4}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n}{smi}
\PYG{g+go}{CCC1(c2ccccc2)C(=O)N(C)C(=N1)O}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{smi}\PYG{o}{=}\PYG{n}{Getmol}\PYG{o}{.}\PYG{n}{GetMolFromNCBI}\PYG{p}{(}\PYG{n}{cid}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{2244}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n}{smi}
\PYG{g+go}{CC(=O)Oc1ccccc1C(=O)O}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{smi}\PYG{o}{=}\PYG{n}{Getmol}\PYG{o}{.}\PYG{n}{GetMolFromKegg}\PYG{p}{(}\PYG{n}{kid}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{D02176}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n}{smi}
\PYG{g+go}{C[N+](C)(C)C[C@H](O)CC(=O)[O\PYGZhy{}]}
\end{Verbatim}


\subsection{Reading molecules}
\label{User_guide:reading-molecules}
The {\hyperref[reference/Getmol:module\string-Getmol]{\sphinxcrossref{\sphinxcode{Getmol}}}} module also provides the tool to read molecules in different formats including SDF, Mol, InChi and Smiles.

Users can read a molecule from string.

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed}\PYG{n+nn}{.}\PYG{n+nn}{PyGetMol}\PYG{n+nn}{.}\PYG{n+nn}{Getmol} \PYG{k}{import} \PYG{n}{ReadMolFromSmile}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{mol} \PYG{o}{=} \PYG{n}{ReadMolFromSmile}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{N[C@@H](CO)C(=O)O}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n}{mol}
\PYG{g+go}{\PYGZlt{}rdkit.Chem.rdchem.Mol object at 0x0D8D3688\PYGZgt{}}
\end{Verbatim}

Users can also read a molecule from a file.

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed}\PYG{n+nn}{.}\PYG{n+nn}{PyGetMol}\PYG{n+nn}{.}\PYG{n+nn}{Getmol} \PYG{k}{import} \PYG{n}{ReadMolFromSDF}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{mol} \PYG{o}{=} \PYG{n}{ReadMolFromSDF}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{./PyBioMed/test/test\PYGZus{}data/test.sdf}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}  \PYG{c+c1}{\PYGZsh{}You should change the path to your own real path}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n}{mol}
\PYG{g+go}{\PYGZlt{}rdkit.Chem.rdmolfiles.SDMolSupplier at 0xd8d03f0\PYGZgt{}}
\end{Verbatim}


\subsection{Getting protein sequence}
\label{User_guide:getting-protein-sequence}
The {\hyperref[reference/GetProtein:module\string-GetProtein]{\sphinxcrossref{\sphinxcode{GetProtein}}}} module provides the tool to get protein sequence by the pdb ID and uniprot ID from website.

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed}\PYG{n+nn}{.}\PYG{n+nn}{PyGetMol} \PYG{k}{import} \PYG{n}{GetProtein}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{GetProtein}\PYG{o}{.}\PYG{n}{GetPDB}\PYG{p}{(}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{1atp}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{seq} \PYG{o}{=} \PYG{n}{GetProtein}\PYG{o}{.}\PYG{n}{GetSeqFromPDB}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{1atp.pdb}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n}{seq}
\PYG{g+go}{GNAAAAKKGSEQESVKEFLAKAKEDFLKKWETPSQNTAQLDQFDRIKTLGTGSFGRVMLVKHKESGNHYAMKILDKQKVVKLKQ}
\PYG{g+go}{IEHTLNEKRILQAVNFPFLVKLEFSFKDNSNLYMVMEYVAGGEMFSHLRRIGRFSEPHARFYAAQIVLTFEYLHSLDLIYRDLK}
\PYG{g+go}{PENLLIDQQGYIQVTDFGFAKRVKGRTWXLCGTPEYLAPEIILSKGYNKAVDWWALGVLIYEMAAGYPPFFADQPIQIYEKIVS}
\PYG{g+go}{GKVRFPSHFSSDLKDLLRNLLQVDLTKRFGNLKNGVNDIKNHKWFATTDWIAIYQRKVEAPFIPKFKGPGDTSNFDDYEEEEIR}
\PYG{g+go}{VXINEKCGKEFTEFTTYADFIASGRTGRRNAIHD}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{seq} \PYG{o}{=} \PYG{n}{GetProtein}\PYG{o}{.}\PYG{n}{GetProteinSequence}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{O00560}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n}{seq}
\PYG{g+go}{MSLYPSLEDLKVDKVIQAQTAFSANPANPAILSEASAPIPHDGNLYPRLYPELSQYMGLSLNEEEIRANVAVVSGAPLQGQLVA}
\PYG{g+go}{RPSSINYMVAPVTGNDVGIRRAEIKQGIREVILCKDQDGKIGLRLKSIDNGIFVQLVQANSPASLVGLRFGDQVLQINGENCAG}
\PYG{g+go}{WSSDKAHKVLKQAFGEKITMTIRDRPFERTITMHKDSTGHVGFIFKNGKITSIVKDSSAARNGLLTEHNICEINGQNVIGLKDS}
\PYG{g+go}{QIADILSTSGTVVTITIMPAFIFEHIIKRMAPSIMKSLMDHTIPEV}
\end{Verbatim}


\subsection{Reading protein sequence}
\label{User_guide:reading-protein-sequence}
\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed}\PYG{n+nn}{.}\PYG{n+nn}{PyGetMol}\PYG{n+nn}{.}\PYG{n+nn}{GetProtein} \PYG{k}{import} \PYG{n}{ReadFasta}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{f} \PYG{o}{=} \PYG{n+nb}{open}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{./PyBioMed/test/test\PYGZus{}data/protein.fasta}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}  \PYG{c+c1}{\PYGZsh{}You should change the path to your own real path}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{protein\PYGZus{}seq} \PYG{o}{=} \PYG{n}{ReadFasta}\PYG{p}{(}\PYG{n}{f}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n}{protein}
\PYG{g+go}{[\PYGZsq{}MLIHQYDHATAQYIASHLADPDPLNDGRWLIPAFATATPLPERPARTWPFFLDGAWVLRPDHRGQRLYRTDTGEAAEIVAAG}
\PYG{g+go}{IAPEAAGLTPTPRPSDEHRWIDGAWQIDPQIVAQRARDAAMREFDLRMASARQANAGRADAYAAGLLSDAEIAVFKAWAIYQMD}
\PYG{g+go}{LVRVVSAASFPDDVQWPAEPDEAAVIEQADGKASAGDAAAA\PYGZsq{},}
\PYG{g+go}{\PYGZsq{}MLIHQYDHATAQYIASHLADPDPLNDGRWLIPAFATATPLPERPARTWPFFLDGAWVLRPDHRGQRLYRTDTGEAAEIVAAGI}
\PYG{g+go}{APEAAGLTPTPRPSDEHRWIDGAWQIDPQIVAQRARDAAMREFDLRMASARQANAGRADAYAAGLLSDAEIAVFKAWAIYQMDL}
\PYG{g+go}{VRVVSAASFPDDVQWPAEPDEAAVIEQADGKASAGDAAAA\PYGZsq{}]}
\end{Verbatim}


\subsection{Getting DNA sequence}
\label{User_guide:getting-dna-sequence}
The {\hyperref[reference/GetDNA:module\string-GetDNA]{\sphinxcrossref{\sphinxcode{GetDNA}}}} module provides the tool to get DNA sequence by the Gene ID from website.

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed}\PYG{n+nn}{.}\PYG{n+nn}{PyGetMol} \PYG{k}{import} \PYG{n}{GetDNA}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{seq} \PYG{o}{=} \PYG{n}{GetDNA}\PYG{o}{.}\PYG{n}{GetDNAFromUniGene}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{AA954964}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n}{seq}
\PYG{g+go}{\PYGZgt{}ENA\textbar{}AA954964\textbar{}AA954964.1 op24b10.s1 Soares\PYGZus{}NFL\PYGZus{}T\PYGZus{}GBC\PYGZus{}S1 Homo sapiens cDNA clone IMAGE:1577755 3\PYGZam{}apos;, mRNA sequence.}
\PYG{g+go}{TTTTAAAATATAAAAGGATAACTTTATTGAATATACAAATTCAAGAGCATTCAATTTTTT}
\PYG{g+go}{TTTAAGATTATGGCATAAGACAGATCAATGGTAATGGTTTATATATCCTATACTTACCAA}
\PYG{g+go}{ACAGATTAGGTAGATATACTGACCTATCAATGCTCAAAATAACAAAATGAATACATGTCC}
\PYG{g+go}{CTAAACTATTTCTGTATTCTATGACTACTAAATGGGAAATCTGTCAGCTGACCACCCACC}
\PYG{g+go}{AGACTTTTTCCCATAGGAAGTTTGATATGCTGTCATTGATATATACCATTTCTGAATATA}
\PYG{g+go}{AACCTCTATCTTGGGTCCTTTTCTCTTTGCCTACTTCATTATCTGTCTTCCCAACCCACC}
\PYG{g+go}{TAAGACTTAGTCAAAACAGGATACAGAGATCTGGATGGCTCTACGCAGAG}
\end{Verbatim}


\subsection{Reading DNA sequence}
\label{User_guide:reading-dna-sequence}
\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed}\PYG{n+nn}{.}\PYG{n+nn}{PyGetMol}\PYG{n+nn}{.}\PYG{n+nn}{GetDNA} \PYG{k}{import} \PYG{n}{ReadFasta}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{f} \PYG{o}{=} \PYG{n+nb}{open}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{./PyBioMed/test/test\PYGZus{}data/example.fasta}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}  \PYG{c+c1}{\PYGZsh{}You should change the path to your own real path}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{dna\PYGZus{}seq} \PYG{o}{=} \PYG{n}{ReadFasta}\PYG{p}{(}\PYG{n}{f}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n}{dna}
\PYG{g+go}{[\PYGZsq{}GACTGAACTGCACTTTGGTTTCATATTATTTGCTC\PYGZsq{}]}
\end{Verbatim}


\section{Pretreating structure}
\label{User_guide:pretreating-structure}
The \sphinxcode{PyPretreat} can pretreat the molecular structure, the protein sequence and the DNA sequence.


\subsection{Pretreating molecules}
\label{User_guide:pretreating-molecules}
The {\hyperref[reference/PyPretreatMol:module\string-PyPretreatMol]{\sphinxcrossref{\sphinxcode{PyPretreatMol}}}} can pretreat the molecular structure. The {\hyperref[reference/PyPretreatMol:module\string-PyPretreatMol]{\sphinxcrossref{\sphinxcode{PyPretreatMol}}}} proivdes the following functions:
\begin{itemize}
\item {} 
Normalization of functional groups to a consistent format.

\item {} 
Recombination of separated charges.

\item {} 
Breaking of bonds to metal atoms.

\item {} 
Competitive reionization to ensure strongest acids ionize first in partially ionize molecules.

\item {} 
Tautomer enumeration and canonicalization.

\item {} 
Neutralization of charges.

\item {} 
Standardization or removal of stereochemistry information.

\item {} 
Filtering of salt and solvent fragments.

\item {} 
Generation of fragment, isotope, charge, tautomer or stereochemistry insensitive parent structures.

\item {} 
Validations to identify molecules with unusual and potentially troublesome characteristics.

\end{itemize}

The user can diconnect metal ion.

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed}\PYG{n+nn}{.}\PYG{n+nn}{PyPretreat}\PYG{n+nn}{.}\PYG{n+nn}{PyPretreatMol} \PYG{k}{import} \PYG{n}{StandardizeMol}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{rdkit} \PYG{k}{import} \PYG{n}{Chem}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{mol} \PYG{o}{=} \PYG{n}{Chem}\PYG{o}{.}\PYG{n}{MolFromSmiles}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{[Na]OC(=O)c1ccc(C[S+2]([O\PYGZhy{}])([O\PYGZhy{}]))cc1}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{sdm} \PYG{o}{=} \PYG{n}{StandardizeMol}\PYG{p}{(}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{mol} \PYG{o}{=} \PYG{n}{sdm}\PYG{o}{.}\PYG{n}{disconnect\PYGZus{}metals}\PYG{p}{(}\PYG{n}{mol}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n}{Chem}\PYG{o}{.}\PYG{n}{MolToSmiles}\PYG{p}{(}\PYG{n}{mol}\PYG{p}{,} \PYG{n}{isomericSmiles}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}
\PYG{g+go}{O=C([O\PYGZhy{}])c1ccc(C[S+2]([O\PYGZhy{}])[O\PYGZhy{}])cc1.[Na+]}
\end{Verbatim}

Pretreat the molecular structure using all functions.

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed}\PYG{n+nn}{.}\PYG{n+nn}{PyPretreat} \PYG{k}{import} \PYG{n}{PyPretreatMol}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{stdsmi} \PYG{o}{=} \PYG{n}{PyPretreatMol}\PYG{o}{.}\PYG{n}{StandardSmi}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{[Na]OC(=O)c1ccc(C[S+2]([O\PYGZhy{}])([O\PYGZhy{}]))cc1}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n}{stdsmi}
\PYG{g+go}{O=C([O\PYGZhy{}])c1ccc(C[S](=O)=O)cc1}
\end{Verbatim}


\subsection{Pretreating protein sequence}
\label{User_guide:pretreating-protein-sequence}
The user can check the protein sequence using the {\hyperref[reference/PyPretreatPro:module\string-PyPretreatPro]{\sphinxcrossref{\sphinxcode{PyPretreatPro}}}}. If the sequence is right, the result is the number of amino acids. If the sequence is wrong, the result is 0.

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed}\PYG{n+nn}{.}\PYG{n+nn}{PyPretreat} \PYG{k}{import} \PYG{n}{PyPretreatPro}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{protein}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{ADGCGVGEGTGQGPMCNCMCMKWVYADEDAADLESDSFADEDASLESDSFPWSNQRVFCSFADEDASU}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n}{PyPretreatPro}\PYG{o}{.}\PYG{n}{ProteinCheck}\PYG{p}{(}\PYG{n}{protein}\PYG{p}{)}
\PYG{g+go}{0}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed}\PYG{n+nn}{.}\PYG{n+nn}{PyPretreat} \PYG{k}{import} \PYG{n}{PyPretreatPro}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{protein}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{ADGCRN}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n}{PyPretreatPro}\PYG{o}{.}\PYG{n}{ProteinCheck}\PYG{p}{(}\PYG{n}{protein}\PYG{p}{)}
\PYG{g+go}{6}
\end{Verbatim}


\subsection{Pretreating DNA sequence}
\label{User_guide:pretreating-dna-sequence}
The user can check the DNA sequence using the {\hyperref[reference/PyPretreatDNA:module\string-PyPretreatDNA]{\sphinxcrossref{\sphinxcode{PyPretreatDNA}}}}. If the sequence is right, the result is True. If the sequence is wrong, the result is the wrong word.

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed}\PYG{n+nn}{.}\PYG{n+nn}{PyPretreat} \PYG{k}{import} \PYG{n}{PyPretreatDNA}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{DNA}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{ATTTAC}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n}{PyPretreatDNA}\PYG{o}{.}\PYG{n}{DNAChecks}\PYG{p}{(}\PYG{n}{DNA}\PYG{p}{)}
\PYG{g+go}{True}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{DNA}\PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{ATCGUA}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n}{PyPretreatDNA}\PYG{o}{.}\PYG{n}{DNAChecks}\PYG{p}{(}\PYG{n}{DNA}\PYG{p}{)}
\PYG{g+go}{U}
\end{Verbatim}


\section{Calculating molecular descriptors}
\label{User_guide:calculating-molecular-descriptors}
The PyBioMed package could calculate a large number of molecular descriptors. These descriptors capture and magnify distinct aspects of chemical structures. Generally speaking, all descriptors could be divided into two classes: descriptors and fingerprints. Descriptors only used the property of molecular topology, including constitutional descriptors, topological descriptors, connectivity indices, E-state indices, Basak information indices, Burden descriptors, autocorrelation descriptors, charge descriptors, molecular properties, kappa shape indices, MOE-type descriptors. Molecular fingerprints contain FP2, FP3, FP4,topological fingerprints, Estate, atompairs, torsions, morgan and MACCS.
\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics[width=10cm]{{single_features}.png}
\caption{The descriptors could be calculated through PyBioMed package}\label{User_guide:id1}\end{figure}


\subsection{Calculating descriptors}
\label{User_guide:calculating-descriptors}
We could import the corresponding module to calculate the molecular descriptors as need. There is 14 modules to compute descriptors. Moreover, a easier way to compute these descriptors is construct a PyMolecule object, which encapsulates all methods for the calculation of descriptors.


\subsubsection{Calculating molecular descriptors via functions}
\label{User_guide:calculating-molecular-descriptors-via-functions}
The \sphinxcode{GetConnectivity()} function in the {\hyperref[reference/connectivity:module\string-connectivity]{\sphinxcrossref{\sphinxcode{connectivity}}}} module can calculate the connectivity descriptors. The result is given in the form of dictionary.

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed}\PYG{n+nn}{.}\PYG{n+nn}{PyMolecule} \PYG{k}{import} \PYG{n}{connectivity}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{rdkit} \PYG{k}{import} \PYG{n}{Chem}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{smi} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CCC1(c2ccccc2)C(=O)N(C)C(=N1)O}\PYG{l+s+s1}{\PYGZsq{}}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{mol} \PYG{o}{=} \PYG{n}{Chem}\PYG{o}{.}\PYG{n}{MolFromSmiles}\PYG{p}{(}\PYG{n}{smi}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{molecular\PYGZus{}descriptor} \PYG{o}{=} \PYG{n}{connectivity}\PYG{o}{.}\PYG{n}{GetConnectivity}\PYG{p}{(}\PYG{n}{mol}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n}{molecular\PYGZus{}descriptor}
\PYG{g+go}{\PYGZob{}\PYGZsq{}Chi3ch\PYGZsq{}: 0.0, \PYGZsq{}knotp\PYGZsq{}: 2.708, \PYGZsq{}dchi3\PYGZsq{}: 3.359, \PYGZsq{}dchi2\PYGZsq{}: 2.895, \PYGZsq{}dchi1\PYGZsq{}: 2.374, \PYGZsq{}dchi0\PYGZsq{}: 2.415, \PYGZsq{}Chi5ch\PYGZsq{}: 0.068, \PYGZsq{}Chiv4\PYGZsq{}: 1.998, \PYGZsq{}Chiv7\PYGZsq{}: 0.259, \PYGZsq{}Chiv6\PYGZsq{}: 0.591, \PYGZsq{}Chiv1\PYGZsq{}: 5.241, \PYGZsq{}Chiv0\PYGZsq{}: 9.344, \PYGZsq{}Chiv3\PYGZsq{}: 3.012, \PYGZsq{}Chiv2\PYGZsq{}: 3.856, \PYGZsq{}Chi4c\PYGZsq{}: 0.083, \PYGZsq{}dchi4\PYGZsq{}: 2.96, \PYGZsq{}Chiv4pc\PYGZsq{}: 1.472, \PYGZsq{}Chiv3c\PYGZsq{}: 0.588, \PYGZsq{}Chiv8\PYGZsq{}: 0.118, \PYGZsq{}Chi3c\PYGZsq{}: 1.264, \PYGZsq{}Chi8\PYGZsq{}: 0.636, \PYGZsq{}Chi9\PYGZsq{}: 0.322, \PYGZsq{}Chi2\PYGZsq{}: 6.751, \PYGZsq{}Chi3\PYGZsq{}: 6.372, \PYGZsq{}Chi0\PYGZsq{}: 11.759, \PYGZsq{}Chi1\PYGZsq{}: 7.615, \PYGZsq{}Chi6\PYGZsq{}: 2.118, \PYGZsq{}Chi7\PYGZsq{}: 1.122, \PYGZsq{}Chi4\PYGZsq{}: 4.959, \PYGZsq{}Chi5\PYGZsq{}: 3.649, \PYGZsq{}Chiv5\PYGZsq{}: 1.244, \PYGZsq{}Chiv4c\PYGZsq{}: 0.04, \PYGZsq{}Chiv9\PYGZsq{}: 0.046, \PYGZsq{}Chi4pc\PYGZsq{}: 3.971, \PYGZsq{}knotpv\PYGZsq{}: 0.884, \PYGZsq{}Chiv5ch\PYGZsq{}: 0.025, \PYGZsq{}Chiv3ch\PYGZsq{}: 0.0, \PYGZsq{}Chiv10\PYGZsq{}: 0.015, \PYGZsq{}Chiv6ch\PYGZsq{}: 0.032, \PYGZsq{}Chi10\PYGZsq{}: 0.135, \PYGZsq{}Chi4ch\PYGZsq{}: 0.0, \PYGZsq{}Chiv4ch\PYGZsq{}: 0.0, \PYGZsq{}mChi1\PYGZsq{}: 0.448, \PYGZsq{}Chi6ch\PYGZsq{}: 0.102\PYGZcb{}}
\end{Verbatim}

The function \sphinxcode{GetTopology()} in the {\hyperref[reference/topology:module\string-topology]{\sphinxcrossref{\sphinxcode{topology}}}} module can calculate all topological descriptors.

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed}\PYG{n+nn}{.}\PYG{n+nn}{PyMolecule} \PYG{k}{import} \PYG{n}{topology}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{rdkit} \PYG{k}{import} \PYG{n}{Chem}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{smi} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CCC1(c2ccccc2)C(=O)N(C)C(=N1)O}\PYG{l+s+s1}{\PYGZsq{}}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{mol} \PYG{o}{=} \PYG{n}{Chem}\PYG{o}{.}\PYG{n}{MolFromSmiles}\PYG{p}{(}\PYG{n}{smi}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{molecular\PYGZus{}descriptor} \PYG{o}{=} \PYG{n}{topology}\PYG{o}{.}\PYG{n}{GetTopology}\PYG{p}{(}\PYG{n}{mol}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{molecular\PYGZus{}descriptor}\PYG{p}{)}
\PYG{g+go}{25}
\end{Verbatim}

The function \sphinxcode{CATS2D()} in the {\hyperref[reference/cats2d:module\string-cats2d]{\sphinxcrossref{\sphinxcode{cats2d}}}} module can calculate all CATS2D descriptors.

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed}\PYG{n+nn}{.}\PYG{n+nn}{PyMolecule}\PYG{n+nn}{.}\PYG{n+nn}{cats2d} \PYG{k}{import} \PYG{n}{CATS2D}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{smi} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CC(N)C(=O)[O\PYGZhy{}]}\PYG{l+s+s1}{\PYGZsq{}}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{mol} \PYG{o}{=} \PYG{n}{Chem}\PYG{o}{.}\PYG{n}{MolFromSmiles}\PYG{p}{(}\PYG{n}{smi}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{cats} \PYG{o}{=} \PYG{n}{CATS2D}\PYG{p}{(}\PYG{n}{mol}\PYG{p}{,}\PYG{n}{PathLength} \PYG{o}{=} \PYG{l+m+mi}{10}\PYG{p}{,}\PYG{n}{scale} \PYG{o}{=} \PYG{l+m+mi}{3}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n}{cats}
\PYG{g+go}{\PYGZob{}\PYGZsq{}CATS\PYGZus{}AP4\PYGZsq{}: 0.0, \PYGZsq{}CATS\PYGZus{}AP3\PYGZsq{}: 1.0, \PYGZsq{}CATS\PYGZus{}AP6\PYGZsq{}: 0.0, \PYGZsq{}CATS\PYGZus{}AA8\PYGZsq{}: 0.0, \PYGZsq{}CATS\PYGZus{}AA9\PYGZsq{}: 0.0, \PYGZsq{}CATS\PYGZus{}AP1\PYGZsq{}: 0.0, ......, \PYGZsq{}CATS\PYGZus{}AP5\PYGZsq{}: 0.0\PYGZcb{}}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{cats}\PYG{p}{)}
\PYG{g+go}{150}
\end{Verbatim}


\subsubsection{Calculating molecular descriptors via PyMolecule object}
\label{User_guide:calculating-molecular-descriptors-via-pymolecule-object}
The \sphinxcode{PyMolecule} class can read molecules in different format including MOL, SMI, InChi and CAS. For example, the user can read a molecule in the format of SMI and calculate the E-state descriptors (316).

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed} \PYG{k}{import} \PYG{n}{Pymolecule}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{smi} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CCC1(c2ccccc2)C(=O)N(C)C(=N1)O}\PYG{l+s+s1}{\PYGZsq{}}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{mol} \PYG{o}{=} \PYG{n}{Pymolecule}\PYG{o}{.}\PYG{n}{PyMolecule}\PYG{p}{(}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{mol}\PYG{o}{.}\PYG{n}{ReadMolFromSmile}\PYG{p}{(}\PYG{n}{smi}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{molecular\PYGZus{}descriptor} \PYG{o}{=} \PYG{n}{mol}\PYG{o}{.}\PYG{n}{GetEstate}\PYG{p}{(}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{molecular\PYGZus{}descriptor}\PYG{p}{)}
\PYG{g+go}{237}
\end{Verbatim}

The object can also read molecules in the format of MOL file and calculate charge descriptors (25).

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed} \PYG{k}{import} \PYG{n}{Pymolecule}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{mol} \PYG{o}{=} \PYG{n}{Pymolecule}\PYG{o}{.}\PYG{n}{PyMolecule}\PYG{p}{(}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{mol}\PYG{o}{.}\PYG{n}{ReadMolFromMol}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{test/test\PYGZus{}data/test.mol}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}   \PYG{c+c1}{\PYGZsh{}change path to the real path in your own computer}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{molecular\PYGZus{}descriptor} \PYG{o}{=} \PYG{n}{mol}\PYG{o}{.}\PYG{n}{GetCharge}\PYG{p}{(}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n}{molecular\PYGZus{}descriptor}
\PYG{g+go}{\PYGZob{}\PYGZsq{}QNmin\PYGZsq{}: 0, \PYGZsq{}QOss\PYGZsq{}: 0.534, \PYGZsq{}Mpc\PYGZsq{}: 0.122, \PYGZsq{}QHss\PYGZsq{}: 0.108, \PYGZsq{}SPP\PYGZsq{}: 0.817, \PYGZsq{}LDI\PYGZsq{}: 0.322, \PYGZsq{}QCmin\PYGZsq{}: \PYGZhy{}0.061, \PYGZsq{}Mac\PYGZsq{}: 0.151, \PYGZsq{}Qass\PYGZsq{}: 0.893, \PYGZsq{}QNss\PYGZsq{}: 0, \PYGZsq{}QCmax\PYGZsq{}: 0.339, \PYGZsq{}QOmax\PYGZsq{}: \PYGZhy{}0.246, \PYGZsq{}Tpc\PYGZsq{}: 1.584, \PYGZsq{}Qmax\PYGZsq{}: 0.339, \PYGZsq{}QOmin\PYGZsq{}: \PYGZhy{}0.478, \PYGZsq{}Tnc\PYGZsq{}: \PYGZhy{}1.584, \PYGZsq{}QHmin\PYGZsq{}: 0.035, \PYGZsq{}QCss\PYGZsq{}: 0.252, \PYGZsq{}QHmax\PYGZsq{}: 0.297, \PYGZsq{}QNmax\PYGZsq{}: 0, \PYGZsq{}Rnc\PYGZsq{}: 0.302, \PYGZsq{}Rpc\PYGZsq{}: 0.214, \PYGZsq{}Qmin\PYGZsq{}: \PYGZhy{}0.478, \PYGZsq{}Tac\PYGZsq{}: 3.167, \PYGZsq{}Mnc\PYGZsq{}: \PYGZhy{}0.198\PYGZcb{}}
\end{Verbatim}

In order to be convenient to users, the object also provides the tool to get molecular structures by the molecular ID from website including NCBI, EBI, CAS, Kegg and Drugbank.

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed} \PYG{k}{import} \PYG{n}{Pymolecule}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{DrugBankID} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{DB01014}\PYG{l+s+s1}{\PYGZsq{}}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{mol} \PYG{o}{=} \PYG{n}{Pymolecule}\PYG{o}{.}\PYG{n}{PyMolecule}\PYG{p}{(}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{smi} \PYG{o}{=} \PYG{n}{mol}\PYG{o}{.}\PYG{n}{GetMolFromDrugbank}\PYG{p}{(}\PYG{n}{DrugBankID}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{mol}\PYG{o}{.}\PYG{n}{ReadMolFromSmile}\PYG{p}{(}\PYG{n}{smi}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{molecular\PYGZus{}descriptor} \PYG{o}{=} \PYG{n}{mol}\PYG{o}{.}\PYG{n}{GetKappa}\PYG{p}{(}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n}{molecular\PYGZus{}descriptor}
\PYG{g+go}{\PYGZob{}\PYGZsq{}phi\PYGZsq{}: 5.989303307692309, \PYGZsq{}kappa1\PYGZsq{}: 22.291, \PYGZsq{}kappa3\PYGZsq{}: 7.51, \PYGZsq{}kappa2\PYGZsq{}: 11.111, \PYGZsq{}kappam1\PYGZsq{}: 18.587, \PYGZsq{}kappam3\PYGZsq{}: 5.395, \PYGZsq{}kappam2\PYGZsq{}: 8.378\PYGZcb{}}
\end{Verbatim}

The code below can calculate all molecular descriptors except fingerprints.

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed} \PYG{k}{import} \PYG{n}{Pymolecule}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{smi} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CCOC=N}\PYG{l+s+s1}{\PYGZsq{}}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{mol} \PYG{o}{=} \PYG{n}{Pymolecule}\PYG{o}{.}\PYG{n}{PyMolecule}\PYG{p}{(}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{mol}\PYG{o}{.}\PYG{n}{ReadMolFromSmile}\PYG{p}{(}\PYG{n}{smi}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{alldes} \PYG{o}{=} \PYG{n}{mol}\PYG{o}{.}\PYG{n}{GetAllDescriptor}\PYG{p}{(}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{alldes}\PYG{p}{)}
\PYG{g+go}{765}
\end{Verbatim}


\subsection{Calculating molecular fingerprints}
\label{User_guide:calculating-molecular-fingerprints}
In the {\hyperref[reference/fingerprint:module\string-fingerprint]{\sphinxcrossref{\sphinxcode{fingerprint}}}} module, there are eighteen types of molecular fingerprints which are defined by abstracting and magnifying different aspects of molecular topology.


\subsubsection{Calculating fingerprint via functions}
\label{User_guide:calculating-fingerprint-via-functions}
The \sphinxcode{CalculateFP2Fingerprint()} function calculates the FP2 fingerprint.

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed}\PYG{n+nn}{.}\PYG{n+nn}{PyMolecule}\PYG{n+nn}{.}\PYG{n+nn}{fingerprint} \PYG{k}{import} \PYG{n}{CalculateFP2Fingerprint}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{import} \PYG{n+nn}{pybel}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{smi} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CCC1(c2ccccc2)C(=O)N(C)C(=N1)O}\PYG{l+s+s1}{\PYGZsq{}}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{mol} \PYG{o}{=} \PYG{n}{pybel}\PYG{o}{.}\PYG{n}{readstring}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{smi}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{smi}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{mol\PYGZus{}fingerprint} \PYG{o}{=} \PYG{n}{CalculateFP2Fingerprint}\PYG{p}{(}\PYG{n}{mol}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{mol\PYGZus{}fingerprint}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{p}{)}
\PYG{g+go}{103}
\end{Verbatim}

The \sphinxcode{CalculateEstateFingerprint()} function calculates the Estate fingerprint.

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed}\PYG{n+nn}{.}\PYG{n+nn}{PyMolecule}\PYG{n+nn}{.}\PYG{n+nn}{fingerprint} \PYG{k}{import} \PYG{n}{CalculateEstateFingerprint}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{smi} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CCC1(c2ccccc2)C(=O)N(C)C(=N1)O}\PYG{l+s+s1}{\PYGZsq{}}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{mol} \PYG{o}{=} \PYG{n}{Chem}\PYG{o}{.}\PYG{n}{MolFromSmiles}\PYG{p}{(}\PYG{n}{smi}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{mol\PYGZus{}fingerprint} \PYG{o}{=} \PYG{n}{CalculateEstateFingerprint}\PYG{p}{(}\PYG{n}{mol}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{mol\PYGZus{}fingerprint}\PYG{p}{[}\PYG{l+m+mi}{2}\PYG{p}{]}\PYG{p}{)}
\PYG{g+go}{79}
\end{Verbatim}

The function \sphinxcode{GhoseCrippenFingerprint()} in the {\hyperref[reference/ghosecrippen:module\string-ghosecrippen]{\sphinxcrossref{\sphinxcode{ghosecrippen}}}} module can calculate all ghosecrippen descriptors.

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed}\PYG{n+nn}{.}\PYG{n+nn}{PyMolecule}\PYG{n+nn}{.}\PYG{n+nn}{ghosecrippen} \PYG{k}{import} \PYG{n}{GhoseCrippenFingerprint}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{smi} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CC(N)C(=O)O}\PYG{l+s+s1}{\PYGZsq{}}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{mol} \PYG{o}{=} \PYG{n}{Chem}\PYG{o}{.}\PYG{n}{MolFromSmiles}\PYG{p}{(}\PYG{n}{smi}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{ghoseFP} \PYG{o}{=} \PYG{n}{GhoseCrippenFingerprint}\PYG{p}{(}\PYG{n}{mol}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n}{ghoseFP}
\PYG{g+go}{\PYGZob{}\PYGZsq{}S3\PYGZsq{}: 0, \PYGZsq{}S2\PYGZsq{}: 0, \PYGZsq{}S1\PYGZsq{}: 0, \PYGZsq{}S4\PYGZsq{}: 0, ......, \PYGZsq{}N9\PYGZsq{}: 0, \PYGZsq{}Hal2\PYGZsq{}: 0\PYGZcb{}}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{ghoseFP}\PYG{p}{)}
\PYG{g+go}{110}
\end{Verbatim}


\subsubsection{Calculating fingerprint via object}
\label{User_guide:calculating-fingerprint-via-object}
The \sphinxcode{PyMolecule} class can calculate eleven kinds of fingerprints. For example, the user can read a molecule in the format of SMI and calculate the ECFP4 fingerprint (1024).

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed} \PYG{k}{import} \PYG{n}{Pymolecule}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{smi} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CCOC=N}\PYG{l+s+s1}{\PYGZsq{}}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{mol} \PYG{o}{=} \PYG{n}{Pymolecule}\PYG{o}{.}\PYG{n}{PyMolecule}\PYG{p}{(}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{mol}\PYG{o}{.}\PYG{n}{ReadMolFromSmile}\PYG{p}{(}\PYG{n}{smi}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{res} \PYG{o}{=} \PYG{n}{mol}\PYG{o}{.}\PYG{n}{GetFingerprint}\PYG{p}{(}\PYG{n}{FPName}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{ECFP4}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n}{res}
\PYG{g+go}{(4294967295L, \PYGZob{}3994088662L: 1, 2246728737L: 1, 3542456614L: 1, 2072128742: 1, 2222711142L: 1, 2669064385L: 1, 3540009223L: 1, 849275503: 1, 2245384272L: 1, 2246703798L: 1, 864674487: 1, 4212523324L: 1, 3340482365L: 1\PYGZcb{}, \PYGZlt{}rdkit.DataStructs.cDataStructs.UIntSparseIntVect object at 0x0CA010D8\PYGZgt{})}
\end{Verbatim}


\section{Calculating protein descriptors}
\label{User_guide:calculating-protein-descriptors}
PyProtein is a tool used for protein feature calculation. PyProtein calculates structural and physicochemical features of proteins and peptides from amino acid sequence. These sequence-derived structural and physicochemical features have been widely used in the development of machine learning models for predicting protein structural and functional classes, post-translational modification, subcellular locations and peptides of specific properties. There are two ways to calculate protein descriptors in the PyProtein module. One is to directly use the corresponding methods, the other one is firstly to construct a {\hyperref[reference/PyProteinclass:module\string-PyProtein]{\sphinxcrossref{\sphinxcode{PyProtein}}}} class and then run their methods to obtain the protein descriptors. It should be noted that the output is a dictionary form, whose keys and values represent the descriptor name and the descriptor value, respectively. The user could clearly understand the meaning of each descriptor.


\subsection{Calculating protein descriptors via functions}
\label{User_guide:calculating-protein-descriptors-via-functions}
The user can input the protein sequence and calculate the protein descriptors using function.

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed}\PYG{n+nn}{.}\PYG{n+nn}{PyProtein} \PYG{k}{import} \PYG{n}{AAComposition}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{protein}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{ADGCGVGEGTGQGPMCNCMCMKWVYADEDAADLESDSFADEDASLESDSFPWSNQRVFCSFADEDAS}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{AAC}\PYG{o}{=}\PYG{n}{AAComposition}\PYG{o}{.}\PYG{n}{CalculateAAComposition}\PYG{p}{(}\PYG{n}{protein}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n}{AAC}
\PYG{g+go}{\PYGZob{}\PYGZsq{}A\PYGZsq{}: 11.94, \PYGZsq{}C\PYGZsq{}: 7.463, \PYGZsq{}E\PYGZsq{}: 8.955, \PYGZsq{}D\PYGZsq{}: 14.925, \PYGZsq{}G\PYGZsq{}: 8.955, \PYGZsq{}F\PYGZsq{}: 5.97, \PYGZsq{}I\PYGZsq{}: 0.0, \PYGZsq{}H\PYGZsq{}: 0.0, \PYGZsq{}K\PYGZsq{}: 1.493, \PYGZsq{}M\PYGZsq{}: 4.478, \PYGZsq{}L\PYGZsq{}: 2.985, \PYGZsq{}N\PYGZsq{}: 2.985, \PYGZsq{}Q\PYGZsq{}: 2.985, \PYGZsq{}P\PYGZsq{}: 2.985, \PYGZsq{}S\PYGZsq{}: 11.94, \PYGZsq{}R\PYGZsq{}: 1.493, \PYGZsq{}T\PYGZsq{}: 1.493, \PYGZsq{}W\PYGZsq{}: 2.985, \PYGZsq{}V\PYGZsq{}: 4.478, \PYGZsq{}Y\PYGZsq{}: 1.493\PYGZcb{}}
\end{Verbatim}

PyBioMed also provides \sphinxcode{getpdb} to get sequence from \href{http://www.rcsb.org/pdb/home/home.do}{PDB} website to calculate protein descriptors.

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed}\PYG{n+nn}{.}\PYG{n+nn}{PyGetMol} \PYG{k}{import} \PYG{n}{GetProtein}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{GetProtein}\PYG{o}{.}\PYG{n}{GetPDB}\PYG{p}{(}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{1atp}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{1efz}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{1f88}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{seq} \PYG{o}{=} \PYG{n}{GetProtein}\PYG{o}{.}\PYG{n}{GetSeqFromPDB}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{1atp.pdb}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n}{seq}
\PYG{g+go}{GNAAAAKKGSEQESVKEFLAKAKEDFLKKWETPSQNTAQLDQFDRIKTLGTGSFGRVMLVKHKESGNHYAMKILDKQKVVKLKQ}
\PYG{g+go}{IEHTLNEKRILQAVNFPFLVKLEFSFKDNSNLYMVMEYVAGGEMFSHLRRIGRFSEPHARFYAAQIVLTFEYLHSLDLIYRDLK}
\PYG{g+go}{PENLLIDQQGYIQVTDFGFAKRVKGRTWXLCGTPEYLAPEIILSKGYNKAVDWWALGVLIYEMAAGYPPFFADQPIQIYEKIVS}
\PYG{g+go}{GKVRFPSHFSSDLKDLLRNLLQVDLTKRFGNLKNGVNDIKNHKWFATTDWIAIYQRKVEAPFIPKFKGPGDTSNFDDYEEEEIR}
\PYG{g+go}{VXINEKCGKEFTEFTTYADFIASGRTGRRNAIHD}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed}\PYG{n+nn}{.}\PYG{n+nn}{PyProtein} \PYG{k}{import} \PYG{n}{CTD}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{protein\PYGZus{}descriptor} \PYG{o}{=} \PYG{n}{CTD}\PYG{o}{.}\PYG{n}{CalculateC}\PYG{p}{(}\PYG{n}{protein}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n}{protein\PYGZus{}descriptor}
\PYG{g+go}{\PYGZob{}\PYGZsq{}\PYGZus{}NormalizedVDWVC2\PYGZsq{}: 0.224, \PYGZsq{}\PYGZus{}PolarizabilityC2\PYGZsq{}: 0.328, \PYGZsq{}\PYGZus{}PolarizabilityC3\PYGZsq{}: 0.179, \PYGZsq{}\PYGZus{}ChargeC1\PYGZsq{}: 0.03, \PYGZsq{}\PYGZus{}PolarizabilityC1\PYGZsq{}: 0.493, \PYGZsq{}\PYGZus{}SecondaryStrC2\PYGZsq{}: 0.239, \PYGZsq{}\PYGZus{}SecondaryStrC3\PYGZsq{}: 0.418, \PYGZsq{}\PYGZus{}NormalizedVDWVC3\PYGZsq{}: 0.179, \PYGZsq{}\PYGZus{}SecondaryStrC1\PYGZsq{}: 0.343, \PYGZsq{}\PYGZus{}SolventAccessibilityC1\PYGZsq{}: 0.448, \PYGZsq{}\PYGZus{}SolventAccessibilityC2\PYGZsq{}: 0.328, \PYGZsq{}\PYGZus{}SolventAccessibilityC3\PYGZsq{}: 0.224, \PYGZsq{}\PYGZus{}NormalizedVDWVC1\PYGZsq{}: 0.522, \PYGZsq{}\PYGZus{}HydrophobicityC3\PYGZsq{}: 0.284, \PYGZsq{}\PYGZus{}HydrophobicityC1\PYGZsq{}: 0.328, \PYGZsq{}\PYGZus{}ChargeC3\PYGZsq{}: 0.239, \PYGZsq{}\PYGZus{}PolarityC2\PYGZsq{}: 0.179, \PYGZsq{}\PYGZus{}PolarityC1\PYGZsq{}: 0.299, \PYGZsq{}\PYGZus{}HydrophobicityC2\PYGZsq{}: 0.388, \PYGZsq{}\PYGZus{}PolarityC3\PYGZsq{}: 0.03, \PYGZsq{}\PYGZus{}ChargeC2\PYGZsq{}: 0.731\PYGZcb{}}
\end{Verbatim}


\subsection{Calculate protein descriptors via object}
\label{User_guide:calculate-protein-descriptors-via-object}
The {\hyperref[reference/PyProteinclass:module\string-PyProtein]{\sphinxcrossref{\sphinxcode{PyProtein}}}} can calculate all kinds of protein descriptors in the PyBioMed.
For example, the {\hyperref[reference/PyProteinclass:module\string-PyProtein]{\sphinxcrossref{\sphinxcode{PyProtein}}}} can calculate DPC.

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed} \PYG{k}{import} \PYG{n}{Pyprotein}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{protein}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{ADGCGVGEGTGQGPMCNCMCMKWVYADEDAADLESDSFADEDASLESDSFPWSNQRVFCSFADEDAS}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{protein\PYGZus{}class} \PYG{o}{=} \PYG{n}{Pyprotein}\PYG{o}{.}\PYG{n}{PyProtein}\PYG{p}{(}\PYG{n}{protein}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{protein\PYGZus{}class}\PYG{o}{.}\PYG{n}{GetDPComp}\PYG{p}{(}\PYG{p}{)}\PYG{p}{)}
\PYG{g+go}{400}
\end{Verbatim}

The {\hyperref[reference/PyProteinclass:module\string-PyProtein]{\sphinxcrossref{\sphinxcode{PyProtein}}}} also provide the tool to get sequence from \href{http://www.uniprot.org/}{Uniprot} through the Uniprot ID.

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed} \PYG{k}{import} \PYG{n}{Pyprotein}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed}\PYG{n+nn}{.}\PYG{n+nn}{PyProtein}\PYG{n+nn}{.}\PYG{n+nn}{GetProteinFromUniprot} \PYG{k}{import} \PYG{n}{GetProteinSequence}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{uniprotID} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{P48039}\PYG{l+s+s1}{\PYGZsq{}}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{protein\PYGZus{}sequence} \PYG{o}{=} \PYG{n}{GetProteinSequence}\PYG{p}{(}\PYG{n}{uniprotID}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n}{protein\PYGZus{}sequence}
\PYG{g+go}{MEDINFASLAPRHGSRPFMGTWNEIGTSQLNGGAFSWSSLWSGIKNFGSSIKSFGNKAWNSNTGQMLRDKLKDQNFQQKVVDGL}
\PYG{g+go}{ASGINGVVDIANQALQNQINQRLENSRQPPVALQQRPPPKVEEVEVEEKLPPLEVAPPLPSKGEKRPRPDLEETLVVESREPPS}
\PYG{g+go}{YEQALKEGASPYPMTKPIGSMARPVYGKESKPVTLELPPPVPTVPPMPAPTLGTAVSRPTAPTVAVATPARRPRGANWQSTLNS}
\PYG{g+go}{IVGLGVKSLKRRRCY}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{protein\PYGZus{}class} \PYG{o}{=} \PYG{n}{Pyprotein}\PYG{o}{.}\PYG{n}{PyProtein}\PYG{p}{(}\PYG{n}{protein\PYGZus{}sequence}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{CTD} \PYG{o}{=} \PYG{n}{protein\PYGZus{}class}\PYG{o}{.}\PYG{n}{GetCTD}\PYG{p}{(}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{CTD}\PYG{p}{)}
\PYG{g+go}{147}
\end{Verbatim}

The {\hyperref[reference/PyProteinclass:module\string-PyProtein]{\sphinxcrossref{\sphinxcode{PyProtein}}}} can calculate all protein descriptors except the tri-peptide composition descriptors.

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed} \PYG{k}{import} \PYG{n}{Pyprotein}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{protein}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{ADGCGVGEGTGQGPMCNCMCMKWVYADEDAADLESDSFADEDASLESDSFPWSNQRVFCSFADEDAS}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{protein\PYGZus{}class} \PYG{o}{=} \PYG{n}{Pyprotein}\PYG{o}{.}\PYG{n}{PyProtein}\PYG{p}{(}\PYG{n}{protein}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{protein\PYGZus{}class}\PYG{o}{.}\PYG{n}{GetALL}\PYG{p}{(}\PYG{p}{)}\PYG{p}{)}
\PYG{g+go}{10049}
\end{Verbatim}


\section{Calculating DNA descriptors}
\label{User_guide:calculating-dna-descriptors}
The PyDNA module can generate various feature vectors for DNA sequences, this module could:
\begin{itemize}
\item {} 
Calculating three nucleic acid composition features describing the local sequence information by means of kmers (subsequences of DNA sequences);

\item {} 
Calculating six autocorrelation features describing the level of correlation between two oligonucleotides along a DNA sequence in terms of their specific physicochemical properties;

\item {} 
Calculating six pseudo nucleotide composition features, which can be used to represent a DNA sequence with a discrete model or vector yet still keep considerable sequence order information, particularly the global or long-range sequence order information, via the physicochemical properties of its constituent oligonucleotides.

\end{itemize}


\subsection{Calculating DNA descriptors via functions}
\label{User_guide:calculating-dna-descriptors-via-functions}
The user can input a DNA sequence and calculate the DNA descriptors using functions.

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed}\PYG{n+nn}{.}\PYG{n+nn}{PyDNA}\PYG{n+nn}{.}\PYG{n+nn}{PyDNAac} \PYG{k}{import} \PYG{n}{GetDAC}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{dac} \PYG{o}{=} \PYG{n}{GetDAC}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{GACTGAACTGCACTTTGGTTTCATATTATTTGCTC}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{phyche\PYGZus{}index}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Twist}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Tilt}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print}\PYG{p}{(}\PYG{n}{dac}\PYG{p}{)}
\PYG{g+go}{\PYGZob{}\PYGZsq{}DAC\PYGZus{}4\PYGZsq{}: \PYGZhy{}0.004, \PYGZsq{}DAC\PYGZus{}1\PYGZsq{}: \PYGZhy{}0.175, \PYGZsq{}DAC\PYGZus{}2\PYGZsq{}: \PYGZhy{}0.185, \PYGZsq{}DAC\PYGZus{}3\PYGZsq{}: \PYGZhy{}0.173\PYGZcb{}}
\end{Verbatim}

The user can check the parameters and calculate the descriptors.

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed}\PYG{n+nn}{.}\PYG{n+nn}{PyDNA} \PYG{k}{import} \PYG{n}{PyDNApsenac}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed}\PYG{n+nn}{.}\PYG{n+nn}{PyDNA}\PYG{n+nn}{.}\PYG{n+nn}{PyDNApsenac} \PYG{k}{import} \PYG{n}{GetPseDNC}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{dnaseq} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{GACTGAACTGCACTTTGGTTTCATATTATTTGCTC}\PYG{l+s+s1}{\PYGZsq{}}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{PyDNApsenac}\PYG{o}{.}\PYG{n}{CheckPsenac}\PYG{p}{(}\PYG{n}{lamada} \PYG{o}{=} \PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{w} \PYG{o}{=} \PYG{l+m+mf}{0.05}\PYG{p}{,} \PYG{n}{k} \PYG{o}{=} \PYG{l+m+mi}{2}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{psednc} \PYG{o}{=} \PYG{n}{GetPseDNC}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{ACCCCA}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{n}{lamada}\PYG{o}{=}\PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{w}\PYG{o}{=}\PYG{l+m+mf}{0.05}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print}\PYG{p}{(}\PYG{n}{psednc}\PYG{p}{)}
\PYG{g+go}{\PYGZob{}\PYGZsq{}PseDNC\PYGZus{}18\PYGZsq{}: 0.0521, \PYGZsq{}PseDNC\PYGZus{}16\PYGZsq{}: 0.0, \PYGZsq{}PseDNC\PYGZus{}17\PYGZsq{}: 0.0391, \PYGZsq{}PseDNC\PYGZus{}14\PYGZsq{}: 0.0, \PYGZsq{}PseDNC\PYGZus{}15\PYGZsq{}: 0.0, \PYGZsq{}PseDNC\PYGZus{}12\PYGZsq{}: 0.0, \PYGZsq{}PseDNC\PYGZus{}13\PYGZsq{}: 0.0, \PYGZsq{}PseDNC\PYGZus{}10\PYGZsq{}: 0.0, \PYGZsq{}PseDNC\PYGZus{}11\PYGZsq{}: 0.0, \PYGZsq{}PseDNC\PYGZus{}4\PYGZsq{}: 0.0, \PYGZsq{}PseDNC\PYGZus{}5\PYGZsq{}: 0.182, \PYGZsq{}PseDNC\PYGZus{}6\PYGZsq{}: 0.545, \PYGZsq{}PseDNC\PYGZus{}7\PYGZsq{}: 0.0, \PYGZsq{}PseDNC\PYGZus{}1\PYGZsq{}: 0.0, \PYGZsq{}PseDNC\PYGZus{}2\PYGZsq{}: 0.182, \PYGZsq{}PseDNC\PYGZus{}3\PYGZsq{}: 0.0, \PYGZsq{}PseDNC\PYGZus{}8\PYGZsq{}: 0.0, \PYGZsq{}PseDNC\PYGZus{}9\PYGZsq{}: 0.0\PYGZcb{}}
\end{Verbatim}


\subsection{Calculating DNA descriptors via object}
\label{User_guide:calculating-dna-descriptors-via-object}
The \sphinxcode{PyDNA} can calculate all kinds of protein descriptors in the PyBioMed.
For example, the \sphinxcode{PyDNA} can calculate SCPseDNC.

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed} \PYG{k}{import} \PYG{n}{Pydna}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{dna} \PYG{o}{=} \PYG{n}{Pydna}\PYG{o}{.}\PYG{n}{PyDNA}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{GACTGAACTGCACTTTGGTTTCATATTATTTGCTC}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{scpsednc} \PYG{o}{=} \PYG{n}{dna}\PYG{o}{.}\PYG{n}{GetSCPseDNC}\PYG{p}{(}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{scpsednc}\PYG{p}{)}
\PYG{g+go}{16}
\end{Verbatim}


\section{Calculating Interaction descriptors}
\label{User_guide:calculating-interaction-descriptors}
The PyInteraction module can generate six types of interaction descriptors indcluding chemical-chemical interaction features, chemical-protein interaction features, chemical-DNA interaction features, protein-protein interaction features, protein-DNA interaction features, and DNA-DNA interaction features by integrating two groups of features.

The user can choose three different types of methods to calculate interaction descriptors. The function \sphinxcode{CalculateInteraction1()} can calculate two interaction features by combining two features.
\begin{equation*}
\begin{split}F_{ab} = \bigl(F_a, F_b\bigr)\end{split}
\end{equation*}
The function \sphinxcode{CalculateInteraction2()} can calculate two interaction features by two multiplied features.
\begin{equation*}
\begin{split}F = \{F(k)= F_a(i) ×F_b(j), i = 1, 2, …, p, j = 1, 2 ,… , p, k = (i-1) ×p+j\}\end{split}
\end{equation*}
The function \sphinxcode{CalculateInteraction3()} can calculate two interaction features by
\begin{equation*}
\begin{split}F=[F_a(i)+F_b(i)),F_a(i)*F_b(i)]\end{split}
\end{equation*}
The function \sphinxcode{CalculateInteraction3()} is only used in the same type of descriptors including chemical-chemical interaction, protein-protein interaction and DNA-DNA interaction.

The user can calculate chemical-chemical features using three methods .
\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics{{CCI}.png}
\caption{The calculation process for chemical-chemical interaction descriptors.}\label{User_guide:id2}\end{figure}

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed}\PYG{n+nn}{.}\PYG{n+nn}{PyInteraction} \PYG{k}{import} \PYG{n}{PyInteraction}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed}\PYG{n+nn}{.}\PYG{n+nn}{PyMolecule} \PYG{k}{import} \PYG{n}{moe}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{rdkit} \PYG{k}{import} \PYG{n}{Chem}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{smis} \PYG{o}{=} \PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CCCC}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CCCCC}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CCCCCC}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CC(N)C(=O)O}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CC(N)C(=O)[O\PYGZhy{}].[Na+]}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{m} \PYG{o}{=} \PYG{n}{Chem}\PYG{o}{.}\PYG{n}{MolFromSmiles}\PYG{p}{(}\PYG{n}{smis}\PYG{p}{[}\PYG{l+m+mi}{3}\PYG{p}{]}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{mol\PYGZus{}des} \PYG{o}{=} \PYG{n}{moe}\PYG{o}{.}\PYG{n}{GetMOE}\PYG{p}{(}\PYG{n}{m}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{mol\PYGZus{}mol\PYGZus{}interaction1} \PYG{o}{=} \PYG{n}{PyInteraction}\PYG{o}{.}\PYG{n}{CalculateInteraction1}\PYG{p}{(}\PYG{n}{mol\PYGZus{}des}\PYG{p}{,}\PYG{n}{mol\PYGZus{}des}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n}{mol\PYGZus{}mol\PYGZus{}interaction1}
\PYG{g+go}{\PYGZob{}\PYGZsq{}slogPVSA6ex\PYGZsq{}: 0.0, \PYGZsq{}PEOEVSA10ex\PYGZsq{}: 0.0,......, \PYGZsq{}EstateVSA9ex\PYGZsq{}: 4.795, \PYGZsq{}slogPVSA2\PYGZsq{}: 4.795, \PYGZsq{}slogPVSA3\PYGZsq{}: 0.0, \PYGZsq{}slogPVSA0\PYGZsq{}: 5.734, \PYGZsq{}slogPVSA1\PYGZsq{}: 17.118, \PYGZsq{}slogPVSA6\PYGZsq{}: 0.0, \PYGZsq{}slogPVSA7\PYGZsq{}: 0.0, \PYGZsq{}slogPVSA4\PYGZsq{}: 6.924, \PYGZsq{}slogPVSA5\PYGZsq{}: 0.0, \PYGZsq{}slogPVSA8\PYGZsq{}: 0.0, \PYGZsq{}slogPVSA9\PYGZsq{}: 0.0\PYGZcb{}}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{mol\PYGZus{}mol\PYGZus{}interaction1}\PYG{p}{)}
\PYG{g+go}{120}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{mol\PYGZus{}mol\PYGZus{}interaction2} \PYG{o}{=} \PYG{n}{PyInteraction}\PYG{o}{.}\PYG{n}{CalculateInteraction2}\PYG{p}{(}\PYG{n}{mol\PYGZus{}des}\PYG{p}{,}\PYG{n}{mol\PYGZus{}des}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{mol\PYGZus{}mol\PYGZus{}interaction2}\PYG{p}{)}
\PYG{g+go}{3600}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{mol\PYGZus{}mol\PYGZus{}interaction3} \PYG{o}{=} \PYG{n}{PyInteraction}\PYG{o}{.}\PYG{n}{CalculateInteraction3}\PYG{p}{(}\PYG{n}{mol\PYGZus{}des}\PYG{p}{,}\PYG{n}{mol\PYGZus{}des}\PYG{p}{)}
\PYG{g+go}{\PYGZob{}\PYGZsq{}EstateVSA9*EstateVSA9\PYGZsq{}: 22.992, \PYGZsq{}EstateVSA9+EstateVSA9\PYGZsq{}: 9.59, \PYGZsq{}PEOEVSA1+PEOEVSA1\PYGZsq{}: 9.59, \PYGZsq{}VSAEstate10*VSAEstate10\PYGZsq{}: 0.0, \PYGZsq{}PEOEVSA3*PEOEVSA3\PYGZsq{}: 0.0, \PYGZsq{}PEOEVSA11*PEOEVSA11\PYGZsq{}: 0.0, \PYGZsq{}PEOEVSA4*PEOEVSA4\PYGZsq{}: 0.0, \PYGZsq{}VSAEstate2+VSAEstate2\PYGZsq{}: 0.0, \PYGZsq{}MRVSA0+MRVSA0\PYGZsq{}: 19.802, \PYGZsq{}MRVSA6+MRVSA6\PYGZsq{}: 0.0......\PYGZcb{}}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{mol\PYGZus{}mol\PYGZus{}interaction3}\PYG{p}{)}
\PYG{g+go}{120}
\end{Verbatim}

The user can calculate chemical-protein feature using two methods.
\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics{{CPI}.png}
\caption{The calculation process for chemical-protein interaction descriptors.}\label{User_guide:id3}\end{figure}

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{rdkit} \PYG{k}{import} \PYG{n}{Chem}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed}\PYG{n+nn}{.}\PYG{n+nn}{PyMolecule} \PYG{k}{import} \PYG{n}{moe}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed}\PYG{n+nn}{.}\PYG{n+nn}{PyInteraction}\PYG{n+nn}{.}\PYG{n+nn}{PyInteraction} \PYG{k}{import} \PYG{n}{CalculateInteraction2}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{smis} \PYG{o}{=} \PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CCCC}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CCCCC}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CCCCCC}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CC(N)C(=O)O}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CC(N)C(=O)[O\PYGZhy{}].[Na+]}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{m} \PYG{o}{=} \PYG{n}{Chem}\PYG{o}{.}\PYG{n}{MolFromSmiles}\PYG{p}{(}\PYG{n}{smis}\PYG{p}{[}\PYG{l+m+mi}{3}\PYG{p}{]}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{mol\PYGZus{}des} \PYG{o}{=} \PYG{n}{moe}\PYG{o}{.}\PYG{n}{GetMOE}\PYG{p}{(}\PYG{n}{m}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed}\PYG{n+nn}{.}\PYG{n+nn}{PyDNA}\PYG{n+nn}{.}\PYG{n+nn}{PyDNApsenac} \PYG{k}{import} \PYG{n}{GetPseDNC}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{protein\PYGZus{}des} \PYG{o}{=} \PYG{n}{GetPseDNC}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{ACCCCA}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{n}{lamada}\PYG{o}{=}\PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{w}\PYG{o}{=}\PYG{l+m+mf}{0.05}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{pro\PYGZus{}mol\PYGZus{}interaction1} \PYG{o}{=} \PYG{n}{PyInteraction}\PYG{o}{.}\PYG{n}{CalculateInteraction1}\PYG{p}{(}\PYG{n}{mol\PYGZus{}des}\PYG{p}{,}\PYG{n}{protein\PYGZus{}des}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{pro\PYGZus{}mol\PYGZus{}interaction1}\PYG{p}{)}
\PYG{g+go}{78}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{pro\PYGZus{}mol\PYGZus{}interaction2} \PYG{o}{=} \PYG{n}{CalculateInteraction2}\PYG{p}{(}\PYG{n}{mol\PYGZus{}des}\PYG{p}{,}\PYG{n}{protein\PYGZus{}des}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{pro\PYGZus{}mol\PYGZus{}interaction2}\PYG{p}{)}
\PYG{g+go}{1080}
\end{Verbatim}

The user can calculate chemical-DNA feature using two methods.

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed}\PYG{n+nn}{.}\PYG{n+nn}{PyDNA} \PYG{k}{import} \PYG{n}{PyDNAac}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{DNA\PYGZus{}des} \PYG{o}{=} \PYG{n}{PyDNAac}\PYG{o}{.}\PYG{n}{GetTCC}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{GACTGAACTGCACTTTGGTTTCATATTATTTGCTC}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{phyche\PYGZus{}index}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Dnase I}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Nucleosome}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{MW\PYGZhy{}kg}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{rdkit} \PYG{k}{import} \PYG{n}{Chem}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{smis} \PYG{o}{=} \PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CCCC}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CCCCC}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CCCCCC}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CC(N)C(=O)O}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CC(N)C(=O)[O\PYGZhy{}].[Na+]}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{m} \PYG{o}{=} \PYG{n}{Chem}\PYG{o}{.}\PYG{n}{MolFromSmiles}\PYG{p}{(}\PYG{n}{smis}\PYG{p}{[}\PYG{l+m+mi}{3}\PYG{p}{]}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{mol\PYGZus{}des} \PYG{o}{=} \PYG{n}{moe}\PYG{o}{.}\PYG{n}{GetMOE}\PYG{p}{(}\PYG{n}{m}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{mol\PYGZus{}DNA\PYGZus{}interaction1} \PYG{o}{=} \PYG{n}{PyInteraction}\PYG{o}{.}\PYG{n}{CalculateInteraction1}\PYG{p}{(}\PYG{n}{mol\PYGZus{}des}\PYG{p}{,}\PYG{n}{DNA\PYGZus{}des}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{mol\PYGZus{}DNA\PYGZus{}interaction1}\PYG{p}{)}
\PYG{g+go}{72}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{mol\PYGZus{}DNA\PYGZus{}interaction2} \PYG{o}{=} \PYG{n}{PyInteraction}\PYG{o}{.}\PYG{n}{CalculateInteraction2}\PYG{p}{(}\PYG{n}{mol\PYGZus{}des}\PYG{p}{,}\PYG{n}{DNA\PYGZus{}des}\PYG{p}{)}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{mol\PYGZus{}DNA\PYGZus{}interaction2}\PYG{p}{)}
\PYG{g+go}{720}
\end{Verbatim}


\chapter{Application}
\label{application:application}\label{application::doc}
The PyBioMed Python package can generate various feature vectors for molecular structure, protein sequences and DNA sequences. The PyBioMed package would be applied to solve many tasks in the field of cheminformatics, bioinformatics and systems biology. We will introduce five examples of its applications including Caco-2 cell permeability, aqueous solubility, drug–target interaction data, protein subcellular location, and nucleosome positioning in genomes. All datasets and Python scripts used in the next five examples can be download on \url{https://github.com/gadsbyfly/PyBioMed/blob/master/doc/download}.
\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics[width=10cm]{{brief}.png}
\caption{The overview of PyBioMed python package. PyBioMed could calculate various molecular descriptors from chemicals, proteins, DNAs/RNAs and their interactions. PyBioMed can also pretreat molecules, protein sequences and DNA sequences.}\label{application:id1}\end{figure}


\section{Application 1 Prediction of Caco-2 Cell Permeability}
\label{application:application-1-prediction-of-caco-2-cell-permeability}
Caco-2 cell monolayer model is a popular surrogate in predicting the in vitro human intestinal permeability of a drug due to its morphological and functional similarity with human enterocytes. Oral administration of drugs is the preferred route and a major goal in the development of new drugs because of its ease and patient compliance. Before an oral drug reaches the systemic circulation, it must pass through intestinal cell membranes via passive diffusion, carrier-mediated uptake or active transport processes. Bioavailability, reflecting the drug proportion in the circulatory system, is a significant index of drug efficacy. Screening for absorption ability is one of the most important parts of assessing oral bioavailability. Caco-2 cell line is a popular surrogate for the human intestinal epithelium to estimate in vivo drug permeability due to their morphological and functional similarities with human enterocytes. To build a Caco-2 cell permeability prediction model, we use the PyBioMed package to calculate molecular features and then the Random Forest (RF) method was applied to build Caco-2 cell permeability classification model. The benchmark data set for building the Caco-2 cell permeability predictor was taken from (NN Wang et al. 2016). The dataset contains 1272 compounds.
\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics[width=10cm]{{caco2}.png}
\caption{The receiver operating characteristic curve of Caco-2 classification.}\label{application:id2}\end{figure}

\begin{Verbatim}[commandchars=\\\{\},numbers=left,firstnumber=1,stepnumber=1]
\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed} \PYG{k+kn}{import} \PYG{n}{Pymolecule}
\PYG{k+kn}{import} \PYG{n+nn}{pandas} \PYG{k+kn}{as} \PYG{n+nn}{pd}
\PYG{k+kn}{from} \PYG{n+nn}{sklearn.ensemble} \PYG{k+kn}{import} \PYG{n}{RandomForestClassifier}
\PYG{k+kn}{from} \PYG{n+nn}{sklearn} \PYG{k+kn}{import} \PYG{n}{metrics}
\PYG{k+kn}{from} \PYG{n+nn}{matplotlib} \PYG{k+kn}{import} \PYG{n}{pyplot} \PYG{k}{as} \PYG{n}{plt}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{c+c1}{\PYGZsh{} load the data}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{n}{train\PYGZus{}set} \PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{read\PYGZus{}excel}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{D:/PyBioMed/PyBioMed\PYGZhy{}1.0/PyBioMed/example/caco2/caco2.xlsx}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{n}{sheetname}\PYG{o}{=}\PYG{l+m+mi}{0}\PYG{p}{)}
\PYG{n}{test\PYGZus{}set} \PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{read\PYGZus{}excel}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{D:/PyBioMed/PyBioMed\PYGZhy{}1.0/PyBioMed/example/caco2/caco2.xlsx}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{n}{sheetname}\PYG{o}{=}\PYG{l+m+mi}{1}\PYG{p}{)}

\PYG{n}{train\PYGZus{}set\PYGZus{}smi} \PYG{o}{=} \PYG{n}{train\PYGZus{}set}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{smi}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}
\PYG{n}{test\PYGZus{}set\PYGZus{}smi} \PYG{o}{=} \PYG{n}{test\PYGZus{}set}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{smi}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}

\PYG{n}{train\PYGZus{}set\PYGZus{}label} \PYG{o}{=} \PYG{n}{train\PYGZus{}set}\PYG{p}{[}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{label}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{]}
\PYG{n}{test\PYGZus{}set\PYGZus{}label} \PYG{o}{=} \PYG{n}{test\PYGZus{}set}\PYG{p}{[}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{label}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{]}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{c+c1}{\PYGZsh{} calculating molecular descriptors to a descriptors dataframe}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{k}{def} \PYG{n+nf}{calculate\PYGZus{}des}\PYG{p}{(}\PYG{n}{smi}\PYG{p}{)}\PYG{p}{:}
        \PYG{n}{des} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{p}{\PYGZcb{}}
        \PYG{n}{drugclass}\PYG{o}{=}\PYG{n}{Pymolecule}\PYG{o}{.}\PYG{n}{PyMolecule}\PYG{p}{(}\PYG{p}{)}
        \PYG{n}{drugclass}\PYG{o}{.}\PYG{n}{ReadMolFromSmile}\PYG{p}{(}\PYG{n}{smi}\PYG{p}{)}
        \PYG{n}{des}\PYG{o}{.}\PYG{n}{update}\PYG{p}{(}\PYG{n}{drugclass}\PYG{o}{.}\PYG{n}{GetMoran}\PYG{p}{(}\PYG{p}{)}\PYG{p}{)}
        \PYG{n}{des}\PYG{o}{.}\PYG{n}{update}\PYG{p}{(}\PYG{n}{drugclass}\PYG{o}{.}\PYG{n}{GetMOE}\PYG{p}{(}\PYG{p}{)}\PYG{p}{)}
        \PYG{k}{return} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{DataFrame}\PYG{p}{(}\PYG{p}{\PYGZob{}}\PYG{n}{smi}\PYG{p}{:}\PYG{p}{(}\PYG{n}{des}\PYG{p}{)}\PYG{p}{\PYGZcb{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{T}

\PYG{n}{train\PYGZus{}set\PYGZus{}des} \PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{concat}\PYG{p}{(}\PYG{n+nb}{map}\PYG{p}{(}\PYG{n}{calculate\PYGZus{}des}\PYG{p}{,}\PYG{n+nb}{list}\PYG{p}{(}\PYG{n}{train\PYGZus{}set\PYGZus{}smi}\PYG{p}{)}\PYG{p}{)}\PYG{p}{)}
\PYG{n}{test\PYGZus{}set\PYGZus{}des} \PYG{o}{=}  \PYG{n}{pd}\PYG{o}{.}\PYG{n}{concat}\PYG{p}{(}\PYG{n+nb}{map}\PYG{p}{(}\PYG{n}{calculate\PYGZus{}des}\PYG{p}{,}\PYG{n+nb}{list}\PYG{p}{(}\PYG{n}{test\PYGZus{}set\PYGZus{}smi}\PYG{p}{)}\PYG{p}{)}\PYG{p}{)}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{c+c1}{\PYGZsh{} building the model and predicting the test set}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{n}{clf} \PYG{o}{=} \PYG{n}{RandomForestClassifier}\PYG{p}{(}\PYG{n}{n\PYGZus{}estimators}\PYG{o}{=}\PYG{l+m+mi}{500}\PYG{p}{,}\PYG{n}{max\PYGZus{}features}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sqrt}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{n\PYGZus{}jobs}\PYG{o}{=}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{max\PYGZus{}depth}\PYG{o}{=}\PYG{n+nb+bp}{None}\PYG{p}{,}\PYG{n}{random\PYGZus{}state}\PYG{o}{=}\PYG{l+m+mi}{0}\PYG{p}{)}
\PYG{n}{clf}\PYG{o}{.}\PYG{n}{fit}\PYG{p}{(}\PYG{n}{train\PYGZus{}set\PYGZus{}des}\PYG{p}{,}\PYG{n}{train\PYGZus{}set\PYGZus{}label}\PYG{p}{)}

\PYG{n}{proba} \PYG{o}{=} \PYG{n}{clf}\PYG{o}{.}\PYG{n}{predict\PYGZus{}proba}\PYG{p}{(}\PYG{n}{test\PYGZus{}set\PYGZus{}des}\PYG{p}{)}\PYG{p}{[}\PYG{p}{:}\PYG{p}{,}\PYG{l+m+mi}{1}\PYG{p}{]}
\PYG{n}{predict\PYGZus{}label} \PYG{o}{=} \PYG{n}{clf}\PYG{o}{.}\PYG{n}{predict}\PYG{p}{(}\PYG{n}{test\PYGZus{}set\PYGZus{}des}\PYG{p}{)}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{c+c1}{\PYGZsh{} Calculating auc score}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{n}{AUC\PYGZus{}score} \PYG{o}{=} \PYG{n+nb}{round}\PYG{p}{(}\PYG{n}{metrics}\PYG{o}{.}\PYG{n}{roc\PYGZus{}auc\PYGZus{}score}\PYG{p}{(}\PYG{n}{test\PYGZus{}set\PYGZus{}label}\PYG{p}{,} \PYG{n}{proba}\PYG{p}{)}\PYG{p}{,}\PYG{l+m+mi}{2}\PYG{p}{)}
\PYG{n}{TPR} \PYG{o}{=} \PYG{n+nb}{round}\PYG{p}{(}\PYG{n}{metrics}\PYG{o}{.}\PYG{n}{recall\PYGZus{}score}\PYG{p}{(}\PYG{n}{test\PYGZus{}set\PYGZus{}label}\PYG{p}{,} \PYG{n}{predict\PYGZus{}label}\PYG{p}{)}\PYG{p}{,}\PYG{l+m+mi}{2}\PYG{p}{)}
\PYG{n}{ACC} \PYG{o}{=} \PYG{n+nb}{round}\PYG{p}{(}\PYG{n}{metrics}\PYG{o}{.}\PYG{n}{accuracy\PYGZus{}score}\PYG{p}{(}\PYG{n}{test\PYGZus{}set\PYGZus{}label}\PYG{p}{,} \PYG{n}{predict\PYGZus{}label}\PYG{p}{)}\PYG{p}{,}\PYG{l+m+mi}{2}\PYG{p}{)}
\PYG{n}{P} \PYG{o}{=} \PYG{n+nb}{float}\PYG{p}{(}\PYG{n}{test\PYGZus{}set\PYGZus{}label}\PYG{o}{.}\PYG{n}{sum}\PYG{p}{(}\PYG{p}{)}\PYG{p}{)}
\PYG{n}{N} \PYG{o}{=} \PYG{n}{test\PYGZus{}set\PYGZus{}label}\PYG{o}{.}\PYG{n}{shape}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]} \PYG{o}{\PYGZhy{}} \PYG{n}{P}
\PYG{n}{SPE} \PYG{o}{=} \PYG{n+nb}{round}\PYG{p}{(}\PYG{p}{(}\PYG{n}{P}\PYG{o}{/}\PYG{n}{N}\PYG{o}{+}\PYG{l+m+mf}{1.0}\PYG{p}{)}\PYG{o}{*}\PYG{n}{ACC}\PYG{o}{\PYGZhy{}}\PYG{n}{TPR}\PYG{o}{*}\PYG{n}{P}\PYG{o}{/}\PYG{n}{N}\PYG{p}{,}\PYG{l+m+mi}{2}\PYG{p}{)}
\PYG{n}{matthews\PYGZus{}corrcoef} \PYG{o}{=} \PYG{n+nb}{round}\PYG{p}{(}\PYG{n}{metrics}\PYG{o}{.}\PYG{n}{matthews\PYGZus{}corrcoef}\PYG{p}{(}\PYG{n}{test\PYGZus{}set\PYGZus{}label}\PYG{p}{,} \PYG{n}{predict\PYGZus{}label}\PYG{p}{)}\PYG{p}{,}\PYG{l+m+mi}{2}\PYG{p}{)}
\PYG{n}{f1\PYGZus{}score} \PYG{o}{=}  \PYG{n+nb}{round}\PYG{p}{(}\PYG{n}{metrics}\PYG{o}{.}\PYG{n}{f1\PYGZus{}score}\PYG{p}{(}\PYG{n}{test\PYGZus{}set\PYGZus{}label}\PYG{p}{,} \PYG{n}{predict\PYGZus{}label}\PYG{p}{)}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{)}
\PYG{n}{fpr\PYGZus{}cv}\PYG{p}{,} \PYG{n}{tpr\PYGZus{}cv}\PYG{p}{,} \PYG{n}{thresholds\PYGZus{}cv} \PYG{o}{=} \PYG{n}{metrics}\PYG{o}{.}\PYG{n}{roc\PYGZus{}curve}\PYG{p}{(}\PYG{n}{test\PYGZus{}set\PYGZus{}label}\PYG{p}{,} \PYG{n}{proba}\PYG{p}{)}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{c+c1}{\PYGZsh{} plotting the auc plot}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{figure}\PYG{p}{(}\PYG{n}{figsize} \PYG{o}{=} \PYG{p}{(}\PYG{l+m+mi}{10}\PYG{p}{,}\PYG{l+m+mi}{7}\PYG{p}{)}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{plot}\PYG{p}{(}\PYG{n}{fpr\PYGZus{}cv}\PYG{p}{,} \PYG{n}{tpr\PYGZus{}cv}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{r}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{label}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{auc = }\PYG{l+s+si}{\PYGZpc{}0.2f}\PYG{l+s+s1}{\PYGZsq{}}\PYG{o}{\PYGZpc{}} \PYG{n}{AUC\PYGZus{}score}\PYG{p}{,} \PYG{n}{lw}\PYG{o}{=}\PYG{l+m+mi}{2}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{xlabel}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{False positive rate}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{fontsize}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mi}{20}\PYG{p}{\PYGZcb{}}\PYG{p}{)}\PYG{p}{;}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{ylabel}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{True positive rate}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{fontsize}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mi}{20}\PYG{p}{\PYGZcb{}}\PYG{p}{)}\PYG{p}{;}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{title}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{ROC of Caco\PYGZhy{}2 Classification}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{fontsize}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mi}{25}\PYG{p}{\PYGZcb{}}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{legend}\PYG{p}{(}\PYG{n}{loc}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{lower right}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}\PYG{n}{numpoints}\PYG{o}{=}\PYG{l+m+mi}{15}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{show}\PYG{p}{(}\PYG{p}{)}
\end{Verbatim}

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sensitivity:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{n}{TPR}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{specificity:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{SPE}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{accuracy:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{ACC}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{AUC:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{AUC\PYGZus{}score}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{MACCS:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{matthews\PYGZus{}corrcoef}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{F1:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{f1\PYGZus{}score}
\PYG{g+go}{sensitivity: 0.91 specificity: 0.8 accuracy: 0.86 AUC: 0.93 MACCS: 0.72 F1: 0.88}
\end{Verbatim}


\section{Application 2 Prediction of aqueous solubility}
\label{application:application-2-prediction-of-aqueous-solubility}
Aqueous solubility is one of the major drug properties to be optimized in drug discovery. Aqueous solubility and membrane permeability are the two key factors that affect a drug’s oral bioavailability. Generally, a drug with high solubility and membrane permeability is considered to have bioavailability problems. Otherwise, it is a problematic candidate or needs careful formulation work. To build an aqueous solubility prediction model, we use the PyBioMed package to calculate molecular features and then the random forest (RF) method was applied to build aqueous solubility regression model. The benchmark data set for building the aqueous solubility regression model was taken from (Junmei Wang et al.). The dataset contains 3637 compounds.
\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics[width=9.5cm]{{solubility}.png}
\caption{The aqueous solubility prediction. The X-axis represents experimental values and the Y-axis represents predicted values.}\label{application:id3}\end{figure}

\begin{Verbatim}[commandchars=\\\{\},numbers=left,firstnumber=1,stepnumber=1]
\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed} \PYG{k+kn}{import} \PYG{n}{Pymolecule}
\PYG{k+kn}{import} \PYG{n+nn}{pandas} \PYG{k+kn}{as} \PYG{n+nn}{pd}
\PYG{k+kn}{import} \PYG{n+nn}{numpy} \PYG{k+kn}{as} \PYG{n+nn}{np}
\PYG{k+kn}{from} \PYG{n+nn}{sklearn} \PYG{k+kn}{import} \PYG{n}{cross\PYGZus{}validation}
\PYG{k+kn}{from} \PYG{n+nn}{sklearn.ensemble} \PYG{k+kn}{import} \PYG{n}{RandomForestRegressor}
\PYG{k+kn}{from} \PYG{n+nn}{matplotlib} \PYG{k+kn}{import} \PYG{n}{pyplot} \PYG{k}{as} \PYG{n}{plt}
\PYG{k+kn}{from} \PYG{n+nn}{sklearn.cross\PYGZus{}validation} \PYG{k+kn}{import} \PYG{n}{train\PYGZus{}test\PYGZus{}split}
\PYG{k+kn}{from} \PYG{n+nn}{sklearn} \PYG{k+kn}{import} \PYG{n}{metrics}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{c+c1}{\PYGZsh{} loading the data}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{n}{solubility\PYGZus{}set} \PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{read\PYGZus{}excel}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{./PyBioMed/example/solubility/Solubility\PYGZhy{}total.xlsx}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{n}{sheetname} \PYG{o}{=} \PYG{l+m+mi}{0}\PYG{p}{)} \PYG{c+c1}{\PYGZsh{}change the path to the real path}
\PYG{n}{smis} \PYG{o}{=} \PYG{n}{solubility\PYGZus{}set}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SMI}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}
\PYG{n}{logS} \PYG{o}{=} \PYG{n}{solubility\PYGZus{}set}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{logS}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{c+c1}{\PYGZsh{} \PYGZsh{}calculating molecular descriptors}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{k}{def} \PYG{n+nf}{calculate\PYGZus{}des}\PYG{p}{(}\PYG{n}{smi}\PYG{p}{)}\PYG{p}{:}
        \PYG{n}{des} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{p}{\PYGZcb{}}
        \PYG{n}{drugclass}\PYG{o}{=}\PYG{n}{Pymolecule}\PYG{o}{.}\PYG{n}{PyMolecule}\PYG{p}{(}\PYG{p}{)}
        \PYG{n}{drugclass}\PYG{o}{.}\PYG{n}{ReadMolFromSmile}\PYG{p}{(}\PYG{n}{smi}\PYG{p}{)}
        \PYG{n}{des}\PYG{o}{.}\PYG{n}{update}\PYG{p}{(}\PYG{n}{drugclass}\PYG{o}{.}\PYG{n}{GetEstate}\PYG{p}{(}\PYG{p}{)}\PYG{p}{)}
        \PYG{n}{des}\PYG{o}{.}\PYG{n}{update}\PYG{p}{(}\PYG{n}{drugclass}\PYG{o}{.}\PYG{n}{GetMOE}\PYG{p}{(}\PYG{p}{)}\PYG{p}{)}
        \PYG{k}{return} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{DataFrame}\PYG{p}{(}\PYG{p}{\PYGZob{}}\PYG{n}{smi}\PYG{p}{:}\PYG{p}{(}\PYG{n}{des}\PYG{p}{)}\PYG{p}{\PYGZcb{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{T}
\PYG{n}{solubility\PYGZus{}set\PYGZus{}des} \PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{concat}\PYG{p}{(}\PYG{n+nb}{map}\PYG{p}{(}\PYG{n}{calculate\PYGZus{}des}\PYG{p}{,}\PYG{n+nb}{list}\PYG{p}{(}\PYG{n}{smis}\PYG{p}{)}\PYG{p}{)}\PYG{p}{)}
\PYG{n}{solubility\PYGZus{}set\PYGZus{}des} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{array}\PYG{p}{(}\PYG{n}{solubility\PYGZus{}set\PYGZus{}des}\PYG{p}{)}
\PYG{n}{logS} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{array}\PYG{p}{(}\PYG{n}{logS}\PYG{p}{)}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{c+c1}{\PYGZsh{} building the model and predict}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{n}{train\PYGZus{}set\PYGZus{}des}\PYG{p}{,} \PYG{n}{test\PYGZus{}set\PYGZus{}des}\PYG{p}{,} \PYG{n}{train\PYGZus{}logS}\PYG{p}{,} \PYG{n}{test\PYGZus{}logS} \PYG{o}{=} \PYG{n}{train\PYGZus{}test\PYGZus{}split}\PYG{p}{(}\PYG{n}{solubility\PYGZus{}set\PYGZus{}des}\PYG{p}{,}
                                                                                                \PYG{n}{logS}\PYG{p}{,} \PYG{n}{test\PYGZus{}size} \PYG{o}{=} \PYG{l+m+mf}{0.33}\PYG{p}{,} \PYG{n}{random\PYGZus{}state} \PYG{o}{=} \PYG{l+m+mi}{42}\PYG{p}{)}

\PYG{n}{kf} \PYG{o}{=} \PYG{n}{cross\PYGZus{}validation}\PYG{o}{.}\PYG{n}{KFold}\PYG{p}{(}\PYG{n}{train\PYGZus{}set\PYGZus{}des}\PYG{o}{.}\PYG{n}{shape}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,} \PYG{n}{n\PYGZus{}folds}\PYG{o}{=}\PYG{l+m+mi}{10}\PYG{p}{,} \PYG{n}{random\PYGZus{}state}\PYG{o}{=}\PYG{l+m+mi}{0}\PYG{p}{)}
\PYG{n}{clf} \PYG{o}{=} \PYG{n}{RandomForestRegressor}\PYG{p}{(}\PYG{n}{n\PYGZus{}estimators}\PYG{o}{=}\PYG{l+m+mi}{500}\PYG{p}{,} \PYG{n}{max\PYGZus{}features}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{auto}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{n\PYGZus{}jobs} \PYG{o}{=} \PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1}\PYG{p}{)}
\PYG{n}{CV\PYGZus{}pred\PYGZus{}logS} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{VALIDATION\PYGZus{}index} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{k}{for} \PYG{n}{train\PYGZus{}index}\PYG{p}{,} \PYG{n}{validation\PYGZus{}index} \PYG{o+ow}{in} \PYG{n}{kf}\PYG{p}{:}
        \PYG{n}{VALIDATION\PYGZus{}index} \PYG{o}{=} \PYG{n}{VALIDATION\PYGZus{}index} \PYG{o}{+} \PYG{n+nb}{list}\PYG{p}{(}\PYG{n}{validation\PYGZus{}index}\PYG{p}{)}
        \PYG{n}{clf}\PYG{o}{.}\PYG{n}{fit}\PYG{p}{(}\PYG{n}{train\PYGZus{}set\PYGZus{}des}\PYG{p}{[}\PYG{n}{train\PYGZus{}index} \PYG{p}{,}\PYG{p}{:}\PYG{p}{]}\PYG{p}{,}\PYG{n}{train\PYGZus{}logS}\PYG{p}{[}\PYG{n}{train\PYGZus{}index}\PYG{p}{]}\PYG{p}{)}
        \PYG{n}{pred\PYGZus{}logS} \PYG{o}{=} \PYG{n}{clf}\PYG{o}{.}\PYG{n}{predict}\PYG{p}{(}\PYG{n}{train\PYGZus{}set\PYGZus{}des}\PYG{p}{[}\PYG{n}{validation\PYGZus{}index}\PYG{p}{,}\PYG{p}{:}\PYG{p}{]}\PYG{p}{)}
        \PYG{n}{CV\PYGZus{}pred\PYGZus{}logS} \PYG{o}{=} \PYG{n}{CV\PYGZus{}pred\PYGZus{}logS} \PYG{o}{+} \PYG{n+nb}{list}\PYG{p}{(}\PYG{n}{pred\PYGZus{}logS}\PYG{p}{)}
\PYG{n}{CV\PYGZus{}true\PYGZus{}logS} \PYG{o}{=} \PYG{n}{train\PYGZus{}logS}\PYG{p}{[}\PYG{n}{VALIDATION\PYGZus{}index}\PYG{p}{]}
\PYG{n}{r2\PYGZus{}CV} \PYG{o}{=} \PYG{n}{metrics}\PYG{o}{.}\PYG{n}{r2\PYGZus{}score}\PYG{p}{(}\PYG{n}{CV\PYGZus{}true\PYGZus{}logS}\PYG{p}{,} \PYG{n}{CV\PYGZus{}pred\PYGZus{}logS}\PYG{p}{)}

\PYG{n}{clf}\PYG{o}{.}\PYG{n}{fit}\PYG{p}{(}\PYG{n}{train\PYGZus{}set\PYGZus{}des}\PYG{p}{,}\PYG{n}{train\PYGZus{}logS}\PYG{p}{)}
\PYG{n}{pred\PYGZus{}logS\PYGZus{}test} \PYG{o}{=} \PYG{n}{clf}\PYG{o}{.}\PYG{n}{predict}\PYG{p}{(}\PYG{n}{test\PYGZus{}set\PYGZus{}des}\PYG{p}{)}
\PYG{n}{r2\PYGZus{}test} \PYG{o}{=} \PYG{n}{metrics}\PYG{o}{.}\PYG{n}{r2\PYGZus{}score}\PYG{p}{(}\PYG{n}{test\PYGZus{}logS}\PYG{p}{,} \PYG{n}{pred\PYGZus{}logS\PYGZus{}test}\PYG{p}{)}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{c+c1}{\PYGZsh{} plotting the figure}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{figure}\PYG{p}{(}\PYG{n}{figsize} \PYG{o}{=} \PYG{p}{(}\PYG{l+m+mi}{10}\PYG{p}{,}\PYG{l+m+mi}{10}\PYG{p}{)}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{plot}\PYG{p}{(}\PYG{n+nb}{range}\PYG{p}{(}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{15}\PYG{p}{,}\PYG{l+m+mi}{5}\PYG{p}{)}\PYG{p}{,}\PYG{n+nb}{range}\PYG{p}{(}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{15}\PYG{p}{,}\PYG{l+m+mi}{5}\PYG{p}{)}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{black}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{plot}\PYG{p}{(}\PYG{n}{CV\PYGZus{}true\PYGZus{}logS}\PYG{p}{,}\PYG{n}{CV\PYGZus{}pred\PYGZus{}logS}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{b.}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{n}{label} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{cross validation}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{alpha} \PYG{o}{=} \PYG{l+m+mf}{0.5} \PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{plot}\PYG{p}{(}\PYG{n}{test\PYGZus{}logS}\PYG{p}{,}\PYG{n}{pred\PYGZus{}logS\PYGZus{}test}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{r.}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{n}{label} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{test set}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{n}{alpha} \PYG{o}{=} \PYG{l+m+mf}{0.5}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{title}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Aqueous Solubility Prediction}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{fontsize}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mi}{25}\PYG{p}{\PYGZcb{}}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{legend}\PYG{p}{(}\PYG{n}{loc}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{lower right}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}\PYG{n}{numpoints}\PYG{o}{=}\PYG{l+m+mi}{1}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{plot}\PYG{p}{(}\PYG{p}{)}
\end{Verbatim}

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CV\PYGZus{}R\PYGZca{}2:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{r2\PYGZus{}cv}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Test\PYGZus{}R\PYGZca{}2:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{r2\PYGZus{}test}\PYG{l+s+s1}{\PYGZsq{}}
\PYG{g+go}{CV\PYGZus{}R\PYGZca{}2: 0.86 Test\PYGZus{}R\PYGZca{}2: 0.84}
\end{Verbatim}


\section{Application 3 Prediction of drug–target interaction from the integration of chemical and protein spaces}
\label{application:application-3-prediction-of-drugtarget-interaction-from-the-integration-of-chemical-and-protein-spaces}
Drug-target interactions (DTIs) are central to current drug discovery processes and public health fields. The rapidly increasing amount of publicly available data in biology and chemistry enables researchers to revisit drug-target interaction problems by systematic integration and analysis of heterogeneous data. To identify the interactions between drugs and targets is of important in drug discovery today. Interaction with ligands can modulate the function of many targets in the processes of signal transport, catalytic reaction and so on. With the enrichment of data repository, automatically prediction of target-protein interactions is an alternative method to facilitate drug discovery. Our previous work (Cao et al, 2014) proved that the calculated features perform well in the prediction of chemical-protein interaction. The benchmark data set for building the drug-target interaction predictor was taken from (Yamanishi, Araki et al. 2008). The dataset contains 6888 samples, among them 2922 drug-protein pairs have interactions which are defined as positive dataset and 3966 drug-protein pairs do not have interactions which are defined as negative dataset. To represent each drug-protein pairs, 150 CATS molecular fingerprints and 147 CTD composition, transition and distribution features of protein, a total number of 297 features were used. The random forest (RF) classifier was employed to build model.
\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics[width=10cm]{{DPI}.png}
\caption{The receiver operating characteristic curve of drug-target interaction classification.}\label{application:id4}\end{figure}

\begin{Verbatim}[commandchars=\\\{\},numbers=left,firstnumber=1,stepnumber=1]
\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed} \PYG{k+kn}{import} \PYG{n}{Pymolecule}
\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed} \PYG{k+kn}{import} \PYG{n}{Pyprotein}
\PYG{k+kn}{import} \PYG{n+nn}{pandas} \PYG{k+kn}{as} \PYG{n+nn}{pd}
\PYG{k+kn}{import} \PYG{n+nn}{numpy} \PYG{k+kn}{as} \PYG{n+nn}{np}
\PYG{k+kn}{from} \PYG{n+nn}{sklearn.ensemble} \PYG{k+kn}{import} \PYG{n}{RandomForestClassifier} \PYG{k}{as} \PYG{n}{RF}
\PYG{k+kn}{from} \PYG{n+nn}{sklearn} \PYG{k+kn}{import} \PYG{n}{cross\PYGZus{}validation}
\PYG{k+kn}{from} \PYG{n+nn}{sklearn} \PYG{k+kn}{import} \PYG{n}{metrics}
\PYG{k+kn}{from} \PYG{n+nn}{matplotlib} \PYG{k+kn}{import} \PYG{n}{pyplot} \PYG{k}{as} \PYG{n}{plt}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{c+c1}{\PYGZsh{} loading the data}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{n}{path} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{input PyBioMed path in your computer}\PYG{l+s+s1}{\PYGZsq{}}  \PYG{c+c1}{\PYGZsh{}input the real path in your own path}
\PYG{n}{smis} \PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{read\PYGZus{}excel}\PYG{p}{(}\PYG{n}{path} \PYG{o}{+} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{example/dpi/DPI\PYGZus{}SMIs.xlsx}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{smis}\PYG{o}{.}\PYG{n}{index} \PYG{o}{=} \PYG{n}{smis}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Drug}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}
\PYG{n}{protein\PYGZus{}seq} \PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{read\PYGZus{}table}\PYG{p}{(}\PYG{n}{path} \PYG{o}{+} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{example/dpi/hsa\PYGZus{}seqs\PYGZus{}all.tsv}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{sep} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+se}{\PYGZbs{}t}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{protein\PYGZus{}seq}\PYG{o}{.}\PYG{n}{index} \PYG{o}{=}  \PYG{n}{protein\PYGZus{}seq}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Protein}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}

\PYG{n}{positive\PYGZus{}pairs} \PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{read\PYGZus{}excel}\PYG{p}{(}\PYG{n}{path} \PYG{o}{+} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{example/dpi/Enzyme.xls}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{positive\PYGZus{}pairs} \PYG{o}{=} \PYG{n+nb}{zip}\PYG{p}{(}\PYG{n+nb}{list}\PYG{p}{(}\PYG{n}{positive\PYGZus{}pairs}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Protein}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{)}\PYG{p}{,} \PYG{n+nb}{list}\PYG{p}{(}\PYG{n}{positive\PYGZus{}pairs}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Drug}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{)}\PYG{p}{)}

\PYG{n}{negative\PYGZus{}pairs} \PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{read\PYGZus{}excel}\PYG{p}{(}\PYG{n}{path} \PYG{o}{+} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{example/dpi/Enzymedecoy.xls}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{negative\PYGZus{}pairs} \PYG{o}{=} \PYG{n+nb}{zip}\PYG{p}{(}\PYG{n+nb}{list}\PYG{p}{(}\PYG{n}{negative\PYGZus{}pairs}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Protein}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{)}\PYG{p}{,} \PYG{n+nb}{list}\PYG{p}{(}\PYG{n}{negative\PYGZus{}pairs}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Drug}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{)}\PYG{p}{)}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{c+c1}{\PYGZsh{} calculating descriptors}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{k}{def} \PYG{n+nf}{calculate\PYGZus{}pair\PYGZus{}des}\PYG{p}{(}\PYG{n}{smi}\PYG{p}{,} \PYG{n}{seq}\PYG{p}{)}\PYG{p}{:}
        \PYG{n}{pair\PYGZus{}des} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{p}{\PYGZcb{}}
        \PYG{n}{drugclass} \PYG{o}{=} \PYG{n}{Pymolecule}\PYG{o}{.}\PYG{n}{PyMolecule}\PYG{p}{(}\PYG{p}{)}
        \PYG{n}{drugclass}\PYG{o}{.}\PYG{n}{ReadMolFromSmile}\PYG{p}{(}\PYG{n}{smi}\PYG{p}{)}
        \PYG{n}{pair\PYGZus{}des}\PYG{o}{.}\PYG{n}{update}\PYG{p}{(}\PYG{n}{drugclass}\PYG{o}{.}\PYG{n}{GetCATS2D}\PYG{p}{(}\PYG{p}{)}\PYG{p}{)}
        \PYG{n}{proclass} \PYG{o}{=} \PYG{n}{Pyprotein}\PYG{o}{.}\PYG{n}{PyProtein}\PYG{p}{(}\PYG{n}{seq}\PYG{p}{)}
        \PYG{n}{pair\PYGZus{}des}\PYG{o}{.}\PYG{n}{update}\PYG{p}{(}\PYG{n}{proclass}\PYG{o}{.}\PYG{n}{GetCTD}\PYG{p}{(}\PYG{p}{)}\PYG{p}{)}
        \PYG{k}{return} \PYG{n}{pair\PYGZus{}des}
\PYG{n}{positive\PYGZus{}pairs\PYGZus{}des} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{p}{\PYGZcb{}}
\PYG{k}{for} \PYG{n}{n}\PYG{p}{,} \PYG{n}{positive\PYGZus{}pair} \PYG{o+ow}{in} \PYG{n+nb}{enumerate}\PYG{p}{(}\PYG{n}{positive\PYGZus{}pairs}\PYG{p}{)}\PYG{p}{:}
        \PYG{k}{try}\PYG{p}{:}
                \PYG{n}{pair\PYGZus{}des} \PYG{o}{=} \PYG{n}{calculate\PYGZus{}pair\PYGZus{}des}\PYG{p}{(}\PYG{n}{smis}\PYG{o}{.}\PYG{n}{ix}\PYG{p}{[}\PYG{n}{positive\PYGZus{}pair}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{p}{]}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{p}{,}\PYG{n}{protein\PYGZus{}seq}\PYG{o}{.}\PYG{n}{ix}\PYG{p}{[}\PYG{n}{positive\PYGZus{}pair}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{]}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{p}{)}
                \PYG{n}{positive\PYGZus{}pairs\PYGZus{}des}\PYG{p}{[}\PYG{n}{n}\PYG{p}{]} \PYG{o}{=} \PYG{n}{pair\PYGZus{}des}
        \PYG{k}{except}\PYG{p}{:}
                \PYG{k}{continue}

\PYG{n}{negative\PYGZus{}pairs\PYGZus{}des} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{p}{\PYGZcb{}}
\PYG{k}{for} \PYG{n}{n}\PYG{p}{,} \PYG{n}{negative\PYGZus{}pair} \PYG{o+ow}{in} \PYG{n+nb}{enumerate}\PYG{p}{(}\PYG{n}{negative\PYGZus{}pairs}\PYG{p}{)}\PYG{p}{:}
        \PYG{k}{try}\PYG{p}{:}
                \PYG{n}{pair\PYGZus{}des} \PYG{o}{=} \PYG{n}{calculate\PYGZus{}pair\PYGZus{}des}\PYG{p}{(}\PYG{n}{smis}\PYG{o}{.}\PYG{n}{ix}\PYG{p}{[}\PYG{n}{negative\PYGZus{}pair}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{p}{]}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{p}{,}\PYG{n}{protein\PYGZus{}seq}\PYG{o}{.}\PYG{n}{ix}\PYG{p}{[}\PYG{n}{negative\PYGZus{}pair}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{]}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{p}{)}
                \PYG{n}{negative\PYGZus{}pairs\PYGZus{}des}\PYG{p}{[}\PYG{n}{n}\PYG{p}{]} \PYG{o}{=} \PYG{n}{pair\PYGZus{}des}
        \PYG{k}{except}\PYG{p}{:}
                \PYG{k}{continue}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{c+c1}{\PYGZsh{} cross\PYGZhy{}validation}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{n}{x} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{array}\PYG{p}{(}\PYG{n}{pd}\PYG{o}{.}\PYG{n}{concat}\PYG{p}{(}\PYG{p}{[}\PYG{n}{pd}\PYG{o}{.}\PYG{n}{DataFrame}\PYG{p}{(}\PYG{n}{positive\PYGZus{}pairs\PYGZus{}des}\PYG{p}{)}\PYG{o}{.}\PYG{n}{T}\PYG{p}{,} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{DataFrame}\PYG{p}{(}\PYG{n}{negative\PYGZus{}pairs\PYGZus{}des}\PYG{p}{)}\PYG{o}{.}\PYG{n}{T}\PYG{p}{]}\PYG{p}{,}
                                                \PYG{n}{join\PYGZus{}axes}\PYG{o}{=}\PYG{p}{[}\PYG{n}{pd}\PYG{o}{.}\PYG{n}{DataFrame}\PYG{p}{(}\PYG{n}{positive\PYGZus{}pairs\PYGZus{}des}\PYG{p}{)}\PYG{o}{.}\PYG{n}{T}\PYG{o}{.}\PYG{n}{columns}\PYG{p}{]}\PYG{p}{,}\PYG{n}{axis} \PYG{o}{=} \PYG{l+m+mi}{0}\PYG{p}{,} \PYG{n}{ignore\PYGZus{}index}\PYG{o}{=}\PYG{n+nb+bp}{True}\PYG{p}{)}\PYG{p}{)}

\PYG{n}{positive\PYGZus{}count}\PYG{p}{,} \PYG{n}{negative\PYGZus{}count} \PYG{o}{=} \PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{positive\PYGZus{}pairs\PYGZus{}des}\PYG{p}{)}\PYG{p}{,} \PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{negative\PYGZus{}pairs\PYGZus{}des}\PYG{p}{)}
\PYG{n}{y} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{array}\PYG{p}{(}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{*}\PYG{n}{positive\PYGZus{}count}\PYG{o}{+} \PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{*}\PYG{n}{negative\PYGZus{}count}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} ROC curve of CV}
\PYG{n}{kf} \PYG{o}{=} \PYG{n}{cross\PYGZus{}validation}\PYG{o}{.}\PYG{n}{KFold}\PYG{p}{(}\PYG{n}{x}\PYG{o}{.}\PYG{n}{shape}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,} \PYG{n}{n\PYGZus{}folds}\PYG{o}{=}\PYG{l+m+mi}{10}\PYG{p}{,} \PYG{n}{shuffle}\PYG{o}{=}\PYG{n+nb+bp}{True}\PYG{p}{,}\PYG{n}{random\PYGZus{}state}\PYG{o}{=}\PYG{l+m+mi}{5}\PYG{p}{)}
\PYG{n}{clf} \PYG{o}{=} \PYG{n}{RF}\PYG{p}{(}\PYG{n}{n\PYGZus{}estimators}\PYG{o}{=}\PYG{l+m+mi}{500}\PYG{p}{,} \PYG{n}{max\PYGZus{}features}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sqrt}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{n\PYGZus{}jobs}\PYG{o}{=}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{oob\PYGZus{}score}\PYG{o}{=}\PYG{n+nb+bp}{True}\PYG{p}{)}
\PYG{n}{CV\PYGZus{}pred\PYGZus{}prob} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{CV\PYGZus{}pred\PYGZus{}label}\PYG{o}{=}\PYG{p}{[}\PYG{p}{]}
\PYG{n}{VALIDATION\PYGZus{}index} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{k}{for} \PYG{n}{train\PYGZus{}index}\PYG{p}{,} \PYG{n}{validation\PYGZus{}index} \PYG{o+ow}{in} \PYG{n}{kf}\PYG{p}{:}
                \PYG{n}{VALIDATION\PYGZus{}index} \PYG{o}{=} \PYG{n}{VALIDATION\PYGZus{}index} \PYG{o}{+} \PYG{n+nb}{list}\PYG{p}{(}\PYG{n}{validation\PYGZus{}index}\PYG{p}{)}
                \PYG{n}{clf}\PYG{o}{.}\PYG{n}{fit}\PYG{p}{(}\PYG{n}{x}\PYG{p}{[}\PYG{n}{train\PYGZus{}index} \PYG{p}{,}\PYG{p}{:}\PYG{p}{]}\PYG{p}{,}\PYG{n}{y}\PYG{p}{[}\PYG{n}{train\PYGZus{}index}\PYG{p}{]}\PYG{p}{)}
                \PYG{n}{pred\PYGZus{}prob} \PYG{o}{=} \PYG{n}{clf}\PYG{o}{.}\PYG{n}{predict\PYGZus{}proba}\PYG{p}{(}\PYG{n}{x}\PYG{p}{[}\PYG{n}{validation\PYGZus{}index}\PYG{p}{,}\PYG{p}{:}\PYG{p}{]}\PYG{p}{)}
                \PYG{n}{pred\PYGZus{}label} \PYG{o}{=} \PYG{n}{clf}\PYG{o}{.}\PYG{n}{predict}\PYG{p}{(}\PYG{n}{x}\PYG{p}{[}\PYG{n}{validation\PYGZus{}index}\PYG{p}{,}\PYG{p}{:}\PYG{p}{]}\PYG{p}{)}
                \PYG{n}{CV\PYGZus{}pred\PYGZus{}prob} \PYG{o}{=} \PYG{n}{CV\PYGZus{}pred\PYGZus{}prob} \PYG{o}{+} \PYG{n+nb}{list}\PYG{p}{(}\PYG{n}{pred\PYGZus{}prob}\PYG{p}{[}\PYG{p}{:}\PYG{p}{,}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{p}{)}
                \PYG{n}{CV\PYGZus{}pred\PYGZus{}label} \PYG{o}{=} \PYG{n}{CV\PYGZus{}pred\PYGZus{}label} \PYG{o}{+} \PYG{n+nb}{list}\PYG{p}{(}\PYG{n}{pred\PYGZus{}label}\PYG{p}{)}
\PYG{n}{fpr\PYGZus{}cv}\PYG{p}{,} \PYG{n}{tpr\PYGZus{}cv}\PYG{p}{,} \PYG{n}{thresholds\PYGZus{}cv} \PYG{o}{=} \PYG{n}{metrics}\PYG{o}{.}\PYG{n}{roc\PYGZus{}curve}\PYG{p}{(}\PYG{n}{y}\PYG{p}{[}\PYG{n}{VALIDATION\PYGZus{}index}\PYG{p}{]}\PYG{p}{,} \PYG{n}{CV\PYGZus{}pred\PYGZus{}prob}\PYG{p}{)}
\PYG{n}{y\PYGZus{}true} \PYG{o}{=} \PYG{n}{y}\PYG{p}{[}\PYG{n}{VALIDATION\PYGZus{}index}\PYG{p}{]}
\PYG{n}{AUC\PYGZus{}score} \PYG{o}{=} \PYG{n}{metrics}\PYG{o}{.}\PYG{n}{roc\PYGZus{}auc\PYGZus{}score}\PYG{p}{(}\PYG{n}{y}\PYG{p}{[}\PYG{n}{VALIDATION\PYGZus{}index}\PYG{p}{]}\PYG{p}{,} \PYG{n}{CV\PYGZus{}pred\PYGZus{}prob}\PYG{p}{)}
\PYG{n}{TPR} \PYG{o}{=} \PYG{n}{metrics}\PYG{o}{.}\PYG{n}{recall\PYGZus{}score}\PYG{p}{(}\PYG{n}{y\PYGZus{}true}\PYG{p}{,} \PYG{n}{CV\PYGZus{}pred\PYGZus{}label}\PYG{p}{)}
\PYG{n}{ACC} \PYG{o}{=} \PYG{n}{metrics}\PYG{o}{.}\PYG{n}{accuracy\PYGZus{}score}\PYG{p}{(}\PYG{n}{y\PYGZus{}true}\PYG{p}{,} \PYG{n}{CV\PYGZus{}pred\PYGZus{}label}\PYG{p}{)}
\PYG{n}{SPE} \PYG{o}{=} \PYG{p}{(}\PYG{n+nb}{float}\PYG{p}{(}\PYG{n}{positive\PYGZus{}count}\PYG{p}{)}\PYG{o}{/}\PYG{n+nb}{float}\PYG{p}{(}\PYG{n}{negative\PYGZus{}count}\PYG{p}{)}\PYG{o}{+}\PYG{l+m+mf}{1.0}\PYG{p}{)}\PYG{o}{*}\PYG{n}{ACC}\PYG{o}{\PYGZhy{}}\PYG{n}{TPR}\PYG{o}{*}\PYG{n+nb}{float}\PYG{p}{(}\PYG{n}{positive\PYGZus{}count}\PYG{p}{)}\PYG{o}{/}\PYG{n+nb}{float}\PYG{p}{(}\PYG{n}{negative\PYGZus{}count}\PYG{p}{)}
\PYG{n}{matthews\PYGZus{}corrcoef} \PYG{o}{=} \PYG{n}{metrics}\PYG{o}{.}\PYG{n}{matthews\PYGZus{}corrcoef}\PYG{p}{(}\PYG{n}{y\PYGZus{}true}\PYG{p}{,} \PYG{n}{CV\PYGZus{}pred\PYGZus{}label}\PYG{p}{)}
\PYG{n}{f1\PYGZus{}score} \PYG{o}{=}  \PYG{n}{metrics}\PYG{o}{.}\PYG{n}{f1\PYGZus{}score}\PYG{p}{(}\PYG{n}{y\PYGZus{}true}\PYG{p}{,} \PYG{n}{CV\PYGZus{}pred\PYGZus{}label}\PYG{p}{)}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{c+c1}{\PYGZsh{} plotting the figure}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{figure}\PYG{p}{(}\PYG{n}{figsize} \PYG{o}{=} \PYG{p}{(}\PYG{l+m+mi}{10}\PYG{p}{,}\PYG{l+m+mi}{7}\PYG{p}{)}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{plot}\PYG{p}{(}\PYG{n}{fpr\PYGZus{}cv}\PYG{p}{,} \PYG{n}{tpr\PYGZus{}cv}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{r}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{label}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{auc = }\PYG{l+s+si}{\PYGZpc{}0.2f}\PYG{l+s+s1}{\PYGZsq{}}\PYG{o}{\PYGZpc{}} \PYG{n}{AUC\PYGZus{}score}\PYG{p}{,} \PYG{n}{lw}\PYG{o}{=}\PYG{l+m+mi}{2}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{xlabel}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{False positive rate}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{fontsize}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mi}{20}\PYG{p}{\PYGZcb{}}\PYG{p}{)}\PYG{p}{;}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{ylabel}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{True positive rate}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{fontsize}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mi}{20}\PYG{p}{\PYGZcb{}}\PYG{p}{)}\PYG{p}{;}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{title}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{ROC of Drug\PYGZhy{}target Interaction Classification}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{fontsize}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mi}{25}\PYG{p}{\PYGZcb{}}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{legend}\PYG{p}{(}\PYG{n}{loc}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{lower right}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}\PYG{n}{numpoints}\PYG{o}{=}\PYG{l+m+mi}{15}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{show}\PYG{p}{(}\PYG{p}{)}
\end{Verbatim}

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sensitivity:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{n}{TPR}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{specificity:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{SPE}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{accuracy:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{ACC}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{AUC:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{AUC\PYGZus{}score}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{MCC:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{matthews\PYGZus{}corrcoef}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{F1:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{f1\PYGZus{}score}
\PYG{g+go}{sensitivity: 0.84 specificity: 0.93 accuracy: 0.89 AUC: 0.95 MCC: 0.78 F1: 0.87}
\end{Verbatim}


\section{Application 4 Prediction of protein subcellular location}
\label{application:application-4-prediction-of-protein-subcellular-location}
To identify the functions of proteins in organism is one of the fundamental goals in cell biology and proteomics. The function of a protein in organism is closely linked to its location in a cell. Determination of protein subcellular location (PSL) by experimental methods is expensive and time-consuming. With the enrichment of data repository, automatically prediction of PSL is an alternative method to facilitate the determination of PSL. To build a PSL prediction model, we use PyProtein in PyBioMed to calculate protein features and then the random forest (RF) method was applied to build PSL classification model. The benchmark data set for building the protein subcellular location predictor was taken from (Jia, Qian et al. 2007). The dataset contains 2568 samples, among them 849 proteins were located at Cytoplasm which is defined as positive dataset and 1619 proteins were located at Nucleus which is defined as negative dataset. For each protein, 20 amino acid composition (AAC), 147 CTD composition, transition and distribution and 30 pseudo amino acid composition (PAAC), a total number of 197 features were calculate through the PyBioMed tool.

To build the classification model, the CSV file containing the calculated descriptors was then converted to sample matrix (x\_train) and a sample label vector (y\_train) is also provided. Then, the python script randomforests.py based on sklearn package was employed to build the classification model (the number of trees is 500, the maximum number of features in each tree is square root of the number of features). The performance of this model was evaluated by using 10-fold cross-validation. The AUC score, accuracy, sensitivity and specificity are 0.90, 0.85, 0.94 and 0.69 respectively
\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics[width=10cm]{{subcell}.png}
\caption{The receiver operating characteristic curve of protein subcellular location classification.}\label{application:id5}\end{figure}

\begin{Verbatim}[commandchars=\\\{\},numbers=left,firstnumber=1,stepnumber=1]
\PYG{k+kn}{import} \PYG{n+nn}{pandas} \PYG{k+kn}{as} \PYG{n+nn}{pd}
\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed.PyProtein.CTD} \PYG{k+kn}{import} \PYG{n}{CalculateCTD}
\PYG{k+kn}{import} \PYG{n+nn}{numpy} \PYG{k+kn}{as} \PYG{n+nn}{np}
\PYG{k+kn}{from} \PYG{n+nn}{sklearn.ensemble} \PYG{k+kn}{import} \PYG{n}{RandomForestClassifier} \PYG{k}{as} \PYG{n}{RF}
\PYG{k+kn}{from} \PYG{n+nn}{sklearn} \PYG{k+kn}{import} \PYG{n}{cross\PYGZus{}validation}
\PYG{k+kn}{from} \PYG{n+nn}{sklearn} \PYG{k+kn}{import} \PYG{n}{metrics}
\PYG{k+kn}{from} \PYG{n+nn}{matplotlib} \PYG{k+kn}{import} \PYG{n}{pyplot} \PYG{k}{as} \PYG{n}{plt}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{c+c1}{\PYGZsh{} loading the data}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{n}{path} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{input PyBioMed path in your computer}\PYG{l+s+s1}{\PYGZsq{}}  \PYG{c+c1}{\PYGZsh{}input the PyBioMed path in your own computer}
\PYG{n}{f} \PYG{o}{=} \PYG{n+nb}{open}\PYG{p}{(}\PYG{n}{path} \PYG{o}{+} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{example/subcell/Cytoplasm\PYGZus{}seq.txt}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{r}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{cytoplasm} \PYG{o}{=} \PYG{p}{[}\PYG{n}{line}\PYG{o}{.}\PYG{n}{replace}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+se}{\PYGZbs{}n}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)} \PYG{k}{for} \PYG{n}{line} \PYG{o+ow}{in} \PYG{n}{f}\PYG{o}{.}\PYG{n}{readlines}\PYG{p}{(}\PYG{p}{)} \PYG{k}{if} \PYG{n}{line} \PYG{o}{!=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+se}{\PYGZbs{}n}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}
\PYG{n}{f}\PYG{o}{.}\PYG{n}{close}\PYG{p}{(}\PYG{p}{)}
\PYG{n}{f} \PYG{o}{=} \PYG{n+nb}{open}\PYG{p}{(}\PYG{n}{path} \PYG{o}{+} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{example/subcell/Nuclear\PYGZus{}seq.txt}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{r}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{nuclear} \PYG{o}{=} \PYG{p}{[}\PYG{n}{line}\PYG{o}{.}\PYG{n}{replace}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+se}{\PYGZbs{}n}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)} \PYG{k}{for} \PYG{n}{line} \PYG{o+ow}{in} \PYG{n}{f}\PYG{o}{.}\PYG{n}{readlines}\PYG{p}{(}\PYG{p}{)} \PYG{k}{if} \PYG{n}{line} \PYG{o}{!=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+se}{\PYGZbs{}n}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}
\PYG{n}{f}\PYG{o}{.}\PYG{n}{close}\PYG{p}{(}\PYG{p}{)}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{c+c1}{\PYGZsh{} calculating the descriptors}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{n}{cytoplasm\PYGZus{}des} \PYG{o}{=} \PYG{n+nb}{dict}\PYG{p}{(}\PYG{n+nb}{zip}\PYG{p}{(}\PYG{n+nb}{range}\PYG{p}{(}\PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{cytoplasm}\PYG{p}{)}\PYG{p}{)}\PYG{p}{,}\PYG{n+nb}{map}\PYG{p}{(}\PYG{n}{CalculateCTD}\PYG{p}{,}\PYG{n}{cytoplasm}\PYG{p}{)}\PYG{p}{)}\PYG{p}{)}
\PYG{n}{nuclear\PYGZus{}des} \PYG{o}{=} \PYG{n+nb}{dict}\PYG{p}{(}\PYG{n+nb}{zip}\PYG{p}{(}\PYG{n+nb}{range}\PYG{p}{(}\PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{nuclear}\PYG{p}{)}\PYG{p}{)}\PYG{p}{,}\PYG{n+nb}{map}\PYG{p}{(}\PYG{n}{CalculateCTD}\PYG{p}{,}\PYG{n}{nuclear}\PYG{p}{)}\PYG{p}{)}\PYG{p}{)}
\PYG{n}{cytoplasm\PYGZus{}des\PYGZus{}df} \PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{DataFrame}\PYG{p}{(}\PYG{n}{cytoplasm\PYGZus{}des}\PYG{p}{)}\PYG{o}{.}\PYG{n}{T}
\PYG{n}{nuclear\PYGZus{}des\PYGZus{}df} \PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{DataFrame}\PYG{p}{(}\PYG{n}{nuclear\PYGZus{}des}\PYG{p}{)}\PYG{o}{.}\PYG{n}{T}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{c+c1}{\PYGZsh{} cross\PYGZhy{}validation}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{n}{x} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{array}\PYG{p}{(}\PYG{n}{pd}\PYG{o}{.}\PYG{n}{concat}\PYG{p}{(}\PYG{p}{[}\PYG{n}{cytoplasm\PYGZus{}des\PYGZus{}df}\PYG{p}{,} \PYG{n}{nuclear\PYGZus{}des\PYGZus{}df}\PYG{p}{]}\PYG{p}{)}\PYG{p}{)}
\PYG{n}{positive\PYGZus{}count}\PYG{p}{,} \PYG{n}{negative\PYGZus{}count} \PYG{o}{=} \PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{cytoplasm\PYGZus{}des}\PYG{p}{)}\PYG{p}{,} \PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{nuclear\PYGZus{}des}\PYG{p}{)}
\PYG{n}{y} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{array}\PYG{p}{(}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{*}\PYG{n}{positive\PYGZus{}count}\PYG{o}{+} \PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{*}\PYG{n}{negative\PYGZus{}count}\PYG{p}{)}
\PYG{n}{kf} \PYG{o}{=} \PYG{n}{cross\PYGZus{}validation}\PYG{o}{.}\PYG{n}{KFold}\PYG{p}{(}\PYG{n}{x}\PYG{o}{.}\PYG{n}{shape}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,} \PYG{n}{n\PYGZus{}folds}\PYG{o}{=}\PYG{l+m+mi}{10}\PYG{p}{,} \PYG{n}{shuffle} \PYG{o}{=} \PYG{n+nb+bp}{True}\PYG{p}{,} \PYG{n}{random\PYGZus{}state}\PYG{o}{=}\PYG{l+m+mi}{5}\PYG{p}{)}
\PYG{n}{clf} \PYG{o}{=} \PYG{n}{RF}\PYG{p}{(}\PYG{n}{n\PYGZus{}estimators}\PYG{o}{=}\PYG{l+m+mi}{500}\PYG{p}{,} \PYG{n}{max\PYGZus{}features}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sqrt}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{n\PYGZus{}jobs}\PYG{o}{=}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{oob\PYGZus{}score}\PYG{o}{=}\PYG{n+nb+bp}{True}\PYG{p}{)}
\PYG{n}{CV\PYGZus{}pred\PYGZus{}prob} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{CV\PYGZus{}pred\PYGZus{}label}\PYG{o}{=}\PYG{p}{[}\PYG{p}{]}
\PYG{n}{VALIDATION\PYGZus{}index} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{kf} \PYG{o}{=} \PYG{n}{cross\PYGZus{}validation}\PYG{o}{.}\PYG{n}{KFold}\PYG{p}{(}\PYG{n}{x}\PYG{o}{.}\PYG{n}{shape}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,} \PYG{n}{n\PYGZus{}folds}\PYG{o}{=}\PYG{l+m+mi}{10}\PYG{p}{,} \PYG{n}{shuffle}\PYG{o}{=}\PYG{n+nb+bp}{True}\PYG{p}{,}\PYG{n}{random\PYGZus{}state}\PYG{o}{=}\PYG{l+m+mi}{5}\PYG{p}{)}
\PYG{n}{clf} \PYG{o}{=} \PYG{n}{RF}\PYG{p}{(}\PYG{n}{n\PYGZus{}estimators}\PYG{o}{=}\PYG{l+m+mi}{500}\PYG{p}{,} \PYG{n}{max\PYGZus{}features}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sqrt}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{n\PYGZus{}jobs}\PYG{o}{=}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{oob\PYGZus{}score}\PYG{o}{=}\PYG{n+nb+bp}{True}\PYG{p}{)}
\PYG{n}{CV\PYGZus{}pred\PYGZus{}prob} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{CV\PYGZus{}pred\PYGZus{}label}\PYG{o}{=}\PYG{p}{[}\PYG{p}{]}
\PYG{n}{VALIDATION\PYGZus{}index} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{k}{for} \PYG{n}{train\PYGZus{}index}\PYG{p}{,} \PYG{n}{validation\PYGZus{}index} \PYG{o+ow}{in} \PYG{n}{kf}\PYG{p}{:}
                \PYG{n}{VALIDATION\PYGZus{}index} \PYG{o}{=} \PYG{n}{VALIDATION\PYGZus{}index} \PYG{o}{+} \PYG{n+nb}{list}\PYG{p}{(}\PYG{n}{validation\PYGZus{}index}\PYG{p}{)}
                \PYG{n}{clf}\PYG{o}{.}\PYG{n}{fit}\PYG{p}{(}\PYG{n}{x}\PYG{p}{[}\PYG{n}{train\PYGZus{}index} \PYG{p}{,}\PYG{p}{:}\PYG{p}{]}\PYG{p}{,}\PYG{n}{y}\PYG{p}{[}\PYG{n}{train\PYGZus{}index}\PYG{p}{]}\PYG{p}{)}
                \PYG{n}{pred\PYGZus{}prob} \PYG{o}{=} \PYG{n}{clf}\PYG{o}{.}\PYG{n}{predict\PYGZus{}proba}\PYG{p}{(}\PYG{n}{x}\PYG{p}{[}\PYG{n}{validation\PYGZus{}index}\PYG{p}{,}\PYG{p}{:}\PYG{p}{]}\PYG{p}{)}
                \PYG{n}{pred\PYGZus{}label} \PYG{o}{=} \PYG{n}{clf}\PYG{o}{.}\PYG{n}{predict}\PYG{p}{(}\PYG{n}{x}\PYG{p}{[}\PYG{n}{validation\PYGZus{}index}\PYG{p}{,}\PYG{p}{:}\PYG{p}{]}\PYG{p}{)}
                \PYG{n}{CV\PYGZus{}pred\PYGZus{}prob} \PYG{o}{=} \PYG{n}{CV\PYGZus{}pred\PYGZus{}prob} \PYG{o}{+} \PYG{n+nb}{list}\PYG{p}{(}\PYG{n}{pred\PYGZus{}prob}\PYG{p}{[}\PYG{p}{:}\PYG{p}{,}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{p}{)}
                \PYG{n}{CV\PYGZus{}pred\PYGZus{}label} \PYG{o}{=} \PYG{n}{CV\PYGZus{}pred\PYGZus{}label} \PYG{o}{+} \PYG{n+nb}{list}\PYG{p}{(}\PYG{n}{pred\PYGZus{}label}\PYG{p}{)}
\PYG{n}{fpr\PYGZus{}cv}\PYG{p}{,} \PYG{n}{tpr\PYGZus{}cv}\PYG{p}{,} \PYG{n}{thresholds\PYGZus{}cv} \PYG{o}{=} \PYG{n}{metrics}\PYG{o}{.}\PYG{n}{roc\PYGZus{}curve}\PYG{p}{(}\PYG{n}{y}\PYG{p}{[}\PYG{n}{VALIDATION\PYGZus{}index}\PYG{p}{]}\PYG{p}{,} \PYG{n}{CV\PYGZus{}pred\PYGZus{}prob}\PYG{p}{)}
\PYG{n}{y\PYGZus{}true} \PYG{o}{=} \PYG{n}{y}\PYG{p}{[}\PYG{n}{VALIDATION\PYGZus{}index}\PYG{p}{]}
\PYG{n}{AUC\PYGZus{}score} \PYG{o}{=} \PYG{n}{metrics}\PYG{o}{.}\PYG{n}{roc\PYGZus{}auc\PYGZus{}score}\PYG{p}{(}\PYG{n}{y}\PYG{p}{[}\PYG{n}{VALIDATION\PYGZus{}index}\PYG{p}{]}\PYG{p}{,} \PYG{n}{CV\PYGZus{}pred\PYGZus{}prob}\PYG{p}{)}
\PYG{n}{TPR} \PYG{o}{=} \PYG{n}{metrics}\PYG{o}{.}\PYG{n}{recall\PYGZus{}score}\PYG{p}{(}\PYG{n}{y\PYGZus{}true}\PYG{p}{,} \PYG{n}{CV\PYGZus{}pred\PYGZus{}label}\PYG{p}{)}
\PYG{n}{ACC} \PYG{o}{=} \PYG{n}{metrics}\PYG{o}{.}\PYG{n}{accuracy\PYGZus{}score}\PYG{p}{(}\PYG{n}{y\PYGZus{}true}\PYG{p}{,} \PYG{n}{CV\PYGZus{}pred\PYGZus{}label}\PYG{p}{)}
\PYG{n}{SPE} \PYG{o}{=} \PYG{p}{(}\PYG{n+nb}{float}\PYG{p}{(}\PYG{n}{positive\PYGZus{}count}\PYG{p}{)}\PYG{o}{/}\PYG{n+nb}{float}\PYG{p}{(}\PYG{n}{negative\PYGZus{}count}\PYG{p}{)}\PYG{o}{+}\PYG{l+m+mf}{1.0}\PYG{p}{)}\PYG{o}{*}\PYG{n}{ACC}\PYG{o}{\PYGZhy{}}\PYG{n}{TPR}\PYG{o}{*}\PYG{n+nb}{float}\PYG{p}{(}\PYG{n}{positive\PYGZus{}count}\PYG{p}{)}\PYG{o}{/}\PYG{n+nb}{float}\PYG{p}{(}\PYG{n}{negative\PYGZus{}count}\PYG{p}{)}
\PYG{n}{matthews\PYGZus{}corrcoef} \PYG{o}{=} \PYG{n}{metrics}\PYG{o}{.}\PYG{n}{matthews\PYGZus{}corrcoef}\PYG{p}{(}\PYG{n}{y\PYGZus{}true}\PYG{p}{,} \PYG{n}{CV\PYGZus{}pred\PYGZus{}label}\PYG{p}{)}
\PYG{n}{f1\PYGZus{}score} \PYG{o}{=}  \PYG{n}{metrics}\PYG{o}{.}\PYG{n}{f1\PYGZus{}score}\PYG{p}{(}\PYG{n}{y\PYGZus{}true}\PYG{p}{,} \PYG{n}{CV\PYGZus{}pred\PYGZus{}label}\PYG{p}{)}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{c+c1}{\PYGZsh{} plotting the figure}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{figure}\PYG{p}{(}\PYG{n}{figsize} \PYG{o}{=} \PYG{p}{(}\PYG{l+m+mi}{10}\PYG{p}{,}\PYG{l+m+mi}{7}\PYG{p}{)}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{plot}\PYG{p}{(}\PYG{n}{fpr\PYGZus{}cv}\PYG{p}{,} \PYG{n}{tpr\PYGZus{}cv}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{r}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{label}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{auc = }\PYG{l+s+si}{\PYGZpc{}0.2f}\PYG{l+s+s1}{\PYGZsq{}}\PYG{o}{\PYGZpc{}} \PYG{n}{AUC\PYGZus{}score}\PYG{p}{,} \PYG{n}{lw}\PYG{o}{=}\PYG{l+m+mi}{2}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{xlabel}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{False positive rate}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{fontsize}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mi}{20}\PYG{p}{\PYGZcb{}}\PYG{p}{)}\PYG{p}{;}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{ylabel}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{True positive rate}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{fontsize}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mi}{20}\PYG{p}{\PYGZcb{}}\PYG{p}{)}\PYG{p}{;}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{title}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{ROC of protein subcellular location Classification}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{fontsize}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mi}{25}\PYG{p}{\PYGZcb{}}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{legend}\PYG{p}{(}\PYG{n}{loc}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{lower right}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}\PYG{n}{numpoints}\PYG{o}{=}\PYG{l+m+mi}{15}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{show}\PYG{p}{(}\PYG{p}{)}
\end{Verbatim}

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sensitivity:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{n}{TPR}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{specificity:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{SPE}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{accuracy:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{ACC}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{AUC:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{AUC\PYGZus{}score}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{MACCS:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{matthews\PYGZus{}corrcoef}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{F1:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{f1\PYGZus{}score}
\PYG{g+go}{sensitivity: 0.67 specificity: 0.92 accuracy: 0.84 AUC: 0.89 MACCS: 0.62 F1: 0.74}
\end{Verbatim}


\section{Application 5 Predicting nucleosome positioning in genomes with dinucleotide-based auto covariance}
\label{application:application-5-predicting-nucleosome-positioning-in-genomes-with-dinucleotide-based-auto-covariance}
Nucleosome positioning participates in many cellular activities and plays significant roles in regulating cellular processes (Guo, et al., 2014). Computational methods that can predict nucleosome positioning based on the DNA sequences is highly desired. Here, a computational predictor was constructed by using dinucleotide-based auto covariance and SVMs, and its performance was evaluated by 10-fold cross-validation. The benchmark data set for the H. sapiens was taken from (Schones, et al., 2008). Since the H. sapiens genome and its nucleosome map contain a huge amount of data, according to Liu’s strategy (Liu, et al., 2011) the nucleosome-forming sequence samples (positive data) and the linkers or nucleosome-inhibiting sequence samples (negative data) were extracted from chromosome (Guo, et al., 2014). A file named ``H\_sapiens\_pos.fasta'' containing 2,273 nucleosome-forming DNA segments is used as the positive dataset, and a file named ``H\_sapiens\_neg.fasta'' containing 2,300 nucleosome-inhibiting DNA segments is used as the negative dataset.
\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics[width=10cm]{{DNA}.png}
\caption{The receiver operating characteristic curve of nucleosome positioning in genomes classification.}\label{application:id6}\end{figure}

\begin{Verbatim}[commandchars=\\\{\},numbers=left,firstnumber=1,stepnumber=1]
\PYG{k+kn}{import} \PYG{n+nn}{pandas} \PYG{k+kn}{as} \PYG{n+nn}{pd}
\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed} \PYG{k+kn}{import} \PYG{n}{Pydna}
\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed.PyGetMol} \PYG{k+kn}{import} \PYG{n}{GetDNA}
\PYG{k+kn}{import} \PYG{n+nn}{numpy} \PYG{k+kn}{as} \PYG{n+nn}{np}
\PYG{k+kn}{from} \PYG{n+nn}{sklearn.ensemble} \PYG{k+kn}{import} \PYG{n}{RandomForestClassifier} \PYG{k}{as} \PYG{n}{RF}
\PYG{k+kn}{from} \PYG{n+nn}{sklearn} \PYG{k+kn}{import} \PYG{n}{cross\PYGZus{}validation}
\PYG{k+kn}{from} \PYG{n+nn}{sklearn} \PYG{k+kn}{import} \PYG{n}{metrics}
\PYG{k+kn}{from} \PYG{n+nn}{matplotlib} \PYG{k+kn}{import} \PYG{n}{pyplot} \PYG{k}{as} \PYG{n}{plt}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{c+c1}{\PYGZsh{} loading data}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{n}{path} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{PyBioMed package real path in your computer}\PYG{l+s+s1}{\PYGZsq{}}   \PYG{c+c1}{\PYGZsh{} input the real path in your own computer}
\PYG{n}{seqs\PYGZus{}pos} \PYG{o}{=} \PYG{n}{GetDNA}\PYG{o}{.}\PYG{n}{ReadFasta}\PYG{p}{(}\PYG{n+nb}{open}\PYG{p}{(}\PYG{n}{path} \PYG{o}{+} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{/example/dna/H\PYGZus{}sapiens\PYGZus{}pos.fasta}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}\PYG{p}{)}
\PYG{n}{seqs\PYGZus{}neg} \PYG{o}{=} \PYG{n}{GetDNA}\PYG{o}{.}\PYG{n}{ReadFasta}\PYG{p}{(}\PYG{n+nb}{open}\PYG{p}{(}\PYG{n}{path} \PYG{o}{+} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{/example/dna/H\PYGZus{}sapiens\PYGZus{}neg.fasta}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}\PYG{p}{)}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{c+c1}{\PYGZsh{} calculating descriptors}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{k}{def} \PYG{n+nf}{calculate\PYGZus{}des}\PYG{p}{(}\PYG{n}{seq}\PYG{p}{)}\PYG{p}{:}
        \PYG{n}{des} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
        \PYG{n}{dnaclass} \PYG{o}{=} \PYG{n}{Pydna}\PYG{o}{.}\PYG{n}{PyDNA}\PYG{p}{(}\PYG{n}{seq}\PYG{p}{)}
        \PYG{n}{des}\PYG{o}{.}\PYG{n}{extend}\PYG{p}{(}\PYG{n}{dnaclass}\PYG{o}{.}\PYG{n}{GetDAC}\PYG{p}{(}\PYG{n}{all\PYGZus{}property}\PYG{o}{=}\PYG{n+nb+bp}{True}\PYG{p}{)}\PYG{o}{.}\PYG{n}{values}\PYG{p}{(}\PYG{p}{)}\PYG{p}{)}
        \PYG{n}{des}\PYG{o}{.}\PYG{n}{extend}\PYG{p}{(}\PYG{n}{dnaclass}\PYG{o}{.}\PYG{n}{GetPseDNC}\PYG{p}{(}\PYG{n}{all\PYGZus{}property}\PYG{o}{=}\PYG{n+nb+bp}{True}\PYG{p}{,}\PYG{n}{lamada}\PYG{o}{=}\PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{w}\PYG{o}{=}\PYG{l+m+mf}{0.05}\PYG{p}{)}\PYG{o}{.}\PYG{n}{values}\PYG{p}{(}\PYG{p}{)}\PYG{p}{)}
        \PYG{n}{des}\PYG{o}{.}\PYG{n}{extend}\PYG{p}{(}\PYG{n}{dnaclass}\PYG{o}{.}\PYG{n}{GetPseKNC}\PYG{p}{(}\PYG{n}{all\PYGZus{}property}\PYG{o}{=}\PYG{n+nb+bp}{True}\PYG{p}{,}\PYG{n}{lamada}\PYG{o}{=}\PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{w}\PYG{o}{=}\PYG{l+m+mf}{0.05}\PYG{p}{)}\PYG{o}{.}\PYG{n}{values}\PYG{p}{(}\PYG{p}{)}\PYG{p}{)}
        \PYG{n}{des}\PYG{o}{.}\PYG{n}{extend}\PYG{p}{(}\PYG{n}{dnaclass}\PYG{o}{.}\PYG{n}{GetSCPseDNC}\PYG{p}{(}\PYG{n}{all\PYGZus{}property}\PYG{o}{=}\PYG{n+nb+bp}{True}\PYG{p}{)}\PYG{o}{.}\PYG{n}{values}\PYG{p}{(}\PYG{p}{)}\PYG{p}{)}
        \PYG{k}{return} \PYG{n}{des}
\PYG{n}{pos\PYGZus{}des} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{k}{for} \PYG{n}{seq\PYGZus{}pos} \PYG{o+ow}{in} \PYG{n}{seqs\PYGZus{}pos}\PYG{p}{:}
        \PYG{n}{pos\PYGZus{}des}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{calculate\PYGZus{}des}\PYG{p}{(}\PYG{n}{seq\PYGZus{}pos}\PYG{p}{)}\PYG{p}{)}
\PYG{n}{neg\PYGZus{}des} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{k}{for} \PYG{n}{seq\PYGZus{}neg} \PYG{o+ow}{in} \PYG{n}{seqs\PYGZus{}neg}\PYG{p}{:}
        \PYG{n}{neg\PYGZus{}des}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{calculate\PYGZus{}des}\PYG{p}{(}\PYG{n}{seq\PYGZus{}neg}\PYG{p}{)}\PYG{p}{)}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{c+c1}{\PYGZsh{} cross validation}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{n}{x} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{array}\PYG{p}{(}\PYG{n}{pos\PYGZus{}des}\PYG{o}{+}\PYG{n}{neg\PYGZus{}des}\PYG{p}{)}
\PYG{n}{positive\PYGZus{}count}\PYG{p}{,} \PYG{n}{negative\PYGZus{}count} \PYG{o}{=} \PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{pos\PYGZus{}des}\PYG{p}{)}\PYG{p}{,} \PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{neg\PYGZus{}des}\PYG{p}{)}
\PYG{n}{y} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{array}\PYG{p}{(}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{*}\PYG{n}{positive\PYGZus{}count}\PYG{o}{+} \PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{*}\PYG{n}{negative\PYGZus{}count}\PYG{p}{)}
\PYG{n}{kf} \PYG{o}{=} \PYG{n}{cross\PYGZus{}validation}\PYG{o}{.}\PYG{n}{KFold}\PYG{p}{(}\PYG{n}{x}\PYG{o}{.}\PYG{n}{shape}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,} \PYG{n}{n\PYGZus{}folds}\PYG{o}{=}\PYG{l+m+mi}{10}\PYG{p}{,} \PYG{n}{random\PYGZus{}state}\PYG{o}{=}\PYG{l+m+mi}{0}\PYG{p}{)}
\PYG{n}{clf} \PYG{o}{=} \PYG{n}{RF}\PYG{p}{(}\PYG{n}{n\PYGZus{}estimators}\PYG{o}{=}\PYG{l+m+mi}{500}\PYG{p}{,} \PYG{n}{max\PYGZus{}features}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sqrt}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{n\PYGZus{}jobs}\PYG{o}{=}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{oob\PYGZus{}score}\PYG{o}{=}\PYG{n+nb+bp}{True}\PYG{p}{)}
\PYG{n}{CV\PYGZus{}pred\PYGZus{}prob} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{CV\PYGZus{}pred\PYGZus{}label}\PYG{o}{=}\PYG{p}{[}\PYG{p}{]}
\PYG{n}{VALIDATION\PYGZus{}index} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{k}{for} \PYG{n}{train\PYGZus{}index}\PYG{p}{,} \PYG{n}{validation\PYGZus{}index} \PYG{o+ow}{in} \PYG{n}{kf}\PYG{p}{:}
                \PYG{n}{VALIDATION\PYGZus{}index} \PYG{o}{=} \PYG{n}{VALIDATION\PYGZus{}index} \PYG{o}{+} \PYG{n+nb}{list}\PYG{p}{(}\PYG{n}{validation\PYGZus{}index}\PYG{p}{)}
                \PYG{n}{clf}\PYG{o}{.}\PYG{n}{fit}\PYG{p}{(}\PYG{n}{x}\PYG{p}{[}\PYG{n}{train\PYGZus{}index} \PYG{p}{,}\PYG{p}{:}\PYG{p}{]}\PYG{p}{,}\PYG{n}{y}\PYG{p}{[}\PYG{n}{train\PYGZus{}index}\PYG{p}{]}\PYG{p}{)}
                \PYG{n}{pred\PYGZus{}prob} \PYG{o}{=} \PYG{n}{clf}\PYG{o}{.}\PYG{n}{predict\PYGZus{}proba}\PYG{p}{(}\PYG{n}{x}\PYG{p}{[}\PYG{n}{validation\PYGZus{}index}\PYG{p}{,}\PYG{p}{:}\PYG{p}{]}\PYG{p}{)}
                \PYG{n}{pred\PYGZus{}label} \PYG{o}{=} \PYG{n}{clf}\PYG{o}{.}\PYG{n}{predict}\PYG{p}{(}\PYG{n}{x}\PYG{p}{[}\PYG{n}{validation\PYGZus{}index}\PYG{p}{,}\PYG{p}{:}\PYG{p}{]}\PYG{p}{)}
                \PYG{n}{CV\PYGZus{}pred\PYGZus{}prob} \PYG{o}{=} \PYG{n}{CV\PYGZus{}pred\PYGZus{}prob} \PYG{o}{+} \PYG{n+nb}{list}\PYG{p}{(}\PYG{n}{pred\PYGZus{}prob}\PYG{p}{[}\PYG{p}{:}\PYG{p}{,}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{p}{)}
                \PYG{n}{CV\PYGZus{}pred\PYGZus{}label} \PYG{o}{=} \PYG{n}{CV\PYGZus{}pred\PYGZus{}label} \PYG{o}{+} \PYG{n+nb}{list}\PYG{p}{(}\PYG{n}{pred\PYGZus{}label}\PYG{p}{)}
\PYG{n}{fpr\PYGZus{}cv}\PYG{p}{,} \PYG{n}{tpr\PYGZus{}cv}\PYG{p}{,} \PYG{n}{thresholds\PYGZus{}cv} \PYG{o}{=} \PYG{n}{metrics}\PYG{o}{.}\PYG{n}{roc\PYGZus{}curve}\PYG{p}{(}\PYG{n}{y}\PYG{p}{[}\PYG{n}{VALIDATION\PYGZus{}index}\PYG{p}{]}\PYG{p}{,} \PYG{n}{CV\PYGZus{}pred\PYGZus{}prob}\PYG{p}{)}
\PYG{c+c1}{\PYGZsh{} Calculate auc score of cv}
\PYG{n}{y\PYGZus{}true} \PYG{o}{=} \PYG{n}{y}\PYG{p}{[}\PYG{n}{VALIDATION\PYGZus{}index}\PYG{p}{]}
\PYG{n}{AUC\PYGZus{}score} \PYG{o}{=} \PYG{n}{metrics}\PYG{o}{.}\PYG{n}{roc\PYGZus{}auc\PYGZus{}score}\PYG{p}{(}\PYG{n}{y}\PYG{p}{[}\PYG{n}{VALIDATION\PYGZus{}index}\PYG{p}{]}\PYG{p}{,} \PYG{n}{CV\PYGZus{}pred\PYGZus{}prob}\PYG{p}{)}
\PYG{n}{TPR} \PYG{o}{=} \PYG{n}{metrics}\PYG{o}{.}\PYG{n}{recall\PYGZus{}score}\PYG{p}{(}\PYG{n}{y\PYGZus{}true}\PYG{p}{,} \PYG{n}{CV\PYGZus{}pred\PYGZus{}label}\PYG{p}{)}
\PYG{n}{ACC} \PYG{o}{=} \PYG{n}{metrics}\PYG{o}{.}\PYG{n}{accuracy\PYGZus{}score}\PYG{p}{(}\PYG{n}{y\PYGZus{}true}\PYG{p}{,} \PYG{n}{CV\PYGZus{}pred\PYGZus{}label}\PYG{p}{)}
\PYG{n}{SPE} \PYG{o}{=} \PYG{p}{(}\PYG{n+nb}{float}\PYG{p}{(}\PYG{n}{positive\PYGZus{}count}\PYG{p}{)}\PYG{o}{/}\PYG{n+nb}{float}\PYG{p}{(}\PYG{n}{negative\PYGZus{}count}\PYG{p}{)}\PYG{o}{+}\PYG{l+m+mf}{1.0}\PYG{p}{)}\PYG{o}{*}\PYG{n}{ACC}\PYG{o}{\PYGZhy{}}\PYG{n}{TPR}\PYG{o}{*}\PYG{n+nb}{float}\PYG{p}{(}\PYG{n}{positive\PYGZus{}count}\PYG{p}{)}\PYG{o}{/}\PYG{n+nb}{float}\PYG{p}{(}\PYG{n}{negative\PYGZus{}count}\PYG{p}{)}
\PYG{n}{matthews\PYGZus{}corrcoef} \PYG{o}{=} \PYG{n}{metrics}\PYG{o}{.}\PYG{n}{matthews\PYGZus{}corrcoef}\PYG{p}{(}\PYG{n}{y\PYGZus{}true}\PYG{p}{,} \PYG{n}{CV\PYGZus{}pred\PYGZus{}label}\PYG{p}{)}
\PYG{n}{f1\PYGZus{}score} \PYG{o}{=}  \PYG{n}{metrics}\PYG{o}{.}\PYG{n}{f1\PYGZus{}score}\PYG{p}{(}\PYG{n}{y\PYGZus{}true}\PYG{p}{,} \PYG{n}{CV\PYGZus{}pred\PYGZus{}label}\PYG{p}{)}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{c+c1}{\PYGZsh{} plotting the figure}
\PYG{c+c1}{\PYGZsh{}==============================================================================}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{figure}\PYG{p}{(}\PYG{n}{figsize} \PYG{o}{=} \PYG{p}{(}\PYG{l+m+mi}{10}\PYG{p}{,}\PYG{l+m+mi}{7}\PYG{p}{)}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{plot}\PYG{p}{(}\PYG{n}{fpr\PYGZus{}cv}\PYG{p}{,} \PYG{n}{tpr\PYGZus{}cv}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{r}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{label}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{auc = }\PYG{l+s+si}{\PYGZpc{}0.2f}\PYG{l+s+s1}{\PYGZsq{}}\PYG{o}{\PYGZpc{}} \PYG{n}{AUC\PYGZus{}score}\PYG{p}{,} \PYG{n}{lw}\PYG{o}{=}\PYG{l+m+mi}{2}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{xlabel}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{False positive rate}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{fontsize}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mi}{20}\PYG{p}{\PYGZcb{}}\PYG{p}{)}\PYG{p}{;}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{ylabel}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{True positive rate}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{fontsize}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mi}{20}\PYG{p}{\PYGZcb{}}\PYG{p}{)}\PYG{p}{;}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{title}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{ROC of Nucleosome Positioning in Genomes Classification}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{fontsize}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mi}{25}\PYG{p}{\PYGZcb{}}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{legend}\PYG{p}{(}\PYG{n}{loc}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{lower right}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}\PYG{n}{numpoints}\PYG{o}{=}\PYG{l+m+mi}{15}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{show}\PYG{p}{(}\PYG{p}{)}
\end{Verbatim}

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n+nb}{print} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sensitivity:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{n}{TPR}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{specificity:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{SPE}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{accuracy:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{ACC}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{AUC:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{AUC\PYGZus{}score}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{MACCS:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{matthews\PYGZus{}corrcoef}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{F1:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{f1\PYGZus{}score}
\PYG{g+go}{sensitivity: 0.82 specificity: 0.80 accuracy: 0.81 AUC: 0.88 MACCS: 0.62 F1: 0.81}
\end{Verbatim}


\chapter{PyBioMed API}
\label{modules::doc}\label{modules:pybiomed-api}

\section{PyDNA}
\label{reference/PyDNA::doc}\label{reference/PyDNA:pydna}

\subsection{PyDNAac module}
\label{reference/PyDNAac::doc}\label{reference/PyDNAac:pydnaac-module}\phantomsection\label{reference/PyDNAac:module-PyDNAac}\index{PyDNAac (module)}
Created on Thu May 19 01:05:29 2016

@author: yzj
\index{CheckAcc() (in module PyDNAac)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAac:PyDNAac.CheckAcc}\pysiglinewithargsret{\sphinxcode{PyDNAac.}\sphinxbfcode{CheckAcc}}{\emph{lag}, \emph{k}}{}
\end{fulllineitems}

\index{GetDAC() (in module PyDNAac)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAac:PyDNAac.GetDAC}\pysiglinewithargsret{\sphinxcode{PyDNAac.}\sphinxbfcode{GetDAC}}{\emph{input\_data}, \emph{**kwargs}}{}
Make DAC dictionary.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{input\_data}} -- file object or sequence list.

\item {} 
\textbf{\texttt{phyche\_index}} -- physicochemical properties list.

\item {} 
\textbf{\texttt{all\_property}} -- bool, choose all physicochemical properties or not.

\item {} 
\textbf{\texttt{extra\_phyche\_index}} -- dict, the key is the dinucleotide (string), and its corresponding value is a list.
It means user-defined phyche\_index.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{GetDACC() (in module PyDNAac)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAac:PyDNAac.GetDACC}\pysiglinewithargsret{\sphinxcode{PyDNAac.}\sphinxbfcode{GetDACC}}{\emph{input\_data}, \emph{**kwargs}}{}
Make DACC dictionary.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{input\_data}} -- file object or sequence list.

\item {} 
\textbf{\texttt{phyche\_index}} -- physicochemical properties list.

\item {} 
\textbf{\texttt{all\_property}} -- bool, choose all physicochemical properties or not.

\item {} 
\textbf{\texttt{extra\_phyche\_index}} -- dict, the key is the dinucleotide (string), and its corresponding value is a list.
It means user-defined phyche\_index.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{GetDCC() (in module PyDNAac)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAac:PyDNAac.GetDCC}\pysiglinewithargsret{\sphinxcode{PyDNAac.}\sphinxbfcode{GetDCC}}{\emph{input\_data}, \emph{**kwargs}}{}
Make DCC vector.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{input\_data}} -- file object or sequence list.

\item {} 
\textbf{\texttt{phyche\_index}} -- physicochemical properties list.

\item {} 
\textbf{\texttt{all\_property}} -- bool, choose all physicochemical properties or not.

\item {} 
\textbf{\texttt{extra\_phyche\_index}} -- dict, the key is the dinucleotide (string), and its corresponding value is a list.
It means user-defined phyche\_index.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{GetTAC() (in module PyDNAac)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAac:PyDNAac.GetTAC}\pysiglinewithargsret{\sphinxcode{PyDNAac.}\sphinxbfcode{GetTAC}}{\emph{input\_data}, \emph{**kwargs}}{}
Make TAC dictionary.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{input\_data}} -- file object or sequence list.

\item {} 
\textbf{\texttt{phyche\_index}} -- physicochemical properties list.

\item {} 
\textbf{\texttt{all\_property}} -- bool, choose all physicochemical properties or not.

\item {} 
\textbf{\texttt{extra\_phyche\_index}} -- dict, the key is the dinucleotide (string), and its corresponding value is a list.
It means user-defined phyche\_index.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{GetTACC() (in module PyDNAac)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAac:PyDNAac.GetTACC}\pysiglinewithargsret{\sphinxcode{PyDNAac.}\sphinxbfcode{GetTACC}}{\emph{input\_data}, \emph{**kwargs}}{}
Make TACC dictionary.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{input\_data}} -- file object or sequence list.

\item {} 
\textbf{\texttt{phyche\_index}} -- physicochemical properties list.

\item {} 
\textbf{\texttt{all\_property}} -- bool, choose all physicochemical properties or not.

\item {} 
\textbf{\texttt{extra\_phyche\_index}} -- dict, the key is the dinucleotide (string), and its corresponding value is a list.
It means user-defined phyche\_index.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{GetTCC() (in module PyDNAac)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAac:PyDNAac.GetTCC}\pysiglinewithargsret{\sphinxcode{PyDNAac.}\sphinxbfcode{GetTCC}}{\emph{input\_data}, \emph{**kwargs}}{}
Make TCC dictionary.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{input\_data}} -- file object or sequence list.

\item {} 
\textbf{\texttt{phyche\_index}} -- physicochemical properties list.

\item {} 
\textbf{\texttt{all\_property}} -- bool, choose all physicochemical properties or not.

\item {} 
\textbf{\texttt{extra\_phyche\_index}} -- dict, the key is the dinucleotide (string), and its corresponding value is a list.
It means user-defined phyche\_index.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{ReadyAcc() (in module PyDNAac)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAac:PyDNAac.ReadyAcc}\pysiglinewithargsret{\sphinxcode{PyDNAac.}\sphinxbfcode{ReadyAcc}}{\emph{input\_data}, \emph{k}, \emph{phyche\_index=None}, \emph{all\_property=False}, \emph{extra\_phyche\_index=None}}{}
\end{fulllineitems}



\subsection{PyDNAacutil module}
\label{reference/PyDNAacutil::doc}\label{reference/PyDNAacutil:module-PyDNAacutil}\label{reference/PyDNAacutil:pydnaacutil-module}\index{PyDNAacutil (module)}
Created on Sun May 22 14:40:13 2016

@author: yzj
\index{ExtendPhycheIndex() (in module PyDNAacutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAacutil:PyDNAacutil.ExtendPhycheIndex}\pysiglinewithargsret{\sphinxcode{PyDNAacutil.}\sphinxbfcode{ExtendPhycheIndex}}{\emph{original\_index}, \emph{extend\_index}}{}
Extend \{phyche:{[}value, ... {]}\}

\end{fulllineitems}

\index{MakeACVector() (in module PyDNAacutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAacutil:PyDNAacutil.MakeACVector}\pysiglinewithargsret{\sphinxcode{PyDNAacutil.}\sphinxbfcode{MakeACVector}}{\emph{sequence\_list}, \emph{lag}, \emph{phyche\_value}, \emph{k}}{}
\end{fulllineitems}

\index{MakeCCVector() (in module PyDNAacutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAacutil:PyDNAacutil.MakeCCVector}\pysiglinewithargsret{\sphinxcode{PyDNAacutil.}\sphinxbfcode{MakeCCVector}}{\emph{sequence\_list}, \emph{lag}, \emph{phyche\_value}, \emph{k}}{}
\end{fulllineitems}



\subsection{PyDNAnac module}
\label{reference/PyDNAnac:module-PyDNAnac}\label{reference/PyDNAnac:pydnanac-module}\label{reference/PyDNAnac::doc}\index{PyDNAnac (module)}
Created on Sun May 22 14:45:46 2016

@author: yzj
\index{CheckNacPara() (in module PyDNAnac)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAnac:PyDNAnac.CheckNacPara}\pysiglinewithargsret{\sphinxcode{PyDNAnac.}\sphinxbfcode{CheckNacPara}}{\emph{k}, \emph{normalize=False}, \emph{upto=False}, \emph{alphabet='ACGT'}}{}
\end{fulllineitems}

\index{GetIdKmer() (in module PyDNAnac)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAnac:PyDNAnac.GetIdKmer}\pysiglinewithargsret{\sphinxcode{PyDNAnac.}\sphinxbfcode{GetIdKmer}}{\emph{data}, \emph{hs}, \emph{non\_hs}, \emph{**kwargs}}{}
Make IDKmer vector.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{data}} -- Need to processed FASTA file.

\item {} 
\textbf{\texttt{hs}} -- Positive FASTA file.

\item {} 
\textbf{\texttt{non\_hs}} -- Negative FASTA file.

\item {} 
\textbf{\texttt{k}} -- int, the k value of kmer, it should be larger than 0.

\item {} 
\textbf{\texttt{upto}} -- bool, whether to generate 1-kmer, 2-kmer, ..., k-mer.

\item {} 
\textbf{\texttt{alphabet}} -- string.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{GetKmer() (in module PyDNAnac)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAnac:PyDNAnac.GetKmer}\pysiglinewithargsret{\sphinxcode{PyDNAnac.}\sphinxbfcode{GetKmer}}{\emph{data}, \emph{**kwargs}}{}
Make a kmer dictionary with options k, upto, revcomp, normalize.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{k}} -- int, the k value of kmer, it should be larger than 0.

\item {} 
\textbf{\texttt{normalize}} -- bool, normalize the result vector or not.

\item {} 
\textbf{\texttt{upto}} -- bool, whether to generate 1-kmer, 2-kmer, ..., k-mer.

\item {} 
\textbf{\texttt{alphabet}} -- string.

\item {} 
\textbf{\texttt{data}} -- file object or sequence list.

\end{itemize}

\item[{Returns}] \leavevmode
kmer vector.

\end{description}\end{quote}

\end{fulllineitems}

\index{GetKmerList() (in module PyDNAnac)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAnac:PyDNAnac.GetKmerList}\pysiglinewithargsret{\sphinxcode{PyDNAnac.}\sphinxbfcode{GetKmerList}}{\emph{k}, \emph{upto}, \emph{alphabet}}{}
Get the kmer list.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{k}} -- int, the k value of kmer, it should be larger than 0.

\item {} 
\textbf{\texttt{upto}} -- bool, whether to generate 1-kmer, 2-kmer, ..., k-mer.

\item {} 
\textbf{\texttt{alphabet}} -- string.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{GetRevcKmer() (in module PyDNAnac)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAnac:PyDNAnac.GetRevcKmer}\pysiglinewithargsret{\sphinxcode{PyDNAnac.}\sphinxbfcode{GetRevcKmer}}{\emph{data}, \emph{**kwargs}}{}
Make a reverse compliment kmer dictionary with options k, upto, normalize.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\textbf{\texttt{data}} -- file object or sequence list.

\item[{Returns}] \leavevmode
reverse compliment kmer vector.

\end{description}\end{quote}

\end{fulllineitems}



\subsection{PyDNAnacutil module}
\label{reference/PyDNAnacutil:module-PyDNAnacutil}\label{reference/PyDNAnacutil:pydnanacutil-module}\label{reference/PyDNAnacutil::doc}\index{PyDNAnacutil (module)}
Created on Sun May 22 14:47:55 2016

@author: yzj
\index{ComputeBinNum() (in module PyDNAnacutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAnacutil:PyDNAnacutil.ComputeBinNum}\pysiglinewithargsret{\sphinxcode{PyDNAnacutil.}\sphinxbfcode{ComputeBinNum}}{\emph{num\_bins}, \emph{position}, \emph{k}, \emph{numbers}}{}
\end{fulllineitems}

\index{ComputeQuantileBoundaries() (in module PyDNAnacutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAnacutil:PyDNAnacutil.ComputeQuantileBoundaries}\pysiglinewithargsret{\sphinxcode{PyDNAnacutil.}\sphinxbfcode{ComputeQuantileBoundaries}}{\emph{num\_bins}, \emph{k\_values}, \emph{number\_filename}}{}
\end{fulllineitems}

\index{Diversity() (in module PyDNAnacutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAnacutil:PyDNAnacutil.Diversity}\pysiglinewithargsret{\sphinxcode{PyDNAnacutil.}\sphinxbfcode{Diversity}}{\emph{vec}}{}
Calculate diversity.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\textbf{\texttt{vec}} -- kmer vec

\item[{Returns}] \leavevmode
Diversity(X)

\end{description}\end{quote}

\end{fulllineitems}

\index{FindRevcomp() (in module PyDNAnacutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAnacutil:PyDNAnacutil.FindRevcomp}\pysiglinewithargsret{\sphinxcode{PyDNAnacutil.}\sphinxbfcode{FindRevcomp}}{\emph{sequence}, \emph{revcomp\_dictionary}}{}
\end{fulllineitems}

\index{Frequency() (in module PyDNAnacutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAnacutil:PyDNAnacutil.Frequency}\pysiglinewithargsret{\sphinxcode{PyDNAnacutil.}\sphinxbfcode{Frequency}}{\emph{tol\_str}, \emph{tar\_str}}{}
Generate the frequency of tar\_str in tol\_str.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{tol\_str}} -- mother string.

\item {} 
\textbf{\texttt{tar\_str}} -- substring.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{IdXS() (in module PyDNAnacutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAnacutil:PyDNAnacutil.IdXS}\pysiglinewithargsret{\sphinxcode{PyDNAnacutil.}\sphinxbfcode{IdXS}}{\emph{vec\_x}, \emph{vec\_s}, \emph{diversity\_s}}{}
Calculate ID(X, S)
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{vec\_x}} -- kmer X

\item {} 
\textbf{\texttt{vec\_s}} -- kmer S

\end{itemize}

\item[{Returns}] \leavevmode
ID(X, S) = Diversity(X + S) - Diversity(X) - Diversity(S)

\end{description}\end{quote}

\end{fulllineitems}

\index{MakeIndex() (in module PyDNAnacutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAnacutil:PyDNAnacutil.MakeIndex}\pysiglinewithargsret{\sphinxcode{PyDNAnacutil.}\sphinxbfcode{MakeIndex}}{\emph{k}}{}
\end{fulllineitems}

\index{MakeIndexUptoK() (in module PyDNAnacutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAnacutil:PyDNAnacutil.MakeIndexUptoK}\pysiglinewithargsret{\sphinxcode{PyDNAnacutil.}\sphinxbfcode{MakeIndexUptoK}}{\emph{k}}{}
\end{fulllineitems}

\index{MakeIndexUptoKRevcomp() (in module PyDNAnacutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAnacutil:PyDNAnacutil.MakeIndexUptoKRevcomp}\pysiglinewithargsret{\sphinxcode{PyDNAnacutil.}\sphinxbfcode{MakeIndexUptoKRevcomp}}{\emph{k}}{}
\end{fulllineitems}

\index{MakeKmerList() (in module PyDNAnacutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAnacutil:PyDNAnacutil.MakeKmerList}\pysiglinewithargsret{\sphinxcode{PyDNAnacutil.}\sphinxbfcode{MakeKmerList}}{\emph{k}, \emph{alphabet}}{}
\end{fulllineitems}

\index{MakeKmerVector() (in module PyDNAnacutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAnacutil:PyDNAnacutil.MakeKmerVector}\pysiglinewithargsret{\sphinxcode{PyDNAnacutil.}\sphinxbfcode{MakeKmerVector}}{\emph{seq\_list}, \emph{kmer\_list}, \emph{rev\_kmer\_list}, \emph{k}, \emph{upto}, \emph{revcomp}, \emph{normalize}}{}
\end{fulllineitems}

\index{MakeRevcompKmerList() (in module PyDNAnacutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAnacutil:PyDNAnacutil.MakeRevcompKmerList}\pysiglinewithargsret{\sphinxcode{PyDNAnacutil.}\sphinxbfcode{MakeRevcompKmerList}}{\emph{kmer\_list}}{}
\end{fulllineitems}

\index{MakeSequenceVector() (in module PyDNAnacutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAnacutil:PyDNAnacutil.MakeSequenceVector}\pysiglinewithargsret{\sphinxcode{PyDNAnacutil.}\sphinxbfcode{MakeSequenceVector}}{\emph{sequence}, \emph{numbers}, \emph{num\_bins}, \emph{revcomp}, \emph{revcomp\_dictionary}, \emph{normalize\_method}, \emph{k\_values}, \emph{mismatch}, \emph{alphabet}, \emph{kmer\_list}, \emph{boundaries}, \emph{pseudocount}}{}
\end{fulllineitems}

\index{MakeUptoKmerList() (in module PyDNAnacutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAnacutil:PyDNAnacutil.MakeUptoKmerList}\pysiglinewithargsret{\sphinxcode{PyDNAnacutil.}\sphinxbfcode{MakeUptoKmerList}}{\emph{k\_values}, \emph{alphabet}}{}
\end{fulllineitems}

\index{NormalizeVector() (in module PyDNAnacutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAnacutil:PyDNAnacutil.NormalizeVector}\pysiglinewithargsret{\sphinxcode{PyDNAnacutil.}\sphinxbfcode{NormalizeVector}}{\emph{normalize\_method}, \emph{k\_values}, \emph{vector}, \emph{kmer\_list}}{}
\end{fulllineitems}

\index{ReadFastaSequence() (in module PyDNAnacutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAnacutil:PyDNAnacutil.ReadFastaSequence}\pysiglinewithargsret{\sphinxcode{PyDNAnacutil.}\sphinxbfcode{ReadFastaSequence}}{\emph{numeric}, \emph{fasta\_file}}{}
\end{fulllineitems}

\index{ReadSequenceAndNumbers() (in module PyDNAnacutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAnacutil:PyDNAnacutil.ReadSequenceAndNumbers}\pysiglinewithargsret{\sphinxcode{PyDNAnacutil.}\sphinxbfcode{ReadSequenceAndNumbers}}{\emph{fasta\_file}, \emph{numbers\_filename}, \emph{numbers\_file}}{}
\end{fulllineitems}

\index{Substitute() (in module PyDNAnacutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAnacutil:PyDNAnacutil.Substitute}\pysiglinewithargsret{\sphinxcode{PyDNAnacutil.}\sphinxbfcode{Substitute}}{\emph{position}, \emph{letter}, \emph{string}}{}
\end{fulllineitems}

\index{cmp() (in module PyDNAnacutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAnacutil:PyDNAnacutil.cmp}\pysiglinewithargsret{\sphinxcode{PyDNAnacutil.}\sphinxbfcode{cmp}}{\emph{a}, \emph{b}}{}
\end{fulllineitems}



\subsection{PyDNApsenac module}
\label{reference/PyDNApsenac::doc}\label{reference/PyDNApsenac:pydnapsenac-module}\label{reference/PyDNApsenac:module-PyDNApsenac}\index{PyDNApsenac (module)}
Created on Thu Jun 02 09:38:15 2016

@author: yzj
\index{CheckPsenac() (in module PyDNApsenac)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNApsenac:PyDNApsenac.CheckPsenac}\pysiglinewithargsret{\sphinxcode{PyDNApsenac.}\sphinxbfcode{CheckPsenac}}{\emph{lamada}, \emph{w}, \emph{k}}{}
Check the validation of parameter lamada, w and k.

\end{fulllineitems}

\index{GetPCPseDNC() (in module PyDNApsenac)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNApsenac:PyDNApsenac.GetPCPseDNC}\pysiglinewithargsret{\sphinxcode{PyDNApsenac.}\sphinxbfcode{GetPCPseDNC}}{\emph{input\_data}, \emph{**kwargs}}{}
Make a PCPseDNC dictionary.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{input\_data}} -- file object or sequence list.

\item {} 
\textbf{\texttt{phyche\_index}} -- physicochemical properties list.

\item {} 
\textbf{\texttt{all\_property}} -- choose all physicochemical properties or not.

\item {} 
\textbf{\texttt{extra\_phyche\_index}} -- \begin{description}
\item[{dict, the key is the dinucleotide (string),}] \leavevmode
the value is its physicochemical property value (list).

\end{description}

It means the user-defined physicochemical indices.


\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{GetPCPseTNC() (in module PyDNApsenac)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNApsenac:PyDNApsenac.GetPCPseTNC}\pysiglinewithargsret{\sphinxcode{PyDNApsenac.}\sphinxbfcode{GetPCPseTNC}}{\emph{input\_data}, \emph{**kwargs}}{}
Make a PCPseDNC dictionary.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{input\_data}} -- file object or sequence list.

\item {} 
\textbf{\texttt{phyche\_index}} -- physicochemical properties list.

\item {} 
\textbf{\texttt{all\_property}} -- choose all physicochemical properties or not.

\item {} 
\textbf{\texttt{extra\_phyche\_index}} -- \begin{description}
\item[{dict, the key is the dinucleotide (string),}] \leavevmode
the value is its physicochemical property value (list).

\end{description}

It means the user-defined physicochemical indices.


\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{GetPseDNC() (in module PyDNApsenac)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNApsenac:PyDNApsenac.GetPseDNC}\pysiglinewithargsret{\sphinxcode{PyDNApsenac.}\sphinxbfcode{GetPseDNC}}{\emph{input\_data}, \emph{**kwargs}}{}
Make PseDNC dictionary.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{input\_data}} -- file type or handle.

\item {} 
\textbf{\texttt{k}} -- k-tuple.

\item {} 
\textbf{\texttt{extra\_phyche\_index}} -- \begin{description}
\item[{dict, the key is the dinucleotide (string),}] \leavevmode
the value is its physicochemical property value (list).

\end{description}

It means the user-defined physicochemical indices.


\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{GetPseKNC() (in module PyDNApsenac)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNApsenac:PyDNApsenac.GetPseKNC}\pysiglinewithargsret{\sphinxcode{PyDNApsenac.}\sphinxbfcode{GetPseKNC}}{\emph{input\_data}, \emph{**kwargs}}{}
Make PseKNC dictionary.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{input\_data}} -- file type or handle.

\item {} 
\textbf{\texttt{k}} -- k-tuple.

\item {} 
\textbf{\texttt{extra\_phyche\_index}} -- \begin{description}
\item[{dict, the key is the dinucleotide (string),}] \leavevmode
the value is its physicochemical property value (list).

\end{description}

It means the user-defined physicochemical indices.


\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{GetSCPseDNC() (in module PyDNApsenac)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNApsenac:PyDNApsenac.GetSCPseDNC}\pysiglinewithargsret{\sphinxcode{PyDNApsenac.}\sphinxbfcode{GetSCPseDNC}}{\emph{input\_data}, \emph{**kwargs}}{}
Make a SCPseDNC dictionary.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{input\_data}} -- file object or sequence list.

\item {} 
\textbf{\texttt{phyche\_index}} -- physicochemical properties list.

\item {} 
\textbf{\texttt{all\_property}} -- choose all physicochemical properties or not.

\item {} 
\textbf{\texttt{extra\_phyche\_index}} -- \begin{description}
\item[{dict, the key is the dinucleotide (string),}] \leavevmode
the value is its physicochemical property value (list).

\end{description}

It means the user-defined physicochemical indices.


\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{GetSCPseTNC() (in module PyDNApsenac)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNApsenac:PyDNApsenac.GetSCPseTNC}\pysiglinewithargsret{\sphinxcode{PyDNApsenac.}\sphinxbfcode{GetSCPseTNC}}{\emph{input\_data}, \emph{**kwargs}}{}
Make a SCPseTNC dictionary.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{input\_data}} -- file object or sequence list.

\item {} 
\textbf{\texttt{phyche\_index}} -- physicochemical properties list.

\item {} 
\textbf{\texttt{all\_property}} -- choose all physicochemical properties or not.

\item {} 
\textbf{\texttt{extra\_phyche\_index}} -- \begin{description}
\item[{dict, the key is the dinucleotide (string),}] \leavevmode
the value is its physicochemical property value (list).

\end{description}

It means the user-defined physicochemical indices.


\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{GetSequenceListAndPhycheValue() (in module PyDNApsenac)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNApsenac:PyDNApsenac.GetSequenceListAndPhycheValue}\pysiglinewithargsret{\sphinxcode{PyDNApsenac.}\sphinxbfcode{GetSequenceListAndPhycheValue}}{\emph{input\_data}, \emph{k}, \emph{phyche\_index}, \emph{extra\_phyche\_index}, \emph{all\_property}}{}
For PseKNC-general make sequence\_list and phyche\_value.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{input\_data}} -- file type or handle.

\item {} 
\textbf{\texttt{k}} -- int, the value of k-tuple.

\item {} 
\textbf{\texttt{k}} -- physicochemical properties list.

\item {} 
\textbf{\texttt{extra\_phyche\_index}} -- \begin{description}
\item[{dict, the key is the dinucleotide (string),}] \leavevmode
the value is its physicochemical property value (list).

\end{description}

It means the user-defined physicochemical indices.


\item {} 
\textbf{\texttt{all\_property}} -- bool, choose all physicochemical properties or not.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{GetSequenceListAndPhycheValuePsednc() (in module PyDNApsenac)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNApsenac:PyDNApsenac.GetSequenceListAndPhycheValuePsednc}\pysiglinewithargsret{\sphinxcode{PyDNApsenac.}\sphinxbfcode{GetSequenceListAndPhycheValuePsednc}}{\emph{input\_data}, \emph{extra\_phyche\_index=None}}{}
For PseDNC, PseKNC, make sequence\_list and phyche\_value.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{input\_data}} -- file type or handle.

\item {} 
\textbf{\texttt{extra\_phyche\_index}} -- \begin{description}
\item[{dict, the key is the dinucleotide (string),}] \leavevmode
the value is its physicochemical property value (list).

\end{description}

It means the user-defined physicochemical indices.


\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{GetSequenceListAndPhycheValuePseknc() (in module PyDNApsenac)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNApsenac:PyDNApsenac.GetSequenceListAndPhycheValuePseknc}\pysiglinewithargsret{\sphinxcode{PyDNApsenac.}\sphinxbfcode{GetSequenceListAndPhycheValuePseknc}}{\emph{input\_data}, \emph{extra\_phyche\_index=None}}{}
For PseDNC, PseKNC, make sequence\_list and phyche\_value.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{input\_data}} -- file type or handle.

\item {} 
\textbf{\texttt{extra\_phyche\_index}} -- \begin{description}
\item[{dict, the key is the dinucleotide (string),}] \leavevmode
the value is its physicochemical property value (list).

\end{description}

It means the user-defined physicochemical indices.


\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}



\subsection{PyDNApsenacutil module}
\label{reference/PyDNApsenacutil::doc}\label{reference/PyDNApsenacutil:pydnapsenacutil-module}\label{reference/PyDNApsenacutil:module-PyDNApsenacutil}\index{PyDNApsenacutil (module)}
Created on Thu Jun 02 10:00:35 2016

@author: yzj
\index{ExtendPhycheIndex() (in module PyDNApsenacutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNApsenacutil:PyDNApsenacutil.ExtendPhycheIndex}\pysiglinewithargsret{\sphinxcode{PyDNApsenacutil.}\sphinxbfcode{ExtendPhycheIndex}}{\emph{original\_index}, \emph{extend\_index}}{}
Extend \{phyche:{[}value, ... {]}\}

\end{fulllineitems}

\index{GetParallelFactor() (in module PyDNApsenacutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNApsenacutil:PyDNApsenacutil.GetParallelFactor}\pysiglinewithargsret{\sphinxcode{PyDNApsenacutil.}\sphinxbfcode{GetParallelFactor}}{\emph{k}, \emph{lamada}, \emph{sequence}, \emph{phyche\_value}}{}
Get the corresponding factor theta list.

\end{fulllineitems}

\index{GetParallelFactorPsednc() (in module PyDNApsenacutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNApsenacutil:PyDNApsenacutil.GetParallelFactorPsednc}\pysiglinewithargsret{\sphinxcode{PyDNApsenacutil.}\sphinxbfcode{GetParallelFactorPsednc}}{\emph{lamada}, \emph{sequence}, \emph{phyche\_value}}{}
Get the corresponding factor theta list.
This def is just for dinucleotide.

\end{fulllineitems}

\index{GetPhycheFactorDic() (in module PyDNApsenacutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNApsenacutil:PyDNApsenacutil.GetPhycheFactorDic}\pysiglinewithargsret{\sphinxcode{PyDNApsenacutil.}\sphinxbfcode{GetPhycheFactorDic}}{\emph{k}}{}
Get all \{nucleotide: {[}(phyche, value), ...{]}\} dict.

\end{fulllineitems}

\index{GetPhycheIndex() (in module PyDNApsenacutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNApsenacutil:PyDNApsenacutil.GetPhycheIndex}\pysiglinewithargsret{\sphinxcode{PyDNApsenacutil.}\sphinxbfcode{GetPhycheIndex}}{\emph{k}, \emph{phyche\_list}}{}
get phyche\_value according phyche\_list.

\end{fulllineitems}

\index{GetSeriesFactor() (in module PyDNApsenacutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNApsenacutil:PyDNApsenacutil.GetSeriesFactor}\pysiglinewithargsret{\sphinxcode{PyDNApsenacutil.}\sphinxbfcode{GetSeriesFactor}}{\emph{k}, \emph{lamada}, \emph{sequence}, \emph{phyche\_value}}{}
Get the corresponding series factor theta list.

\end{fulllineitems}

\index{MakeOldPsekncVector() (in module PyDNApsenacutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNApsenacutil:PyDNApsenacutil.MakeOldPsekncVector}\pysiglinewithargsret{\sphinxcode{PyDNApsenacutil.}\sphinxbfcode{MakeOldPsekncVector}}{\emph{sequence\_list}, \emph{lamada}, \emph{w}, \emph{k}, \emph{phyche\_value}, \emph{theta\_type=1}}{}
Generate the pseknc vector.

\end{fulllineitems}

\index{MakePsekncVector() (in module PyDNApsenacutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNApsenacutil:PyDNApsenacutil.MakePsekncVector}\pysiglinewithargsret{\sphinxcode{PyDNApsenacutil.}\sphinxbfcode{MakePsekncVector}}{\emph{sequence\_list}, \emph{lamada}, \emph{w}, \emph{k}, \emph{phyche\_value}, \emph{theta\_type=1}}{}
Generate the pseknc vector.

\end{fulllineitems}

\index{ParallelCorFunction() (in module PyDNApsenacutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNApsenacutil:PyDNApsenacutil.ParallelCorFunction}\pysiglinewithargsret{\sphinxcode{PyDNApsenacutil.}\sphinxbfcode{ParallelCorFunction}}{\emph{nucleotide1}, \emph{nucleotide2}, \emph{phyche\_index}}{}
Get the cFactor.(Type1)

\end{fulllineitems}

\index{SeriesCorFunction() (in module PyDNApsenacutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNApsenacutil:PyDNApsenacutil.SeriesCorFunction}\pysiglinewithargsret{\sphinxcode{PyDNApsenacutil.}\sphinxbfcode{SeriesCorFunction}}{\emph{nucleotide1}, \emph{nucleotide2}, \emph{big\_lamada}, \emph{phyche\_value}}{}
Get the series correlation Factor(Type 2).

\end{fulllineitems}



\subsection{PyDNAutil module}
\label{reference/PyDNAutil::doc}\label{reference/PyDNAutil:pydnautil-module}\label{reference/PyDNAutil:module-PyDNAutil}\index{PyDNAutil (module)}
Created on Wed May 18 14:06:37 2016

@author: yzj
\index{ALPHABET (in module PyDNAutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAutil:PyDNAutil.ALPHABET}\pysigline{\sphinxcode{PyDNAutil.}\sphinxbfcode{ALPHABET}\sphinxstrong{ = `ACGT'}}
Used for process original data.

\end{fulllineitems}

\index{ConvertPhycheIndexToDict() (in module PyDNAutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAutil:PyDNAutil.ConvertPhycheIndexToDict}\pysiglinewithargsret{\sphinxcode{PyDNAutil.}\sphinxbfcode{ConvertPhycheIndexToDict}}{\emph{phyche\_index}}{}
\end{fulllineitems}

\index{DNAChecks() (in module PyDNAutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAutil:PyDNAutil.DNAChecks}\pysiglinewithargsret{\sphinxcode{PyDNAutil.}\sphinxbfcode{DNAChecks}}{\emph{s}}{}
\end{fulllineitems}

\index{Frequency() (in module PyDNAutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAutil:PyDNAutil.Frequency}\pysiglinewithargsret{\sphinxcode{PyDNAutil.}\sphinxbfcode{Frequency}}{\emph{tol\_str}, \emph{tar\_str}}{}
Generate the frequency of tar\_str in tol\_str.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{tol\_str}} -- mother string.

\item {} 
\textbf{\texttt{tar\_str}} -- substring.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{GeneratePhycheValue() (in module PyDNAutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAutil:PyDNAutil.GeneratePhycheValue}\pysiglinewithargsret{\sphinxcode{PyDNAutil.}\sphinxbfcode{GeneratePhycheValue}}{\emph{k}, \emph{phyche\_index=None}, \emph{all\_property=False}, \emph{extra\_phyche\_index=None}}{}
Combine the user selected phyche\_list, is\_all\_property and 
extra\_phyche\_index to a new standard phyche\_value.
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#

\end{fulllineitems}

\index{GetData() (in module PyDNAutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAutil:PyDNAutil.GetData}\pysiglinewithargsret{\sphinxcode{PyDNAutil.}\sphinxbfcode{GetData}}{\emph{input\_data}, \emph{desc=False}}{}
Get sequence data from file or list with check.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{input\_data}} -- type file or list

\item {} 
\textbf{\texttt{desc}} -- with this option, the return value will be a Seq object list(it only works in file object).

\end{itemize}

\item[{Returns}] \leavevmode
sequence data or shutdown.

\end{description}\end{quote}

\end{fulllineitems}

\index{GetSequenceCheckDna() (in module PyDNAutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAutil:PyDNAutil.GetSequenceCheckDna}\pysiglinewithargsret{\sphinxcode{PyDNAutil.}\sphinxbfcode{GetSequenceCheckDna}}{\emph{f}}{}
Read the fasta file.

Input: f: HANDLE to input. e.g. sys.stdin, or open(\textless{}file\textgreater{})

\end{fulllineitems}

\index{IsFasta() (in module PyDNAutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAutil:PyDNAutil.IsFasta}\pysiglinewithargsret{\sphinxcode{PyDNAutil.}\sphinxbfcode{IsFasta}}{\emph{seq}}{}
Judge the Seq object is in FASTA format.
Two situation:
1. No seq name.
2. Seq name is illegal.
3. No sequence.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\textbf{\texttt{seq}} -- Seq object.

\end{description}\end{quote}

\end{fulllineitems}

\index{IsSequenceList() (in module PyDNAutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAutil:PyDNAutil.IsSequenceList}\pysiglinewithargsret{\sphinxcode{PyDNAutil.}\sphinxbfcode{IsSequenceList}}{\emph{sequence\_list}}{}
Judge the sequence list is within the scope of alphabet and 
change the lowercase to capital.
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#

\end{fulllineitems}

\index{IsUnderAlphabet() (in module PyDNAutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAutil:PyDNAutil.IsUnderAlphabet}\pysiglinewithargsret{\sphinxcode{PyDNAutil.}\sphinxbfcode{IsUnderAlphabet}}{\emph{s}, \emph{alphabet}}{}
Judge the string is within the scope of the alphabet or not.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{s}} -- The string.

\item {} 
\textbf{\texttt{alphabet}} -- alphabet.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{NormalizeIndex() (in module PyDNAutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAutil:PyDNAutil.NormalizeIndex}\pysiglinewithargsret{\sphinxcode{PyDNAutil.}\sphinxbfcode{NormalizeIndex}}{\emph{phyche\_index}, \emph{is\_convert\_dict=False}}{}
\end{fulllineitems}

\index{ReadFasta() (in module PyDNAutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAutil:PyDNAutil.ReadFasta}\pysiglinewithargsret{\sphinxcode{PyDNAutil.}\sphinxbfcode{ReadFasta}}{\emph{f}}{}
Read a fasta file.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\textbf{\texttt{f}} -- HANDLE to input. e.g. sys.stdin, or open(\textless{}file\textgreater{})

\end{description}\end{quote}

\end{fulllineitems}

\index{ReadFastaCheckDna() (in module PyDNAutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAutil:PyDNAutil.ReadFastaCheckDna}\pysiglinewithargsret{\sphinxcode{PyDNAutil.}\sphinxbfcode{ReadFastaCheckDna}}{\emph{f}}{}
Read the fasta file, and check its legality.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\textbf{\texttt{f}} -- HANDLE to input. e.g. sys.stdin, or open(\textless{}file\textgreater{})

\end{description}\end{quote}

\end{fulllineitems}

\index{ReadFastaYield() (in module PyDNAutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAutil:PyDNAutil.ReadFastaYield}\pysiglinewithargsret{\sphinxcode{PyDNAutil.}\sphinxbfcode{ReadFastaYield}}{\emph{f}}{}
Yields a Seq object.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\textbf{\texttt{f}} -- HANDLE to input. e.g. sys.stdin, or open(\textless{}file\textgreater{})

\end{description}\end{quote}

\end{fulllineitems}

\index{Seq (class in PyDNAutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAutil:PyDNAutil.Seq}\pysiglinewithargsret{\sphinxstrong{class }\sphinxcode{PyDNAutil.}\sphinxbfcode{Seq}}{\emph{name}, \emph{seq}, \emph{no}}{}
\end{fulllineitems}

\index{StandardDeviation() (in module PyDNAutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAutil:PyDNAutil.StandardDeviation}\pysiglinewithargsret{\sphinxcode{PyDNAutil.}\sphinxbfcode{StandardDeviation}}{\emph{value\_list}}{}
\end{fulllineitems}

\index{WriteLibsvm() (in module PyDNAutil)}

\begin{fulllineitems}
\phantomsection\label{reference/PyDNAutil:PyDNAutil.WriteLibsvm}\pysiglinewithargsret{\sphinxcode{PyDNAutil.}\sphinxbfcode{WriteLibsvm}}{\emph{vector\_list}, \emph{label\_list}, \emph{write\_file}}{}
\end{fulllineitems}



\section{PyMolecule}
\label{reference/PyMolecule:pymolecule}\label{reference/PyMolecule::doc}

\subsection{AtomProperty module}
\label{reference/AtomProperty:module-AtomProperty}\label{reference/AtomProperty::doc}\label{reference/AtomProperty:atomproperty-module}\index{AtomProperty (module)}
You can freely use and distribute it. If you hava

any problem, you could contact with us timely!

Authors: Zhijiang Yao and Dongsheng Cao.

Date: 2016.06.04

Email: \href{mailto:gadsby@163.com}{gadsby@163.com}

Z: atomic number
L: principal quantum number
Zv: number of valence electrons
Rv: van der Waals atomic radius
Rc: covalent radius
m: atomic mass
V: van der Waals vloume
En: Sanderson electronegativity
alapha: atomic polarizability (10e-24 cm3)
IP: ionization potential (eV)
EA: electron affinity (eV)
\index{GetAbsoluteAtomicProperty() (in module AtomProperty)}

\begin{fulllineitems}
\phantomsection\label{reference/AtomProperty:AtomProperty.GetAbsoluteAtomicProperty}\pysiglinewithargsret{\sphinxcode{AtomProperty.}\sphinxbfcode{GetAbsoluteAtomicProperty}}{\emph{element='C'}, \emph{propertyname='m'}}{}
Get the absolute property value with propertyname for the given atom.

\end{fulllineitems}

\index{GetRelativeAtomicProperty() (in module AtomProperty)}

\begin{fulllineitems}
\phantomsection\label{reference/AtomProperty:AtomProperty.GetRelativeAtomicProperty}\pysiglinewithargsret{\sphinxcode{AtomProperty.}\sphinxbfcode{GetRelativeAtomicProperty}}{\emph{element='C'}, \emph{propertyname='m'}}{}
Get the absolute property value with propertyname for the given atom.

\end{fulllineitems}



\subsection{AtomTypes module}
\label{reference/AtomTypes:atomtypes-module}\label{reference/AtomTypes::doc}\label{reference/AtomTypes:module-AtomTypes}\index{AtomTypes (module)}
You can freely use and distribute it. If you hava

any problem, you could contact with us timely!

Authors: Zhijiang Yao and Dongsheng Cao.

Date: 2016.06.04

Email: \href{mailto:gadsby@163.com}{gadsby@163.com}

contains SMARTS definitions and calculators for EState atom types

defined in: Hall and Kier JCICS \_35\_ 1039-1045 (1995)  Table 1
\index{BuildPatts() (in module AtomTypes)}

\begin{fulllineitems}
\phantomsection\label{reference/AtomTypes:AtomTypes.BuildPatts}\pysiglinewithargsret{\sphinxcode{AtomTypes.}\sphinxbfcode{BuildPatts}}{\emph{rawV=None}}{}
Internal Use Only

\end{fulllineitems}

\index{GetAtomLabel() (in module AtomTypes)}

\begin{fulllineitems}
\phantomsection\label{reference/AtomTypes:AtomTypes.GetAtomLabel}\pysiglinewithargsret{\sphinxcode{AtomTypes.}\sphinxbfcode{GetAtomLabel}}{\emph{mol}}{}
Obtain the atom index in a molecule for the above given atom types

\end{fulllineitems}

\index{TypeAtoms() (in module AtomTypes)}

\begin{fulllineitems}
\phantomsection\label{reference/AtomTypes:AtomTypes.TypeAtoms}\pysiglinewithargsret{\sphinxcode{AtomTypes.}\sphinxbfcode{TypeAtoms}}{\emph{mol}}{}
assigns each atom in a molecule to an EState type

\textbf{Returns:}
\begin{quote}

list of tuples (atoms can possibly match multiple patterns) with atom types
\end{quote}

\end{fulllineitems}



\subsection{basak module}
\label{reference/basak:module-basak}\label{reference/basak::doc}\label{reference/basak:basak-module}\index{basak (module)}
topological structure. You can get 21 molecular connectivity descriptors.

You can freely use and distribute it. If you hava  any problem, you could

contact with us timely!

Authors: Zhijiang Yao and Dongsheng Cao.

Date: 2016.06.04

Email: \href{mailto:gadsby@163}{gadsby@163}.


\bigskip\hrule{}\bigskip

\index{CalculateBasakCIC0() (in module basak)}

\begin{fulllineitems}
\phantomsection\label{reference/basak:basak.CalculateBasakCIC0}\pysiglinewithargsret{\sphinxcode{basak.}\sphinxbfcode{CalculateBasakCIC0}}{\emph{mol}}{}
Obtain the complementary information content with order 0

proposed by Basak

\end{fulllineitems}

\index{CalculateBasakCIC1() (in module basak)}

\begin{fulllineitems}
\phantomsection\label{reference/basak:basak.CalculateBasakCIC1}\pysiglinewithargsret{\sphinxcode{basak.}\sphinxbfcode{CalculateBasakCIC1}}{\emph{mol}}{}
Obtain the complementary information content with order 1 proposed

by Basak.

\end{fulllineitems}

\index{CalculateBasakCIC2() (in module basak)}

\begin{fulllineitems}
\phantomsection\label{reference/basak:basak.CalculateBasakCIC2}\pysiglinewithargsret{\sphinxcode{basak.}\sphinxbfcode{CalculateBasakCIC2}}{\emph{mol}}{}
Obtain the complementary information content with order 2 proposed

by Basak.

\end{fulllineitems}

\index{CalculateBasakCIC3() (in module basak)}

\begin{fulllineitems}
\phantomsection\label{reference/basak:basak.CalculateBasakCIC3}\pysiglinewithargsret{\sphinxcode{basak.}\sphinxbfcode{CalculateBasakCIC3}}{\emph{mol}}{}
Obtain the complementary information content with order 3 proposed

by Basak.

\end{fulllineitems}

\index{CalculateBasakCIC4() (in module basak)}

\begin{fulllineitems}
\phantomsection\label{reference/basak:basak.CalculateBasakCIC4}\pysiglinewithargsret{\sphinxcode{basak.}\sphinxbfcode{CalculateBasakCIC4}}{\emph{mol}}{}
Obtain the complementary information content with order 4 proposed

by Basak.

\end{fulllineitems}

\index{CalculateBasakCIC5() (in module basak)}

\begin{fulllineitems}
\phantomsection\label{reference/basak:basak.CalculateBasakCIC5}\pysiglinewithargsret{\sphinxcode{basak.}\sphinxbfcode{CalculateBasakCIC5}}{\emph{mol}}{}
Obtain the complementary information content with order 5 proposed

by Basak.

\end{fulllineitems}

\index{CalculateBasakCIC6() (in module basak)}

\begin{fulllineitems}
\phantomsection\label{reference/basak:basak.CalculateBasakCIC6}\pysiglinewithargsret{\sphinxcode{basak.}\sphinxbfcode{CalculateBasakCIC6}}{\emph{mol}}{}
Obtain the complementary information content with order 6 proposed

by Basak.

\end{fulllineitems}

\index{CalculateBasakIC0() (in module basak)}

\begin{fulllineitems}
\phantomsection\label{reference/basak:basak.CalculateBasakIC0}\pysiglinewithargsret{\sphinxcode{basak.}\sphinxbfcode{CalculateBasakIC0}}{\emph{mol}}{}
Obtain the information content with order 0 proposed by Basak

\end{fulllineitems}

\index{CalculateBasakIC1() (in module basak)}

\begin{fulllineitems}
\phantomsection\label{reference/basak:basak.CalculateBasakIC1}\pysiglinewithargsret{\sphinxcode{basak.}\sphinxbfcode{CalculateBasakIC1}}{\emph{mol}}{}
Obtain the information content with order 1 proposed by Basak

\end{fulllineitems}

\index{CalculateBasakIC2() (in module basak)}

\begin{fulllineitems}
\phantomsection\label{reference/basak:basak.CalculateBasakIC2}\pysiglinewithargsret{\sphinxcode{basak.}\sphinxbfcode{CalculateBasakIC2}}{\emph{mol}}{}
Obtain the information content with order 2 proposed by Basak

\end{fulllineitems}

\index{CalculateBasakIC3() (in module basak)}

\begin{fulllineitems}
\phantomsection\label{reference/basak:basak.CalculateBasakIC3}\pysiglinewithargsret{\sphinxcode{basak.}\sphinxbfcode{CalculateBasakIC3}}{\emph{mol}}{}
Obtain the information content with order 3 proposed by Basak

\end{fulllineitems}

\index{CalculateBasakIC4() (in module basak)}

\begin{fulllineitems}
\phantomsection\label{reference/basak:basak.CalculateBasakIC4}\pysiglinewithargsret{\sphinxcode{basak.}\sphinxbfcode{CalculateBasakIC4}}{\emph{mol}}{}
Obtain the information content with order 4 proposed by Basak

\end{fulllineitems}

\index{CalculateBasakIC5() (in module basak)}

\begin{fulllineitems}
\phantomsection\label{reference/basak:basak.CalculateBasakIC5}\pysiglinewithargsret{\sphinxcode{basak.}\sphinxbfcode{CalculateBasakIC5}}{\emph{mol}}{}
Obtain the information content with order 5 proposed by Basak

\end{fulllineitems}

\index{CalculateBasakIC6() (in module basak)}

\begin{fulllineitems}
\phantomsection\label{reference/basak:basak.CalculateBasakIC6}\pysiglinewithargsret{\sphinxcode{basak.}\sphinxbfcode{CalculateBasakIC6}}{\emph{mol}}{}
Obtain the information content with order 6 proposed by Basak

\end{fulllineitems}

\index{CalculateBasakSIC0() (in module basak)}

\begin{fulllineitems}
\phantomsection\label{reference/basak:basak.CalculateBasakSIC0}\pysiglinewithargsret{\sphinxcode{basak.}\sphinxbfcode{CalculateBasakSIC0}}{\emph{mol}}{}
Obtain the structural information content with order 0

proposed by Basak

\end{fulllineitems}

\index{CalculateBasakSIC1() (in module basak)}

\begin{fulllineitems}
\phantomsection\label{reference/basak:basak.CalculateBasakSIC1}\pysiglinewithargsret{\sphinxcode{basak.}\sphinxbfcode{CalculateBasakSIC1}}{\emph{mol}}{}
Obtain the structural information content with order 1

proposed by Basak.

\end{fulllineitems}

\index{CalculateBasakSIC2() (in module basak)}

\begin{fulllineitems}
\phantomsection\label{reference/basak:basak.CalculateBasakSIC2}\pysiglinewithargsret{\sphinxcode{basak.}\sphinxbfcode{CalculateBasakSIC2}}{\emph{mol}}{}
Obtain the structural information content with order 2 proposed

by Basak.

\end{fulllineitems}

\index{CalculateBasakSIC3() (in module basak)}

\begin{fulllineitems}
\phantomsection\label{reference/basak:basak.CalculateBasakSIC3}\pysiglinewithargsret{\sphinxcode{basak.}\sphinxbfcode{CalculateBasakSIC3}}{\emph{mol}}{}
Obtain the structural information content with order 3 proposed

by Basak.

\end{fulllineitems}

\index{CalculateBasakSIC4() (in module basak)}

\begin{fulllineitems}
\phantomsection\label{reference/basak:basak.CalculateBasakSIC4}\pysiglinewithargsret{\sphinxcode{basak.}\sphinxbfcode{CalculateBasakSIC4}}{\emph{mol}}{}
Obtain the structural information content with order 4 proposed

by Basak.

\end{fulllineitems}

\index{CalculateBasakSIC5() (in module basak)}

\begin{fulllineitems}
\phantomsection\label{reference/basak:basak.CalculateBasakSIC5}\pysiglinewithargsret{\sphinxcode{basak.}\sphinxbfcode{CalculateBasakSIC5}}{\emph{mol}}{}
Obtain the structural information content with order 5 proposed

by Basak.

\end{fulllineitems}

\index{CalculateBasakSIC6() (in module basak)}

\begin{fulllineitems}
\phantomsection\label{reference/basak:basak.CalculateBasakSIC6}\pysiglinewithargsret{\sphinxcode{basak.}\sphinxbfcode{CalculateBasakSIC6}}{\emph{mol}}{}
Obtain the structural information content with order 6 proposed

by Basak.

\end{fulllineitems}

\index{Getbasak() (in module basak)}

\begin{fulllineitems}
\phantomsection\label{reference/basak:basak.Getbasak}\pysiglinewithargsret{\sphinxcode{basak.}\sphinxbfcode{Getbasak}}{\emph{mol}}{}
\end{fulllineitems}



\subsection{bcut module}
\label{reference/bcut:bcut-module}\label{reference/bcut:module-bcut}\label{reference/bcut::doc}\index{bcut (module)}

\bigskip\hrule{}\bigskip


The calculation of Burden eigvenvalue descriptors. You can get 64

molecular decriptors. You can freely use and distribute it. If you hava

any problem, you could contact with us timely!

Authors: Zhijiang Yao and Dongsheng Cao.

Date: 2016.06.04

Email: \href{mailto:gadsby@163.com}{gadsby@163.com}


\bigskip\hrule{}\bigskip

\index{CalculateBurdenElectronegativity() (in module bcut)}

\begin{fulllineitems}
\phantomsection\label{reference/bcut:bcut.CalculateBurdenElectronegativity}\pysiglinewithargsret{\sphinxcode{bcut.}\sphinxbfcode{CalculateBurdenElectronegativity}}{\emph{mol}}{}
Calculate Burden descriptors based on atomic electronegativity.

\end{fulllineitems}

\index{CalculateBurdenMass() (in module bcut)}

\begin{fulllineitems}
\phantomsection\label{reference/bcut:bcut.CalculateBurdenMass}\pysiglinewithargsret{\sphinxcode{bcut.}\sphinxbfcode{CalculateBurdenMass}}{\emph{mol}}{}
Calculate Burden descriptors based on atomic mass.

\end{fulllineitems}

\index{CalculateBurdenPolarizability() (in module bcut)}

\begin{fulllineitems}
\phantomsection\label{reference/bcut:bcut.CalculateBurdenPolarizability}\pysiglinewithargsret{\sphinxcode{bcut.}\sphinxbfcode{CalculateBurdenPolarizability}}{\emph{mol}}{}
Calculate Burden descriptors based on polarizability.

\end{fulllineitems}

\index{CalculateBurdenVDW() (in module bcut)}

\begin{fulllineitems}
\phantomsection\label{reference/bcut:bcut.CalculateBurdenVDW}\pysiglinewithargsret{\sphinxcode{bcut.}\sphinxbfcode{CalculateBurdenVDW}}{\emph{mol}}{}
Calculate Burden descriptors based on atomic vloumes

\end{fulllineitems}

\index{GetBurden() (in module bcut)}

\begin{fulllineitems}
\phantomsection\label{reference/bcut:bcut.GetBurden}\pysiglinewithargsret{\sphinxcode{bcut.}\sphinxbfcode{GetBurden}}{\emph{mol}}{}
Calculate all 64 Burden descriptors

\end{fulllineitems}



\subsection{cats2d module}
\label{reference/cats2d::doc}\label{reference/cats2d:module-cats2d}\label{reference/cats2d:cats2d-module}\index{cats2d (module)}
\# CATS2D  Potential Pharmacophore Point (PPP) definitions as describes in
\# Pharmacophores and Pharmacophore Searches 2006 (Eds. T. Langer and R.D. Hoffmann), Chapter 3:
\# Alignment-free Pharmacophore Patterns - A Correlattion-vector Approach.
\# The last lipophilic pattern on page 55 of the book is realized as a graph search and not
\# as a SMARTS search. Therefore, the list contains only two lipophilic SMARTS patterns.
\# The format is tab separated and contains in the first column the PPP type (D = H-bond donor,
\# A = H-bond acceptor, P = positive, N = negative, L = lipophilic). The second column of each entry
\# contains the SMARTS pattern(s). The last entry is a description of the molecular feature

D       {[}OH{]}    Oxygen atom of an OH group
D       {[}\#7H,\#7H2{]}      Nitrogen atom of an NH or NH2 group
A       {[}O{]}     Oxygen atom
A       {[}\#7H0{]}  Nitrogen atom not adjacent to a hydrogen atom
P       {[}\emph{+{]}    atom with a positive charge
P       {[}\#7H2{]}  Nitrogen atom of an NH2 group
N       {[}}-{]}    Atom with a negative charge
N       {[}C\&D2\&\$(C(=O)O),P\&D2\&\$(P(=O)O),S\&D2\&\$(S(=O)O){]}  Carbon, sulfur or phosphorus atom of a COOH, SOOH or POOH group. This pattern is realized by an graph algorithm
L       {[}Cl,Br,I{]}       Chlorine, bromine, or iodine atom
L       {[}S;D2;\$(S(C)(C)){]}       Sulfur atom adjacent to exactly two carbon atoms

Created on Thu Sep  1 20:13:38 2016

Authors: Zhijiang Yao and Dongsheng Cao.

Email: \href{mailto:gadsby@163.com}{gadsby@163.com} and \href{mailto:oriental-cds@163.com}{oriental-cds@163.com}
\index{AssignAtomType() (in module cats2d)}

\begin{fulllineitems}
\phantomsection\label{reference/cats2d:cats2d.AssignAtomType}\pysiglinewithargsret{\sphinxcode{cats2d.}\sphinxbfcode{AssignAtomType}}{\emph{mol}}{}
Assign the atoms in the mol object into each of the PPP type

according to PPP list definition.

\end{fulllineitems}

\index{CATS2D() (in module cats2d)}

\begin{fulllineitems}
\phantomsection\label{reference/cats2d:cats2d.CATS2D}\pysiglinewithargsret{\sphinxcode{cats2d.}\sphinxbfcode{CATS2D}}{\emph{mol}, \emph{PathLength=10}, \emph{scale=3}}{}
The main program for calculating the CATS descriptors.

CATS: chemically advanced template serach

----\textgreater{} CATS\_DA0 ....

Usage:
\begin{quote}

result=CATS2D(mol,PathLength = 10,scale = 1)

Input: mol is a molecule object.
\begin{quote}

PathLength is the max topological distance between two atoms.

scale is the normalization method (descriptor scaling method)

scale = 1 indicates that no normalization. That is to say: the

values of the vector represent raw counts (``counts'').

scale = 2 indicates that division by the number of non-hydrogen

atoms (heavy atoms) in the molecule.

scale = 3 indicates that division of each of 15 possible PPP pairs

by the added occurrences of the two respective PPPs.
\end{quote}

Output: result is a dict format with the definitions of each descritor.
\end{quote}

\end{fulllineitems}

\index{ContructLFromGraphSearch() (in module cats2d)}

\begin{fulllineitems}
\phantomsection\label{reference/cats2d:cats2d.ContructLFromGraphSearch}\pysiglinewithargsret{\sphinxcode{cats2d.}\sphinxbfcode{ContructLFromGraphSearch}}{\emph{mol}}{}
The last lipophilic pattern on page 55 of the book is realized as a graph 
search and not as a SMARTS search.

``L'' carbon atom adjacent only to carbon atoms.

\end{fulllineitems}

\index{FormCATSDict() (in module cats2d)}

\begin{fulllineitems}
\phantomsection\label{reference/cats2d:cats2d.FormCATSDict}\pysiglinewithargsret{\sphinxcode{cats2d.}\sphinxbfcode{FormCATSDict}}{\emph{AtomDict}, \emph{CATSLabel}}{}
Construt the CATS dict.

\end{fulllineitems}

\index{FormCATSLabel() (in module cats2d)}

\begin{fulllineitems}
\phantomsection\label{reference/cats2d:cats2d.FormCATSLabel}\pysiglinewithargsret{\sphinxcode{cats2d.}\sphinxbfcode{FormCATSLabel}}{\emph{PathLength=10}}{}
Construct the CATS label such as AA0, AA1,....AP3,.......

The result is a list format.

A   acceptor;
P   positive;
N   negative;
L   lipophilic;
D   donor;
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#

\end{fulllineitems}

\index{MatchAtomType() (in module cats2d)}

\begin{fulllineitems}
\phantomsection\label{reference/cats2d:cats2d.MatchAtomType}\pysiglinewithargsret{\sphinxcode{cats2d.}\sphinxbfcode{MatchAtomType}}{\emph{IndexList}, \emph{AtomTypeDict}}{}
Mapping two atoms with a certain distance into their atom types

such as AA,AL, DP,LD etc.

\end{fulllineitems}



\subsection{charge module}
\label{reference/charge:module-charge}\label{reference/charge:charge-module}\label{reference/charge::doc}\index{charge (module)}

\bigskip\hrule{}\bigskip


The calculation of Charge descriptors based on Gasteiger/Marseli partial

charges(25). You can freely use and distribute it. If you hava  any problem,

you could contact with us timely!

Authors: Zhijiang Yao and Dongsheng Cao.

Date: 2016.06.04

Email: \href{mailto:gadsby@163.com}{gadsby@163.com}


\bigskip\hrule{}\bigskip

\index{CalculateAllMaxNCharge() (in module charge)}

\begin{fulllineitems}
\phantomsection\label{reference/charge:charge.CalculateAllMaxNCharge}\pysiglinewithargsret{\sphinxcode{charge.}\sphinxbfcode{CalculateAllMaxNCharge}}{\emph{mol}}{}
Most negative charge on all atoms

--\textgreater{}Qmin

Usage:
\begin{quote}

result=CalculateAllMaxNCharge(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateAllMaxPCharge() (in module charge)}

\begin{fulllineitems}
\phantomsection\label{reference/charge:charge.CalculateAllMaxPCharge}\pysiglinewithargsret{\sphinxcode{charge.}\sphinxbfcode{CalculateAllMaxPCharge}}{\emph{mol}}{}
Most positive charge on ALL atoms

--\textgreater{}Qmax

Usage:
\begin{quote}

result=CalculateAllMaxPCharge(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateAllSumSquareCharge() (in module charge)}

\begin{fulllineitems}
\phantomsection\label{reference/charge:charge.CalculateAllSumSquareCharge}\pysiglinewithargsret{\sphinxcode{charge.}\sphinxbfcode{CalculateAllSumSquareCharge}}{\emph{mol}}{}
The sum of square charges on all atoms

--\textgreater{}Qass

Usage:
\begin{quote}

result=CalculateAllSumSquareCharge(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateCMaxNCharge() (in module charge)}

\begin{fulllineitems}
\phantomsection\label{reference/charge:charge.CalculateCMaxNCharge}\pysiglinewithargsret{\sphinxcode{charge.}\sphinxbfcode{CalculateCMaxNCharge}}{\emph{mol}}{}
Most negative charge on C atoms

--\textgreater{}QCmin

Usage:
\begin{quote}

result=CalculateCMaxNCharge(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateCMaxPCharge() (in module charge)}

\begin{fulllineitems}
\phantomsection\label{reference/charge:charge.CalculateCMaxPCharge}\pysiglinewithargsret{\sphinxcode{charge.}\sphinxbfcode{CalculateCMaxPCharge}}{\emph{mol}}{}
Most positive charge on C atoms

--\textgreater{}QCmax

Usage:
\begin{quote}

result=CalculateCMaxPCharge(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateCSumSquareCharge() (in module charge)}

\begin{fulllineitems}
\phantomsection\label{reference/charge:charge.CalculateCSumSquareCharge}\pysiglinewithargsret{\sphinxcode{charge.}\sphinxbfcode{CalculateCSumSquareCharge}}{\emph{mol}}{}
The sum of square charges on all C atoms

--\textgreater{}QCss

Usage:
\begin{quote}

result=CalculateCSumSquareCharge(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateHMaxNCharge() (in module charge)}

\begin{fulllineitems}
\phantomsection\label{reference/charge:charge.CalculateHMaxNCharge}\pysiglinewithargsret{\sphinxcode{charge.}\sphinxbfcode{CalculateHMaxNCharge}}{\emph{mol}}{}
Most negative charge on H atoms

--\textgreater{}QHmin

Usage:
\begin{quote}

result=CalculateHMaxNCharge(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateHMaxPCharge() (in module charge)}

\begin{fulllineitems}
\phantomsection\label{reference/charge:charge.CalculateHMaxPCharge}\pysiglinewithargsret{\sphinxcode{charge.}\sphinxbfcode{CalculateHMaxPCharge}}{\emph{mol}}{}
Most positive charge on H atoms

--\textgreater{}QHmax

Usage:
\begin{quote}

result=CalculateHMaxPCharge(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateHSumSquareCharge() (in module charge)}

\begin{fulllineitems}
\phantomsection\label{reference/charge:charge.CalculateHSumSquareCharge}\pysiglinewithargsret{\sphinxcode{charge.}\sphinxbfcode{CalculateHSumSquareCharge}}{\emph{mol}}{}
The sum of square charges on all H atoms

--\textgreater{}QHss

Usage:
\begin{quote}

result=CalculateHSumSquareCharge(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateLocalDipoleIndex() (in module charge)}

\begin{fulllineitems}
\phantomsection\label{reference/charge:charge.CalculateLocalDipoleIndex}\pysiglinewithargsret{\sphinxcode{charge.}\sphinxbfcode{CalculateLocalDipoleIndex}}{\emph{mol}}{}
Calculation of local dipole index (D)

--\textgreater{}LDI

Usage:
\begin{quote}

result=CalculateLocalDipoleIndex(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateMeanAbsoulteCharge() (in module charge)}

\begin{fulllineitems}
\phantomsection\label{reference/charge:charge.CalculateMeanAbsoulteCharge}\pysiglinewithargsret{\sphinxcode{charge.}\sphinxbfcode{CalculateMeanAbsoulteCharge}}{\emph{mol}}{}
The average absolute charge

--\textgreater{}Mac

Usage:
\begin{quote}

result=CalculateMeanAbsoulteCharge(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateMeanNCharge() (in module charge)}

\begin{fulllineitems}
\phantomsection\label{reference/charge:charge.CalculateMeanNCharge}\pysiglinewithargsret{\sphinxcode{charge.}\sphinxbfcode{CalculateMeanNCharge}}{\emph{mol}}{}
The average negative charge

--\textgreater{}Mnc

Usage:
\begin{quote}

result=CalculateMeanNCharge(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateMeanPCharge() (in module charge)}

\begin{fulllineitems}
\phantomsection\label{reference/charge:charge.CalculateMeanPCharge}\pysiglinewithargsret{\sphinxcode{charge.}\sphinxbfcode{CalculateMeanPCharge}}{\emph{mol}}{}
The average postive charge

--\textgreater{}Mpc

Usage:
\begin{quote}

result=CalculateMeanPCharge(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateNMaxNCharge() (in module charge)}

\begin{fulllineitems}
\phantomsection\label{reference/charge:charge.CalculateNMaxNCharge}\pysiglinewithargsret{\sphinxcode{charge.}\sphinxbfcode{CalculateNMaxNCharge}}{\emph{mol}}{}
Most negative charge on N atoms

--\textgreater{}QNmin

Usage:
\begin{quote}

result=CalculateNMaxNCharge(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateNMaxPCharge() (in module charge)}

\begin{fulllineitems}
\phantomsection\label{reference/charge:charge.CalculateNMaxPCharge}\pysiglinewithargsret{\sphinxcode{charge.}\sphinxbfcode{CalculateNMaxPCharge}}{\emph{mol}}{}
Most positive charge on N atoms

--\textgreater{}QNmax

Usage:
\begin{quote}

result=CalculateNMaxPCharge(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateNSumSquareCharge() (in module charge)}

\begin{fulllineitems}
\phantomsection\label{reference/charge:charge.CalculateNSumSquareCharge}\pysiglinewithargsret{\sphinxcode{charge.}\sphinxbfcode{CalculateNSumSquareCharge}}{\emph{mol}}{}
The sum of square charges on all N atoms

--\textgreater{}QNss

Usage:
\begin{quote}

result=CalculateNSumSquareCharge(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateOMaxNCharge() (in module charge)}

\begin{fulllineitems}
\phantomsection\label{reference/charge:charge.CalculateOMaxNCharge}\pysiglinewithargsret{\sphinxcode{charge.}\sphinxbfcode{CalculateOMaxNCharge}}{\emph{mol}}{}
Most negative charge on O atoms

--\textgreater{}QOmin

Usage:
\begin{quote}

result=CalculateOMaxNCharge(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateOMaxPCharge() (in module charge)}

\begin{fulllineitems}
\phantomsection\label{reference/charge:charge.CalculateOMaxPCharge}\pysiglinewithargsret{\sphinxcode{charge.}\sphinxbfcode{CalculateOMaxPCharge}}{\emph{mol}}{}
Most positive charge on O atoms

--\textgreater{}QOmax

Usage:
\begin{quote}

result=CalculateOMaxPCharge(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateOSumSquareCharge() (in module charge)}

\begin{fulllineitems}
\phantomsection\label{reference/charge:charge.CalculateOSumSquareCharge}\pysiglinewithargsret{\sphinxcode{charge.}\sphinxbfcode{CalculateOSumSquareCharge}}{\emph{mol}}{}
The sum of square charges on all O atoms

--\textgreater{}QOss

Usage:
\begin{quote}

result=CalculateOSumSquareCharge(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateRelativeNCharge() (in module charge)}

\begin{fulllineitems}
\phantomsection\label{reference/charge:charge.CalculateRelativeNCharge}\pysiglinewithargsret{\sphinxcode{charge.}\sphinxbfcode{CalculateRelativeNCharge}}{\emph{mol}}{}
The partial charge of the most negative atom divided

by the total negative charge.

--\textgreater{}Rnc

Usage:
\begin{quote}

result=CalculateRelativeNCharge(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateRelativePCharge() (in module charge)}

\begin{fulllineitems}
\phantomsection\label{reference/charge:charge.CalculateRelativePCharge}\pysiglinewithargsret{\sphinxcode{charge.}\sphinxbfcode{CalculateRelativePCharge}}{\emph{mol}}{}
The partial charge of the most positive atom divided by

the total positive charge.

--\textgreater{}Rpc

Usage:
\begin{quote}

result=CalculateRelativePCharge(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateSubmolPolarityPara() (in module charge)}

\begin{fulllineitems}
\phantomsection\label{reference/charge:charge.CalculateSubmolPolarityPara}\pysiglinewithargsret{\sphinxcode{charge.}\sphinxbfcode{CalculateSubmolPolarityPara}}{\emph{mol}}{}
Calculation of submolecular polarity parameter(SPP)

--\textgreater{}SPP

Usage:
\begin{quote}

result=CalculateSubmolPolarityPara(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateTotalAbsoulteCharge() (in module charge)}

\begin{fulllineitems}
\phantomsection\label{reference/charge:charge.CalculateTotalAbsoulteCharge}\pysiglinewithargsret{\sphinxcode{charge.}\sphinxbfcode{CalculateTotalAbsoulteCharge}}{\emph{mol}}{}
The total absolute charge

--\textgreater{}Tac

Usage:
\begin{quote}

result=CalculateTotalAbsoulteCharge(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateTotalNCharge() (in module charge)}

\begin{fulllineitems}
\phantomsection\label{reference/charge:charge.CalculateTotalNCharge}\pysiglinewithargsret{\sphinxcode{charge.}\sphinxbfcode{CalculateTotalNCharge}}{\emph{mol}}{}
The total negative charge

--\textgreater{}Tnc

Usage:
\begin{quote}

result=CalculateTotalNCharge(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateTotalPCharge() (in module charge)}

\begin{fulllineitems}
\phantomsection\label{reference/charge:charge.CalculateTotalPCharge}\pysiglinewithargsret{\sphinxcode{charge.}\sphinxbfcode{CalculateTotalPCharge}}{\emph{mol}}{}
The total postive charge

--\textgreater{}Tpc

Usage:
\begin{quote}

result=CalculateTotalPCharge(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{GetCharge() (in module charge)}

\begin{fulllineitems}
\phantomsection\label{reference/charge:charge.GetCharge}\pysiglinewithargsret{\sphinxcode{charge.}\sphinxbfcode{GetCharge}}{\emph{mol}}{}
Get the dictionary of constitutional descriptors for given moelcule mol

Usage:
\begin{quote}

result=GetCharge(mol)

Input: mol is a molecule object.

Output: result is a dict form containing all charge descriptors.
\end{quote}

\end{fulllineitems}



\subsection{connectivity module}
\label{reference/connectivity::doc}\label{reference/connectivity:connectivity-module}\label{reference/connectivity:module-connectivity}\index{connectivity (module)}
structure. You can get 44 molecular connectivity descriptors. You can freely

use and distribute it. If you hava  any problem, you could contact with us timely!

Authors: Zhijiang Yao and Dongsheng Cao.

Date: 2016.06.04

Email: \href{mailto:gadsby@163.com}{gadsby@163.com} and \href{mailto:oriental-cds@163.com}{oriental-cds@163.com}


\bigskip\hrule{}\bigskip

\index{CalculateChi0() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateChi0}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateChi0}}{\emph{mol}}{}
Calculation of molecular connectivity chi index for path order 0

----\textgreater{}Chi0

Usage:
\begin{quote}

result=CalculateChi0(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateChi1() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateChi1}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateChi1}}{\emph{mol}}{}
Calculation of molecular connectivity chi index for path order 1

(i.e.,Radich)

----\textgreater{}Chi1

Usage:
\begin{quote}

result=CalculateChi1(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateChi10p() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateChi10p}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateChi10p}}{\emph{mol}}{}
Calculation of molecular connectivity chi index for path order 10

----\textgreater{}Chi10

Usage:
\begin{quote}

result=CalculateChi10p(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateChi2() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateChi2}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateChi2}}{\emph{mol}}{}
Calculation of molecular connectivity chi index for path order 2

----\textgreater{}Chi2

Usage:
\begin{quote}

result=CalculateChi2(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateChi3c() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateChi3c}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateChi3c}}{\emph{mol}}{}
Calculation of molecular connectivity chi index for cluster

----\textgreater{}Chi3c

Usage:
\begin{quote}

result=CalculateChi3c(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateChi3ch() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateChi3ch}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateChi3ch}}{\emph{mol}}{}
Calculation of molecular connectivity chi index for cycles of 3

----\textgreater{}Chi3ch

Usage:
\begin{quote}

result=CalculateChi3ch(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateChi3p() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateChi3p}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateChi3p}}{\emph{mol}}{}
Calculation of molecular connectivity chi index for path order 3

----\textgreater{}Chi3

Usage:
\begin{quote}

result=CalculateChi3p(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateChi4c() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateChi4c}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateChi4c}}{\emph{mol}}{}
Calculation of molecular connectivity chi index for cluster

----\textgreater{}Chi4c

Usage:
\begin{quote}

result=CalculateChi4c(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateChi4ch() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateChi4ch}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateChi4ch}}{\emph{mol}}{}
Calculation of molecular connectivity chi index for cycles of 4

----\textgreater{}Chi4ch

Usage:
\begin{quote}

result=CalculateChi4ch(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateChi4p() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateChi4p}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateChi4p}}{\emph{mol}}{}
Calculation of molecular connectivity chi index for path order 4

----\textgreater{}Chi4

Usage:
\begin{quote}

result=CalculateChi4p(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateChi4pc() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateChi4pc}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateChi4pc}}{\emph{mol}}{}
Calculation of molecular connectivity chi index for path/cluster

----\textgreater{}Chi4pc

Usage:
\begin{quote}

result=CalculateChi4pc(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateChi5ch() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateChi5ch}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateChi5ch}}{\emph{mol}}{}
Calculation of molecular connectivity chi index for cycles of 5

----\textgreater{}Chi5ch

Usage:
\begin{quote}

result=CalculateChi5ch(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateChi5p() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateChi5p}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateChi5p}}{\emph{mol}}{}
Calculation of molecular connectivity chi index for path order 5

----\textgreater{}Chi5

Usage:
\begin{quote}

result=CalculateChi5p(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateChi6ch() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateChi6ch}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateChi6ch}}{\emph{mol}}{}
Calculation of molecular connectivity chi index for cycles of 6

----\textgreater{}Chi6ch

Usage:
\begin{quote}

result=CalculateChi6ch(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateChi6p() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateChi6p}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateChi6p}}{\emph{mol}}{}
Calculation of molecular connectivity chi index for path order 6

----\textgreater{}Chi6

Usage:
\begin{quote}

result=CalculateChi6p(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateChi7p() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateChi7p}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateChi7p}}{\emph{mol}}{}
Calculation of molecular connectivity chi index for path order 7

----\textgreater{}Chi7

Usage:
\begin{quote}

result=CalculateChi7p(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateChi8p() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateChi8p}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateChi8p}}{\emph{mol}}{}
Calculation of molecular connectivity chi index for path order 8

----\textgreater{}Chi8

Usage:
\begin{quote}

result=CalculateChi8p(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateChi9p() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateChi9p}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateChi9p}}{\emph{mol}}{}
Calculation of molecular connectivity chi index for path order 9

----\textgreater{}Chi9

Usage:
\begin{quote}

result=CalculateChi9p(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateChiv0() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateChiv0}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateChiv0}}{\emph{mol}}{}
Calculation of valence molecular connectivity chi index for

path order 0

----\textgreater{}Chiv0

Usage:
\begin{quote}

result=CalculateChiv0(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateChiv1() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateChiv1}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateChiv1}}{\emph{mol}}{}
Calculation of valence molecular connectivity chi index for

path order 1

----\textgreater{}Chiv1

Usage:
\begin{quote}

result=CalculateChiv1(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateChiv10p() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateChiv10p}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateChiv10p}}{\emph{mol}}{}
Calculation of valence molecular connectivity chi index for

path order 10

----\textgreater{}Chiv10

Usage:
\begin{quote}

result=CalculateChiv10p(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateChiv2() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateChiv2}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateChiv2}}{\emph{mol}}{}
Calculation of valence molecular connectivity chi index for

path order 2

----\textgreater{}Chiv2

Usage:
\begin{quote}

result=CalculateChiv2(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateChiv3c() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateChiv3c}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateChiv3c}}{\emph{mol}}{}
Calculation of valence molecular connectivity chi index for cluster

----\textgreater{}Chiv3c

Usage:
\begin{quote}

result=CalculateChiv3c(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateChiv3ch() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateChiv3ch}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateChiv3ch}}{\emph{mol}}{}
Calculation of valence molecular connectivity chi index

for cycles of 3

----\textgreater{}Chiv3ch

Usage:
\begin{quote}

result=CalculateChiv3ch(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateChiv3p() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateChiv3p}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateChiv3p}}{\emph{mol}}{}
Calculation of valence molecular connectivity chi index for

path order 3

----\textgreater{}Chiv3

Usage:
\begin{quote}

result=CalculateChiv3p(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateChiv4c() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateChiv4c}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateChiv4c}}{\emph{mol}}{}
Calculation of valence molecular connectivity chi index for cluster

----\textgreater{}Chiv4c

Usage:
\begin{quote}

result=CalculateChiv4c(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateChiv4ch() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateChiv4ch}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateChiv4ch}}{\emph{mol}}{}
Calculation of valence molecular connectivity chi index for

cycles of 4

----\textgreater{}Chiv4ch

Usage:
\begin{quote}

result=CalculateChiv4ch(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateChiv4p() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateChiv4p}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateChiv4p}}{\emph{mol}}{}
Calculation of valence molecular connectivity chi index for

path order 4

----\textgreater{}Chiv4

Usage:
\begin{quote}

result=CalculateChiv4p(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateChiv4pc() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateChiv4pc}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateChiv4pc}}{\emph{mol}}{}
Calculation of valence molecular connectivity chi index for

path/cluster

----\textgreater{}Chiv4pc

Usage:
\begin{quote}

result=CalculateChiv4pc(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateChiv5ch() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateChiv5ch}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateChiv5ch}}{\emph{mol}}{}
Calculation of valence molecular connectivity chi index for

cycles of 5

----\textgreater{}Chiv5ch

Usage:
\begin{quote}

result=CalculateChiv5ch(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateChiv5p() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateChiv5p}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateChiv5p}}{\emph{mol}}{}
Calculation of valence molecular connectivity chi index for

path order 5

----\textgreater{}Chiv5

Usage:
\begin{quote}

result=CalculateChiv5p(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateChiv6ch() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateChiv6ch}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateChiv6ch}}{\emph{mol}}{}
Calculation of valence molecular connectivity chi index for

cycles of 6

----\textgreater{}Chiv6ch

Usage:
\begin{quote}

result=CalculateChiv6ch(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateChiv6p() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateChiv6p}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateChiv6p}}{\emph{mol}}{}
Calculation of valence molecular connectivity chi index for

path order 6

----\textgreater{}Chiv6

Usage:
\begin{quote}

result=CalculateChiv6p(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateChiv7p() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateChiv7p}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateChiv7p}}{\emph{mol}}{}
Calculation of valence molecular connectivity chi index for

path order 7

----\textgreater{}Chiv7

Usage:
\begin{quote}

result=CalculateChiv7p(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateChiv8p() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateChiv8p}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateChiv8p}}{\emph{mol}}{}
Calculation of valence molecular connectivity chi index for

path order 8

----\textgreater{}Chiv8

Usage:
\begin{quote}

result=CalculateChiv8p(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateChiv9p() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateChiv9p}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateChiv9p}}{\emph{mol}}{}
Calculation of valence molecular connectivity chi index for

path order 9

----\textgreater{}Chiv9

Usage:
\begin{quote}

result=CalculateChiv9p(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateDeltaChi0() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateDeltaChi0}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateDeltaChi0}}{\emph{mol}}{}
Calculation of the difference between chi0v and chi0

----\textgreater{}dchi0

Usage:
\begin{quote}

result=CalculateDeltaChi0(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateDeltaChi1() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateDeltaChi1}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateDeltaChi1}}{\emph{mol}}{}
Calculation of the difference between chi1v and chi1

----\textgreater{}dchi1

Usage:
\begin{quote}

result=CalculateDeltaChi1(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateDeltaChi2() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateDeltaChi2}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateDeltaChi2}}{\emph{mol}}{}
Calculation of the difference between chi2v and chi2

----\textgreater{}dchi2

Usage:
\begin{quote}

result=CalculateDeltaChi2(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateDeltaChi3() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateDeltaChi3}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateDeltaChi3}}{\emph{mol}}{}
Calculation of the difference between chi3v and chi3

----\textgreater{}dchi3

Usage:
\begin{quote}

result=CalculateDeltaChi3(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateDeltaChi3c4pc() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateDeltaChi3c4pc}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateDeltaChi3c4pc}}{\emph{mol}}{}
Calculation of the difference between chi3c and chi4pc

----\textgreater{}knotp

Usage:
\begin{quote}

result=CalculateDeltaChi3c4pc(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateDeltaChi4() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateDeltaChi4}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateDeltaChi4}}{\emph{mol}}{}
Calculation of the difference between chi4v and chi4

----\textgreater{}dchi4

Usage:
\begin{quote}

result=CalculateDeltaChi4(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateDeltaChiv3c4pc() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateDeltaChiv3c4pc}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateDeltaChiv3c4pc}}{\emph{mol}}{}
Calculation of the difference between chiv3c and chiv4pc

----\textgreater{}knotpv

Usage:
\begin{quote}

result=CalculateDeltaChiv3c4pc(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateMeanRandic() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.CalculateMeanRandic}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{CalculateMeanRandic}}{\emph{mol}}{}
Calculation of mean chi1 (Randic) connectivity index.

----\textgreater{}mchi1

Usage:
\begin{quote}

result=CalculateMeanRandic(mol)

Input: mol is a molecule object.

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{GetConnectivity() (in module connectivity)}

\begin{fulllineitems}
\phantomsection\label{reference/connectivity:connectivity.GetConnectivity}\pysiglinewithargsret{\sphinxcode{connectivity.}\sphinxbfcode{GetConnectivity}}{\emph{mol}}{}
Get the dictionary of connectivity descriptors for given moelcule mol

Usage:
\begin{quote}

result=GetConnectivity(mol)

Input: mol is a molecule object.

Output: result is a dict form containing all connectivity indices
\end{quote}

\end{fulllineitems}



\subsection{constitution module}
\label{reference/constitution:constitution-module}\label{reference/constitution:module-constitution}\label{reference/constitution::doc}\index{constitution (module)}
structure. You can get 30 molecular connectivity descriptors. You can freely

use and distribute it. If you hava  any problem, you could contact with us timely!

Authors: Zhijiang Yao and Dongsheng Cao.

Date: 2016.06.04

Email: \href{mailto:gadsby@163.com}{gadsby@163.com} and \href{mailto:oriental-cds@163.com}{oriental-cds@163.com}


\bigskip\hrule{}\bigskip

\index{CalculateAllAtomNumber() (in module constitution)}

\begin{fulllineitems}
\phantomsection\label{reference/constitution:constitution.CalculateAllAtomNumber}\pysiglinewithargsret{\sphinxcode{constitution.}\sphinxbfcode{CalculateAllAtomNumber}}{\emph{mol}}{}
Calculation of all atom counts in a molecule

----\textgreater{}nta

Usage:
\begin{quote}

result=CalculateAllAtomNumber(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateAromaticBondNumber() (in module constitution)}

\begin{fulllineitems}
\phantomsection\label{reference/constitution:constitution.CalculateAromaticBondNumber}\pysiglinewithargsret{\sphinxcode{constitution.}\sphinxbfcode{CalculateAromaticBondNumber}}{\emph{mol}}{}
Calculation of aromatic bond counts in a molecule

----\textgreater{}naro

Usage:
\begin{quote}

result=CalculateAromaticBondNumber(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateAverageMolWeight() (in module constitution)}

\begin{fulllineitems}
\phantomsection\label{reference/constitution:constitution.CalculateAverageMolWeight}\pysiglinewithargsret{\sphinxcode{constitution.}\sphinxbfcode{CalculateAverageMolWeight}}{\emph{mol}}{}
Calcualtion of average molecular weight

Note that not including H

----\textgreater{}AWeight

Usage:
\begin{quote}

result=CalculateAverageMolWeight(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateBromineNumber() (in module constitution)}

\begin{fulllineitems}
\phantomsection\label{reference/constitution:constitution.CalculateBromineNumber}\pysiglinewithargsret{\sphinxcode{constitution.}\sphinxbfcode{CalculateBromineNumber}}{\emph{mol}}{}
Calculation of Bromine counts in a molecule

----\textgreater{}ncobr

Usage:
\begin{quote}

result=CalculateBromineNumber(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateCarbonNumber() (in module constitution)}

\begin{fulllineitems}
\phantomsection\label{reference/constitution:constitution.CalculateCarbonNumber}\pysiglinewithargsret{\sphinxcode{constitution.}\sphinxbfcode{CalculateCarbonNumber}}{\emph{mol}}{}
Calculation of Carbon number in a molecule

----\textgreater{}ncarb

Usage:
\begin{quote}

result=CalculateCarbonNumber(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateChlorinNumber() (in module constitution)}

\begin{fulllineitems}
\phantomsection\label{reference/constitution:constitution.CalculateChlorinNumber}\pysiglinewithargsret{\sphinxcode{constitution.}\sphinxbfcode{CalculateChlorinNumber}}{\emph{mol}}{}
Calculation of Chlorin counts in a molecule

----\textgreater{}ncocl

Usage:
\begin{quote}

result=CalculateChlorinNumber(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateDoubleBondNumber() (in module constitution)}

\begin{fulllineitems}
\phantomsection\label{reference/constitution:constitution.CalculateDoubleBondNumber}\pysiglinewithargsret{\sphinxcode{constitution.}\sphinxbfcode{CalculateDoubleBondNumber}}{\emph{mol}}{}
Calculation of double bond counts in a molecule

----\textgreater{}ndb

Usage:
\begin{quote}

result=CalculateDoubleBondNumber(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateFluorinNumber() (in module constitution)}

\begin{fulllineitems}
\phantomsection\label{reference/constitution:constitution.CalculateFluorinNumber}\pysiglinewithargsret{\sphinxcode{constitution.}\sphinxbfcode{CalculateFluorinNumber}}{\emph{mol}}{}
Calculation of Fluorin counts in a molecule

----\textgreater{}ncof

Usage:
\begin{quote}

result=CalculateFluorinNumber(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateHacceptorNumber() (in module constitution)}

\begin{fulllineitems}
\phantomsection\label{reference/constitution:constitution.CalculateHacceptorNumber}\pysiglinewithargsret{\sphinxcode{constitution.}\sphinxbfcode{CalculateHacceptorNumber}}{\emph{mol}}{}
Calculation of Hydrogen bond acceptor counts in a molecule

----\textgreater{}naccr

Usage:
\begin{quote}

result=CalculateHacceptorNumber(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateHalogenNumber() (in module constitution)}

\begin{fulllineitems}
\phantomsection\label{reference/constitution:constitution.CalculateHalogenNumber}\pysiglinewithargsret{\sphinxcode{constitution.}\sphinxbfcode{CalculateHalogenNumber}}{\emph{mol}}{}
Calculation of Halogen counts in a molecule

----\textgreater{}nhal

Usage:
\begin{quote}

result=CalculateHalogenNumber(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateHdonorNumber() (in module constitution)}

\begin{fulllineitems}
\phantomsection\label{reference/constitution:constitution.CalculateHdonorNumber}\pysiglinewithargsret{\sphinxcode{constitution.}\sphinxbfcode{CalculateHdonorNumber}}{\emph{mol}}{}
Calculation of Hydrongen bond donor counts in a molecule

----\textgreater{}ndonr

Usage:
\begin{quote}

result=CalculateHdonorNumber(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateHeavyAtomNumber() (in module constitution)}

\begin{fulllineitems}
\phantomsection\label{reference/constitution:constitution.CalculateHeavyAtomNumber}\pysiglinewithargsret{\sphinxcode{constitution.}\sphinxbfcode{CalculateHeavyAtomNumber}}{\emph{mol}}{}
Calculation of Heavy atom counts in a molecule

----\textgreater{}nhev

Usage:
\begin{quote}

result=CalculateHeavyAtomNumber(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateHeteroNumber() (in module constitution)}

\begin{fulllineitems}
\phantomsection\label{reference/constitution:constitution.CalculateHeteroNumber}\pysiglinewithargsret{\sphinxcode{constitution.}\sphinxbfcode{CalculateHeteroNumber}}{\emph{mol}}{}
Calculation of Hetero counts in a molecule

----\textgreater{}nhet

Usage:
\begin{quote}

result=CalculateHeteroNumber(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateHydrogenNumber() (in module constitution)}

\begin{fulllineitems}
\phantomsection\label{reference/constitution:constitution.CalculateHydrogenNumber}\pysiglinewithargsret{\sphinxcode{constitution.}\sphinxbfcode{CalculateHydrogenNumber}}{\emph{mol}}{}
Calculation of Number of Hydrogen in a molecule

----\textgreater{}nhyd

Usage:
\begin{quote}

result=CalculateHydrogenNumber(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateIodineNumber() (in module constitution)}

\begin{fulllineitems}
\phantomsection\label{reference/constitution:constitution.CalculateIodineNumber}\pysiglinewithargsret{\sphinxcode{constitution.}\sphinxbfcode{CalculateIodineNumber}}{\emph{mol}}{}
Calculation of Iodine counts in a molecule

----\textgreater{}ncoi

Usage:
\begin{quote}

result=CalculateIodineNumber(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateMolWeight() (in module constitution)}

\begin{fulllineitems}
\phantomsection\label{reference/constitution:constitution.CalculateMolWeight}\pysiglinewithargsret{\sphinxcode{constitution.}\sphinxbfcode{CalculateMolWeight}}{\emph{mol}}{}
Calculation of molecular weight

Note that not including H

----\textgreater{}Weight

Usage:
\begin{quote}

result=CalculateMolWeight(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateNitrogenNumber() (in module constitution)}

\begin{fulllineitems}
\phantomsection\label{reference/constitution:constitution.CalculateNitrogenNumber}\pysiglinewithargsret{\sphinxcode{constitution.}\sphinxbfcode{CalculateNitrogenNumber}}{\emph{mol}}{}
Calculation of Nitrogen counts in a molecule

----\textgreater{}nnitro

Usage:
\begin{quote}

result=CalculateNitrogenNumber(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateOxygenNumber() (in module constitution)}

\begin{fulllineitems}
\phantomsection\label{reference/constitution:constitution.CalculateOxygenNumber}\pysiglinewithargsret{\sphinxcode{constitution.}\sphinxbfcode{CalculateOxygenNumber}}{\emph{mol}}{}
Calculation of Oxygen counts in a molecule

----\textgreater{}noxy

Usage:
\begin{quote}

result=CalculateOxygenNumber(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculatePath1() (in module constitution)}

\begin{fulllineitems}
\phantomsection\label{reference/constitution:constitution.CalculatePath1}\pysiglinewithargsret{\sphinxcode{constitution.}\sphinxbfcode{CalculatePath1}}{\emph{mol}}{}
\end{fulllineitems}

\index{CalculatePath2() (in module constitution)}

\begin{fulllineitems}
\phantomsection\label{reference/constitution:constitution.CalculatePath2}\pysiglinewithargsret{\sphinxcode{constitution.}\sphinxbfcode{CalculatePath2}}{\emph{mol}}{}
\end{fulllineitems}

\index{CalculatePath3() (in module constitution)}

\begin{fulllineitems}
\phantomsection\label{reference/constitution:constitution.CalculatePath3}\pysiglinewithargsret{\sphinxcode{constitution.}\sphinxbfcode{CalculatePath3}}{\emph{mol}}{}
\end{fulllineitems}

\index{CalculatePath4() (in module constitution)}

\begin{fulllineitems}
\phantomsection\label{reference/constitution:constitution.CalculatePath4}\pysiglinewithargsret{\sphinxcode{constitution.}\sphinxbfcode{CalculatePath4}}{\emph{mol}}{}
\end{fulllineitems}

\index{CalculatePath5() (in module constitution)}

\begin{fulllineitems}
\phantomsection\label{reference/constitution:constitution.CalculatePath5}\pysiglinewithargsret{\sphinxcode{constitution.}\sphinxbfcode{CalculatePath5}}{\emph{mol}}{}
\end{fulllineitems}

\index{CalculatePath6() (in module constitution)}

\begin{fulllineitems}
\phantomsection\label{reference/constitution:constitution.CalculatePath6}\pysiglinewithargsret{\sphinxcode{constitution.}\sphinxbfcode{CalculatePath6}}{\emph{mol}}{}
\end{fulllineitems}

\index{CalculatePhosphorNumber() (in module constitution)}

\begin{fulllineitems}
\phantomsection\label{reference/constitution:constitution.CalculatePhosphorNumber}\pysiglinewithargsret{\sphinxcode{constitution.}\sphinxbfcode{CalculatePhosphorNumber}}{\emph{mol}}{}
Calcualtion of Phosphor number in a molecule

----\textgreater{}nphos

Usage:
\begin{quote}

result=CalculatePhosphorNumber(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateRingNumber() (in module constitution)}

\begin{fulllineitems}
\phantomsection\label{reference/constitution:constitution.CalculateRingNumber}\pysiglinewithargsret{\sphinxcode{constitution.}\sphinxbfcode{CalculateRingNumber}}{\emph{mol}}{}
Calculation of ring counts in a molecule

----\textgreater{}nring

Usage:
\begin{quote}

result=CalculateRingNumber(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateRotationBondNumber() (in module constitution)}

\begin{fulllineitems}
\phantomsection\label{reference/constitution:constitution.CalculateRotationBondNumber}\pysiglinewithargsret{\sphinxcode{constitution.}\sphinxbfcode{CalculateRotationBondNumber}}{\emph{mol}}{}
Calculation of rotation bonds counts in a molecule

----\textgreater{}nrot

Note that this is the same as calculation of single bond

counts in a molecule.

Usage:
\begin{quote}

result=CalculateRotationBondNumber(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateSingleBondNumber() (in module constitution)}

\begin{fulllineitems}
\phantomsection\label{reference/constitution:constitution.CalculateSingleBondNumber}\pysiglinewithargsret{\sphinxcode{constitution.}\sphinxbfcode{CalculateSingleBondNumber}}{\emph{mol}}{}
Calculation of single bond counts in a molecule

----\textgreater{}nsb

Usage:
\begin{quote}

result=CalculateSingleBondNumber(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateSulfurNumber() (in module constitution)}

\begin{fulllineitems}
\phantomsection\label{reference/constitution:constitution.CalculateSulfurNumber}\pysiglinewithargsret{\sphinxcode{constitution.}\sphinxbfcode{CalculateSulfurNumber}}{\emph{mol}}{}
Calculation of Sulfur counts in a molecule

----\textgreater{}nsulph

Usage:
\begin{quote}

result=CalculateSulfurNumber(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateTripleBondNumber() (in module constitution)}

\begin{fulllineitems}
\phantomsection\label{reference/constitution:constitution.CalculateTripleBondNumber}\pysiglinewithargsret{\sphinxcode{constitution.}\sphinxbfcode{CalculateTripleBondNumber}}{\emph{mol}}{}
Calculation of triple bond counts in a molecule

----\textgreater{}ntb

Usage:
\begin{quote}

result=CalculateTripleBondNumber(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{GetConstitutional() (in module constitution)}

\begin{fulllineitems}
\phantomsection\label{reference/constitution:constitution.GetConstitutional}\pysiglinewithargsret{\sphinxcode{constitution.}\sphinxbfcode{GetConstitutional}}{\emph{mol}}{}
Get the dictionary of constitutional descriptors for given moelcule mol

Usage:
\begin{quote}

result=GetConstitutional(mol)

Input: mol is a molecule object.

Output: result is a dict form containing all constitutional values.
\end{quote}

\end{fulllineitems}



\subsection{estate module}
\label{reference/estate:estate-module}\label{reference/estate:module-estate}\label{reference/estate::doc}\index{estate (module)}
and Hall's paper. If you have any question please contact me via email.

Authors: Zhijiang Yao and Dongsheng Cao.

Date: 2016.06.04

Email: \href{mailto:gadsby@163.com}{gadsby@163.com} and \href{mailto:oriental-cds@163.com}{oriental-cds@163.com}


\bigskip\hrule{}\bigskip

\index{CalculateEstateFingerprint() (in module estate)}

\begin{fulllineitems}
\phantomsection\label{reference/estate:estate.CalculateEstateFingerprint}\pysiglinewithargsret{\sphinxcode{estate.}\sphinxbfcode{CalculateEstateFingerprint}}{\emph{mol}}{}
The Calculation of EState Fingerprints.

It is the number of times each possible atom type is hit.

Usage:
\begin{quote}

result=CalculateEstateFingerprint(mol)

Input: mol is a molecule object.

Output: result is a dict form containing 79 estate fragments.
\end{quote}

\end{fulllineitems}

\index{CalculateEstateValue() (in module estate)}

\begin{fulllineitems}
\phantomsection\label{reference/estate:estate.CalculateEstateValue}\pysiglinewithargsret{\sphinxcode{estate.}\sphinxbfcode{CalculateEstateValue}}{\emph{mol}}{}
The Calculate of EState Values.

It is the sum of the Estate indices for atoms of each type.

Usage:
\begin{quote}

result=CalculateEstateValue(mol)

Input: mol is a molecule object.

Output: result is a dict form containing 79 estate values.
\end{quote}

\end{fulllineitems}

\index{CalculateMaxAtomTypeEState() (in module estate)}

\begin{fulllineitems}
\phantomsection\label{reference/estate:estate.CalculateMaxAtomTypeEState}\pysiglinewithargsret{\sphinxcode{estate.}\sphinxbfcode{CalculateMaxAtomTypeEState}}{\emph{mol}}{}
Calculation of maximum of E-State value of specified atom type

res----\textgreater{}dict type

Usage:
\begin{quote}

result=CalculateMaxAtomTypeEState(mol)

Input: mol is a molecule object.

Output: result is a dict form containing 79 max estate values.
\end{quote}

\end{fulllineitems}

\index{CalculateMinAtomTypeEState() (in module estate)}

\begin{fulllineitems}
\phantomsection\label{reference/estate:estate.CalculateMinAtomTypeEState}\pysiglinewithargsret{\sphinxcode{estate.}\sphinxbfcode{CalculateMinAtomTypeEState}}{\emph{mol}}{}
Calculation of minimum of E-State value of specified atom type

res----\textgreater{}dict type

Usage:
\begin{quote}

result=CalculateMinAtomTypeEState(mol)

Input: mol is a molecule object.

Output: result is a dict form containing 79 min estate values.
\end{quote}

\end{fulllineitems}

\index{GetEstate() (in module estate)}

\begin{fulllineitems}
\phantomsection\label{reference/estate:estate.GetEstate}\pysiglinewithargsret{\sphinxcode{estate.}\sphinxbfcode{GetEstate}}{\emph{mol}}{}
Obtain all descriptors related to Estate.

Usage:
\begin{quote}

result=GetEstate(mol)

Input: mol is a molecule object.

Output: result is a dict form containing all estate values.
\end{quote}

\end{fulllineitems}



\subsection{fingerprint module}
\label{reference/fingerprint:module-fingerprint}\label{reference/fingerprint::doc}\label{reference/fingerprint:fingerprint-module}\index{fingerprint (module)}
fingerprint system. If you have any question please contact me via email.

2016.11.15

@author: Zhijiang Yao and Dongsheng Cao

Email: \href{mailto:gadsby@163.com}{gadsby@163.com} and \href{mailto:oriental-cds@163.com}{oriental-cds@163.com}


\bigskip\hrule{}\bigskip

\index{CalculateAtomPairsFingerprint() (in module fingerprint)}

\begin{fulllineitems}
\phantomsection\label{reference/fingerprint:fingerprint.CalculateAtomPairsFingerprint}\pysiglinewithargsret{\sphinxcode{fingerprint.}\sphinxbfcode{CalculateAtomPairsFingerprint}}{\emph{mol}}{}
Calculate atom pairs fingerprints

Usage:
\begin{quote}

result=CalculateAtomPairsFingerprint(mol)

Input: mol is a molecule object.

Output: result is a tuple form. The first is the number of

fingerprints. The second is a dict form whose keys are the

position which this molecule has some substructure. The third

is the DataStructs which is used for calculating the similarity.
\end{quote}

\end{fulllineitems}

\index{CalculateDaylightFingerprint() (in module fingerprint)}

\begin{fulllineitems}
\phantomsection\label{reference/fingerprint:fingerprint.CalculateDaylightFingerprint}\pysiglinewithargsret{\sphinxcode{fingerprint.}\sphinxbfcode{CalculateDaylightFingerprint}}{\emph{mol}}{}
Calculate Daylight-like fingerprint or topological fingerprint

(2048 bits).

Usage:
\begin{quote}

result=CalculateDaylightFingerprint(mol)

Input: mol is a molecule object.

Output: result is a tuple form. The first is the number of

fingerprints. The second is a dict form whose keys are the

position which this molecule has some substructure. The third

is the DataStructs which is used for calculating the similarity.
\end{quote}

\end{fulllineitems}

\index{CalculateECFP2Fingerprint() (in module fingerprint)}

\begin{fulllineitems}
\phantomsection\label{reference/fingerprint:fingerprint.CalculateECFP2Fingerprint}\pysiglinewithargsret{\sphinxcode{fingerprint.}\sphinxbfcode{CalculateECFP2Fingerprint}}{\emph{mol}, \emph{radius=1}}{}
Calculate ECFP2

Usage:
\begin{quote}

result=CalculateECFP2Fingerprint(mol)

Input: mol is a molecule object.

radius is a radius.

Output: result is a tuple form. The first is the vector of

fingerprints. The second is a dict form whose keys are the

position which this molecule has some substructure. The third

is the DataStructs which is used for calculating the similarity.
\end{quote}

\end{fulllineitems}

\index{CalculateECFP4Fingerprint() (in module fingerprint)}

\begin{fulllineitems}
\phantomsection\label{reference/fingerprint:fingerprint.CalculateECFP4Fingerprint}\pysiglinewithargsret{\sphinxcode{fingerprint.}\sphinxbfcode{CalculateECFP4Fingerprint}}{\emph{mol}, \emph{radius=2}}{}
Calculate ECFP4

Usage:
\begin{quote}

result=CalculateECFP4Fingerprint(mol)

Input: mol is a molecule object.

radius is a radius.

Output: result is a tuple form. The first is the vector of

fingerprints. The second is a dict form whose keys are the

position which this molecule has some substructure. The third

is the DataStructs which is used for calculating the similarity.
\end{quote}

\end{fulllineitems}

\index{CalculateECFP6Fingerprint() (in module fingerprint)}

\begin{fulllineitems}
\phantomsection\label{reference/fingerprint:fingerprint.CalculateECFP6Fingerprint}\pysiglinewithargsret{\sphinxcode{fingerprint.}\sphinxbfcode{CalculateECFP6Fingerprint}}{\emph{mol}, \emph{radius=3}}{}
Calculate ECFP6

Usage:
\begin{quote}

result=CalculateECFP6Fingerprint(mol)

Input: mol is a molecule object.

radius is a radius.

Output: result is a tuple form. The first is the vector of

fingerprints. The second is a dict form whose keys are the

position which this molecule has some substructure. The third

is the DataStructs which is used for calculating the similarity.
\end{quote}

\end{fulllineitems}

\index{CalculateEstateFingerprint() (in module fingerprint)}

\begin{fulllineitems}
\phantomsection\label{reference/fingerprint:fingerprint.CalculateEstateFingerprint}\pysiglinewithargsret{\sphinxcode{fingerprint.}\sphinxbfcode{CalculateEstateFingerprint}}{\emph{mol}}{}
Calculate E-state fingerprints (79 bits).

Usage:
\begin{quote}

result=CalculateEstateFingerprint(mol)

Input: mol is a molecule object.

Output: result is a tuple form. The first is the number of

fingerprints. The second is a dict form whose keys are the

position which this molecule has some substructure. The third

is the DataStructs which is used for calculating the similarity.
\end{quote}

\end{fulllineitems}

\index{CalculateFCFP2Fingerprint() (in module fingerprint)}

\begin{fulllineitems}
\phantomsection\label{reference/fingerprint:fingerprint.CalculateFCFP2Fingerprint}\pysiglinewithargsret{\sphinxcode{fingerprint.}\sphinxbfcode{CalculateFCFP2Fingerprint}}{\emph{mol}, \emph{radius=1}, \emph{nBits=1024}}{}
Calculate FCFP2

Usage:
\begin{quote}

result=CalculateFCFP2Fingerprint(mol)

Input: mol is a molecule object.

radius is a radius.

Output: result is a tuple form. The first is the vector of

fingerprints. The second is a dict form whose keys are the

position which this molecule has some substructure. The third

is the DataStructs which is used for calculating the similarity.
\end{quote}

\end{fulllineitems}

\index{CalculateFCFP4Fingerprint() (in module fingerprint)}

\begin{fulllineitems}
\phantomsection\label{reference/fingerprint:fingerprint.CalculateFCFP4Fingerprint}\pysiglinewithargsret{\sphinxcode{fingerprint.}\sphinxbfcode{CalculateFCFP4Fingerprint}}{\emph{mol}, \emph{radius=2}, \emph{nBits=1024}}{}
Calculate FCFP4

Usage:
\begin{quote}

result=CalculateFCFP4Fingerprint(mol)

Input: mol is a molecule object.

radius is a radius.

Output: result is a tuple form. The first is the vector of

fingerprints. The second is a dict form whose keys are the

position which this molecule has some substructure. The third

is the DataStructs which is used for calculating the similarity.
\end{quote}

\end{fulllineitems}

\index{CalculateFCFP6Fingerprint() (in module fingerprint)}

\begin{fulllineitems}
\phantomsection\label{reference/fingerprint:fingerprint.CalculateFCFP6Fingerprint}\pysiglinewithargsret{\sphinxcode{fingerprint.}\sphinxbfcode{CalculateFCFP6Fingerprint}}{\emph{mol}, \emph{radius=3}, \emph{nBits=1024}}{}
Calculate FCFP6

Usage:
\begin{quote}

result=CalculateFCFP4Fingerprint(mol)

Input: mol is a molecule object.

radius is a radius.

Output: result is a tuple form. The first is the vector of

fingerprints. The second is a dict form whose keys are the

position which this molecule has some substructure. The third

is the DataStructs which is used for calculating the similarity.
\end{quote}

\end{fulllineitems}

\index{CalculateFP2Fingerprint() (in module fingerprint)}

\begin{fulllineitems}
\phantomsection\label{reference/fingerprint:fingerprint.CalculateFP2Fingerprint}\pysiglinewithargsret{\sphinxcode{fingerprint.}\sphinxbfcode{CalculateFP2Fingerprint}}{\emph{mol}}{}
Calculate FP2 fingerprints (1024 bits).

Usage:
\begin{quote}

result=CalculateFP2Fingerprint(mol)

Input: mol is a molecule object.

Output: result is a tuple form. The first is the number of

fingerprints. The second is a dict form whose keys are the

position which this molecule has some substructure. The third

is the DataStructs which is used for calculating the similarity.
\end{quote}

\end{fulllineitems}

\index{CalculateFP3Fingerprint() (in module fingerprint)}

\begin{fulllineitems}
\phantomsection\label{reference/fingerprint:fingerprint.CalculateFP3Fingerprint}\pysiglinewithargsret{\sphinxcode{fingerprint.}\sphinxbfcode{CalculateFP3Fingerprint}}{\emph{mol}}{}
Calculate FP3 fingerprints (210 bits).

Usage:
\begin{quote}

result=CalculateFP3Fingerprint(mol)

Input: mol is a molecule object.

Output: result is a tuple form. The first is the number of

fingerprints. The second is a dict form whose keys are the

position which this molecule has some substructure. The third

is the DataStructs which is used for calculating the similarity.
\end{quote}

\end{fulllineitems}

\index{CalculateFP4Fingerprint() (in module fingerprint)}

\begin{fulllineitems}
\phantomsection\label{reference/fingerprint:fingerprint.CalculateFP4Fingerprint}\pysiglinewithargsret{\sphinxcode{fingerprint.}\sphinxbfcode{CalculateFP4Fingerprint}}{\emph{mol}}{}
Calculate FP4 fingerprints (307 bits).

Usage:
\begin{quote}

result=CalculateFP4Fingerprint(mol)

Input: mol is a molecule object.

Output: result is a tuple form. The first is the number of

fingerprints. The second is a dict form whose keys are the

position which this molecule has some substructure. The third

is the DataStructs which is used for calculating the similarity.
\end{quote}

\end{fulllineitems}

\index{CalculateGhoseCrippenFingerprint() (in module fingerprint)}

\begin{fulllineitems}
\phantomsection\label{reference/fingerprint:fingerprint.CalculateGhoseCrippenFingerprint}\pysiglinewithargsret{\sphinxcode{fingerprint.}\sphinxbfcode{CalculateGhoseCrippenFingerprint}}{\emph{mol}, \emph{count=False}}{}
\end{fulllineitems}

\index{CalculateMACCSFingerprint() (in module fingerprint)}

\begin{fulllineitems}
\phantomsection\label{reference/fingerprint:fingerprint.CalculateMACCSFingerprint}\pysiglinewithargsret{\sphinxcode{fingerprint.}\sphinxbfcode{CalculateMACCSFingerprint}}{\emph{mol}}{}
Calculate MACCS keys (166 bits).

Usage:
\begin{quote}

result=CalculateMACCSFingerprint(mol)

Input: mol is a molecule object.

Output: result is a tuple form. The first is the number of

fingerprints. The second is a dict form whose keys are the

position which this molecule has some substructure. The third

is the DataStructs which is used for calculating the similarity.
\end{quote}

\end{fulllineitems}

\index{CalculateMorganFingerprint() (in module fingerprint)}

\begin{fulllineitems}
\phantomsection\label{reference/fingerprint:fingerprint.CalculateMorganFingerprint}\pysiglinewithargsret{\sphinxcode{fingerprint.}\sphinxbfcode{CalculateMorganFingerprint}}{\emph{mol}, \emph{radius=2}}{}
Calculate Morgan

Usage:
\begin{quote}

result=CalculateMorganFingerprint(mol)

Input: mol is a molecule object.

radius is a radius.

Output: result is a tuple form. The first is the number of

fingerprints. The second is a dict form whose keys are the

position which this molecule has some substructure. The third

is the DataStructs which is used for calculating the similarity.
\end{quote}

\end{fulllineitems}

\index{CalculatePharm2D2pointFingerprint() (in module fingerprint)}

\begin{fulllineitems}
\phantomsection\label{reference/fingerprint:fingerprint.CalculatePharm2D2pointFingerprint}\pysiglinewithargsret{\sphinxcode{fingerprint.}\sphinxbfcode{CalculatePharm2D2pointFingerprint}}{\emph{mol}, \emph{featFactory=\textless{}rdkit.Chem.rdMolChemicalFeatures.MolChemicalFeatureFactory object at 0x045D0FB8\textgreater{}}}{}
\end{fulllineitems}

\index{CalculatePharm2D3pointFingerprint() (in module fingerprint)}

\begin{fulllineitems}
\phantomsection\label{reference/fingerprint:fingerprint.CalculatePharm2D3pointFingerprint}\pysiglinewithargsret{\sphinxcode{fingerprint.}\sphinxbfcode{CalculatePharm2D3pointFingerprint}}{\emph{mol}, \emph{featFactory=\textless{}rdkit.Chem.rdMolChemicalFeatures.MolChemicalFeatureFactory object at 0x045D0FB8\textgreater{}}}{}
\end{fulllineitems}

\index{CalculateSimilarityPybel() (in module fingerprint)}

\begin{fulllineitems}
\phantomsection\label{reference/fingerprint:fingerprint.CalculateSimilarityPybel}\pysiglinewithargsret{\sphinxcode{fingerprint.}\sphinxbfcode{CalculateSimilarityPybel}}{\emph{fp1}, \emph{fp2}}{}
Calculate Tanimoto similarity between two molecules.

Usage:
\begin{quote}

result=CalculateSimilarityPybel(fp1,fp2)

Input: fp1 and fp2 are two DataStructs.

Output: result is a Tanimoto similarity value.
\end{quote}

\end{fulllineitems}

\index{CalculateSimilarityRdkit() (in module fingerprint)}

\begin{fulllineitems}
\phantomsection\label{reference/fingerprint:fingerprint.CalculateSimilarityRdkit}\pysiglinewithargsret{\sphinxcode{fingerprint.}\sphinxbfcode{CalculateSimilarityRdkit}}{\emph{fp1}, \emph{fp2}, \emph{similarity='Tanimoto'}}{}
Calculate similarity between two molecules.

Usage:
\begin{quote}

result=CalculateSimilarity(fp1,fp2)
Users can choose 11 different types:
Tanimoto, Dice, Cosine, Sokal, Russel,
RogotGoldberg, AllBit, Kulczynski, 
McConnaughey, Asymmetric, BraunBlanquet       
Input: fp1 and fp2 are two DataStructs.

Output: result is a similarity value.
\end{quote}

\end{fulllineitems}

\index{CalculateTopologicalTorsionFingerprint() (in module fingerprint)}

\begin{fulllineitems}
\phantomsection\label{reference/fingerprint:fingerprint.CalculateTopologicalTorsionFingerprint}\pysiglinewithargsret{\sphinxcode{fingerprint.}\sphinxbfcode{CalculateTopologicalTorsionFingerprint}}{\emph{mol}}{}
Calculate Topological Torsion Fingerprints

Usage:
\begin{quote}

result=CalculateTopologicalTorsionFingerprint(mol)

Input: mol is a molecule object.

Output: result is a tuple form. The first is the number of

fingerprints. The second is a dict form whose keys are the

position which this molecule has some substructure. The third

is the DataStructs which is used for calculating the similarity.
\end{quote}

\end{fulllineitems}



\subsection{geary module}
\label{reference/geary::doc}\label{reference/geary:module-geary}\label{reference/geary:geary-module}\index{geary (module)}
structure. You can get 32 molecular autocorrelation descriptors. You can

freely use and distribute it. If you hava  any problem, you could contact

with us timely!

Authors: Zhijiang Yao and Dongsheng Cao.

Date: 2016.06.04

Email: \href{mailto:gadsby@163.com}{gadsby@163.com} and \href{mailto:oriental-cds@163.com}{oriental-cds@163.com}


\bigskip\hrule{}\bigskip

\index{CalculateGearyAutoElectronegativity() (in module geary)}

\begin{fulllineitems}
\phantomsection\label{reference/geary:geary.CalculateGearyAutoElectronegativity}\pysiglinewithargsret{\sphinxcode{geary.}\sphinxbfcode{CalculateGearyAutoElectronegativity}}{\emph{mol}}{}
Calculation of Geary autocorrelation descriptors based on

carbon-scaled atomic Sanderson electronegativity.

Usage:

res=CalculateGearyAutoElectronegativity(mol)

Input: mol is a molecule object.

Output: res is a dict form containing eight geary autocorrealtion

\end{fulllineitems}

\index{CalculateGearyAutoMass() (in module geary)}

\begin{fulllineitems}
\phantomsection\label{reference/geary:geary.CalculateGearyAutoMass}\pysiglinewithargsret{\sphinxcode{geary.}\sphinxbfcode{CalculateGearyAutoMass}}{\emph{mol}}{}
Calculation of Geary autocorrelation descriptors based on

carbon-scaled atomic mass.

Usage:

res=CalculateMoranAutoMass(mol)

Input: mol is a molecule object.

Output: res is a dict form containing eight geary autocorrealtion

\end{fulllineitems}

\index{CalculateGearyAutoPolarizability() (in module geary)}

\begin{fulllineitems}
\phantomsection\label{reference/geary:geary.CalculateGearyAutoPolarizability}\pysiglinewithargsret{\sphinxcode{geary.}\sphinxbfcode{CalculateGearyAutoPolarizability}}{\emph{mol}}{}
Calculation of Geary autocorrelation descriptors based on

carbon-scaled atomic polarizability.

Usage:

res=CalculateGearyAutoPolarizability(mol)

Input: mol is a molecule object.

Output: res is a dict form containing eight geary autocorrealtion

\end{fulllineitems}

\index{CalculateGearyAutoVolume() (in module geary)}

\begin{fulllineitems}
\phantomsection\label{reference/geary:geary.CalculateGearyAutoVolume}\pysiglinewithargsret{\sphinxcode{geary.}\sphinxbfcode{CalculateGearyAutoVolume}}{\emph{mol}}{}
Calculation of Geary autocorrelation descriptors based on

carbon-scaled atomic van der Waals volume.

Usage:

res=CalculateGearyAutoVolume(mol)

Input: mol is a molecule object.

Output: res is a dict form containing eight geary autocorrealtion

\end{fulllineitems}

\index{GetGearyAuto() (in module geary)}

\begin{fulllineitems}
\phantomsection\label{reference/geary:geary.GetGearyAuto}\pysiglinewithargsret{\sphinxcode{geary.}\sphinxbfcode{GetGearyAuto}}{\emph{mol}}{}
Calcualate all Geary autocorrelation descriptors.

(carbon-scaled atomic mass, carbon-scaled atomic van der Waals volume,

carbon-scaled atomic Sanderson electronegativity,

carbon-scaled atomic polarizability)

Usage:

res=GetGearyAuto(mol)

Input: mol is a molecule object.

Output: res is a dict form containing all geary autocorrealtion

\end{fulllineitems}



\subsection{ghosecrippen module}
\label{reference/ghosecrippen::doc}\label{reference/ghosecrippen:ghosecrippen-module}\label{reference/ghosecrippen:module-ghosecrippen}\index{ghosecrippen (module)}
This module is to calculate the ghosecrippen descriptor. If you

have any question please contact me via email.

Authors: Zhijiang Yao and Dongsheng Cao.

Date: 2016.06.04

Email: \href{mailto:gadsby@163.com}{gadsby@163.com} and \href{mailto:oriental-cds@163.com}{oriental-cds@163.com}
\index{GhoseCrippenFingerprint() (in module ghosecrippen)}

\begin{fulllineitems}
\phantomsection\label{reference/ghosecrippen:ghosecrippen.GhoseCrippenFingerprint}\pysiglinewithargsret{\sphinxcode{ghosecrippen.}\sphinxbfcode{GhoseCrippenFingerprint}}{\emph{mol}, \emph{count=False}}{}
Ghose-Crippen substructures or counts based on the definitions of

SMARTS from Ghose-Crippen's paper. (110 dimension)

\end{fulllineitems}



\subsection{kappa module}
\label{reference/kappa:module-kappa}\label{reference/kappa::doc}\label{reference/kappa:kappa-module}\index{kappa (module)}
structure. You can get 7 molecular kappa descriptors. You can

freely use and distribute it. If you hava  any problem, you could contact

with us timely!

Authors: Zhijiang Yao and Dongsheng Cao.

Date: 2016.06.04

Email: \href{mailto:gadsby@163.com}{gadsby@163.com} and \href{mailto:oriental-cds@163.com}{oriental-cds@163.com}


\bigskip\hrule{}\bigskip

\index{CalculateFlexibility() (in module kappa)}

\begin{fulllineitems}
\phantomsection\label{reference/kappa:kappa.CalculateFlexibility}\pysiglinewithargsret{\sphinxcode{kappa.}\sphinxbfcode{CalculateFlexibility}}{\emph{mol}}{}
Calculation of Kier molecular flexibility index

----\textgreater{}phi

Usage:
\begin{quote}

result=CalculateFlexibility(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateKappa1() (in module kappa)}

\begin{fulllineitems}
\phantomsection\label{reference/kappa:kappa.CalculateKappa1}\pysiglinewithargsret{\sphinxcode{kappa.}\sphinxbfcode{CalculateKappa1}}{\emph{mol}}{}
Calculation of molecular shape index for one bonded fragment

----\textgreater{}kappa1

Usage:
\begin{quote}

result=CalculateKappa1(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateKappa2() (in module kappa)}

\begin{fulllineitems}
\phantomsection\label{reference/kappa:kappa.CalculateKappa2}\pysiglinewithargsret{\sphinxcode{kappa.}\sphinxbfcode{CalculateKappa2}}{\emph{mol}}{}
Calculation of molecular shape index for two bonded fragment

----\textgreater{}kappa2

Usage:
\begin{quote}

result=CalculateKappa2(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateKappa3() (in module kappa)}

\begin{fulllineitems}
\phantomsection\label{reference/kappa:kappa.CalculateKappa3}\pysiglinewithargsret{\sphinxcode{kappa.}\sphinxbfcode{CalculateKappa3}}{\emph{mol}}{}
Calculation of molecular shape index for three bonded fragment

----\textgreater{}kappa3

Usage:
\begin{quote}

result=CalculateKappa3(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateKappaAlapha1() (in module kappa)}

\begin{fulllineitems}
\phantomsection\label{reference/kappa:kappa.CalculateKappaAlapha1}\pysiglinewithargsret{\sphinxcode{kappa.}\sphinxbfcode{CalculateKappaAlapha1}}{\emph{mol}}{}
Calculation of molecular shape index for one bonded fragment

with Alapha

----\textgreater{}kappam1

Usage:
\begin{quote}

result=CalculateKappaAlapha1(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateKappaAlapha2() (in module kappa)}

\begin{fulllineitems}
\phantomsection\label{reference/kappa:kappa.CalculateKappaAlapha2}\pysiglinewithargsret{\sphinxcode{kappa.}\sphinxbfcode{CalculateKappaAlapha2}}{\emph{mol}}{}
Calculation of molecular shape index for two bonded fragment

with Alapha

----\textgreater{}kappam2

Usage:
\begin{quote}

result=CalculateKappaAlapha2(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateKappaAlapha3() (in module kappa)}

\begin{fulllineitems}
\phantomsection\label{reference/kappa:kappa.CalculateKappaAlapha3}\pysiglinewithargsret{\sphinxcode{kappa.}\sphinxbfcode{CalculateKappaAlapha3}}{\emph{mol}}{}
Calculation of molecular shape index for three bonded fragment

with Alapha

----\textgreater{}kappam3

Usage:
\begin{quote}

result=CalculateKappaAlapha3(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{GetKappa() (in module kappa)}

\begin{fulllineitems}
\phantomsection\label{reference/kappa:kappa.GetKappa}\pysiglinewithargsret{\sphinxcode{kappa.}\sphinxbfcode{GetKappa}}{\emph{mol}}{}
Calculation of all kappa values.

Usage:
\begin{quote}

result=GetKappa(mol)

Input: mol is a molecule object.

Output: result is a dcit form containing 6 kappa values.
\end{quote}

\end{fulllineitems}



\subsection{moe module}
\label{reference/moe:module-moe}\label{reference/moe::doc}\label{reference/moe:moe-module}\index{moe (module)}
include LabuteASA, TPSA, slogPVSA, MRVSA, PEOEVSA, EstateVSA and VSAEstate,

respectively (60).

If you have any question about these indices please contact me via email.

Authors: Zhijiang Yao and Dongsheng Cao.

Date: 2016.06.04

Email: \href{mailto:gadsby@163.com}{gadsby@163.com} and \href{mailto:oriental-cds@163.com}{oriental-cds@163.com}


\bigskip\hrule{}\bigskip

\index{CalculateEstateVSA() (in module moe)}

\begin{fulllineitems}
\phantomsection\label{reference/moe:moe.CalculateEstateVSA}\pysiglinewithargsret{\sphinxcode{moe.}\sphinxbfcode{CalculateEstateVSA}}{\emph{mol}, \emph{bins=None}}{}
MOE-type descriptors using Estate indices and surface area

contributions.

estateBins={[}-0.390,0.290,0.717,1.165,1.540,1.807,2.05,4.69,9.17,15.0{]}

You can specify your own bins to compute some descriptors

Usage:
\begin{quote}

result=CalculateEstateVSA(mol)

Input: mol is a molecule object

Output: result is a dict form
\end{quote}

\end{fulllineitems}

\index{CalculateLabuteASA() (in module moe)}

\begin{fulllineitems}
\phantomsection\label{reference/moe:moe.CalculateLabuteASA}\pysiglinewithargsret{\sphinxcode{moe.}\sphinxbfcode{CalculateLabuteASA}}{\emph{mol}}{}
Calculation of Labute's Approximate Surface Area (ASA from MOE)

Usage:
\begin{quote}

result=CalculateLabuteASA(mol)

Input: mol is a molecule object

Output: result is a dict form
\end{quote}

\end{fulllineitems}

\index{CalculatePEOEVSA() (in module moe)}

\begin{fulllineitems}
\phantomsection\label{reference/moe:moe.CalculatePEOEVSA}\pysiglinewithargsret{\sphinxcode{moe.}\sphinxbfcode{CalculatePEOEVSA}}{\emph{mol}, \emph{bins=None}}{}
MOE-type descriptors using partial charges and surface

area contributions.

chgBins={[}-.3,-.25,-.20,-.15,-.10,-.05,0,.05,.10,.15,.20,.25,.30{]}

You can specify your own bins to compute some descriptors

Usage:
\begin{quote}

result=CalculatePEOEVSA(mol)

Input: mol is a molecule object

Output: result is a dict form
\end{quote}

\end{fulllineitems}

\index{CalculateSLOGPVSA() (in module moe)}

\begin{fulllineitems}
\phantomsection\label{reference/moe:moe.CalculateSLOGPVSA}\pysiglinewithargsret{\sphinxcode{moe.}\sphinxbfcode{CalculateSLOGPVSA}}{\emph{mol}, \emph{bins=None}}{}
MOE-type descriptors using LogP contributions and surface

area contributions.

logpBins={[}-0.4,-0.2,0,0.1,0.15,0.2,0.25,0.3,0.4,0.5,0.6{]}

You can specify your own bins to compute some descriptors.

Usage:
\begin{quote}

result=CalculateSLOGPVSA(mol)

Input: mol is a molecule object

Output: result is a dict form
\end{quote}

\end{fulllineitems}

\index{CalculateSMRVSA() (in module moe)}

\begin{fulllineitems}
\phantomsection\label{reference/moe:moe.CalculateSMRVSA}\pysiglinewithargsret{\sphinxcode{moe.}\sphinxbfcode{CalculateSMRVSA}}{\emph{mol}, \emph{bins=None}}{}
MOE-type descriptors using MR contributions and surface

area contributions.

mrBins={[}1.29, 1.82, 2.24, 2.45, 2.75, 3.05, 3.63,3.8,4.0{]}

You can specify your own bins to compute some descriptors.

Usage:
\begin{quote}

result=CalculateSMRVSA(mol)

Input: mol is a molecule object

Output: result is a dict form
\end{quote}

\end{fulllineitems}

\index{CalculateTPSA() (in module moe)}

\begin{fulllineitems}
\phantomsection\label{reference/moe:moe.CalculateTPSA}\pysiglinewithargsret{\sphinxcode{moe.}\sphinxbfcode{CalculateTPSA}}{\emph{mol}}{}
Calculation of topological polar surface area based on fragments.

Implementation based on the Daylight contrib program tpsa.

Usage:
\begin{quote}

result=CalculateTPSA(mol)

Input: mol is a molecule object

Output: result is a dict form
\end{quote}

\end{fulllineitems}

\index{CalculateVSAEstate() (in module moe)}

\begin{fulllineitems}
\phantomsection\label{reference/moe:moe.CalculateVSAEstate}\pysiglinewithargsret{\sphinxcode{moe.}\sphinxbfcode{CalculateVSAEstate}}{\emph{mol}, \emph{bins=None}}{}
MOE-type descriptors using Estate indices and surface

area contributions.

vsaBins={[}4.78,5.00,5.410,5.740,6.00,6.07,6.45,7.00,11.0{]}

You can specify your own bins to compute some descriptors

Usage:
\begin{quote}

result=CalculateVSAEstate(mol)

Input: mol is a molecule object

Output: result is a dict form
\end{quote}

\end{fulllineitems}

\index{GetMOE() (in module moe)}

\begin{fulllineitems}
\phantomsection\label{reference/moe:moe.GetMOE}\pysiglinewithargsret{\sphinxcode{moe.}\sphinxbfcode{GetMOE}}{\emph{mol}}{}
The calculation of MOE-type descriptors (ALL).

Usage:
\begin{quote}

result=GetMOE(mol)

Input: mol is a molecule object

Output: result is a dict form
\end{quote}

\end{fulllineitems}



\subsection{molproperty module}
\label{reference/molproperty:module-molproperty}\label{reference/molproperty::doc}\label{reference/molproperty:molproperty-module}\index{molproperty (module)}
type of approaches(6), including: LogP; LogP2; MR; TPSA, UI and Hy.You can

freely use and distribute it. If you hava  any problem, you could contact

with us timely!

Authors: Zhijiang Yao and Dongsheng Cao.

Date: 2016.06.04

Email: \href{mailto:gadsby@163.com}{gadsby@163.com} and \href{mailto:oriental-cds@163.com}{oriental-cds@163.com}


\bigskip\hrule{}\bigskip

\index{CalculateHydrophilicityFactor() (in module molproperty)}

\begin{fulllineitems}
\phantomsection\label{reference/molproperty:molproperty.CalculateHydrophilicityFactor}\pysiglinewithargsret{\sphinxcode{molproperty.}\sphinxbfcode{CalculateHydrophilicityFactor}}{\emph{mol}}{}
Calculation of hydrophilicity factor. The hydrophilicity

index is described in more detail on page 225 of the

Handbook of Molecular Descriptors (Todeschini and Consonni 2000).

----\textgreater{}Hy

Usage:
\begin{quote}

result=CalculateHydrophilicityFactor(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateMolLogP() (in module molproperty)}

\begin{fulllineitems}
\phantomsection\label{reference/molproperty:molproperty.CalculateMolLogP}\pysiglinewithargsret{\sphinxcode{molproperty.}\sphinxbfcode{CalculateMolLogP}}{\emph{mol}}{}
Cacluation of LogP value based on Crippen method

----\textgreater{}LogP

Usage:
\begin{quote}

result=CalculateMolLogP(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateMolLogP2() (in module molproperty)}

\begin{fulllineitems}
\phantomsection\label{reference/molproperty:molproperty.CalculateMolLogP2}\pysiglinewithargsret{\sphinxcode{molproperty.}\sphinxbfcode{CalculateMolLogP2}}{\emph{mol}}{}
Cacluation of LogP\textasciicircum{}2 value based on Crippen method

----\textgreater{}LogP2

Usage:
\begin{quote}

result=CalculateMolLogP2(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateMolMR() (in module molproperty)}

\begin{fulllineitems}
\phantomsection\label{reference/molproperty:molproperty.CalculateMolMR}\pysiglinewithargsret{\sphinxcode{molproperty.}\sphinxbfcode{CalculateMolMR}}{\emph{mol}}{}
Cacluation of molecular refraction value based on Crippen method

----\textgreater{}MR

Usage:
\begin{quote}

result=CalculateMolMR(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateTPSA() (in module molproperty)}

\begin{fulllineitems}
\phantomsection\label{reference/molproperty:molproperty.CalculateTPSA}\pysiglinewithargsret{\sphinxcode{molproperty.}\sphinxbfcode{CalculateTPSA}}{\emph{mol}}{}
calculates the polar surface area of a molecule based upon fragments

Algorithm in:
\begin{enumerate}
\setcounter{enumi}{15}
\item {} 
Ertl, B. Rohde, P. Selzer

\end{enumerate}

Fast Calculation of Molecular Polar Surface Area as a Sum of

Fragment-based Contributions and Its Application to the Prediction

of Drug Transport Properties, J.Med.Chem. 43, 3714-3717, 2000

Implementation based on the Daylight contrib program tpsa.

----\textgreater{}TPSA

Usage:
\begin{quote}

result=CalculateTPSA(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateUnsaturationIndex() (in module molproperty)}

\begin{fulllineitems}
\phantomsection\label{reference/molproperty:molproperty.CalculateUnsaturationIndex}\pysiglinewithargsret{\sphinxcode{molproperty.}\sphinxbfcode{CalculateUnsaturationIndex}}{\emph{mol}}{}
Calculation of unsaturation index.

----\textgreater{}UI

Usage:
\begin{quote}

result=CalculateUnsaturationIndex(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateXlogP() (in module molproperty)}

\begin{fulllineitems}
\phantomsection\label{reference/molproperty:molproperty.CalculateXlogP}\pysiglinewithargsret{\sphinxcode{molproperty.}\sphinxbfcode{CalculateXlogP}}{\emph{mol}}{}
Calculation of Wang octanol water partition coefficient.

----\textgreater{}XLogP

Usage:
\begin{quote}

result=CalculateXlogP(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{CalculateXlogP2() (in module molproperty)}

\begin{fulllineitems}
\phantomsection\label{reference/molproperty:molproperty.CalculateXlogP2}\pysiglinewithargsret{\sphinxcode{molproperty.}\sphinxbfcode{CalculateXlogP2}}{\emph{mol}}{}
Calculation of Wang octanol water partition coefficient (XLogP\textasciicircum{}2).

----\textgreater{}XLogP2

Usage:
\begin{quote}

result=CalculateMolLogP(mol)

Input: mol is a molecule object.

Output: result is a numeric value.
\end{quote}

\end{fulllineitems}

\index{GetMolecularProperty() (in module molproperty)}

\begin{fulllineitems}
\phantomsection\label{reference/molproperty:molproperty.GetMolecularProperty}\pysiglinewithargsret{\sphinxcode{molproperty.}\sphinxbfcode{GetMolecularProperty}}{\emph{mol}}{}
Get the dictionary of constitutional descriptors for

given moelcule mol

Usage:
\begin{quote}

result=GetMolecularProperty(mol)

Input: mol is a molecule object.

Output: result is a dict form containing 6 molecular properties.
\end{quote}

\end{fulllineitems}



\subsection{moran module}
\label{reference/moran:moran-module}\label{reference/moran:module-moran}\label{reference/moran::doc}\index{moran (module)}

\bigskip\hrule{}\bigskip


The calculation of Moran autocorrelation descriptors. You can get 32 molecular

decriptors. You can freely use and distribute it. If you hava  any problem,

you could contact with us timely!

Authors: Zhijiang Yao and Dongsheng Cao.

Date: 2016.06.04

Email: \href{mailto:gadsby@163.com}{gadsby@163.com} and \href{mailto:oriental-cds@163.com}{oriental-cds@163.com}


\bigskip\hrule{}\bigskip

\index{CalculateMoranAutoElectronegativity() (in module moran)}

\begin{fulllineitems}
\phantomsection\label{reference/moran:moran.CalculateMoranAutoElectronegativity}\pysiglinewithargsret{\sphinxcode{moran.}\sphinxbfcode{CalculateMoranAutoElectronegativity}}{\emph{mol}}{}
Calculation of Moran autocorrelation descriptors based on

carbon-scaled atomic Sanderson electronegativity.

Usage:

res=CalculateMoranAutoElectronegativity(mol)

Input: mol is a molecule object.

Output: res is a dict form containing eight moran autocorrealtion

\end{fulllineitems}

\index{CalculateMoranAutoMass() (in module moran)}

\begin{fulllineitems}
\phantomsection\label{reference/moran:moran.CalculateMoranAutoMass}\pysiglinewithargsret{\sphinxcode{moran.}\sphinxbfcode{CalculateMoranAutoMass}}{\emph{mol}}{}
Calculation of Moran autocorrelation descriptors based on

carbon-scaled atomic mass.

Usage:

res=CalculateMoranAutoMass(mol)

Input: mol is a molecule object.

Output: res is a dict form containing eight moran autocorrealtion

\end{fulllineitems}

\index{CalculateMoranAutoPolarizability() (in module moran)}

\begin{fulllineitems}
\phantomsection\label{reference/moran:moran.CalculateMoranAutoPolarizability}\pysiglinewithargsret{\sphinxcode{moran.}\sphinxbfcode{CalculateMoranAutoPolarizability}}{\emph{mol}}{}
Calculation of Moran autocorrelation descriptors based on

carbon-scaled atomic polarizability.

Usage:

res=CalculateMoranAutoPolarizability(mol)

Input: mol is a molecule object.

Output: res is a dict form containing eight moran autocorrealtion

\end{fulllineitems}

\index{CalculateMoranAutoVolume() (in module moran)}

\begin{fulllineitems}
\phantomsection\label{reference/moran:moran.CalculateMoranAutoVolume}\pysiglinewithargsret{\sphinxcode{moran.}\sphinxbfcode{CalculateMoranAutoVolume}}{\emph{mol}}{}
Calculation of Moran autocorrelation descriptors based on

carbon-scaled atomic van der Waals volume.

Usage:

res=CalculateMoranAutoVolume(mol)

Input: mol is a molecule object.

Output: res is a dict form containing eight moran autocorrealtion

\end{fulllineitems}

\index{GetMoranAuto() (in module moran)}

\begin{fulllineitems}
\phantomsection\label{reference/moran:moran.GetMoranAuto}\pysiglinewithargsret{\sphinxcode{moran.}\sphinxbfcode{GetMoranAuto}}{\emph{mol}}{}
Calcualate all Moran autocorrelation descriptors.

(carbon-scaled atomic mass, carbon-scaled atomic van der Waals volume,

carbon-scaled atomic Sanderson electronegativity,

carbon-scaled atomic polarizability)

Usage:

res=GetMoranAuto(mol)

Input: mol is a molecule object.

Output: res is a dict form containing all moran autocorrealtion

\end{fulllineitems}



\subsection{moreaubroto module}
\label{reference/moreaubroto::doc}\label{reference/moreaubroto:moreaubroto-module}\label{reference/moreaubroto:module-moreaubroto}\index{moreaubroto (module)}

\bigskip\hrule{}\bigskip


The calculation of Moreau-Broto autocorrelation descriptors. You can get 32

molecular decriptors. You can freely use and distribute it. If you hava

any problem, you could contact with us timely!

Authors: Zhijiang Yao and Dongsheng Cao.

Date: 2016.06.04

Email: \href{mailto:gadsby@163.com}{gadsby@163.com} and \href{mailto:oriental-cds@163.com}{oriental-cds@163.com}


\bigskip\hrule{}\bigskip

\index{CalculateMoreauBrotoAutoElectronegativity() (in module moreaubroto)}

\begin{fulllineitems}
\phantomsection\label{reference/moreaubroto:moreaubroto.CalculateMoreauBrotoAutoElectronegativity}\pysiglinewithargsret{\sphinxcode{moreaubroto.}\sphinxbfcode{CalculateMoreauBrotoAutoElectronegativity}}{\emph{mol}}{}
Calculation of Moreau-Broto autocorrelation descriptors based on

carbon-scaled atomic Sanderson electronegativity.

Usage:

res=CalculateMoreauBrotoAutoElectronegativity(mol)

Input: mol is a molcule object.

Output: res is a dict form containing eight moreau broto autocorrealtion

\end{fulllineitems}

\index{CalculateMoreauBrotoAutoMass() (in module moreaubroto)}

\begin{fulllineitems}
\phantomsection\label{reference/moreaubroto:moreaubroto.CalculateMoreauBrotoAutoMass}\pysiglinewithargsret{\sphinxcode{moreaubroto.}\sphinxbfcode{CalculateMoreauBrotoAutoMass}}{\emph{mol}}{}
Calculation of Moreau-Broto autocorrelation descriptors based on

carbon-scaled atomic mass.

Usage:

res=CalculateMoreauBrotoAutoMass(mol)

Input: mol is a molecule object.

Output: res is a dict form containing eight moreau broto autocorrealtion

\end{fulllineitems}

\index{CalculateMoreauBrotoAutoPolarizability() (in module moreaubroto)}

\begin{fulllineitems}
\phantomsection\label{reference/moreaubroto:moreaubroto.CalculateMoreauBrotoAutoPolarizability}\pysiglinewithargsret{\sphinxcode{moreaubroto.}\sphinxbfcode{CalculateMoreauBrotoAutoPolarizability}}{\emph{mol}}{}
Calculation of Moreau-Broto autocorrelation descriptors based on

carbon-scaled atomic polarizability.

res=CalculateMoreauBrotoAutoPolarizability(mol)

Input: mol is a molcule object.

Output: res is a dict form containing eight moreau broto autocorrealtion

\end{fulllineitems}

\index{CalculateMoreauBrotoAutoVolume() (in module moreaubroto)}

\begin{fulllineitems}
\phantomsection\label{reference/moreaubroto:moreaubroto.CalculateMoreauBrotoAutoVolume}\pysiglinewithargsret{\sphinxcode{moreaubroto.}\sphinxbfcode{CalculateMoreauBrotoAutoVolume}}{\emph{mol}}{}
Calculation of Moreau-Broto autocorrelation descriptors based on

carbon-scaled atomic van der Waals volume.

Usage:

res=CalculateMoreauBrotoAutoVolume(mol)

Input: mol is a molcule object.

Output: res is a dict form containing eight moreau broto autocorrealtion

\end{fulllineitems}

\index{GetMoreauBrotoAuto() (in module moreaubroto)}

\begin{fulllineitems}
\phantomsection\label{reference/moreaubroto:moreaubroto.GetMoreauBrotoAuto}\pysiglinewithargsret{\sphinxcode{moreaubroto.}\sphinxbfcode{GetMoreauBrotoAuto}}{\emph{mol}}{}
Calcualate all Moreau-Broto autocorrelation descriptors.

(carbon-scaled atomic mass, carbon-scaled atomic van der Waals volume,

carbon-scaled atomic Sanderson electronegativity,

carbon-scaled atomic polarizability)

Usage:

res=GetMoreauBrotoAuto(mol)

Input: mol is a molecule object.

Output: res is a dict form containing all moreau broto autocorrelation

\end{fulllineitems}



\subsection{topology module}
\label{reference/topology::doc}\label{reference/topology:module-topology}\label{reference/topology:topology-module}\index{topology (module)}
structure. You can get 25 molecular topological descriptors. You can freely

use and distribute it. If you hava  any problem, you could contact with us timely!

Authors: Zhijiang Yao and Dongsheng Cao.

Date: 2016.06.04

Email: \href{mailto:gadsby@163.com}{gadsby@163.com} and \href{mailto:oriental-cds@163.com}{oriental-cds@163.com}


\bigskip\hrule{}\bigskip

\index{CalculateArithmeticTopoIndex() (in module topology)}

\begin{fulllineitems}
\phantomsection\label{reference/topology:topology.CalculateArithmeticTopoIndex}\pysiglinewithargsret{\sphinxcode{topology.}\sphinxbfcode{CalculateArithmeticTopoIndex}}{\emph{mol}}{}
Arithmetic topological index by Narumi

----\textgreater{}Arto

Usage:
\begin{quote}

result=CalculateArithmeticTopoIndex(mol)

Input: mol is a molecule object

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateBalaban() (in module topology)}

\begin{fulllineitems}
\phantomsection\label{reference/topology:topology.CalculateBalaban}\pysiglinewithargsret{\sphinxcode{topology.}\sphinxbfcode{CalculateBalaban}}{\emph{mol}}{}
Calculation of Balaban index in a molecule

----\textgreater{}J

Usage:
\begin{quote}

result=CalculateBalaban(mol)

Input: mol is a molecule object

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateBertzCT() (in module topology)}

\begin{fulllineitems}
\phantomsection\label{reference/topology:topology.CalculateBertzCT}\pysiglinewithargsret{\sphinxcode{topology.}\sphinxbfcode{CalculateBertzCT}}{\emph{mol}}{}
A topological index meant to quantify ``complexity'' of molecules.

Consists of a sum of two terms, one representing the complexity

of the bonding, the other representing the complexity of the

distribution of heteroatoms.

From S. H. Bertz, J. Am. Chem. Soc., vol 103, 3599-3601 (1981)

----\textgreater{}BertzCT(log value)

Usage:
\begin{quote}

result=CalculateBertzCT(mol)

Input: mol is a molecule object

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateDiameter() (in module topology)}

\begin{fulllineitems}
\phantomsection\label{reference/topology:topology.CalculateDiameter}\pysiglinewithargsret{\sphinxcode{topology.}\sphinxbfcode{CalculateDiameter}}{\emph{mol}}{}
Calculation of diameter, which is   Largest value

in the distance matrix {[}Petitjean 1992{]}.

----\textgreater{}diametert

Usage:
\begin{quote}

result=CalculateDiameter(mol)

Input: mol is a molecule object

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateGeometricTopoIndex() (in module topology)}

\begin{fulllineitems}
\phantomsection\label{reference/topology:topology.CalculateGeometricTopoIndex}\pysiglinewithargsret{\sphinxcode{topology.}\sphinxbfcode{CalculateGeometricTopoIndex}}{\emph{mol}}{}
Geometric topological index by Narumi

----\textgreater{}Geto

Usage:
\begin{quote}

result=CalculateGeometricTopoIndex(mol)

Input: mol is a molecule object

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateGraphDistance() (in module topology)}

\begin{fulllineitems}
\phantomsection\label{reference/topology:topology.CalculateGraphDistance}\pysiglinewithargsret{\sphinxcode{topology.}\sphinxbfcode{CalculateGraphDistance}}{\emph{mol}}{}
Calculation of graph distance index

----\textgreater{}Tigdi(log value)

Usage:
\begin{quote}

result=CalculateGraphDistance(mol)

Input: mol is a molecule object

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateGutmanTopo() (in module topology)}

\begin{fulllineitems}
\phantomsection\label{reference/topology:topology.CalculateGutmanTopo}\pysiglinewithargsret{\sphinxcode{topology.}\sphinxbfcode{CalculateGutmanTopo}}{\emph{mol}}{}
Calculation of Gutman molecular topological index based on

simple vertex degree

----\textgreater{}GMTI(log value)

Usage:
\begin{quote}

result=CalculateGutmanTopo(mol)

Input: mol is a molecule object

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateHarary() (in module topology)}

\begin{fulllineitems}
\phantomsection\label{reference/topology:topology.CalculateHarary}\pysiglinewithargsret{\sphinxcode{topology.}\sphinxbfcode{CalculateHarary}}{\emph{mol}}{}
Calculation of Harary number

----\textgreater{}Thara

Usage:
\begin{quote}

result=CalculateHarary(mol)

Input: mol is a molecule object

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateHarmonicTopoIndex() (in module topology)}

\begin{fulllineitems}
\phantomsection\label{reference/topology:topology.CalculateHarmonicTopoIndex}\pysiglinewithargsret{\sphinxcode{topology.}\sphinxbfcode{CalculateHarmonicTopoIndex}}{\emph{mol}}{}
Calculation of harmonic topological index proposed by Narnumi.

----\textgreater{}Hato

Usage:
\begin{quote}

result=CalculateHarmonicTopoIndex(mol)

Input: mol is a molecule object

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateIpc() (in module topology)}

\begin{fulllineitems}
\phantomsection\label{reference/topology:topology.CalculateIpc}\pysiglinewithargsret{\sphinxcode{topology.}\sphinxbfcode{CalculateIpc}}{\emph{mol}}{}
This returns the information content of the coefficients of the

characteristic polynomial of the adjacency matrix of a

hydrogen-suppressed graph of a molecule.

`avg = 1' returns the information content divided by the total

population.

From D. Bonchev \& N. Trinajstic, J. Chem. Phys. vol 67,

4517-4533 (1977)
\begin{quote}

----\textgreater{}Ipc(log value)
\end{quote}

Usage:
\begin{quote}

result=CalculateIpc(mol)

Input: mol is a molecule object

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateMZagreb1() (in module topology)}

\begin{fulllineitems}
\phantomsection\label{reference/topology:topology.CalculateMZagreb1}\pysiglinewithargsret{\sphinxcode{topology.}\sphinxbfcode{CalculateMZagreb1}}{\emph{mol}}{}
Calculation of Modified Zagreb index with order 1 in a molecule

----\textgreater{}MZM1

Usage:
\begin{quote}

result=CalculateMZagreb1(mol)

Input: mol is a molecule object

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateMZagreb2() (in module topology)}

\begin{fulllineitems}
\phantomsection\label{reference/topology:topology.CalculateMZagreb2}\pysiglinewithargsret{\sphinxcode{topology.}\sphinxbfcode{CalculateMZagreb2}}{\emph{mol}}{}
Calculation of Modified Zagreb index with order 2 in a molecule

----\textgreater{}MZM2

Usage:
\begin{quote}

result=CalculateMZagreb2(mol)

Input: mol is a molecule object

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateMeanWeiner() (in module topology)}

\begin{fulllineitems}
\phantomsection\label{reference/topology:topology.CalculateMeanWeiner}\pysiglinewithargsret{\sphinxcode{topology.}\sphinxbfcode{CalculateMeanWeiner}}{\emph{mol}}{}
Calculation of Mean Weiner number in a molecule

----\textgreater{}AW

Usage:
\begin{quote}

result=CalculateWeiner(mol)

Input: mol is a molecule object

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculatePetitjean() (in module topology)}

\begin{fulllineitems}
\phantomsection\label{reference/topology:topology.CalculatePetitjean}\pysiglinewithargsret{\sphinxcode{topology.}\sphinxbfcode{CalculatePetitjean}}{\emph{mol}}{}
Calculation of Petitjean based on topology.

Value of (diameter - radius) / diameter as defined in {[}Petitjean 1992{]}.

----\textgreater{}petitjeant

Usage:
\begin{quote}

result=CalculatePetitjean(mol)

Input: mol is a molecule object

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculatePlatt() (in module topology)}

\begin{fulllineitems}
\phantomsection\label{reference/topology:topology.CalculatePlatt}\pysiglinewithargsret{\sphinxcode{topology.}\sphinxbfcode{CalculatePlatt}}{\emph{mol}}{}
Calculation of Platt number in a molecule

----\textgreater{}Platt

Usage:
\begin{quote}

result=CalculatePlatt(mol)

Input: mol is a molecule object

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculatePoglianiIndex() (in module topology)}

\begin{fulllineitems}
\phantomsection\label{reference/topology:topology.CalculatePoglianiIndex}\pysiglinewithargsret{\sphinxcode{topology.}\sphinxbfcode{CalculatePoglianiIndex}}{\emph{mol}}{}
Calculation of Poglicani index

The Pogliani index (Dz) is the sum over all non-hydrogen atoms

of a modified vertex degree calculated as the ratio

of the number of valence electrons over the principal

quantum number of an atom {[}L. Pogliani, J.Phys.Chem.

1996, 100, 18065-18077{]}.

----\textgreater{}DZ

Usage:
\begin{quote}

result=CalculatePoglianiIndex(mol)

Input: mol is a molecule object

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculatePolarityNumber() (in module topology)}

\begin{fulllineitems}
\phantomsection\label{reference/topology:topology.CalculatePolarityNumber}\pysiglinewithargsret{\sphinxcode{topology.}\sphinxbfcode{CalculatePolarityNumber}}{\emph{mol}}{}
Calculation of Polarity number.

It is the number of pairs of vertexes at

distance matrix equal to 3

----\textgreater{}Pol

Usage:
\begin{quote}

result=CalculatePolarityNumber(mol)

Input: mol is a molecule object

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateQuadratic() (in module topology)}

\begin{fulllineitems}
\phantomsection\label{reference/topology:topology.CalculateQuadratic}\pysiglinewithargsret{\sphinxcode{topology.}\sphinxbfcode{CalculateQuadratic}}{\emph{mol}}{}
Calculation of Quadratic index in a molecule

----\textgreater{}Qindex

Usage:
\begin{quote}

result=CalculateQuadratic(mol)

Input: mol is a molecule object

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateRadius() (in module topology)}

\begin{fulllineitems}
\phantomsection\label{reference/topology:topology.CalculateRadius}\pysiglinewithargsret{\sphinxcode{topology.}\sphinxbfcode{CalculateRadius}}{\emph{mol}}{}
Calculation of radius based on topology.

It is :If ri is the largest matrix entry in row i of the distance

matrix D,then the radius is defined as the smallest of the ri

{[}Petitjean 1992{]}.

----\textgreater{}radiust

Usage:
\begin{quote}

result=CalculateRadius(mol)

Input: mol is a molecule object

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateSchiultz() (in module topology)}

\begin{fulllineitems}
\phantomsection\label{reference/topology:topology.CalculateSchiultz}\pysiglinewithargsret{\sphinxcode{topology.}\sphinxbfcode{CalculateSchiultz}}{\emph{mol}}{}
Calculation of Schiultz number

----\textgreater{}Tsch(log value)

Usage:
\begin{quote}

result=CalculateSchiultz(mol)

Input: mol is a molecule object

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateSimpleTopoIndex() (in module topology)}

\begin{fulllineitems}
\phantomsection\label{reference/topology:topology.CalculateSimpleTopoIndex}\pysiglinewithargsret{\sphinxcode{topology.}\sphinxbfcode{CalculateSimpleTopoIndex}}{\emph{mol}}{}
Calculation of the logarithm of the simple topological index by Narumi,

which is defined as the product of the vertex degree.

----\textgreater{}Sito

Usage:
\begin{quote}

result=CalculateSimpleTopoIndex(mol)

Input: mol is a molecule object

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateWeiner() (in module topology)}

\begin{fulllineitems}
\phantomsection\label{reference/topology:topology.CalculateWeiner}\pysiglinewithargsret{\sphinxcode{topology.}\sphinxbfcode{CalculateWeiner}}{\emph{mol}}{}
Calculation of Weiner number in a molecule

----\textgreater{}W

Usage:
\begin{quote}

result=CalculateWeiner(mol)

Input: mol is a molecule object

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateXuIndex() (in module topology)}

\begin{fulllineitems}
\phantomsection\label{reference/topology:topology.CalculateXuIndex}\pysiglinewithargsret{\sphinxcode{topology.}\sphinxbfcode{CalculateXuIndex}}{\emph{mol}}{}
Calculation of Xu index

----\textgreater{}Xu

Usage:
\begin{quote}

result=CalculateXuIndex(mol)

Input: mol is a molecule object

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateZagreb1() (in module topology)}

\begin{fulllineitems}
\phantomsection\label{reference/topology:topology.CalculateZagreb1}\pysiglinewithargsret{\sphinxcode{topology.}\sphinxbfcode{CalculateZagreb1}}{\emph{mol}}{}
Calculation of Zagreb index with order 1 in a molecule

----\textgreater{}ZM1

Usage:
\begin{quote}

result=CalculateZagreb1(mol)

Input: mol is a molecule object

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{CalculateZagreb2() (in module topology)}

\begin{fulllineitems}
\phantomsection\label{reference/topology:topology.CalculateZagreb2}\pysiglinewithargsret{\sphinxcode{topology.}\sphinxbfcode{CalculateZagreb2}}{\emph{mol}}{}
Calculation of Zagreb index with order 2 in a molecule

----\textgreater{}ZM2

Usage:
\begin{quote}

result=CalculateZagreb2(mol)

Input: mol is a molecule object

Output: result is a numeric value
\end{quote}

\end{fulllineitems}

\index{GetTopology() (in module topology)}

\begin{fulllineitems}
\phantomsection\label{reference/topology:topology.GetTopology}\pysiglinewithargsret{\sphinxcode{topology.}\sphinxbfcode{GetTopology}}{\emph{mol}}{}
Get the dictionary of constitutional descriptors for given

moelcule mol

Usage:
\begin{quote}

result=CalculateWeiner(mol)

Input: mol is a molecule object

Output: result is a dict form containing all topological indices.
\end{quote}

\end{fulllineitems}



\subsection{Scaffolds module}
\label{reference/Scaffolds::doc}\label{reference/Scaffolds:scaffolds-module}\label{reference/Scaffolds:module-Scaffolds}\index{Scaffolds (module)}
If you have any question please contact me via email.

Bemis, G. W.; Murcko, M. A. “The Properties of Known Drugs. 1. Molecular Frameworks.” 
J. Med. Chem. 39:2887-93 (1996).

2016.11.15

@author: Zhijiang Yao and Dongsheng Cao

Email: \href{mailto:gadsby@163.com}{gadsby@163.com} and \href{mailto:oriental-cds@163.com}{oriental-cds@163.com}


\bigskip\hrule{}\bigskip

\index{GetScaffold() (in module Scaffolds)}

\begin{fulllineitems}
\phantomsection\label{reference/Scaffolds:Scaffolds.GetScaffold}\pysiglinewithargsret{\sphinxcode{Scaffolds.}\sphinxbfcode{GetScaffold}}{\emph{mol}, \emph{generic\_framework=False}}{}
Calculate Scaffold

Usage:
\begin{quote}

result = GetScaffold(mol)

Input: mol is a molecule object.

generic\_framework is boolean value. If the generic\_framework is True, the

result returns a generic framework.

Output: result is a string form of the molecule's scaffold.
\end{quote}

\end{fulllineitems}

\index{GetUniqueScaffold() (in module Scaffolds)}

\begin{fulllineitems}
\phantomsection\label{reference/Scaffolds:Scaffolds.GetUniqueScaffold}\pysiglinewithargsret{\sphinxcode{Scaffolds.}\sphinxbfcode{GetUniqueScaffold}}{\emph{mols}, \emph{generic\_framework=False}}{}
Calculate molecules' unique scaffolds

Usage:
\begin{quote}

result = GetUniqueScaffold(mols)

Input: mols is molecules object.

generic\_framework is boolean value. If the generic\_framework is True, the

result returns a generic framework dictionary.

Output: result is a tuple form. The first is the list of

unique scaffolds. The second is a dict form whose keys are the

scaffolds and the values are the scaffolds' count.
\end{quote}

\end{fulllineitems}



\section{PyProtein}
\label{reference/PyProtein:pyprotein}\label{reference/PyProtein::doc}

\subsection{PyProtein module}
\label{reference/PyProteinclass:module-PyProtein}\label{reference/PyProteinclass:pyprotein-module}\label{reference/PyProteinclass::doc}\index{PyProtein (module)}
A class used for computing different types of protein descriptors!

You can freely use and distribute it. If you have any problem,

you could contact with us timely.

Authors: Zhijiang Yao and Dongsheng Cao.

Date: 2016.06.04

Email: \href{mailto:gadsby@163.com}{gadsby@163.com}
\index{PyProtein (class in PyProtein)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinclass:PyProtein.PyProtein}\pysiglinewithargsret{\sphinxstrong{class }\sphinxcode{PyProtein.}\sphinxbfcode{PyProtein}}{\emph{ProteinSequence='`}}{}
This GetProDes class aims at collecting all descriptor calcualtion modules into a simple class.
\index{AALetter (PyProtein.PyProtein attribute)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinclass:PyProtein.PyProtein.AALetter}\pysigline{\sphinxbfcode{AALetter}\sphinxstrong{ = {[}'A', `R', `N', `D', `C', `E', `Q', `G', `H', `I', `L', `K', `M', `F', `P', `S', `T', `W', `Y', `V'{]}}}
\end{fulllineitems}

\index{GetAAComp() (PyProtein.PyProtein method)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinclass:PyProtein.PyProtein.GetAAComp}\pysiglinewithargsret{\sphinxbfcode{GetAAComp}}{}{}
amino acid compositon descriptors (20)

Usage:

result = GetAAComp()

\end{fulllineitems}

\index{GetAAindex1() (PyProtein.PyProtein method)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinclass:PyProtein.PyProtein.GetAAindex1}\pysiglinewithargsret{\sphinxbfcode{GetAAindex1}}{\emph{name}, \emph{path='.'}}{}
Get the amino acid property values from aaindex1

Usage:

result=GetAAIndex1(name)

Input: name is the name of amino acid property (e.g., KRIW790103)

Output: result is a dict form containing the properties of 20 amino acids

\end{fulllineitems}

\index{GetAAindex23() (PyProtein.PyProtein method)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinclass:PyProtein.PyProtein.GetAAindex23}\pysiglinewithargsret{\sphinxbfcode{GetAAindex23}}{\emph{name}, \emph{path='.'}}{}
Get the amino acid property values from aaindex2 and aaindex3

Usage:

result=GetAAIndex23(name)

Input: name is the name of amino acid property (e.g.,TANS760101,GRAR740104)

Output: result is a dict form containing the properties of 400 amino acid pairs

\end{fulllineitems}

\index{GetALL() (PyProtein.PyProtein method)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinclass:PyProtein.PyProtein.GetALL}\pysiglinewithargsret{\sphinxbfcode{GetALL}}{}{}
Calcualte all descriptors except tri-peptide descriptors

\end{fulllineitems}

\index{GetAPAAC() (PyProtein.PyProtein method)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinclass:PyProtein.PyProtein.GetAPAAC}\pysiglinewithargsret{\sphinxbfcode{GetAPAAC}}{\emph{lamda=10}, \emph{weight=0.5}}{}
Amphiphilic (Type II) Pseudo amino acid composition descriptors

default is 30

Usage:

result = GetAPAAC(lamda=10,weight=0.5)

lamda factor reflects the rank of correlation and is a non-Negative integer, such as 15.

Note that (1)lamda should NOT be larger than the length of input protein sequence;
\begin{enumerate}
\setcounter{enumi}{1}
\item {} 
lamda must be non-Negative integer, such as 0, 1, 2, ...; (3) when lamda =0, the

\end{enumerate}

output of PseAA server is the 20-D amino acid composition.

weight factor is designed for the users to put weight on the additional PseAA components

with respect to the conventional AA components. The user can select any value within the

region from 0.05 to 0.7 for the weight factor.

\end{fulllineitems}

\index{GetCTD() (PyProtein.PyProtein method)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinclass:PyProtein.PyProtein.GetCTD}\pysiglinewithargsret{\sphinxbfcode{GetCTD}}{}{}
Composition Transition Distribution descriptors (147)

Usage:

result = GetCTD()

\end{fulllineitems}

\index{GetDPComp() (PyProtein.PyProtein method)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinclass:PyProtein.PyProtein.GetDPComp}\pysiglinewithargsret{\sphinxbfcode{GetDPComp}}{}{}
dipeptide composition descriptors (400)

Usage:

result = GetDPComp()

\end{fulllineitems}

\index{GetGearyAuto() (PyProtein.PyProtein method)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinclass:PyProtein.PyProtein.GetGearyAuto}\pysiglinewithargsret{\sphinxbfcode{GetGearyAuto}}{}{}
Geary autocorrelation descriptors (240)

Usage:

result = GetGearyAuto()

\end{fulllineitems}

\index{GetGearyAutop() (PyProtein.PyProtein method)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinclass:PyProtein.PyProtein.GetGearyAutop}\pysiglinewithargsret{\sphinxbfcode{GetGearyAutop}}{\emph{AAP=\{\}}, \emph{AAPName='p'}}{}
Geary autocorrelation descriptors for the given property (30)

Usage:

result = GetGearyAutop(AAP=\{\},AAPName='p')

AAP is a dict containing physicochemical properities of 20 amino acids

\end{fulllineitems}

\index{GetMoranAuto() (PyProtein.PyProtein method)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinclass:PyProtein.PyProtein.GetMoranAuto}\pysiglinewithargsret{\sphinxbfcode{GetMoranAuto}}{}{}
Moran autocorrelation descriptors (240)

Usage:

result = GetMoranAuto()

\end{fulllineitems}

\index{GetMoranAutop() (PyProtein.PyProtein method)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinclass:PyProtein.PyProtein.GetMoranAutop}\pysiglinewithargsret{\sphinxbfcode{GetMoranAutop}}{\emph{AAP=\{\}}, \emph{AAPName='p'}}{}
Moran autocorrelation descriptors for the given property (30)

Usage:

result = GetMoranAutop(AAP=\{\},AAPName='p')

AAP is a dict containing physicochemical properities of 20 amino acids

\end{fulllineitems}

\index{GetMoreauBrotoAuto() (PyProtein.PyProtein method)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinclass:PyProtein.PyProtein.GetMoreauBrotoAuto}\pysiglinewithargsret{\sphinxbfcode{GetMoreauBrotoAuto}}{}{}
Normalized Moreau-Broto autocorrelation descriptors (240)

Usage:

result = GetMoreauBrotoAuto()

\end{fulllineitems}

\index{GetMoreauBrotoAutop() (PyProtein.PyProtein method)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinclass:PyProtein.PyProtein.GetMoreauBrotoAutop}\pysiglinewithargsret{\sphinxbfcode{GetMoreauBrotoAutop}}{\emph{AAP=\{\}}, \emph{AAPName='p'}}{}
Normalized Moreau-Broto autocorrelation descriptors for the given property (30)

Usage:

result = GetMoreauBrotoAutop(AAP=\{\},AAPName='p')

AAP is a dict containing physicochemical properities of 20 amino acids

\end{fulllineitems}

\index{GetPAAC() (PyProtein.PyProtein method)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinclass:PyProtein.PyProtein.GetPAAC}\pysiglinewithargsret{\sphinxbfcode{GetPAAC}}{\emph{lamda=10}, \emph{weight=0.05}}{}
Type I Pseudo amino acid composition descriptors (default is 30)

Usage:

result = GetPAAC(lamda=10,weight=0.05)

lamda factor reflects the rank of correlation and is a non-Negative integer, such as 15.

Note that (1)lamda should NOT be larger than the length of input protein sequence;
\begin{enumerate}
\setcounter{enumi}{1}
\item {} 
lamda must be non-Negative integer, such as 0, 1, 2, ...; (3) when lamda =0, the

\end{enumerate}

output of PseAA server is the 20-D amino acid composition.

weight factor is designed for the users to put weight on the additional PseAA components

with respect to the conventional AA components. The user can select any value within the

region from 0.05 to 0.7 for the weight factor.

\end{fulllineitems}

\index{GetPAACp() (PyProtein.PyProtein method)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinclass:PyProtein.PyProtein.GetPAACp}\pysiglinewithargsret{\sphinxbfcode{GetPAACp}}{\emph{lamda=10}, \emph{weight=0.05}, \emph{AAP={[}{]}}}{}
Type I Pseudo amino acid composition descriptors for the given properties (default is 30)

Usage:

result = GetPAACp(lamda=10,weight=0.05,AAP={[}{]})

lamda factor reflects the rank of correlation and is a non-Negative integer, such as 15.

Note that (1)lamda should NOT be larger than the length of input protein sequence;
\begin{enumerate}
\setcounter{enumi}{1}
\item {} 
lamda must be non-Negative integer, such as 0, 1, 2, ...; (3) when lamda =0, the

\end{enumerate}

output of PseAA server is the 20-D amino acid composition.

weight factor is designed for the users to put weight on the additional PseAA components

with respect to the conventional AA components. The user can select any value within the

region from 0.05 to 0.7 for the weight factor.

AAP is a list form containing the properties, each of which is a dict form.

\end{fulllineitems}

\index{GetQSO() (PyProtein.PyProtein method)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinclass:PyProtein.PyProtein.GetQSO}\pysiglinewithargsret{\sphinxbfcode{GetQSO}}{\emph{maxlag=30}, \emph{weight=0.1}}{}
Quasi sequence order descriptors  default is 50

result = GetQSO(maxlag=30, weight=0.1)

maxlag is the maximum lag and the length of the protein should be larger

than maxlag. default is 45.

\end{fulllineitems}

\index{GetQSOp() (PyProtein.PyProtein method)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinclass:PyProtein.PyProtein.GetQSOp}\pysiglinewithargsret{\sphinxbfcode{GetQSOp}}{\emph{maxlag=30}, \emph{weight=0.1}, \emph{distancematrix=\{\}}}{}
Quasi sequence order descriptors  default is 50

result = GetQSO(maxlag=30, weight=0.1)

maxlag is the maximum lag and the length of the protein should be larger

than maxlag. default is 45.

distancematrix is a dict form containing 400 distance values

\end{fulllineitems}

\index{GetSOCN() (PyProtein.PyProtein method)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinclass:PyProtein.PyProtein.GetSOCN}\pysiglinewithargsret{\sphinxbfcode{GetSOCN}}{\emph{maxlag=45}}{}
Sequence order coupling numbers  default is 45

Usage:

result = GetSOCN(maxlag=45)

maxlag is the maximum lag and the length of the protein should be larger

than maxlag. default is 45.

\end{fulllineitems}

\index{GetSOCNp() (PyProtein.PyProtein method)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinclass:PyProtein.PyProtein.GetSOCNp}\pysiglinewithargsret{\sphinxbfcode{GetSOCNp}}{\emph{maxlag=45}, \emph{distancematrix=\{\}}}{}
Sequence order coupling numbers  default is 45

Usage:

result = GetSOCN(maxlag=45)

maxlag is the maximum lag and the length of the protein should be larger

than maxlag. default is 45.

distancematrix is a dict form containing 400 distance values

\end{fulllineitems}

\index{GetSubSeq() (PyProtein.PyProtein method)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinclass:PyProtein.PyProtein.GetSubSeq}\pysiglinewithargsret{\sphinxbfcode{GetSubSeq}}{\emph{ToAA='S'}, \emph{window=3}}{}
obtain the sub sequences wit length 2*window+1, whose central point is ToAA

Usage:

result = GetSubSeq(ToAA='S',window=3)

ToAA is the central (query point) amino acid in the sub-sequence.

window is the span.

\end{fulllineitems}

\index{GetTPComp() (PyProtein.PyProtein method)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinclass:PyProtein.PyProtein.GetTPComp}\pysiglinewithargsret{\sphinxbfcode{GetTPComp}}{}{}
tri-peptide composition descriptors (8000)

Usage:

result = GetTPComp()

\end{fulllineitems}

\index{GetTriad() (PyProtein.PyProtein method)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinclass:PyProtein.PyProtein.GetTriad}\pysiglinewithargsret{\sphinxbfcode{GetTriad}}{}{}
Calculate the conjoint triad features from protein sequence.

Useage:

res = GetTriad()

Output is a dict form containing all 343 conjoint triad features.

\end{fulllineitems}

\index{Version (PyProtein.PyProtein attribute)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinclass:PyProtein.PyProtein.Version}\pysigline{\sphinxbfcode{Version}\sphinxstrong{ = 1.0}}
\end{fulllineitems}


\end{fulllineitems}



\subsection{AAComposition module}
\label{reference/AAComposition:aacomposition-module}\label{reference/AAComposition:module-AAComposition}\label{reference/AAComposition::doc}\index{AAComposition (module)}

\bigskip\hrule{}\bigskip


The module is used for computing the composition of amino acids, dipetide and

3-mers (tri-peptide) for a given protein sequence. You can get 8420 descriptors

for a given protein sequence. You can freely use and distribute it. If you hava

any problem, you could contact with us timely!

References:

{[}1{]}: Reczko, M. and Bohr, H. (1994) The DEF data base of sequence based protein

fold class predictions. Nucleic Acids Res, 22, 3616-3619.

{[}2{]}: Hua, S. and Sun, Z. (2001) Support vector machine approach for protein

subcellular localization prediction. Bioinformatics, 17, 721-728.

{[}3{]}:Grassmann, J., Reczko, M., Suhai, S. and Edler, L. (1999) Protein fold class

prediction: new methods of statistical classification. Proc Int Conf Intell Syst Mol

Biol, 106-112.

Authors: Dongsheng Cao and Yizeng Liang.

Date: 2012.3.27

Email: \href{mailto:oriental-cds@163.com}{oriental-cds@163.com}


\bigskip\hrule{}\bigskip

\index{CalculateAAComposition() (in module AAComposition)}

\begin{fulllineitems}
\phantomsection\label{reference/AAComposition:AAComposition.CalculateAAComposition}\pysiglinewithargsret{\sphinxcode{AAComposition.}\sphinxbfcode{CalculateAAComposition}}{\emph{ProteinSequence}}{}
Calculate the composition of Amino acids

for a given protein sequence.

Usage:

result=CalculateAAComposition(protein)

Input: protein is a pure protein sequence.

Output: result is a dict form containing the composition of

\end{fulllineitems}

\index{CalculateAADipeptideComposition() (in module AAComposition)}

\begin{fulllineitems}
\phantomsection\label{reference/AAComposition:AAComposition.CalculateAADipeptideComposition}\pysiglinewithargsret{\sphinxcode{AAComposition.}\sphinxbfcode{CalculateAADipeptideComposition}}{\emph{ProteinSequence}}{}
Calculate the composition of AADs, dipeptide and 3-mers for a

given protein sequence.

Usage:

result=CalculateAADipeptideComposition(protein)

Input: protein is a pure protein sequence.

Output: result is a dict form containing all composition values of

\end{fulllineitems}

\index{CalculateDipeptideComposition() (in module AAComposition)}

\begin{fulllineitems}
\phantomsection\label{reference/AAComposition:AAComposition.CalculateDipeptideComposition}\pysiglinewithargsret{\sphinxcode{AAComposition.}\sphinxbfcode{CalculateDipeptideComposition}}{\emph{ProteinSequence}}{}
Calculate the composition of dipeptidefor a given protein sequence.

Usage:

result=CalculateDipeptideComposition(protein)

Input: protein is a pure protein sequence.

Output: result is a dict form containing the composition of

\end{fulllineitems}

\index{GetSpectrumDict() (in module AAComposition)}

\begin{fulllineitems}
\phantomsection\label{reference/AAComposition:AAComposition.GetSpectrumDict}\pysiglinewithargsret{\sphinxcode{AAComposition.}\sphinxbfcode{GetSpectrumDict}}{\emph{proteinsequence}}{}
Calcualte the spectrum descriptors of 3-mers for a given protein.

Usage:

result=GetSpectrumDict(protein)

Input: protein is a pure protein sequence.

Output: result is a dict form containing the composition values of 8000

\end{fulllineitems}

\index{Getkmers() (in module AAComposition)}

\begin{fulllineitems}
\phantomsection\label{reference/AAComposition:AAComposition.Getkmers}\pysiglinewithargsret{\sphinxcode{AAComposition.}\sphinxbfcode{Getkmers}}{}{}
Get the amino acid list of 3-mers.

Usage:

result=Getkmers()

Output: result is a list form containing 8000 tri-peptides.

\end{fulllineitems}



\subsection{AAIndex module}
\label{reference/AAIndex:aaindex-module}\label{reference/AAIndex:module-AAIndex}\label{reference/AAIndex::doc}\index{AAIndex (module)}
This module is used for obtaining the properties of amino acids or their pairs

from the aaindex database. You can freely use and distribute it. If you hava

any problem, you could contact with us timely!

Authors: Zhijiang Yao and Dongsheng Cao.

Date: 2016.06.04

Email: \href{mailto:gadsby@163.com}{gadsby@163.com}
\index{GetAAIndex1() (in module AAIndex)}

\begin{fulllineitems}
\phantomsection\label{reference/AAIndex:AAIndex.GetAAIndex1}\pysiglinewithargsret{\sphinxcode{AAIndex.}\sphinxbfcode{GetAAIndex1}}{\emph{name}, \emph{path='.'}}{}
Get the amino acid property values from aaindex1

Usage:

result=GetAAIndex1(name)

Input: name is the name of amino acid property (e.g., KRIW790103)

Output: result is a dict form containing the properties of 20 amino acids

\end{fulllineitems}

\index{GetAAIndex23() (in module AAIndex)}

\begin{fulllineitems}
\phantomsection\label{reference/AAIndex:AAIndex.GetAAIndex23}\pysiglinewithargsret{\sphinxcode{AAIndex.}\sphinxbfcode{GetAAIndex23}}{\emph{name}, \emph{path='.'}}{}
Get the amino acid property values from aaindex2 and aaindex3

Usage:

result=GetAAIndex23(name)

Input: name is the name of amino acid property (e.g.,TANS760101,GRAR740104)

Output: result is a dict form containing the properties of 400 amino acid pairs

\end{fulllineitems}

\index{MatrixRecord (class in AAIndex)}

\begin{fulllineitems}
\phantomsection\label{reference/AAIndex:AAIndex.MatrixRecord}\pysigline{\sphinxstrong{class }\sphinxcode{AAIndex.}\sphinxbfcode{MatrixRecord}}
Bases: {\hyperref[reference/AAIndex:AAIndex.Record]{\sphinxcrossref{\sphinxcode{AAIndex.Record}}}}

Matrix record for mutation matrices or pair-wise contact potentials
\index{extend() (AAIndex.MatrixRecord method)}

\begin{fulllineitems}
\phantomsection\label{reference/AAIndex:AAIndex.MatrixRecord.extend}\pysiglinewithargsret{\sphinxbfcode{extend}}{\emph{row}}{}
\end{fulllineitems}

\index{get() (AAIndex.MatrixRecord method)}

\begin{fulllineitems}
\phantomsection\label{reference/AAIndex:AAIndex.MatrixRecord.get}\pysiglinewithargsret{\sphinxbfcode{get}}{\emph{aai}, \emph{aaj}, \emph{d=None}}{}
\end{fulllineitems}

\index{median() (AAIndex.MatrixRecord method)}

\begin{fulllineitems}
\phantomsection\label{reference/AAIndex:AAIndex.MatrixRecord.median}\pysiglinewithargsret{\sphinxbfcode{median}}{}{}
\end{fulllineitems}


\end{fulllineitems}

\index{Record (class in AAIndex)}

\begin{fulllineitems}
\phantomsection\label{reference/AAIndex:AAIndex.Record}\pysigline{\sphinxstrong{class }\sphinxcode{AAIndex.}\sphinxbfcode{Record}}
Amino acid index (AAindex) Record
\index{aakeys (AAIndex.Record attribute)}

\begin{fulllineitems}
\phantomsection\label{reference/AAIndex:AAIndex.Record.aakeys}\pysigline{\sphinxbfcode{aakeys}\sphinxstrong{ = `ARNDCQEGHILKMFPSTWYV'}}
\end{fulllineitems}

\index{extend() (AAIndex.Record method)}

\begin{fulllineitems}
\phantomsection\label{reference/AAIndex:AAIndex.Record.extend}\pysiglinewithargsret{\sphinxbfcode{extend}}{\emph{row}}{}
\end{fulllineitems}

\index{get() (AAIndex.Record method)}

\begin{fulllineitems}
\phantomsection\label{reference/AAIndex:AAIndex.Record.get}\pysiglinewithargsret{\sphinxbfcode{get}}{\emph{aai}, \emph{aaj=None}, \emph{d=None}}{}
\end{fulllineitems}

\index{median() (AAIndex.Record method)}

\begin{fulllineitems}
\phantomsection\label{reference/AAIndex:AAIndex.Record.median}\pysiglinewithargsret{\sphinxbfcode{median}}{}{}
\end{fulllineitems}


\end{fulllineitems}

\index{get() (in module AAIndex)}

\begin{fulllineitems}
\phantomsection\label{reference/AAIndex:AAIndex.get}\pysiglinewithargsret{\sphinxcode{AAIndex.}\sphinxbfcode{get}}{\emph{key}}{}
Get record for key

\end{fulllineitems}

\index{grep() (in module AAIndex)}

\begin{fulllineitems}
\phantomsection\label{reference/AAIndex:AAIndex.grep}\pysiglinewithargsret{\sphinxcode{AAIndex.}\sphinxbfcode{grep}}{\emph{pattern}}{}
Search for pattern in title and description of all records (case
insensitive) and print results on standard output.

\end{fulllineitems}

\index{init() (in module AAIndex)}

\begin{fulllineitems}
\phantomsection\label{reference/AAIndex:AAIndex.init}\pysiglinewithargsret{\sphinxcode{AAIndex.}\sphinxbfcode{init}}{\emph{path=None}, \emph{index=`123'}}{}
Read in the aaindex files. You need to run this (once) before you can
access any records. If the files are not within the current directory,
you need to specify the correct directory path. By default all three
aaindex files are read in.

\end{fulllineitems}

\index{init\_from\_file() (in module AAIndex)}

\begin{fulllineitems}
\phantomsection\label{reference/AAIndex:AAIndex.init_from_file}\pysiglinewithargsret{\sphinxcode{AAIndex.}\sphinxbfcode{init\_from\_file}}{\emph{filename}, \emph{type=\textless{}class AAIndex.Record at 0x0380B998\textgreater{}}}{}
\end{fulllineitems}

\index{search() (in module AAIndex)}

\begin{fulllineitems}
\phantomsection\label{reference/AAIndex:AAIndex.search}\pysiglinewithargsret{\sphinxcode{AAIndex.}\sphinxbfcode{search}}{\emph{pattern}, \emph{searchtitle=True}, \emph{casesensitive=False}}{}
Search for pattern in description and title (optional) of all records and
return matched records as list. By default search case insensitive.

\end{fulllineitems}



\subsection{Autocorrelation module}
\label{reference/Autocorrelation::doc}\label{reference/Autocorrelation:autocorrelation-module}\label{reference/Autocorrelation:module-Autocorrelation}\index{Autocorrelation (module)}

\bigskip\hrule{}\bigskip


This module is used for computing the Autocorrelation descriptors based different
\begin{quote}

properties of AADs.You can also input your properties of AADs, then it can help you
\end{quote}

to compute Autocorrelation descriptors based on the property of AADs. Currently, You

can get 720 descriptors for a given protein sequence based on our provided physicochemical

properties of AADs. You can freely use and distribute it. If you hava  any problem,

you could contact with us timely!

References:

{[}1{]}: \url{http://www.genome.ad.jp/dbget/aaindex.html}

{[}2{]}:Feng, Z.P. and Zhang, C.T. (2000) Prediction of membrane protein types based on

the hydrophobic index of amino acids. J Protein Chem, 19, 269-275.

{[}3{]}:Horne, D.S. (1988) Prediction of protein helix content from an autocorrelation

analysis of sequence hydrophobicities. Biopolymers, 27, 451-477.

{[}4{]}:Sokal, R.R. and Thomson, B.A. (2006) Population structure inferred by local

spatial autocorrelation: an Usage from an Amerindian tribal population. Am J

Phys Anthropol, 129, 121-131.

Authors: Zhijiang Yao and Dongsheng Cao.

Date: 2016.06.04

Email: \href{mailto:gadsby@163.com}{gadsby@163.com}


\bigskip\hrule{}\bigskip

\index{CalculateAutoTotal() (in module Autocorrelation)}

\begin{fulllineitems}
\phantomsection\label{reference/Autocorrelation:Autocorrelation.CalculateAutoTotal}\pysiglinewithargsret{\sphinxcode{Autocorrelation.}\sphinxbfcode{CalculateAutoTotal}}{\emph{ProteinSequence}}{}
A method used for computing all autocorrelation descriptors based on 8 properties of AADs.

Usage:

result=CalculateGearyAutoTotal(protein)

Input: protein is a pure protein sequence.

Output: result is a dict form containing 30*8*3=720 normalized Moreau Broto, Moran, and Geary

\end{fulllineitems}

\index{CalculateEachGearyAuto() (in module Autocorrelation)}

\begin{fulllineitems}
\phantomsection\label{reference/Autocorrelation:Autocorrelation.CalculateEachGearyAuto}\pysiglinewithargsret{\sphinxcode{Autocorrelation.}\sphinxbfcode{CalculateEachGearyAuto}}{\emph{ProteinSequence}, \emph{AAP}, \emph{AAPName}}{}
you can use the function to compute GearyAuto

descriptors for different properties based on AADs.

Usage:

result=CalculateEachGearyAuto(protein,AAP,AAPName)

Input: protein is a pure protein sequence.

AAP is a dict form containing the properties of 20 amino acids (e.g., \_AvFlexibility).

AAPName is a string used for indicating the property (e.g., `\_AvFlexibility').

Output: result is a dict form containing 30 Geary autocorrelation

\end{fulllineitems}

\index{CalculateEachMoranAuto() (in module Autocorrelation)}

\begin{fulllineitems}
\phantomsection\label{reference/Autocorrelation:Autocorrelation.CalculateEachMoranAuto}\pysiglinewithargsret{\sphinxcode{Autocorrelation.}\sphinxbfcode{CalculateEachMoranAuto}}{\emph{ProteinSequence}, \emph{AAP}, \emph{AAPName}}{}
you can use the function to compute MoranAuto

descriptors for different properties based on AADs.

Usage:

result=CalculateEachMoranAuto(protein,AAP,AAPName)

Input: protein is a pure protein sequence.

AAP is a dict form containing the properties of 20 amino acids (e.g., \_AvFlexibility).

AAPName is a string used for indicating the property (e.g., `\_AvFlexibility').

Output: result is a dict form containing 30 Moran autocorrelation

\end{fulllineitems}

\index{CalculateEachNormalizedMoreauBrotoAuto() (in module Autocorrelation)}

\begin{fulllineitems}
\phantomsection\label{reference/Autocorrelation:Autocorrelation.CalculateEachNormalizedMoreauBrotoAuto}\pysiglinewithargsret{\sphinxcode{Autocorrelation.}\sphinxbfcode{CalculateEachNormalizedMoreauBrotoAuto}}{\emph{ProteinSequence}, \emph{AAP}, \emph{AAPName}}{}
you can use the function to compute MoreauBrotoAuto

descriptors for different properties based on AADs.

Usage:

result=CalculateEachNormalizedMoreauBrotoAuto(protein,AAP,AAPName)

Input: protein is a pure protein sequence.

AAP is a dict form containing the properties of 20 amino acids (e.g., \_AvFlexibility).

AAPName is a string used for indicating the property (e.g., `\_AvFlexibility').

Output: result is a dict form containing 30 Normalized Moreau-Broto autocorrelation

\end{fulllineitems}

\index{CalculateGearyAuto() (in module Autocorrelation)}

\begin{fulllineitems}
\phantomsection\label{reference/Autocorrelation:Autocorrelation.CalculateGearyAuto}\pysiglinewithargsret{\sphinxcode{Autocorrelation.}\sphinxbfcode{CalculateGearyAuto}}{\emph{ProteinSequence}, \emph{AAProperty}, \emph{AAPropertyName}}{}
A method used for computing GearyAuto for all properties

Usage:

result=CalculateGearyAuto(protein,AAP,AAPName)

Input: protein is a pure protein sequence.

AAProperty is a list or tuple form containing the properties of 20 amino acids (e.g., \_AAProperty).

AAPName is a list or tuple form used for indicating the property (e.g., `\_AAPropertyName').

Output: result is a dict form containing 30*p Geary autocorrelation

\end{fulllineitems}

\index{CalculateGearyAutoAvFlexibility() (in module Autocorrelation)}

\begin{fulllineitems}
\phantomsection\label{reference/Autocorrelation:Autocorrelation.CalculateGearyAutoAvFlexibility}\pysiglinewithargsret{\sphinxcode{Autocorrelation.}\sphinxbfcode{CalculateGearyAutoAvFlexibility}}{\emph{ProteinSequence}}{}
Calculte the GearyAuto Autocorrelation descriptors based on

AvFlexibility.

Usage:
result=CalculateGearyAutoAvFlexibility(protein)

Input: protein is a pure protein sequence.

Output: result is a dict form containing 30 Geary Autocorrelation

\end{fulllineitems}

\index{CalculateGearyAutoFreeEnergy() (in module Autocorrelation)}

\begin{fulllineitems}
\phantomsection\label{reference/Autocorrelation:Autocorrelation.CalculateGearyAutoFreeEnergy}\pysiglinewithargsret{\sphinxcode{Autocorrelation.}\sphinxbfcode{CalculateGearyAutoFreeEnergy}}{\emph{ProteinSequence}}{}
Calculte the GearyAuto Autocorrelation descriptors based on

FreeEnergy.

Usage:

result=CalculateGearyAutoFreeEnergy(protein)

Input: protein is a pure protein sequence.

Output: result is a dict form containing 30 Geary Autocorrelation

\end{fulllineitems}

\index{CalculateGearyAutoHydrophobicity() (in module Autocorrelation)}

\begin{fulllineitems}
\phantomsection\label{reference/Autocorrelation:Autocorrelation.CalculateGearyAutoHydrophobicity}\pysiglinewithargsret{\sphinxcode{Autocorrelation.}\sphinxbfcode{CalculateGearyAutoHydrophobicity}}{\emph{ProteinSequence}}{}
Calculte the GearyAuto Autocorrelation descriptors based on

hydrophobicity.

Usage:

result=CalculateGearyAutoHydrophobicity(protein)

Input: protein is a pure protein sequence.

Output: result is a dict form containing 30 Geary Autocorrelation

\end{fulllineitems}

\index{CalculateGearyAutoMutability() (in module Autocorrelation)}

\begin{fulllineitems}
\phantomsection\label{reference/Autocorrelation:Autocorrelation.CalculateGearyAutoMutability}\pysiglinewithargsret{\sphinxcode{Autocorrelation.}\sphinxbfcode{CalculateGearyAutoMutability}}{\emph{ProteinSequence}}{}
Calculte the GearyAuto Autocorrelation descriptors based on

Mutability.

Usage:

result=CalculateGearyAutoMutability(protein)

Input: protein is a pure protein sequence.

Output: result is a dict form containing 30 Geary Autocorrelation

\end{fulllineitems}

\index{CalculateGearyAutoPolarizability() (in module Autocorrelation)}

\begin{fulllineitems}
\phantomsection\label{reference/Autocorrelation:Autocorrelation.CalculateGearyAutoPolarizability}\pysiglinewithargsret{\sphinxcode{Autocorrelation.}\sphinxbfcode{CalculateGearyAutoPolarizability}}{\emph{ProteinSequence}}{}
Calculte the GearyAuto Autocorrelation descriptors based on

Polarizability.

Usage:

result=CalculateGearyAutoPolarizability(protein)

Input: protein is a pure protein sequence.

Output: result is a dict form containing 30 Geary Autocorrelation

\end{fulllineitems}

\index{CalculateGearyAutoResidueASA() (in module Autocorrelation)}

\begin{fulllineitems}
\phantomsection\label{reference/Autocorrelation:Autocorrelation.CalculateGearyAutoResidueASA}\pysiglinewithargsret{\sphinxcode{Autocorrelation.}\sphinxbfcode{CalculateGearyAutoResidueASA}}{\emph{ProteinSequence}}{}
Calculte the GearyAuto Autocorrelation descriptors based on

ResidueASA.

Usage:

result=CalculateGearyAutoResidueASA(protein)

Input: protein is a pure protein sequence.

Output: result is a dict form containing 30 Geary Autocorrelation

\end{fulllineitems}

\index{CalculateGearyAutoResidueVol() (in module Autocorrelation)}

\begin{fulllineitems}
\phantomsection\label{reference/Autocorrelation:Autocorrelation.CalculateGearyAutoResidueVol}\pysiglinewithargsret{\sphinxcode{Autocorrelation.}\sphinxbfcode{CalculateGearyAutoResidueVol}}{\emph{ProteinSequence}}{}
Calculte the GearyAuto Autocorrelation descriptors based on

ResidueVol.

Usage:

result=CalculateGearyAutoResidueVol(protein)

Input: protein is a pure protein sequence.

Output: result is a dict form containing 30 Geary Autocorrelation

\end{fulllineitems}

\index{CalculateGearyAutoSteric() (in module Autocorrelation)}

\begin{fulllineitems}
\phantomsection\label{reference/Autocorrelation:Autocorrelation.CalculateGearyAutoSteric}\pysiglinewithargsret{\sphinxcode{Autocorrelation.}\sphinxbfcode{CalculateGearyAutoSteric}}{\emph{ProteinSequence}}{}
Calculte the GearyAuto Autocorrelation descriptors based on

Steric.

Usage:

result=CalculateGearyAutoSteric(protein)

Input: protein is a pure protein sequence.

Output: result is a dict form containing 30 Geary Autocorrelation

\end{fulllineitems}

\index{CalculateGearyAutoTotal() (in module Autocorrelation)}

\begin{fulllineitems}
\phantomsection\label{reference/Autocorrelation:Autocorrelation.CalculateGearyAutoTotal}\pysiglinewithargsret{\sphinxcode{Autocorrelation.}\sphinxbfcode{CalculateGearyAutoTotal}}{\emph{ProteinSequence}}{}
A method used for computing Geary autocorrelation descriptors based on 8 properties of AADs.

Usage:

result=CalculateGearyAutoTotal(protein)

Input: protein is a pure protein sequence.

Output: result is a dict form containing 30*8=240 Geary

\end{fulllineitems}

\index{CalculateMoranAuto() (in module Autocorrelation)}

\begin{fulllineitems}
\phantomsection\label{reference/Autocorrelation:Autocorrelation.CalculateMoranAuto}\pysiglinewithargsret{\sphinxcode{Autocorrelation.}\sphinxbfcode{CalculateMoranAuto}}{\emph{ProteinSequence}, \emph{AAProperty}, \emph{AAPropertyName}}{}
A method used for computing MoranAuto for all properties

Usage:

result=CalculateMoranAuto(protein,AAP,AAPName)

Input: protein is a pure protein sequence.

AAProperty is a list or tuple form containing the properties of 20 amino acids (e.g., \_AAProperty).

AAPName is a list or tuple form used for indicating the property (e.g., `\_AAPropertyName').

Output: result is a dict form containing 30*p Moran autocorrelation

\end{fulllineitems}

\index{CalculateMoranAutoAvFlexibility() (in module Autocorrelation)}

\begin{fulllineitems}
\phantomsection\label{reference/Autocorrelation:Autocorrelation.CalculateMoranAutoAvFlexibility}\pysiglinewithargsret{\sphinxcode{Autocorrelation.}\sphinxbfcode{CalculateMoranAutoAvFlexibility}}{\emph{ProteinSequence}}{}
Calculte the MoranAuto Autocorrelation descriptors based on

AvFlexibility.

Usage:

result=CalculateMoranAutoAvFlexibility(protein)

Input: protein is a pure protein sequence.

Output: result is a dict form containing 30 Moran Autocorrelation

\end{fulllineitems}

\index{CalculateMoranAutoFreeEnergy() (in module Autocorrelation)}

\begin{fulllineitems}
\phantomsection\label{reference/Autocorrelation:Autocorrelation.CalculateMoranAutoFreeEnergy}\pysiglinewithargsret{\sphinxcode{Autocorrelation.}\sphinxbfcode{CalculateMoranAutoFreeEnergy}}{\emph{ProteinSequence}}{}
Calculte the MoranAuto Autocorrelation descriptors based on

FreeEnergy.

Usage:

result=CalculateMoranAutoFreeEnergy(protein)

Input: protein is a pure protein sequence.

Output: result is a dict form containing 30 Moran Autocorrelation

\end{fulllineitems}

\index{CalculateMoranAutoHydrophobicity() (in module Autocorrelation)}

\begin{fulllineitems}
\phantomsection\label{reference/Autocorrelation:Autocorrelation.CalculateMoranAutoHydrophobicity}\pysiglinewithargsret{\sphinxcode{Autocorrelation.}\sphinxbfcode{CalculateMoranAutoHydrophobicity}}{\emph{ProteinSequence}}{}
Calculte the MoranAuto Autocorrelation descriptors based on hydrophobicity.

Usage:

result=CalculateMoranAutoHydrophobicity(protein)

Input: protein is a pure protein sequence.

Output: result is a dict form containing 30 Moran Autocorrelation

\end{fulllineitems}

\index{CalculateMoranAutoMutability() (in module Autocorrelation)}

\begin{fulllineitems}
\phantomsection\label{reference/Autocorrelation:Autocorrelation.CalculateMoranAutoMutability}\pysiglinewithargsret{\sphinxcode{Autocorrelation.}\sphinxbfcode{CalculateMoranAutoMutability}}{\emph{ProteinSequence}}{}
Calculte the MoranAuto Autocorrelation descriptors based on

Mutability.

Usage:

result=CalculateMoranAutoMutability(protein)

Input: protein is a pure protein sequence.

Output: result is a dict form containing 30 Moran Autocorrelation

\end{fulllineitems}

\index{CalculateMoranAutoPolarizability() (in module Autocorrelation)}

\begin{fulllineitems}
\phantomsection\label{reference/Autocorrelation:Autocorrelation.CalculateMoranAutoPolarizability}\pysiglinewithargsret{\sphinxcode{Autocorrelation.}\sphinxbfcode{CalculateMoranAutoPolarizability}}{\emph{ProteinSequence}}{}
Calculte the MoranAuto Autocorrelation descriptors based on

Polarizability.

Usage:

result=CalculateMoranAutoPolarizability(protein)

Input: protein is a pure protein sequence.

Output: result is a dict form containing 30 Moran Autocorrelation

\end{fulllineitems}

\index{CalculateMoranAutoResidueASA() (in module Autocorrelation)}

\begin{fulllineitems}
\phantomsection\label{reference/Autocorrelation:Autocorrelation.CalculateMoranAutoResidueASA}\pysiglinewithargsret{\sphinxcode{Autocorrelation.}\sphinxbfcode{CalculateMoranAutoResidueASA}}{\emph{ProteinSequence}}{}
Calculte the MoranAuto Autocorrelation descriptors based on

ResidueASA.

Usage:

result=CalculateMoranAutoResidueASA(protein)

Input: protein is a pure protein sequence.

Output: result is a dict form containing 30 Moran Autocorrelation

\end{fulllineitems}

\index{CalculateMoranAutoResidueVol() (in module Autocorrelation)}

\begin{fulllineitems}
\phantomsection\label{reference/Autocorrelation:Autocorrelation.CalculateMoranAutoResidueVol}\pysiglinewithargsret{\sphinxcode{Autocorrelation.}\sphinxbfcode{CalculateMoranAutoResidueVol}}{\emph{ProteinSequence}}{}
Calculte the MoranAuto Autocorrelation descriptors based on

ResidueVol.

Usage:

result=CalculateMoranAutoResidueVol(protein)

Input: protein is a pure protein sequence.

Output: result is a dict form containing 30 Moran Autocorrelation

\end{fulllineitems}

\index{CalculateMoranAutoSteric() (in module Autocorrelation)}

\begin{fulllineitems}
\phantomsection\label{reference/Autocorrelation:Autocorrelation.CalculateMoranAutoSteric}\pysiglinewithargsret{\sphinxcode{Autocorrelation.}\sphinxbfcode{CalculateMoranAutoSteric}}{\emph{ProteinSequence}}{}
Calculte the MoranAuto Autocorrelation descriptors based on

AutoSteric.

Usage:

result=CalculateMoranAutoSteric(protein)

Input: protein is a pure protein sequence.

Output: result is a dict form containing 30 Moran Autocorrelation

\end{fulllineitems}

\index{CalculateMoranAutoTotal() (in module Autocorrelation)}

\begin{fulllineitems}
\phantomsection\label{reference/Autocorrelation:Autocorrelation.CalculateMoranAutoTotal}\pysiglinewithargsret{\sphinxcode{Autocorrelation.}\sphinxbfcode{CalculateMoranAutoTotal}}{\emph{ProteinSequence}}{}
A method used for computing Moran autocorrelation descriptors based on 8 properties of AADs.

Usage:

result=CalculateMoranAutoTotal(protein)

Input: protein is a pure protein sequence.

Output: result is a dict form containing 30*8=240 Moran

\end{fulllineitems}

\index{CalculateNormalizedMoreauBrotoAuto() (in module Autocorrelation)}

\begin{fulllineitems}
\phantomsection\label{reference/Autocorrelation:Autocorrelation.CalculateNormalizedMoreauBrotoAuto}\pysiglinewithargsret{\sphinxcode{Autocorrelation.}\sphinxbfcode{CalculateNormalizedMoreauBrotoAuto}}{\emph{ProteinSequence}, \emph{AAProperty}, \emph{AAPropertyName}}{}
A method used for computing MoreauBrotoAuto for all properties.

Usage:

result=CalculateNormalizedMoreauBrotoAuto(protein,AAP,AAPName)

Input: protein is a pure protein sequence.

AAProperty is a list or tuple form containing the properties of 20 amino acids (e.g., \_AAProperty).

AAPName is a list or tuple form used for indicating the property (e.g., `\_AAPropertyName').

Output: result is a dict form containing 30*p Normalized Moreau-Broto autocorrelation

\end{fulllineitems}

\index{CalculateNormalizedMoreauBrotoAutoAvFlexibility() (in module Autocorrelation)}

\begin{fulllineitems}
\phantomsection\label{reference/Autocorrelation:Autocorrelation.CalculateNormalizedMoreauBrotoAutoAvFlexibility}\pysiglinewithargsret{\sphinxcode{Autocorrelation.}\sphinxbfcode{CalculateNormalizedMoreauBrotoAutoAvFlexibility}}{\emph{ProteinSequence}}{}
Calculte the NormalizedMoreauBorto Autocorrelation descriptors based on

AvFlexibility.

Usage:

result=CalculateNormalizedMoreauBrotoAutoAvFlexibility(protein)

Input: protein is a pure protein sequence.

Output: result is a dict form containing 30 Normalized Moreau-Broto Autocorrelation

\end{fulllineitems}

\index{CalculateNormalizedMoreauBrotoAutoFreeEnergy() (in module Autocorrelation)}

\begin{fulllineitems}
\phantomsection\label{reference/Autocorrelation:Autocorrelation.CalculateNormalizedMoreauBrotoAutoFreeEnergy}\pysiglinewithargsret{\sphinxcode{Autocorrelation.}\sphinxbfcode{CalculateNormalizedMoreauBrotoAutoFreeEnergy}}{\emph{ProteinSequence}}{}
Calculte the NormalizedMoreauBorto Autocorrelation descriptors based on

FreeEnergy.

Usage:

result=CalculateNormalizedMoreauBrotoAutoFreeEnergy(protein)

Input: protein is a pure protein sequence.

Output: result is a dict form containing 30 Normalized Moreau-Broto Autocorrelation

\end{fulllineitems}

\index{CalculateNormalizedMoreauBrotoAutoHydrophobicity() (in module Autocorrelation)}

\begin{fulllineitems}
\phantomsection\label{reference/Autocorrelation:Autocorrelation.CalculateNormalizedMoreauBrotoAutoHydrophobicity}\pysiglinewithargsret{\sphinxcode{Autocorrelation.}\sphinxbfcode{CalculateNormalizedMoreauBrotoAutoHydrophobicity}}{\emph{ProteinSequence}}{}
Calculte the NormalizedMoreauBorto Autocorrelation descriptors based on

hydrophobicity.

Usage:

result=CalculateNormalizedMoreauBrotoAutoHydrophobicity(protein)

Input: protein is a pure protein sequence.

Output: result is a dict form containing 30 Normalized Moreau-Broto Autocorrelation

\end{fulllineitems}

\index{CalculateNormalizedMoreauBrotoAutoMutability() (in module Autocorrelation)}

\begin{fulllineitems}
\phantomsection\label{reference/Autocorrelation:Autocorrelation.CalculateNormalizedMoreauBrotoAutoMutability}\pysiglinewithargsret{\sphinxcode{Autocorrelation.}\sphinxbfcode{CalculateNormalizedMoreauBrotoAutoMutability}}{\emph{ProteinSequence}}{}
Calculte the NormalizedMoreauBorto Autocorrelation descriptors based on Mutability.

Usage:

result=CalculateNormalizedMoreauBrotoAutoMutability(protein)

Input: protein is a pure protein sequence.

Output: result is a dict form containing 30 Normalized Moreau-Broto Autocorrelation

\end{fulllineitems}

\index{CalculateNormalizedMoreauBrotoAutoPolarizability() (in module Autocorrelation)}

\begin{fulllineitems}
\phantomsection\label{reference/Autocorrelation:Autocorrelation.CalculateNormalizedMoreauBrotoAutoPolarizability}\pysiglinewithargsret{\sphinxcode{Autocorrelation.}\sphinxbfcode{CalculateNormalizedMoreauBrotoAutoPolarizability}}{\emph{ProteinSequence}}{}
Calculte the NormalizedMoreauBorto Autocorrelation descriptors based on

Polarizability.

Usage:

result=CalculateNormalizedMoreauBrotoAutoPolarizability(protein)

Input: protein is a pure protein sequence.

Output: result is a dict form containing 30 Normalized Moreau-Broto Autocorrelation

\end{fulllineitems}

\index{CalculateNormalizedMoreauBrotoAutoResidueASA() (in module Autocorrelation)}

\begin{fulllineitems}
\phantomsection\label{reference/Autocorrelation:Autocorrelation.CalculateNormalizedMoreauBrotoAutoResidueASA}\pysiglinewithargsret{\sphinxcode{Autocorrelation.}\sphinxbfcode{CalculateNormalizedMoreauBrotoAutoResidueASA}}{\emph{ProteinSequence}}{}
Calculte the NormalizedMoreauBorto Autocorrelation descriptors based on

ResidueASA.

Usage:

result=CalculateNormalizedMoreauBrotoAutoResidueASA(protein)

Input: protein is a pure protein sequence.

Output: result is a dict form containing 30 Normalized Moreau-Broto Autocorrelation

\end{fulllineitems}

\index{CalculateNormalizedMoreauBrotoAutoResidueVol() (in module Autocorrelation)}

\begin{fulllineitems}
\phantomsection\label{reference/Autocorrelation:Autocorrelation.CalculateNormalizedMoreauBrotoAutoResidueVol}\pysiglinewithargsret{\sphinxcode{Autocorrelation.}\sphinxbfcode{CalculateNormalizedMoreauBrotoAutoResidueVol}}{\emph{ProteinSequence}}{}
Calculte the NormalizedMoreauBorto Autocorrelation descriptors based on

ResidueVol.

Usage:

result=CalculateNormalizedMoreauBrotoAutoResidueVol(protein)

Input: protein is a pure protein sequence.

Output: result is a dict form containing 30 Normalized Moreau-Broto Autocorrelation

\end{fulllineitems}

\index{CalculateNormalizedMoreauBrotoAutoSteric() (in module Autocorrelation)}

\begin{fulllineitems}
\phantomsection\label{reference/Autocorrelation:Autocorrelation.CalculateNormalizedMoreauBrotoAutoSteric}\pysiglinewithargsret{\sphinxcode{Autocorrelation.}\sphinxbfcode{CalculateNormalizedMoreauBrotoAutoSteric}}{\emph{ProteinSequence}}{}
Calculte the NormalizedMoreauBorto Autocorrelation descriptors based on Steric.

Usage:

result=CalculateNormalizedMoreauBrotoAutoSteric(protein)

Input: protein is a pure protein sequence.

Output: result is a dict form containing 30 Normalized Moreau-Broto Autocorrelation

\end{fulllineitems}

\index{CalculateNormalizedMoreauBrotoAutoTotal() (in module Autocorrelation)}

\begin{fulllineitems}
\phantomsection\label{reference/Autocorrelation:Autocorrelation.CalculateNormalizedMoreauBrotoAutoTotal}\pysiglinewithargsret{\sphinxcode{Autocorrelation.}\sphinxbfcode{CalculateNormalizedMoreauBrotoAutoTotal}}{\emph{ProteinSequence}}{}
A method used for computing normalized Moreau Broto autocorrelation descriptors based

on 8 proterties of AADs.

Usage:

result=CalculateNormalizedMoreauBrotoAutoTotal(protein)

Input: protein is a pure protein sequence.

Output: result is a dict form containing 30*8=240 normalized Moreau Broto

\end{fulllineitems}

\index{NormalizeEachAAP() (in module Autocorrelation)}

\begin{fulllineitems}
\phantomsection\label{reference/Autocorrelation:Autocorrelation.NormalizeEachAAP}\pysiglinewithargsret{\sphinxcode{Autocorrelation.}\sphinxbfcode{NormalizeEachAAP}}{\emph{AAP}}{}
All of the amino acid indices are centralized and

standardized before the calculation.

Usage:

result=NormalizeEachAAP(AAP)

Input: AAP is a dict form containing the properties of 20 amino acids.

Output: result is the a dict form containing the normalized properties

\end{fulllineitems}



\subsection{CTD module}
\label{reference/CTD::doc}\label{reference/CTD:ctd-module}\label{reference/CTD:module-CTD}\index{CTD (module)}

\bigskip\hrule{}\bigskip


This module is used for computing the composition, transition and distribution

descriptors based on the different properties of AADs. The AADs with the same

properties is marked as the same number. You can get 147 descriptors for a given

protein sequence. You can freely use and distribute it. If you hava  any problem,

you could contact with us timely!

References:

{[}1{]}: Inna Dubchak, Ilya Muchink, Stephen R.Holbrook and Sung-Hou Kim. Prediction

of protein folding class using global description of amino acid sequence. Proc.Natl.

Acad.Sci.USA, 1995, 92, 8700-8704.

{[}2{]}:Inna Dubchak, Ilya Muchink, Christopher Mayor, Igor Dralyuk and Sung-Hou Kim.

Recognition of a Protein Fold in the Context of the SCOP classification. Proteins:

Structure, Function and Genetics,1999,35,401-407.

Authors: Zhijiang Yao and Dongsheng Cao.

Date: 2016.06.04

Email: \href{mailto:gadsby@163.com}{gadsby@163.com}


\bigskip\hrule{}\bigskip

\index{CalculateC() (in module CTD)}

\begin{fulllineitems}
\phantomsection\label{reference/CTD:CTD.CalculateC}\pysiglinewithargsret{\sphinxcode{CTD.}\sphinxbfcode{CalculateC}}{\emph{ProteinSequence}}{}
Calculate all composition descriptors based seven different properties of AADs.

Usage:

result=CalculateC(protein)

Input:protein is a pure protein sequence.

\end{fulllineitems}

\index{CalculateCTD() (in module CTD)}

\begin{fulllineitems}
\phantomsection\label{reference/CTD:CTD.CalculateCTD}\pysiglinewithargsret{\sphinxcode{CTD.}\sphinxbfcode{CalculateCTD}}{\emph{ProteinSequence}}{}
Calculate all CTD descriptors based seven different properties of AADs.

Usage:

result=CalculateCTD(protein)

Input:protein is a pure protein sequence.

\end{fulllineitems}

\index{CalculateComposition() (in module CTD)}

\begin{fulllineitems}
\phantomsection\label{reference/CTD:CTD.CalculateComposition}\pysiglinewithargsret{\sphinxcode{CTD.}\sphinxbfcode{CalculateComposition}}{\emph{ProteinSequence}, \emph{AAProperty}, \emph{AAPName}}{}
A method used for computing composition descriptors.

Usage:

result=CalculateComposition(protein,AAProperty,AAPName)

Input: protein is a pure protein sequence.

AAProperty is a dict form containing classifciation of amino acids such as \_Polarizability.

AAPName is a string used for indicating a AAP name.

\end{fulllineitems}

\index{CalculateCompositionCharge() (in module CTD)}

\begin{fulllineitems}
\phantomsection\label{reference/CTD:CTD.CalculateCompositionCharge}\pysiglinewithargsret{\sphinxcode{CTD.}\sphinxbfcode{CalculateCompositionCharge}}{\emph{ProteinSequence}}{}
A method used for calculating composition descriptors based on Charge of

AADs.

Usage:

result=CalculateCompositionCharge(protein)

Input:protein is a pure protein sequence.

\end{fulllineitems}

\index{CalculateCompositionHydrophobicity() (in module CTD)}

\begin{fulllineitems}
\phantomsection\label{reference/CTD:CTD.CalculateCompositionHydrophobicity}\pysiglinewithargsret{\sphinxcode{CTD.}\sphinxbfcode{CalculateCompositionHydrophobicity}}{\emph{ProteinSequence}}{}
A method used for calculating composition descriptors based on Hydrophobicity of

AADs.

Usage:

result=CalculateCompositionHydrophobicity(protein)

Input:protein is a pure protein sequence.

\end{fulllineitems}

\index{CalculateCompositionNormalizedVDWV() (in module CTD)}

\begin{fulllineitems}
\phantomsection\label{reference/CTD:CTD.CalculateCompositionNormalizedVDWV}\pysiglinewithargsret{\sphinxcode{CTD.}\sphinxbfcode{CalculateCompositionNormalizedVDWV}}{\emph{ProteinSequence}}{}
A method used for calculating composition descriptors based on NormalizedVDWV of

AADs.

Usage:

result=CalculateCompositionNormalizedVDWV(protein)

Input:protein is a pure protein sequence.

\end{fulllineitems}

\index{CalculateCompositionPolarity() (in module CTD)}

\begin{fulllineitems}
\phantomsection\label{reference/CTD:CTD.CalculateCompositionPolarity}\pysiglinewithargsret{\sphinxcode{CTD.}\sphinxbfcode{CalculateCompositionPolarity}}{\emph{ProteinSequence}}{}
A method used for calculating composition descriptors based on Polarity of

AADs.

Usage:

result=CalculateCompositionPolarity(protein)

Input:protein is a pure protein sequence.

\end{fulllineitems}

\index{CalculateCompositionPolarizability() (in module CTD)}

\begin{fulllineitems}
\phantomsection\label{reference/CTD:CTD.CalculateCompositionPolarizability}\pysiglinewithargsret{\sphinxcode{CTD.}\sphinxbfcode{CalculateCompositionPolarizability}}{\emph{ProteinSequence}}{}
A method used for calculating composition descriptors based on Polarizability of

AADs.

Usage:

result=CalculateCompositionPolarizability(protein)

Input:protein is a pure protein sequence.

\end{fulllineitems}

\index{CalculateCompositionSecondaryStr() (in module CTD)}

\begin{fulllineitems}
\phantomsection\label{reference/CTD:CTD.CalculateCompositionSecondaryStr}\pysiglinewithargsret{\sphinxcode{CTD.}\sphinxbfcode{CalculateCompositionSecondaryStr}}{\emph{ProteinSequence}}{}
A method used for calculating composition descriptors based on SecondaryStr of

AADs.

Usage:

result=CalculateCompositionSecondaryStr(protein)

Input:protein is a pure protein sequence.

\end{fulllineitems}

\index{CalculateCompositionSolventAccessibility() (in module CTD)}

\begin{fulllineitems}
\phantomsection\label{reference/CTD:CTD.CalculateCompositionSolventAccessibility}\pysiglinewithargsret{\sphinxcode{CTD.}\sphinxbfcode{CalculateCompositionSolventAccessibility}}{\emph{ProteinSequence}}{}
A method used for calculating composition descriptors based on SolventAccessibility

of  AADs.

Usage:

result=CalculateCompositionSolventAccessibility(protein)

Input:protein is a pure protein sequence.

\end{fulllineitems}

\index{CalculateD() (in module CTD)}

\begin{fulllineitems}
\phantomsection\label{reference/CTD:CTD.CalculateD}\pysiglinewithargsret{\sphinxcode{CTD.}\sphinxbfcode{CalculateD}}{\emph{ProteinSequence}}{}
Calculate all distribution descriptors based seven different properties of AADs.

Usage:

result=CalculateD(protein)

Input:protein is a pure protein sequence.

\end{fulllineitems}

\index{CalculateDistribution() (in module CTD)}

\begin{fulllineitems}
\phantomsection\label{reference/CTD:CTD.CalculateDistribution}\pysiglinewithargsret{\sphinxcode{CTD.}\sphinxbfcode{CalculateDistribution}}{\emph{ProteinSequence}, \emph{AAProperty}, \emph{AAPName}}{}
A method used for computing distribution descriptors.

Usage:

result=CalculateDistribution(protein,AAProperty,AAPName)

Input:protein is a pure protein sequence.

AAProperty is a dict form containing classifciation of amino acids such as \_Polarizability.

AAPName is a string used for indicating a AAP name.

\end{fulllineitems}

\index{CalculateDistributionCharge() (in module CTD)}

\begin{fulllineitems}
\phantomsection\label{reference/CTD:CTD.CalculateDistributionCharge}\pysiglinewithargsret{\sphinxcode{CTD.}\sphinxbfcode{CalculateDistributionCharge}}{\emph{ProteinSequence}}{}
A method used for calculating Distribution descriptors based on Charge of

AADs.

Usage:

result=CalculateDistributionCharge(protein)

Input:protein is a pure protein sequence.

\end{fulllineitems}

\index{CalculateDistributionHydrophobicity() (in module CTD)}

\begin{fulllineitems}
\phantomsection\label{reference/CTD:CTD.CalculateDistributionHydrophobicity}\pysiglinewithargsret{\sphinxcode{CTD.}\sphinxbfcode{CalculateDistributionHydrophobicity}}{\emph{ProteinSequence}}{}
A method used for calculating Distribution descriptors based on Hydrophobicity of

AADs.

Usage:

result=CalculateDistributionHydrophobicity(protein)

Input:protein is a pure protein sequence.

\end{fulllineitems}

\index{CalculateDistributionNormalizedVDWV() (in module CTD)}

\begin{fulllineitems}
\phantomsection\label{reference/CTD:CTD.CalculateDistributionNormalizedVDWV}\pysiglinewithargsret{\sphinxcode{CTD.}\sphinxbfcode{CalculateDistributionNormalizedVDWV}}{\emph{ProteinSequence}}{}
A method used for calculating Distribution descriptors based on NormalizedVDWV of

AADs.

Usage:

result=CalculateDistributionNormalizedVDWV(protein)

Input:protein is a pure protein sequence.

\end{fulllineitems}

\index{CalculateDistributionPolarity() (in module CTD)}

\begin{fulllineitems}
\phantomsection\label{reference/CTD:CTD.CalculateDistributionPolarity}\pysiglinewithargsret{\sphinxcode{CTD.}\sphinxbfcode{CalculateDistributionPolarity}}{\emph{ProteinSequence}}{}
A method used for calculating Distribution descriptors based on Polarity of

AADs.

Usage:

result=CalculateDistributionPolarity(protein)

Input:protein is a pure protein sequence.

\end{fulllineitems}

\index{CalculateDistributionPolarizability() (in module CTD)}

\begin{fulllineitems}
\phantomsection\label{reference/CTD:CTD.CalculateDistributionPolarizability}\pysiglinewithargsret{\sphinxcode{CTD.}\sphinxbfcode{CalculateDistributionPolarizability}}{\emph{ProteinSequence}}{}
A method used for calculating Distribution descriptors based on Polarizability of

AADs.

Usage:

result=CalculateDistributionPolarizability(protein)

Input:protein is a pure protein sequence.

\end{fulllineitems}

\index{CalculateDistributionSecondaryStr() (in module CTD)}

\begin{fulllineitems}
\phantomsection\label{reference/CTD:CTD.CalculateDistributionSecondaryStr}\pysiglinewithargsret{\sphinxcode{CTD.}\sphinxbfcode{CalculateDistributionSecondaryStr}}{\emph{ProteinSequence}}{}
A method used for calculating Distribution descriptors based on SecondaryStr of

AADs.

Usage:

result=CalculateDistributionSecondaryStr(protein)

Input:protein is a pure protein sequence.

\end{fulllineitems}

\index{CalculateDistributionSolventAccessibility() (in module CTD)}

\begin{fulllineitems}
\phantomsection\label{reference/CTD:CTD.CalculateDistributionSolventAccessibility}\pysiglinewithargsret{\sphinxcode{CTD.}\sphinxbfcode{CalculateDistributionSolventAccessibility}}{\emph{ProteinSequence}}{}
A method used for calculating Distribution descriptors based on SolventAccessibility

of  AADs.

Usage:

result=CalculateDistributionSolventAccessibility(protein)

Input:protein is a pure protein sequence.

\end{fulllineitems}

\index{CalculateT() (in module CTD)}

\begin{fulllineitems}
\phantomsection\label{reference/CTD:CTD.CalculateT}\pysiglinewithargsret{\sphinxcode{CTD.}\sphinxbfcode{CalculateT}}{\emph{ProteinSequence}}{}
Calculate all transition descriptors based seven different properties of AADs.

Usage:

result=CalculateT(protein)

Input:protein is a pure protein sequence.

\end{fulllineitems}

\index{CalculateTransition() (in module CTD)}

\begin{fulllineitems}
\phantomsection\label{reference/CTD:CTD.CalculateTransition}\pysiglinewithargsret{\sphinxcode{CTD.}\sphinxbfcode{CalculateTransition}}{\emph{ProteinSequence}, \emph{AAProperty}, \emph{AAPName}}{}
A method used for computing transition descriptors

Usage:

result=CalculateTransition(protein,AAProperty,AAPName)

Input:protein is a pure protein sequence.

AAProperty is a dict form containing classifciation of amino acids such as \_Polarizability.

AAPName is a string used for indicating a AAP name.

\end{fulllineitems}

\index{CalculateTransitionCharge() (in module CTD)}

\begin{fulllineitems}
\phantomsection\label{reference/CTD:CTD.CalculateTransitionCharge}\pysiglinewithargsret{\sphinxcode{CTD.}\sphinxbfcode{CalculateTransitionCharge}}{\emph{ProteinSequence}}{}
A method used for calculating Transition descriptors based on Charge of

AADs.

Usage:

result=CalculateTransitionCharge(protein)

Input:protein is a pure protein sequence.

\end{fulllineitems}

\index{CalculateTransitionHydrophobicity() (in module CTD)}

\begin{fulllineitems}
\phantomsection\label{reference/CTD:CTD.CalculateTransitionHydrophobicity}\pysiglinewithargsret{\sphinxcode{CTD.}\sphinxbfcode{CalculateTransitionHydrophobicity}}{\emph{ProteinSequence}}{}
A method used for calculating Transition descriptors based on Hydrophobicity of

AADs.

Usage:

result=CalculateTransitionHydrophobicity(protein)

Input:protein is a pure protein sequence.

\end{fulllineitems}

\index{CalculateTransitionNormalizedVDWV() (in module CTD)}

\begin{fulllineitems}
\phantomsection\label{reference/CTD:CTD.CalculateTransitionNormalizedVDWV}\pysiglinewithargsret{\sphinxcode{CTD.}\sphinxbfcode{CalculateTransitionNormalizedVDWV}}{\emph{ProteinSequence}}{}
A method used for calculating Transition descriptors based on NormalizedVDWV of

AADs.

Usage:

result=CalculateTransitionNormalizedVDWV(protein)

Input:protein is a pure protein sequence.

\end{fulllineitems}

\index{CalculateTransitionPolarity() (in module CTD)}

\begin{fulllineitems}
\phantomsection\label{reference/CTD:CTD.CalculateTransitionPolarity}\pysiglinewithargsret{\sphinxcode{CTD.}\sphinxbfcode{CalculateTransitionPolarity}}{\emph{ProteinSequence}}{}
A method used for calculating Transition descriptors based on Polarity of

AADs.

Usage:

result=CalculateTransitionPolarity(protein)

Input:protein is a pure protein sequence.

\end{fulllineitems}

\index{CalculateTransitionPolarizability() (in module CTD)}

\begin{fulllineitems}
\phantomsection\label{reference/CTD:CTD.CalculateTransitionPolarizability}\pysiglinewithargsret{\sphinxcode{CTD.}\sphinxbfcode{CalculateTransitionPolarizability}}{\emph{ProteinSequence}}{}
A method used for calculating Transition descriptors based on Polarizability of

AADs.

Usage:

result=CalculateTransitionPolarizability(protein)

Input:protein is a pure protein sequence.

\end{fulllineitems}

\index{CalculateTransitionSecondaryStr() (in module CTD)}

\begin{fulllineitems}
\phantomsection\label{reference/CTD:CTD.CalculateTransitionSecondaryStr}\pysiglinewithargsret{\sphinxcode{CTD.}\sphinxbfcode{CalculateTransitionSecondaryStr}}{\emph{ProteinSequence}}{}
A method used for calculating Transition descriptors based on SecondaryStr of

AADs.

Usage:

result=CalculateTransitionSecondaryStr(protein)

Input:protein is a pure protein sequence.

\end{fulllineitems}

\index{CalculateTransitionSolventAccessibility() (in module CTD)}

\begin{fulllineitems}
\phantomsection\label{reference/CTD:CTD.CalculateTransitionSolventAccessibility}\pysiglinewithargsret{\sphinxcode{CTD.}\sphinxbfcode{CalculateTransitionSolventAccessibility}}{\emph{ProteinSequence}}{}
A method used for calculating Transition descriptors based on SolventAccessibility

of  AADs.

Usage:

result=CalculateTransitionSolventAccessibility(protein)

Input:protein is a pure protein sequence.

\end{fulllineitems}

\index{StringtoNum() (in module CTD)}

\begin{fulllineitems}
\phantomsection\label{reference/CTD:CTD.StringtoNum}\pysiglinewithargsret{\sphinxcode{CTD.}\sphinxbfcode{StringtoNum}}{\emph{ProteinSequence}, \emph{AAProperty}}{}
Tranform the protein sequence into the string form such as 32123223132121123.

Usage:

result=StringtoNum(protein,AAProperty)

Input: protein is a pure protein sequence.

AAProperty is a dict form containing classifciation of amino acids such as \_Polarizability.

\end{fulllineitems}



\subsection{ConjointTriad module}
\label{reference/ConjointTriad:conjointtriad-module}\label{reference/ConjointTriad:module-ConjointTriad}\label{reference/ConjointTriad::doc}\index{ConjointTriad (module)}
protein sequence information. You can get 7*7*7=343 features.You can freely

use and distribute it. If you hava any problem, you could contact with us timely!

Reference:

Juwen Shen, Jian Zhang, Xiaomin Luo, Weiliang Zhu, Kunqian Yu, Kaixian Chen,

Yixue Li, Huanliang Jiang. Predicting proten-protein interactions based only

on sequences inforamtion. PNAS. 2007 (104) 4337-4341.

Authors: Zhijiang Yao and Dongsheng Cao.

Date: 2016.06.04

Email: \href{mailto:gadsby@163.com}{gadsby@163.com}


\bigskip\hrule{}\bigskip

\index{CalculateConjointTriad() (in module ConjointTriad)}

\begin{fulllineitems}
\phantomsection\label{reference/ConjointTriad:ConjointTriad.CalculateConjointTriad}\pysiglinewithargsret{\sphinxcode{ConjointTriad.}\sphinxbfcode{CalculateConjointTriad}}{\emph{proteinsequence}}{}
Calculate the conjoint triad features from protein sequence.

Useage:

res = CalculateConjointTriad(protein)

Input: protein is a pure protein sequence.

Output is a dict form containing all 343 conjoint triad features.

\end{fulllineitems}



\subsection{GetProteinFromUniprot module}
\label{reference/GetProteinFromUniprot:module-GetProteinFromUniprot}\label{reference/GetProteinFromUniprot:getproteinfromuniprot-module}\label{reference/GetProteinFromUniprot::doc}\index{GetProteinFromUniprot (module)}

\bigskip\hrule{}\bigskip


This module is used to download the protein sequence from the uniprot (\url{http://www.uniprot.org/})

website. You can only need input a protein ID or prepare a file (ID.txt) related to ID. You can
\begin{quote}

obtain a .txt (ProteinSequence.txt) file saving protein sequence you need.  You can freely use

and distribute it. If you hava  any problem, you could contact with us timely!
\end{quote}

Authors: Zhijiang Yao and Dongsheng Cao.

Date: 2016.06.04

Email: \href{mailto:gadsby@163.com}{gadsby@163.com}


\bigskip\hrule{}\bigskip

\index{GetProteinSequence() (in module GetProteinFromUniprot)}

\begin{fulllineitems}
\phantomsection\label{reference/GetProteinFromUniprot:GetProteinFromUniprot.GetProteinSequence}\pysiglinewithargsret{\sphinxcode{GetProteinFromUniprot.}\sphinxbfcode{GetProteinSequence}}{\emph{ProteinID}}{}
Get the protein sequence from the uniprot website by ID.

Usage:

result=GetProteinSequence(ProteinID)

Input: ProteinID is a string indicating ID such as ``P48039''.

\end{fulllineitems}

\index{GetProteinSequenceFromTxt() (in module GetProteinFromUniprot)}

\begin{fulllineitems}
\phantomsection\label{reference/GetProteinFromUniprot:GetProteinFromUniprot.GetProteinSequenceFromTxt}\pysiglinewithargsret{\sphinxcode{GetProteinFromUniprot.}\sphinxbfcode{GetProteinSequenceFromTxt}}{\emph{path}, \emph{openfile}, \emph{savefile}}{}
Get the protein sequence from the uniprot website by the file containing ID.

Usage:

result=GetProteinSequenceFromTxt(path,openfile,savefile)

Input: path is a directory path containing the ID file such as ``/home/orient/protein/''

openfile is the ID file such as ``proteinID.txt''

\end{fulllineitems}



\subsection{GetSubSeq module}
\label{reference/GetSubSeq:module-GetSubSeq}\label{reference/GetSubSeq:getsubseq-module}\label{reference/GetSubSeq::doc}\index{GetSubSeq (module)}

\bigskip\hrule{}\bigskip


The prediction of functional sites (e.g.,methylation) of proteins usually needs to

split the total protein into a set of segments around specific amino acid. Given a

specific window size p, we can obtain all segments of length equal to (2*p+1) very

easily. Note that the output of the method is a list form. You can freely use and

distribute it. If you have any problem, you could contact with us timely.

Authors: Zhijiang Yao and Dongsheng Cao.

Date: 2016.06.04

Email: \href{mailto:gadsby@163.com}{gadsby@163.com}


\bigskip\hrule{}\bigskip

\index{GetSubSequence() (in module GetSubSeq)}

\begin{fulllineitems}
\phantomsection\label{reference/GetSubSeq:GetSubSeq.GetSubSequence}\pysiglinewithargsret{\sphinxcode{GetSubSeq.}\sphinxbfcode{GetSubSequence}}{\emph{ProteinSequence}, \emph{ToAA='S'}, \emph{window=3}}{}
Get all 2*window+1 sub-sequences whose cener is ToAA in a protein.

Usage:

result=GetSubSequence(protein,ToAA,window)

Input:protein is a pure problem sequence.

ToAA is the central (query point) amino acid in the sub-sequence.

window is the span.

\end{fulllineitems}



\subsection{ProCheck module}
\label{reference/ProCheck::doc}\label{reference/ProCheck:module-ProCheck}\label{reference/ProCheck:procheck-module}\index{ProCheck (module)}
sequence. You can freely use and distribute it. If you hava any problem, you could

contact with us timely!

Authors: Zhijiang Yao and Dongsheng Cao.

Date: 2016.06.04

Email: \href{mailto:gadsby@163.com}{gadsby@163.com}


\bigskip\hrule{}\bigskip

\index{ProteinCheck() (in module ProCheck)}

\begin{fulllineitems}
\phantomsection\label{reference/ProCheck:ProCheck.ProteinCheck}\pysiglinewithargsret{\sphinxcode{ProCheck.}\sphinxbfcode{ProteinCheck}}{\emph{ProteinSequence}}{}
Check whether the protein sequence is a valid amino acid sequence or not

Usage:

result=ProteinCheck(protein)

Input: protein is a pure protein sequence.

Output: if the check is no problem, result will return the length of protein.

\end{fulllineitems}



\subsection{PseudoAAC module}
\label{reference/PseudoAAC:module-PseudoAAC}\label{reference/PseudoAAC:pseudoaac-module}\label{reference/PseudoAAC::doc}\index{PseudoAAC (module)}

\bigskip\hrule{}\bigskip


Instead of using the conventional 20-D amino acid composition to represent the sample

of a protein, Prof. Kuo-Chen Chou proposed the pseudo amino acid (PseAA) composition

in order for inluding the sequence-order information. Based on the concept of Chou's

pseudo amino acid composition, the server PseAA was designed in a flexible way, allowing

users to generate various kinds of pseudo amino acid composition for a given protein

sequence by selecting different parameters and their combinations. This module aims at

computing two types of PseAA descriptors: Type I and Type II.

You can freely use and distribute it. If you have any problem, you could contact

with us timely.

References:

{[}1{]}: Kuo-Chen Chou. Prediction of Protein Cellular Attributes Using Pseudo-Amino Acid

Composition. PROTEINS: Structure, Function, and Genetics, 2001, 43: 246-255.

{[}2{]}: \url{http://www.csbio.sjtu.edu.cn/bioinf/PseAAC/}

{[}3{]}: \url{http://www.csbio.sjtu.edu.cn/bioinf/PseAAC/type2.htm}

{[}4{]}: Kuo-Chen Chou. Using amphiphilic pseudo amino acid composition to predict enzyme

subfamily classes. Bioinformatics, 2005,21,10-19.

Authors: Zhijiang Yao and Dongsheng Cao.

Date: 2016.06.04

Email: \href{mailto:gadsby@163.com}{gadsby@163.com}

The hydrophobicity values are from JACS, 1962, 84: 4240-4246. (C. Tanford).

The hydrophilicity values are from PNAS, 1981, 78:3824-3828 (T.P.Hopp \& K.R.Woods).

The side-chain mass for each of the 20 amino acids.

CRC Handbook of Chemistry and Physics, 66th ed., CRC Press, Boca Raton, Florida (1985).

R.M.C. Dawson, D.C. Elliott, W.H. Elliott, K.M. Jones, Data for Biochemical Research 3rd ed.,

Clarendon Press Oxford (1986).


\bigskip\hrule{}\bigskip

\index{GetAAComposition() (in module PseudoAAC)}

\begin{fulllineitems}
\phantomsection\label{reference/PseudoAAC:PseudoAAC.GetAAComposition}\pysiglinewithargsret{\sphinxcode{PseudoAAC.}\sphinxbfcode{GetAAComposition}}{\emph{ProteinSequence}}{}
Calculate the composition of Amino acids

for a given protein sequence.

Usage:

result=CalculateAAComposition(protein)

Input: protein is a pure protein sequence.

Output: result is a dict form containing the composition of

\end{fulllineitems}

\index{GetAPseudoAAC() (in module PseudoAAC)}

\begin{fulllineitems}
\phantomsection\label{reference/PseudoAAC:PseudoAAC.GetAPseudoAAC}\pysiglinewithargsret{\sphinxcode{PseudoAAC.}\sphinxbfcode{GetAPseudoAAC}}{\emph{ProteinSequence}, \emph{lamda=30}, \emph{weight=0.5}}{}
Computing all of type II pseudo-amino acid compostion descriptors based on the given

properties. Note that the number of PAAC strongly depends on the lamda value. if lamda

= 20, we can obtain 20+20=40 PAAC descriptors. The size of these values depends on the

choice of lamda and weight simultaneously.

Usage:

result=GetAPseudoAAC(protein,lamda,weight)

Input: protein is a pure protein sequence.

lamda factor reflects the rank of correlation and is a non-Negative integer, such as 15.

Note that (1)lamda should NOT be larger than the length of input protein sequence;
\begin{enumerate}
\setcounter{enumi}{1}
\item {} 
lamda must be non-Negative integer, such as 0, 1, 2, ...; (3) when lamda =0, the

\end{enumerate}

output of PseAA server is the 20-D amino acid composition.

weight factor is designed for the users to put weight on the additional PseAA components

with respect to the conventional AA components. The user can select any value within the

region from 0.05 to 0.7 for the weight factor.

\end{fulllineitems}

\index{GetAPseudoAAC1() (in module PseudoAAC)}

\begin{fulllineitems}
\phantomsection\label{reference/PseudoAAC:PseudoAAC.GetAPseudoAAC1}\pysiglinewithargsret{\sphinxcode{PseudoAAC.}\sphinxbfcode{GetAPseudoAAC1}}{\emph{ProteinSequence}, \emph{lamda=30}, \emph{weight=0.5}}{}
Computing the first 20 of type II pseudo-amino acid compostion descriptors based on

\end{fulllineitems}

\index{GetAPseudoAAC2() (in module PseudoAAC)}

\begin{fulllineitems}
\phantomsection\label{reference/PseudoAAC:PseudoAAC.GetAPseudoAAC2}\pysiglinewithargsret{\sphinxcode{PseudoAAC.}\sphinxbfcode{GetAPseudoAAC2}}{\emph{ProteinSequence}, \emph{lamda=30}, \emph{weight=0.5}}{}
Computing the last lamda of type II pseudo-amino acid compostion descriptors based on

\end{fulllineitems}

\index{GetCorrelationFunction() (in module PseudoAAC)}

\begin{fulllineitems}
\phantomsection\label{reference/PseudoAAC:PseudoAAC.GetCorrelationFunction}\pysiglinewithargsret{\sphinxcode{PseudoAAC.}\sphinxbfcode{GetCorrelationFunction}}{\emph{Ri='S'}, \emph{Rj='D'}, \emph{AAP={[}{]}}}{}
Computing the correlation between two given amino acids using the given

properties.

Usage:

result=GetCorrelationFunction(Ri,Rj,AAP)

Input: Ri and Rj are the amino acids, respectively.

AAP is a list form containing the properties, each of which is a dict form.

\end{fulllineitems}

\index{GetPseudoAAC() (in module PseudoAAC)}

\begin{fulllineitems}
\phantomsection\label{reference/PseudoAAC:PseudoAAC.GetPseudoAAC}\pysiglinewithargsret{\sphinxcode{PseudoAAC.}\sphinxbfcode{GetPseudoAAC}}{\emph{ProteinSequence}, \emph{lamda=30}, \emph{weight=0.05}, \emph{AAP={[}{]}}}{}
Computing all of type I pseudo-amino acid compostion descriptors based on the given

properties. Note that the number of PAAC strongly depends on the lamda value. if lamda

= 20, we can obtain 20+20=40 PAAC descriptors. The size of these values depends on the

choice of lamda and weight simultaneously. You must specify some properties into AAP.

Usage:

result=GetPseudoAAC(protein,lamda,weight)

Input: protein is a pure protein sequence.

lamda factor reflects the rank of correlation and is a non-Negative integer, such as 15.

Note that (1)lamda should NOT be larger than the length of input protein sequence;
\begin{enumerate}
\setcounter{enumi}{1}
\item {} 
lamda must be non-Negative integer, such as 0, 1, 2, ...; (3) when lamda =0, the

\end{enumerate}

output of PseAA server is the 20-D amino acid composition.

weight factor is designed for the users to put weight on the additional PseAA components

with respect to the conventional AA components. The user can select any value within the

region from 0.05 to 0.7 for the weight factor.

AAP is a list form containing the properties, each of which is a dict form.

\end{fulllineitems}

\index{GetPseudoAAC1() (in module PseudoAAC)}

\begin{fulllineitems}
\phantomsection\label{reference/PseudoAAC:PseudoAAC.GetPseudoAAC1}\pysiglinewithargsret{\sphinxcode{PseudoAAC.}\sphinxbfcode{GetPseudoAAC1}}{\emph{ProteinSequence}, \emph{lamda=30}, \emph{weight=0.05}, \emph{AAP={[}{]}}}{}
Computing the first 20 of type I pseudo-amino acid compostion descriptors based on the given

\end{fulllineitems}

\index{GetPseudoAAC2() (in module PseudoAAC)}

\begin{fulllineitems}
\phantomsection\label{reference/PseudoAAC:PseudoAAC.GetPseudoAAC2}\pysiglinewithargsret{\sphinxcode{PseudoAAC.}\sphinxbfcode{GetPseudoAAC2}}{\emph{ProteinSequence}, \emph{lamda=30}, \emph{weight=0.05}, \emph{AAP={[}{]}}}{}
Computing the last lamda of type I pseudo-amino acid compostion descriptors based on the given

\end{fulllineitems}

\index{GetSequenceOrderCorrelationFactor() (in module PseudoAAC)}

\begin{fulllineitems}
\phantomsection\label{reference/PseudoAAC:PseudoAAC.GetSequenceOrderCorrelationFactor}\pysiglinewithargsret{\sphinxcode{PseudoAAC.}\sphinxbfcode{GetSequenceOrderCorrelationFactor}}{\emph{ProteinSequence}, \emph{k=1}, \emph{AAP={[}{]}}}{}
Computing the Sequence order correlation factor with gap equal to k based on

the given properities.

Usage:

result=GetSequenceOrderCorrelationFactor(protein,k,AAP)

Input: protein is a pure protein sequence.

k is the gap.

AAP is a list form containing the properties, each of which is a dict form.

\end{fulllineitems}

\index{GetSequenceOrderCorrelationFactorForAPAAC() (in module PseudoAAC)}

\begin{fulllineitems}
\phantomsection\label{reference/PseudoAAC:PseudoAAC.GetSequenceOrderCorrelationFactorForAPAAC}\pysiglinewithargsret{\sphinxcode{PseudoAAC.}\sphinxbfcode{GetSequenceOrderCorrelationFactorForAPAAC}}{\emph{ProteinSequence}, \emph{k=1}}{}
Computing the Sequence order correlation factor with gap equal to k based on

{[}\_Hydrophobicity,\_hydrophilicity{]} for APAAC (type II PseAAC) .

Usage:

result=GetSequenceOrderCorrelationFactorForAPAAC(protein,k)

Input: protein is a pure protein sequence.

k is the gap.

\end{fulllineitems}

\index{NormalizeEachAAP() (in module PseudoAAC)}

\begin{fulllineitems}
\phantomsection\label{reference/PseudoAAC:PseudoAAC.NormalizeEachAAP}\pysiglinewithargsret{\sphinxcode{PseudoAAC.}\sphinxbfcode{NormalizeEachAAP}}{\emph{AAP}}{}
All of the amino acid indices are centralized and

standardized before the calculation.

Usage:

result=NormalizeEachAAP(AAP)

Input: AAP is a dict form containing the properties of 20 amino acids.

Output: result is the a dict form containing the normalized properties

\end{fulllineitems}



\subsection{PyProteinAAComposition module}
\label{reference/PyProteinAAComposition::doc}\label{reference/PyProteinAAComposition:pyproteinaacomposition-module}\label{reference/PyProteinAAComposition:module-PyProteinAAComposition}\index{PyProteinAAComposition (module)}
Created on Thu Jun 02 10:00:35 2016

@author: yzj
\index{CalculateAAComposition() (in module PyProteinAAComposition)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinAAComposition:PyProteinAAComposition.CalculateAAComposition}\pysiglinewithargsret{\sphinxcode{PyProteinAAComposition.}\sphinxbfcode{CalculateAAComposition}}{\emph{ProteinSequence}}{}
Calculate the composition of Amino acids

for a given protein sequence.

Usage:

result=CalculateAAComposition(protein)

Input: protein is a pure protein sequence.

Output: result is a dict form containing the composition of

\end{fulllineitems}

\index{CalculateAADipeptideComposition() (in module PyProteinAAComposition)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinAAComposition:PyProteinAAComposition.CalculateAADipeptideComposition}\pysiglinewithargsret{\sphinxcode{PyProteinAAComposition.}\sphinxbfcode{CalculateAADipeptideComposition}}{\emph{ProteinSequence}}{}
Calculate the composition of AADs, dipeptide and 3-mers for a

given protein sequence.

Usage:

result=CalculateAADipeptideComposition(protein)

Input: protein is a pure protein sequence.

Output: result is a dict form containing all composition values of

\end{fulllineitems}

\index{CalculateDipeptideComposition() (in module PyProteinAAComposition)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinAAComposition:PyProteinAAComposition.CalculateDipeptideComposition}\pysiglinewithargsret{\sphinxcode{PyProteinAAComposition.}\sphinxbfcode{CalculateDipeptideComposition}}{\emph{ProteinSequence}}{}
Calculate the composition of dipeptidefor a given protein sequence.

Usage:

result=CalculateDipeptideComposition(protein)

Input: protein is a pure protein sequence.

Output: result is a dict form containing the composition of

\end{fulllineitems}

\index{GetSpectrumDict() (in module PyProteinAAComposition)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinAAComposition:PyProteinAAComposition.GetSpectrumDict}\pysiglinewithargsret{\sphinxcode{PyProteinAAComposition.}\sphinxbfcode{GetSpectrumDict}}{\emph{proteinsequence}}{}
Calcualte the spectrum descriptors of 3-mers for a given protein.

Usage:

result=GetSpectrumDict(protein)

Input: protein is a pure protein sequence.

Output: result is a dict form containing the composition values of 8000

\end{fulllineitems}

\index{Getkmers() (in module PyProteinAAComposition)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinAAComposition:PyProteinAAComposition.Getkmers}\pysiglinewithargsret{\sphinxcode{PyProteinAAComposition.}\sphinxbfcode{Getkmers}}{}{}
Get the amino acid list of 3-mers.

Usage:

result=Getkmers()

Output: result is a list form containing 8000 tri-peptides.

\end{fulllineitems}



\subsection{PyProteinAAIndex module}
\label{reference/PyProteinAAIndex::doc}\label{reference/PyProteinAAIndex:pyproteinaaindex-module}\label{reference/PyProteinAAIndex:module-PyProteinAAIndex}\index{PyProteinAAIndex (module)}
Created on Thu Jun 02 15:38:19 2016

@author: yzj
\index{GetAAIndex1() (in module PyProteinAAIndex)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinAAIndex:PyProteinAAIndex.GetAAIndex1}\pysiglinewithargsret{\sphinxcode{PyProteinAAIndex.}\sphinxbfcode{GetAAIndex1}}{\emph{name}, \emph{path='.'}}{}
Get the amino acid property values from aaindex1

Usage:

result=GetAAIndex1(name)

Input: name is the name of amino acid property (e.g., KRIW790103)

Output: result is a dict form containing the properties of 20 amino acids

\end{fulllineitems}

\index{GetAAIndex23() (in module PyProteinAAIndex)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinAAIndex:PyProteinAAIndex.GetAAIndex23}\pysiglinewithargsret{\sphinxcode{PyProteinAAIndex.}\sphinxbfcode{GetAAIndex23}}{\emph{name}, \emph{path='.'}}{}
Get the amino acid property values from aaindex2 and aaindex3

Usage:

result=GetAAIndex23(name)

Input: name is the name of amino acid property (e.g.,TANS760101,GRAR740104)

Output: result is a dict form containing the properties of 400 amino acid pairs

\end{fulllineitems}

\index{MatrixRecord (class in PyProteinAAIndex)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinAAIndex:PyProteinAAIndex.MatrixRecord}\pysigline{\sphinxstrong{class }\sphinxcode{PyProteinAAIndex.}\sphinxbfcode{MatrixRecord}}
Bases: {\hyperref[reference/PyProteinAAIndex:PyProteinAAIndex.Record]{\sphinxcrossref{\sphinxcode{PyProteinAAIndex.Record}}}}

Matrix record for mutation matrices or pair-wise contact potentials
\index{extend() (PyProteinAAIndex.MatrixRecord method)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinAAIndex:PyProteinAAIndex.MatrixRecord.extend}\pysiglinewithargsret{\sphinxbfcode{extend}}{\emph{row}}{}
\end{fulllineitems}

\index{get() (PyProteinAAIndex.MatrixRecord method)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinAAIndex:PyProteinAAIndex.MatrixRecord.get}\pysiglinewithargsret{\sphinxbfcode{get}}{\emph{aai}, \emph{aaj}, \emph{d=None}}{}
\end{fulllineitems}

\index{median() (PyProteinAAIndex.MatrixRecord method)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinAAIndex:PyProteinAAIndex.MatrixRecord.median}\pysiglinewithargsret{\sphinxbfcode{median}}{}{}
\end{fulllineitems}


\end{fulllineitems}

\index{Record (class in PyProteinAAIndex)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinAAIndex:PyProteinAAIndex.Record}\pysigline{\sphinxstrong{class }\sphinxcode{PyProteinAAIndex.}\sphinxbfcode{Record}}
Amino acid index (AAindex) Record
\index{aakeys (PyProteinAAIndex.Record attribute)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinAAIndex:PyProteinAAIndex.Record.aakeys}\pysigline{\sphinxbfcode{aakeys}\sphinxstrong{ = `ARNDCQEGHILKMFPSTWYV'}}
\end{fulllineitems}

\index{extend() (PyProteinAAIndex.Record method)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinAAIndex:PyProteinAAIndex.Record.extend}\pysiglinewithargsret{\sphinxbfcode{extend}}{\emph{row}}{}
\end{fulllineitems}

\index{get() (PyProteinAAIndex.Record method)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinAAIndex:PyProteinAAIndex.Record.get}\pysiglinewithargsret{\sphinxbfcode{get}}{\emph{aai}, \emph{aaj=None}, \emph{d=None}}{}
\end{fulllineitems}

\index{median() (PyProteinAAIndex.Record method)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinAAIndex:PyProteinAAIndex.Record.median}\pysiglinewithargsret{\sphinxbfcode{median}}{}{}
\end{fulllineitems}


\end{fulllineitems}

\index{get() (in module PyProteinAAIndex)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinAAIndex:PyProteinAAIndex.get}\pysiglinewithargsret{\sphinxcode{PyProteinAAIndex.}\sphinxbfcode{get}}{\emph{key}}{}
Get record for key

\end{fulllineitems}

\index{grep() (in module PyProteinAAIndex)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinAAIndex:PyProteinAAIndex.grep}\pysiglinewithargsret{\sphinxcode{PyProteinAAIndex.}\sphinxbfcode{grep}}{\emph{pattern}}{}
Search for pattern in title and description of all records (case
insensitive) and print results on standard output.

\end{fulllineitems}

\index{init() (in module PyProteinAAIndex)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinAAIndex:PyProteinAAIndex.init}\pysiglinewithargsret{\sphinxcode{PyProteinAAIndex.}\sphinxbfcode{init}}{\emph{path=None}, \emph{index=`123'}}{}
Read in the aaindex files. You need to run this (once) before you can
access any records. If the files are not within the current directory,
you need to specify the correct directory path. By default all three
aaindex files are read in.

\end{fulllineitems}

\index{init\_from\_file() (in module PyProteinAAIndex)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinAAIndex:PyProteinAAIndex.init_from_file}\pysiglinewithargsret{\sphinxcode{PyProteinAAIndex.}\sphinxbfcode{init\_from\_file}}{\emph{filename}, \emph{type=\textless{}class PyProteinAAIndex.Record\textgreater{}}}{}
\end{fulllineitems}

\index{search() (in module PyProteinAAIndex)}

\begin{fulllineitems}
\phantomsection\label{reference/PyProteinAAIndex:PyProteinAAIndex.search}\pysiglinewithargsret{\sphinxcode{PyProteinAAIndex.}\sphinxbfcode{search}}{\emph{pattern}, \emph{searchtitle=True}, \emph{casesensitive=False}}{}
Search for pattern in description and title (optional) of all records and
return matched records as list. By default search case insensitive.

\end{fulllineitems}



\subsection{QuasiSequenceOrder module}
\label{reference/QuasiSequenceOrder:module-QuasiSequenceOrder}\label{reference/QuasiSequenceOrder:quasisequenceorder-module}\label{reference/QuasiSequenceOrder::doc}\index{QuasiSequenceOrder (module)}
given protein sequence. We can obtain two types of descriptors: Sequence-order-coupling

number and quasi-sequence-order descriptors. Two distance matrixes between 20 amino acids

are employed. You can freely use and distribute it. If you have any problem, please contact

us immediately.

References:

{[}1{]}:Kuo-Chen Chou. Prediction of Protein Subcellar Locations by Incorporating

Quasi-Sequence-Order Effect. Biochemical and Biophysical Research Communications

2000, 278, 477-483.

{[}2{]}: Kuo-Chen Chou and Yu-Dong Cai. Prediction of Protein sucellular locations by

GO-FunD-PseAA predictor, Biochemical and Biophysical Research Communications,

2004, 320, 1236-1239.

{[}3{]}:Gisbert Schneider and Paul wrede. The Rational Design of Amino Acid

Sequences by Artifical Neural Networks and Simulated Molecular Evolution: Do

Novo Design of an Idealized Leader Cleavge Site. Biophys Journal, 1994, 66,

335-344.

Authors: Zhijiang Yao and Dongsheng Cao.

Date: 2016.06.04

Email: \href{mailto:gadsby@163.com}{gadsby@163.com}


\bigskip\hrule{}\bigskip

\index{GetAAComposition() (in module QuasiSequenceOrder)}

\begin{fulllineitems}
\phantomsection\label{reference/QuasiSequenceOrder:QuasiSequenceOrder.GetAAComposition}\pysiglinewithargsret{\sphinxcode{QuasiSequenceOrder.}\sphinxbfcode{GetAAComposition}}{\emph{ProteinSequence}}{}
Calculate the composition of Amino acids

for a given protein sequence.

Usage:

result=CalculateAAComposition(protein)

Input: protein is a pure protein sequence.

Output: result is a dict form containing the composition of

\end{fulllineitems}

\index{GetQuasiSequenceOrder() (in module QuasiSequenceOrder)}

\begin{fulllineitems}
\phantomsection\label{reference/QuasiSequenceOrder:QuasiSequenceOrder.GetQuasiSequenceOrder}\pysiglinewithargsret{\sphinxcode{QuasiSequenceOrder.}\sphinxbfcode{GetQuasiSequenceOrder}}{\emph{ProteinSequence}, \emph{maxlag=30}, \emph{weight=0.1}}{}
Computing quasi-sequence-order descriptors for a given protein.

{[}1{]}:Kuo-Chen Chou. Prediction of Protein Subcellar Locations by

Incorporating Quasi-Sequence-Order Effect. Biochemical and Biophysical

Research Communications 2000, 278, 477-483.

Usage:

result = GetQuasiSequenceOrder(protein,maxlag,weight)

Input: protein is a pure protein sequence

maxlag is the maximum lag and the length of the protein should be larger

than maxlag. default is 30.

weight is a weight factor.  please see reference 1 for its choice. default is 0.1.

\end{fulllineitems}

\index{GetQuasiSequenceOrder1() (in module QuasiSequenceOrder)}

\begin{fulllineitems}
\phantomsection\label{reference/QuasiSequenceOrder:QuasiSequenceOrder.GetQuasiSequenceOrder1}\pysiglinewithargsret{\sphinxcode{QuasiSequenceOrder.}\sphinxbfcode{GetQuasiSequenceOrder1}}{\emph{ProteinSequence}, \emph{maxlag=30}, \emph{weight=0.1}, \emph{distancematrix=\{\}}}{}
Computing the first 20 quasi-sequence-order descriptors for

a given protein sequence.

Usage:

result = GetQuasiSequenceOrder1(protein,maxlag,weigt)

\end{fulllineitems}

\index{GetQuasiSequenceOrder1Grant() (in module QuasiSequenceOrder)}

\begin{fulllineitems}
\phantomsection\label{reference/QuasiSequenceOrder:QuasiSequenceOrder.GetQuasiSequenceOrder1Grant}\pysiglinewithargsret{\sphinxcode{QuasiSequenceOrder.}\sphinxbfcode{GetQuasiSequenceOrder1Grant}}{\emph{ProteinSequence}, \emph{maxlag=30}, \emph{weight=0.1}, \emph{distancematrix=Distancematrix}}{}
Computing the first 20 quasi-sequence-order descriptors for

a given protein sequence.

Usage:

result = GetQuasiSequenceOrder1Grant(protein,maxlag,weigt)

\end{fulllineitems}

\index{GetQuasiSequenceOrder1SW() (in module QuasiSequenceOrder)}

\begin{fulllineitems}
\phantomsection\label{reference/QuasiSequenceOrder:QuasiSequenceOrder.GetQuasiSequenceOrder1SW}\pysiglinewithargsret{\sphinxcode{QuasiSequenceOrder.}\sphinxbfcode{GetQuasiSequenceOrder1SW}}{\emph{ProteinSequence}, \emph{maxlag=30}, \emph{weight=0.1}, \emph{distancematrix=Distancematrix}}{}
Computing the first 20 quasi-sequence-order descriptors for

a given protein sequence.

Usage:

result = GetQuasiSequenceOrder1SW(protein,maxlag,weigt)

\end{fulllineitems}

\index{GetQuasiSequenceOrder2() (in module QuasiSequenceOrder)}

\begin{fulllineitems}
\phantomsection\label{reference/QuasiSequenceOrder:QuasiSequenceOrder.GetQuasiSequenceOrder2}\pysiglinewithargsret{\sphinxcode{QuasiSequenceOrder.}\sphinxbfcode{GetQuasiSequenceOrder2}}{\emph{ProteinSequence}, \emph{maxlag=30}, \emph{weight=0.1}, \emph{distancematrix=\{\}}}{}
Computing the last maxlag quasi-sequence-order descriptors for

a given protein sequence.

Usage:

result = GetQuasiSequenceOrder2(protein,maxlag,weigt)

\end{fulllineitems}

\index{GetQuasiSequenceOrder2Grant() (in module QuasiSequenceOrder)}

\begin{fulllineitems}
\phantomsection\label{reference/QuasiSequenceOrder:QuasiSequenceOrder.GetQuasiSequenceOrder2Grant}\pysiglinewithargsret{\sphinxcode{QuasiSequenceOrder.}\sphinxbfcode{GetQuasiSequenceOrder2Grant}}{\emph{ProteinSequence}, \emph{maxlag=30}, \emph{weight=0.1}, \emph{distancematrix=Distancematrix}}{}
Computing the last maxlag quasi-sequence-order descriptors for

a given protein sequence.

Usage:

result = GetQuasiSequenceOrder2Grant(protein,maxlag,weigt)

\end{fulllineitems}

\index{GetQuasiSequenceOrder2SW() (in module QuasiSequenceOrder)}

\begin{fulllineitems}
\phantomsection\label{reference/QuasiSequenceOrder:QuasiSequenceOrder.GetQuasiSequenceOrder2SW}\pysiglinewithargsret{\sphinxcode{QuasiSequenceOrder.}\sphinxbfcode{GetQuasiSequenceOrder2SW}}{\emph{ProteinSequence}, \emph{maxlag=30}, \emph{weight=0.1}, \emph{distancematrix=Distancematrix}}{}
Computing the last maxlag quasi-sequence-order descriptors for

a given protein sequence.

Usage:

result = GetQuasiSequenceOrder2SW(protein,maxlag,weigt)

\end{fulllineitems}

\index{GetQuasiSequenceOrderp() (in module QuasiSequenceOrder)}

\begin{fulllineitems}
\phantomsection\label{reference/QuasiSequenceOrder:QuasiSequenceOrder.GetQuasiSequenceOrderp}\pysiglinewithargsret{\sphinxcode{QuasiSequenceOrder.}\sphinxbfcode{GetQuasiSequenceOrderp}}{\emph{ProteinSequence}, \emph{maxlag=30}, \emph{weight=0.1}, \emph{distancematrix=\{\}}}{}
Computing quasi-sequence-order descriptors for a given protein.

{[}1{]}:Kuo-Chen Chou. Prediction of Protein Subcellar Locations by

Incorporating Quasi-Sequence-Order Effect. Biochemical and Biophysical

Research Communications 2000, 278, 477-483.

Usage:

result = GetQuasiSequenceOrderp(protein,maxlag,weight,distancematrix)

Input: protein is a pure protein sequence

maxlag is the maximum lag and the length of the protein should be larger

than maxlag. default is 30.

weight is a weight factor.  please see reference 1 for its choice. default is 0.1.

distancematrix is a dict form containing 400 distance values

Output: result is a dict form containing all quasi-sequence-order descriptors

\end{fulllineitems}

\index{GetSequenceOrderCouplingNumber() (in module QuasiSequenceOrder)}

\begin{fulllineitems}
\phantomsection\label{reference/QuasiSequenceOrder:QuasiSequenceOrder.GetSequenceOrderCouplingNumber}\pysiglinewithargsret{\sphinxcode{QuasiSequenceOrder.}\sphinxbfcode{GetSequenceOrderCouplingNumber}}{\emph{ProteinSequence}, \emph{d=1}, \emph{distancematrix=Distancematrix}}{}
Computing the dth-rank sequence order coupling number for a protein.

Usage:

result = GetSequenceOrderCouplingNumber(protein,d)

Input: protein is a pure protein sequence.

d is the gap between two amino acids.

Output: result is numeric value.

\end{fulllineitems}

\index{GetSequenceOrderCouplingNumberGrant() (in module QuasiSequenceOrder)}

\begin{fulllineitems}
\phantomsection\label{reference/QuasiSequenceOrder:QuasiSequenceOrder.GetSequenceOrderCouplingNumberGrant}\pysiglinewithargsret{\sphinxcode{QuasiSequenceOrder.}\sphinxbfcode{GetSequenceOrderCouplingNumberGrant}}{\emph{ProteinSequence}, \emph{maxlag=30}, \emph{distancematrix=Distancematrix}}{}
Computing the sequence order coupling numbers from 1 to maxlag

for a given protein sequence based on the Grantham chemical distance

matrix.

Usage:

result = GetSequenceOrderCouplingNumberGrant(protein, maxlag,distancematrix)

Input: protein is a pure protein sequence

maxlag is the maximum lag and the length of the protein should be larger

than maxlag. default is 30.

distancematrix is a dict form containing Grantham chemical distance

matrix. omitted!

Output: result is a dict form containing all sequence order coupling numbers

\end{fulllineitems}

\index{GetSequenceOrderCouplingNumberSW() (in module QuasiSequenceOrder)}

\begin{fulllineitems}
\phantomsection\label{reference/QuasiSequenceOrder:QuasiSequenceOrder.GetSequenceOrderCouplingNumberSW}\pysiglinewithargsret{\sphinxcode{QuasiSequenceOrder.}\sphinxbfcode{GetSequenceOrderCouplingNumberSW}}{\emph{ProteinSequence}, \emph{maxlag=30}, \emph{distancematrix=Distancematrix}}{}
Computing the sequence order coupling numbers from 1 to maxlag

for a given protein sequence based on the Schneider-Wrede physicochemical

distance matrix

Usage:

result = GetSequenceOrderCouplingNumberSW(protein, maxlag,distancematrix)

Input: protein is a pure protein sequence

maxlag is the maximum lag and the length of the protein should be larger

than maxlag. default is 30.

distancematrix is a dict form containing Schneider-Wrede physicochemical

distance matrix. omitted!

Output: result is a dict form containing all sequence order coupling numbers based

\end{fulllineitems}

\index{GetSequenceOrderCouplingNumberTotal() (in module QuasiSequenceOrder)}

\begin{fulllineitems}
\phantomsection\label{reference/QuasiSequenceOrder:QuasiSequenceOrder.GetSequenceOrderCouplingNumberTotal}\pysiglinewithargsret{\sphinxcode{QuasiSequenceOrder.}\sphinxbfcode{GetSequenceOrderCouplingNumberTotal}}{\emph{ProteinSequence}, \emph{maxlag=30}}{}
Computing the sequence order coupling numbers from 1 to maxlag
for a given protein sequence.
Usage:
result = GetSequenceOrderCouplingNumberTotal(protein, maxlag)

Input: protein is a pure protein sequence

maxlag is the maximum lag and the length of the protein should be larger

than maxlag. default is 30.

\end{fulllineitems}

\index{GetSequenceOrderCouplingNumberp() (in module QuasiSequenceOrder)}

\begin{fulllineitems}
\phantomsection\label{reference/QuasiSequenceOrder:QuasiSequenceOrder.GetSequenceOrderCouplingNumberp}\pysiglinewithargsret{\sphinxcode{QuasiSequenceOrder.}\sphinxbfcode{GetSequenceOrderCouplingNumberp}}{\emph{ProteinSequence}, \emph{maxlag=30}, \emph{distancematrix=\{\}}}{}
Computing the sequence order coupling numbers from 1 to maxlag

for a given protein sequence based on the user-defined property.

Usage:

result = GetSequenceOrderCouplingNumberp(protein, maxlag,distancematrix)

Input: protein is a pure protein sequence

maxlag is the maximum lag and the length of the protein should be larger

than maxlag. default is 30.

distancematrix is the a dict form containing 400 distance values

Output: result is a dict form containing all sequence order coupling numbers based

\end{fulllineitems}



\section{PyPretreat}
\label{reference/PyPretreat::doc}\label{reference/PyPretreat:pypretreat}

\subsection{PyPretreatDNA module}
\label{reference/PyPretreatDNA:module-PyPretreatDNA}\label{reference/PyPretreatDNA:pypretreatdna-module}\label{reference/PyPretreatDNA::doc}\index{PyPretreatDNA (module)}
Created on Wed May 18 14:06:37 2016

@author: yzj
\index{ALPHABET (in module PyPretreatDNA)}

\begin{fulllineitems}
\phantomsection\label{reference/PyPretreatDNA:PyPretreatDNA.ALPHABET}\pysigline{\sphinxcode{PyPretreatDNA.}\sphinxbfcode{ALPHABET}\sphinxstrong{ = `ACGT'}}
Used for process original data.

\end{fulllineitems}

\index{ConvertPhycheIndexToDict() (in module PyPretreatDNA)}

\begin{fulllineitems}
\phantomsection\label{reference/PyPretreatDNA:PyPretreatDNA.ConvertPhycheIndexToDict}\pysiglinewithargsret{\sphinxcode{PyPretreatDNA.}\sphinxbfcode{ConvertPhycheIndexToDict}}{\emph{phyche\_index}}{}
\end{fulllineitems}

\index{DNAChecks() (in module PyPretreatDNA)}

\begin{fulllineitems}
\phantomsection\label{reference/PyPretreatDNA:PyPretreatDNA.DNAChecks}\pysiglinewithargsret{\sphinxcode{PyPretreatDNA.}\sphinxbfcode{DNAChecks}}{\emph{s}}{}
\end{fulllineitems}

\index{Frequency() (in module PyPretreatDNA)}

\begin{fulllineitems}
\phantomsection\label{reference/PyPretreatDNA:PyPretreatDNA.Frequency}\pysiglinewithargsret{\sphinxcode{PyPretreatDNA.}\sphinxbfcode{Frequency}}{\emph{tol\_str}, \emph{tar\_str}}{}
Generate the frequency of tar\_str in tol\_str.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{tol\_str}} -- mother string.

\item {} 
\textbf{\texttt{tar\_str}} -- substring.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{GeneratePhycheValue() (in module PyPretreatDNA)}

\begin{fulllineitems}
\phantomsection\label{reference/PyPretreatDNA:PyPretreatDNA.GeneratePhycheValue}\pysiglinewithargsret{\sphinxcode{PyPretreatDNA.}\sphinxbfcode{GeneratePhycheValue}}{\emph{k}, \emph{phyche\_index=None}, \emph{all\_property=False}, \emph{extra\_phyche\_index=None}}{}
Combine the user selected phyche\_list, is\_all\_property and 
extra\_phyche\_index to a new standard phyche\_value.
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#

\end{fulllineitems}

\index{GetData() (in module PyPretreatDNA)}

\begin{fulllineitems}
\phantomsection\label{reference/PyPretreatDNA:PyPretreatDNA.GetData}\pysiglinewithargsret{\sphinxcode{PyPretreatDNA.}\sphinxbfcode{GetData}}{\emph{input\_data}, \emph{desc=False}}{}
Get sequence data from file or list with check.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{input\_data}} -- type file or list

\item {} 
\textbf{\texttt{desc}} -- with this option, the return value will be a Seq object list(it only works in file object).

\end{itemize}

\item[{Returns}] \leavevmode
sequence data or shutdown.

\end{description}\end{quote}

\end{fulllineitems}

\index{GetSequenceCheckDna() (in module PyPretreatDNA)}

\begin{fulllineitems}
\phantomsection\label{reference/PyPretreatDNA:PyPretreatDNA.GetSequenceCheckDna}\pysiglinewithargsret{\sphinxcode{PyPretreatDNA.}\sphinxbfcode{GetSequenceCheckDna}}{\emph{f}}{}
Read the fasta file.

Input: f: HANDLE to input. e.g. sys.stdin, or open(\textless{}file\textgreater{})

\end{fulllineitems}

\index{IsFasta() (in module PyPretreatDNA)}

\begin{fulllineitems}
\phantomsection\label{reference/PyPretreatDNA:PyPretreatDNA.IsFasta}\pysiglinewithargsret{\sphinxcode{PyPretreatDNA.}\sphinxbfcode{IsFasta}}{\emph{seq}}{}
Judge the Seq object is in FASTA format.
Two situation:
1. No seq name.
2. Seq name is illegal.
3. No sequence.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\textbf{\texttt{seq}} -- Seq object.

\end{description}\end{quote}

\end{fulllineitems}

\index{IsSequenceList() (in module PyPretreatDNA)}

\begin{fulllineitems}
\phantomsection\label{reference/PyPretreatDNA:PyPretreatDNA.IsSequenceList}\pysiglinewithargsret{\sphinxcode{PyPretreatDNA.}\sphinxbfcode{IsSequenceList}}{\emph{sequence\_list}}{}
Judge the sequence list is within the scope of alphabet and 
change the lowercase to capital.
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#

\end{fulllineitems}

\index{IsUnderAlphabet() (in module PyPretreatDNA)}

\begin{fulllineitems}
\phantomsection\label{reference/PyPretreatDNA:PyPretreatDNA.IsUnderAlphabet}\pysiglinewithargsret{\sphinxcode{PyPretreatDNA.}\sphinxbfcode{IsUnderAlphabet}}{\emph{s}, \emph{alphabet}}{}
Judge the string is within the scope of the alphabet or not.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{s}} -- The string.

\item {} 
\textbf{\texttt{alphabet}} -- alphabet.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{NormalizeIndex() (in module PyPretreatDNA)}

\begin{fulllineitems}
\phantomsection\label{reference/PyPretreatDNA:PyPretreatDNA.NormalizeIndex}\pysiglinewithargsret{\sphinxcode{PyPretreatDNA.}\sphinxbfcode{NormalizeIndex}}{\emph{phyche\_index}, \emph{is\_convert\_dict=False}}{}
\end{fulllineitems}

\index{ReadFasta() (in module PyPretreatDNA)}

\begin{fulllineitems}
\phantomsection\label{reference/PyPretreatDNA:PyPretreatDNA.ReadFasta}\pysiglinewithargsret{\sphinxcode{PyPretreatDNA.}\sphinxbfcode{ReadFasta}}{\emph{f}}{}
Read a fasta file.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\textbf{\texttt{f}} -- HANDLE to input. e.g. sys.stdin, or open(\textless{}file\textgreater{})

\end{description}\end{quote}

\end{fulllineitems}

\index{ReadFastaCheckDna() (in module PyPretreatDNA)}

\begin{fulllineitems}
\phantomsection\label{reference/PyPretreatDNA:PyPretreatDNA.ReadFastaCheckDna}\pysiglinewithargsret{\sphinxcode{PyPretreatDNA.}\sphinxbfcode{ReadFastaCheckDna}}{\emph{f}}{}
Read the fasta file, and check its legality.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\textbf{\texttt{f}} -- HANDLE to input. e.g. sys.stdin, or open(\textless{}file\textgreater{})

\end{description}\end{quote}

\end{fulllineitems}

\index{ReadFastaYield() (in module PyPretreatDNA)}

\begin{fulllineitems}
\phantomsection\label{reference/PyPretreatDNA:PyPretreatDNA.ReadFastaYield}\pysiglinewithargsret{\sphinxcode{PyPretreatDNA.}\sphinxbfcode{ReadFastaYield}}{\emph{f}}{}
Yields a Seq object.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\textbf{\texttt{f}} -- HANDLE to input. e.g. sys.stdin, or open(\textless{}file\textgreater{})

\end{description}\end{quote}

\end{fulllineitems}

\index{Seq (class in PyPretreatDNA)}

\begin{fulllineitems}
\phantomsection\label{reference/PyPretreatDNA:PyPretreatDNA.Seq}\pysiglinewithargsret{\sphinxstrong{class }\sphinxcode{PyPretreatDNA.}\sphinxbfcode{Seq}}{\emph{name}, \emph{seq}, \emph{no}}{}
\end{fulllineitems}

\index{StandardDeviation() (in module PyPretreatDNA)}

\begin{fulllineitems}
\phantomsection\label{reference/PyPretreatDNA:PyPretreatDNA.StandardDeviation}\pysiglinewithargsret{\sphinxcode{PyPretreatDNA.}\sphinxbfcode{StandardDeviation}}{\emph{value\_list}}{}
\end{fulllineitems}

\index{WriteLibsvm() (in module PyPretreatDNA)}

\begin{fulllineitems}
\phantomsection\label{reference/PyPretreatDNA:PyPretreatDNA.WriteLibsvm}\pysiglinewithargsret{\sphinxcode{PyPretreatDNA.}\sphinxbfcode{WriteLibsvm}}{\emph{vector\_list}, \emph{label\_list}, \emph{write\_file}}{}
\end{fulllineitems}



\subsection{PyPretreatMol module}
\label{reference/PyPretreatMol:module-PyPretreatMol}\label{reference/PyPretreatMol::doc}\label{reference/PyPretreatMol:pypretreatmol-module}\index{PyPretreatMol (module)}\index{StandardMol() (in module PyPretreatMol)}

\begin{fulllineitems}
\phantomsection\label{reference/PyPretreatMol:PyPretreatMol.StandardMol}\pysiglinewithargsret{\sphinxcode{PyPretreatMol.}\sphinxbfcode{StandardMol}}{\emph{mol}}{}
The function for performing standardization of molecules and deriving parent molecules.
The function contains derive fragment, charge, tautomer, isotope and stereo parent molecules.
The primary usage is:

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{n}{mol1} \PYG{o}{=} \PYG{n}{Chem}\PYG{o}{.}\PYG{n}{MolFromSmiles}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{C1=CC=CC=C1}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{mol2} \PYG{o}{=} \PYG{n}{s}\PYG{o}{.}\PYG{n}{standardize}\PYG{p}{(}\PYG{n}{mol1}\PYG{p}{)}
\end{Verbatim}

\end{fulllineitems}

\index{StandardSmi() (in module PyPretreatMol)}

\begin{fulllineitems}
\phantomsection\label{reference/PyPretreatMol:PyPretreatMol.StandardSmi}\pysiglinewithargsret{\sphinxcode{PyPretreatMol.}\sphinxbfcode{StandardSmi}}{\emph{smi}}{}
The function for performing standardization of molecules and deriving parent molecules.
The function contains derive fragment, charge, tautomer, isotope and stereo parent molecules.
The primary usage is:

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{n}{smi} \PYG{o}{=} \PYG{n}{StandardSmi}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{C[n+]1c([N\PYGZhy{}](C))cccc1}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{Verbatim}

\end{fulllineitems}

\index{StandardizeMol (class in PyPretreatMol)}

\begin{fulllineitems}
\phantomsection\label{reference/PyPretreatMol:PyPretreatMol.StandardizeMol}\pysiglinewithargsret{\sphinxstrong{class }\sphinxcode{PyPretreatMol.}\sphinxbfcode{StandardizeMol}}{\emph{normalizations=(Normalization(u'Nitro to N+(O-)=O', u'{[}*:1{]}{[}N,P,As,Sb:2{]}(={[}O,S,Se,Te:3{]})={[}O,S,Se,Te:4{]}\textgreater{}\textgreater{}{[}*:1{]}{[}*+1:2{]}({[}*-1:3{]})={[}*:4{]}'), Normalization(u'Sulfone to S(=O)(=O)', u'{[}S+2:1{]}({[}O-:2{]})({[}O-:3{]})\textgreater{}\textgreater{}{[}S+0:1{]}(={[}O-0:2{]})(={[}O-0:3{]})'), Normalization(u'Pyridine oxide to n+O-`, u'{[}n:1{]}={[}O:2{]}\textgreater{}\textgreater{}{[}n+:1{]}{[}O-:2{]}'), Normalization(u'Azide to N=N+=N-`, u'{[}*,H:1{]}{[}N:2{]}={[}N:3{]}\#{[}N:4{]}\textgreater{}\textgreater{}{[}*,H:1{]}{[}N:2{]}={[}N+:3{]}={[}N-:4{]}'), Normalization(u'Diazo/azo to =N+=N-`, u'{[}*:1{]}={[}N:2{]}\#{[}N:3{]}\textgreater{}\textgreater{}{[}*:1{]}={[}N+:2{]}={[}N-:3{]}'), Normalization(u'Sulfoxide to -S+(O-)-`, u'{[}!O:1{]}{[}S+0;X3:2{]}(={[}O:3{]}){[}!O:4{]}\textgreater{}\textgreater{}{[}*:1{]}{[}S+1:2{]}({[}O-:3{]}){[}*:4{]}'), Normalization(u'Phosphate to P(O-)=O', u'{[}O,S,Se,Te;-1:1{]}{[}P+;D4:2{]}{[}O,S,Se,Te;-1:3{]}\textgreater{}\textgreater{}{[}*+0:1{]}={[}P+0;D5:2{]}{[}*-1:3{]}'), Normalization(u'Amidinium to C(=NH2+)NH2', u'{[}C,S;X3+1:1{]}({[}NX3:2{]}){[}NX3!H0:3{]}\textgreater{}\textgreater{}{[}*+0:1{]}({[}N:2{]})={[}N+:3{]}'), Normalization(u'Normalize hydrazine-diazonium', u'{[}CX4:1{]}{[}NX3H:2{]}-{[}NX3H:3{]}{[}CX4:4{]}{[}NX2+:5{]}\#{[}NX1:6{]}\textgreater{}\textgreater{}{[}CX4:1{]}{[}NH0:2{]}={[}NH+:3{]}{[}C:4{]}{[}N+0:5{]}={[}NH:6{]}'), Normalization(u'Recombine 1,3-separated charges', u'{[}N,P,As,Sb,O,S,Se,Te;-1:1{]}-{[}A:2{]}={[}N,P,As,Sb,O,S,Se,Te;+1:3{]}\textgreater{}\textgreater{}{[}*-0:1{]}={[}*:2{]}-{[}*+0:3{]}'), Normalization(u'Recombine 1,3-separated charges', u'{[}n,o,p,s;-1:1{]}:{[}a:2{]}={[}N,O,P,S;+1:3{]}\textgreater{}\textgreater{}{[}*-0:1{]}:{[}*:2{]}-{[}*+0:3{]}'), Normalization(u'Recombine 1,3-separated charges', u'{[}N,O,P,S;-1:1{]}-{[}a:2{]}:{[}n,o,p,s;+1:3{]}\textgreater{}\textgreater{}{[}*-0:1{]}={[}*:2{]}:{[}*+0:3{]}'), Normalization(u'Recombine 1,5-separated charges', u'{[}N,P,As,Sb,O,S,Se,Te;-1:1{]}-{[}A+0:2{]}={[}A:3{]}-{[}A:4{]}={[}N,P,As,Sb,O,S,Se,Te;+1:5{]}\textgreater{}\textgreater{}{[}*-0:1{]}={[}*:2{]}-{[}*:3{]}={[}*:4{]}-{[}*+0:5{]}'), Normalization(u'Recombine 1,5-separated charges', u'{[}n,o,p,s;-1:1{]}:{[}a:2{]}:{[}a:3{]}:{[}c:4{]}={[}N,O,P,S;+1:5{]}\textgreater{}\textgreater{}{[}*-0:1{]}:{[}*:2{]}:{[}*:3{]}:{[}c:4{]}-{[}*+0:5{]}'), Normalization(u'Recombine 1,5-separated charges', u'{[}N,O,P,S;-1:1{]}-{[}c:2{]}:{[}a:3{]}:{[}a:4{]}:{[}n,o,p,s;+1:5{]}\textgreater{}\textgreater{}{[}*-0:1{]}={[}c:2{]}:{[}*:3{]}:{[}*:4{]}:{[}*+0:5{]}'), Normalization(u'Normalize 1,3 conjugated cation', u'{[}N,O;+0!H0:1{]}-{[}A:2{]}={[}N!\$(*{[}O-{]}),O;+1H0:3{]}\textgreater{}\textgreater{}{[}*+1:1{]}={[}*:2{]}-{[}*+0:3{]}'), Normalization(u'Normalize 1,3 conjugated cation', u'{[}n;+0!H0:1{]}:{[}c:2{]}={[}N!\$(*{[}O-{]}),O;+1H0:3{]}\textgreater{}\textgreater{}{[}*+1:1{]}:{[}*:2{]}-{[}*+0:3{]}'), Normalization(u'Normalize 1,3 conjugated cation', u'{[}N,O;+0!H0:1{]}-{[}c:2{]}:{[}n!\$(*{[}O-{]}),o;+1H0:3{]}\textgreater{}\textgreater{}{[}*+1:1{]}={[}*:2{]}:{[}*+0:3{]}'), Normalization(u'Normalize 1,5 conjugated cation', u'{[}N,O;+0!H0:1{]}-{[}A:2{]}={[}A:3{]}-{[}A:4{]}={[}N!\$(*{[}O-{]}),O;+1H0:5{]}\textgreater{}\textgreater{}{[}*+1:1{]}={[}*:2{]}-{[}*:3{]}={[}*:4{]}-{[}*+0:5{]}'), Normalization(u'Normalize 1,5 conjugated cation', u'{[}n;+0!H0:1{]}:{[}a:2{]}:{[}a:3{]}:{[}c:4{]}={[}N!\$(*{[}O-{]}),O;+1H0:5{]}\textgreater{}\textgreater{}{[}n+1:1{]}:{[}*:2{]}:{[}*:3{]}:{[}*:4{]}-{[}*+0:5{]}'), Normalization(u'Normalize 1,5 conjugated cation', u'{[}N,O;+0!H0:1{]}-{[}c:2{]}:{[}a:3{]}:{[}a:4{]}:{[}n!\$(*{[}O-{]}),o;+1H0:5{]}\textgreater{}\textgreater{}{[}*+1:1{]}={[}c:2{]}:{[}*:3{]}:{[}*:4{]}:{[}*+0:5{]}'), Normalization(u'Normalize 1,5 conjugated cation', u'{[}n;+0!H0:1{]}1:{[}a:2{]}:{[}a:3{]}:{[}a:4{]}:{[}n!\$(*{[}O-{]});+1H0:5{]}1\textgreater{}\textgreater{}{[}n+1:1{]}1:{[}*:2{]}:{[}*:3{]}:{[}*:4{]}:{[}n+0:5{]}1'), Normalization(u'Normalize 1,5 conjugated cation', u'{[}n;+0!H0:1{]}:{[}a:2{]}:{[}a:3{]}:{[}a:4{]}:{[}n!\$(*{[}O-{]});+1H0:5{]}\textgreater{}\textgreater{}{[}n+1:1{]}:{[}*:2{]}:{[}*:3{]}:{[}*:4{]}:{[}n+0:5{]}'), Normalization(u'Charge normalization', u'{[}F,Cl,Br,I,At;-1:1{]}={[}O:2{]}\textgreater{}\textgreater{}{[}*-0:1{]}{[}O-:2{]}'), Normalization(u'Charge recombination', u'{[}N,P,As,Sb;-1:1{]}={[}C+;v3:2{]}\textgreater{}\textgreater{}{[}*+0:1{]}\#{[}C+0:2{]}')), acid\_base\_pairs=(AcidBasePair(u'-OSO3H', u'OS(=O)(=O){[}OH{]}', u'OS(=O)(=O){[}O-{]}'), AcidBasePair(u'u2013SO3H', u'{[}!O{]}S(=O)(=O){[}OH{]}', u'{[}!O{]}S(=O)(=O){[}O-{]}'), AcidBasePair(u'-OSO2H', u'O{[}SD3{]}(=O){[}OH{]}', u'O{[}SD3{]}(=O){[}O-{]}'), AcidBasePair(u'-SO2H', u'{[}!O{]}{[}SD3{]}(=O){[}OH{]}', u'{[}!O{]}{[}SD3{]}(=O){[}O-{]}'), AcidBasePair(u'-OPO3H2', u'OP(=O)({[}OH{]}){[}OH{]}', u'OP(=O)({[}OH{]}){[}O-{]}'), AcidBasePair(u'-PO3H2', u'{[}!O{]}P(=O)({[}OH{]}){[}OH{]}', u'{[}!O{]}P(=O)({[}OH{]}){[}O-{]}'), AcidBasePair(u'-CO2H', u'C(=O){[}OH{]}', u'C(=O){[}O-{]}'), AcidBasePair(u'thiophenol', u'c{[}SH{]}', u'c{[}S-{]}'), AcidBasePair(u'(-OPO3H)-`, u'OP(=O)({[}OH{]}){[}O-{]}', u'OP(=O)({[}O-{]}){[}O-{]}'), AcidBasePair(u'(-PO3H)-`, u'{[}!O{]}P(=O)({[}OH{]}){[}O-{]}', u'{[}!O{]}P(=O)({[}O-{]}){[}O-{]}'), AcidBasePair(u'phthalimide', u'O=C2c1ccccc1C(=O){[}NH{]}2', u'O=C2c1ccccc1C(=O){[}N-{]}2'), AcidBasePair(u'CO3H (peracetyl)', u'C(=O)O{[}OH{]}', u'C(=O)O{[}O-{]}'), AcidBasePair(u'alpha-carbon-hydrogen-nitro group', u'O=N(O){[}CH{]}', u'O=N(O){[}C-{]}'), AcidBasePair(u'-SO2NH2', u'S(=O)(=O){[}NH2{]}', u'S(=O)(=O){[}NH-{]}'), AcidBasePair(u'-OBO2H2', u'OB({[}OH{]}){[}OH{]}', u'OB({[}OH{]}){[}O-{]}'), AcidBasePair(u'-BO2H2', u'{[}!O{]}B({[}OH{]}){[}OH{]}', u'{[}!O{]}B({[}OH{]}){[}O-{]}'), AcidBasePair(u'phenol', u'c{[}OH{]}', u'c{[}O-{]}'), AcidBasePair(u'SH (aliphatic)', u'C{[}SH{]}', u'C{[}S-{]}'), AcidBasePair(u'(-OBO2H)-`, u'OB({[}OH{]}){[}O-{]}', u'OB({[}O-{]}){[}O-{]}'), AcidBasePair(u'(-BO2H)-`, u'{[}!O{]}B({[}OH{]}){[}O-{]}', u'{[}!O{]}B({[}O-{]}){[}O-{]}'), AcidBasePair(u'cyclopentadiene', u'{[}CH2{]}1C=CC=C1', u'{[}C-{]}1C=CC=C1'), AcidBasePair(u'-CONH2', u'C(=O){[}NH2{]}', u'C(=O){[}NH-{]}'), AcidBasePair(u'imidazole', u'c1cnc{[}n{]}1', u'c1cnc{[}n-{]}1'), AcidBasePair(u'-OH', u'{[}CX4{]}{[}OH{]}', u'{[}CX4{]}{[}O-{]}'), AcidBasePair(u'alpha-carbon-hydrogen-keto group', u'O=C{[}CH{]}', u'O=C{[}C-{]}'), AcidBasePair(u'alpha-carbon-hydrogen-acetyl ester group', u'OC(=O){[}CH{]}', u'OC(=O){[}C-{]}'), AcidBasePair(u'sp carbon hydrogen', u'C\#{[}CH{]}', u'C\#{[}C-{]}'), AcidBasePair(u'alpha-carbon-hydrogen-sulfone group', u'CS(=O)(=O)C{[}CH{]}', u'CS(=O)(=O)C{[}C-{]}'), AcidBasePair(u'alpha-carbon-hydrogen-sulfoxide group', u'C{[}SD3{]}(=O)C{[}CH{]}', u'C{[}SD3{]}(=O)C{[}C-{]}'), AcidBasePair(u'-NH2', u'{[}CX4{]}{[}NH2{]}', u'{[}CX4{]}{[}NH-{]}'), AcidBasePair(u'benzyl hydrogen', u'c{[}CD4H{]}', u'c{[}CD3-{]}'), AcidBasePair(u'sp2-carbon hydrogen', u'{[}CX3{]}={[}CX3H{]}', u'{[}CX3{]}={[}CX2-{]}'), AcidBasePair(u'sp3-carbon hydrogen', u'{[}CX4H{]}', u'{[}CX3-{]}')), tautomer\_transforms=(TautomerTransform(u`1,3 (thio)keto/enol f', u'{[}CX4!H0{]}{[}C{]}={[}O,S,Se,Te;X1{]}', {[}{]}, {[}{]}), TautomerTransform(u`1,3 (thio)keto/enol r', u'{[}O,S,Se,Te;X2!H0{]}{[}C{]}={[}C{]}', {[}{]}, {[}{]}), TautomerTransform(u`1,5 (thio)keto/enol f', u'{[}CX4,NX3;!H0{]}{[}C{]}={[}C{]}{[}CH0{]}={[}O,S,Se,Te;X1{]}', {[}{]}, {[}{]}), TautomerTransform(u`1,5 (thio)keto/enol r', u'{[}O,S,Se,Te;X2!H0{]}{[}CH0{]}=,:{[}C{]}{[}C{]}=,:{[}C,N{]}', {[}{]}, {[}{]}), TautomerTransform(u'aliphatic imine f', u'{[}CX4!H0{]}{[}C{]}={[}NX2{]}', {[}{]}, {[}{]}), TautomerTransform(u'aliphatic imine r', u'{[}NX3!H0{]}{[}C{]}={[}CX3{]}', {[}{]}, {[}{]}), TautomerTransform(u'special imine f', u'{[}N!H0{]}{[}C{]}={[}CX3R0{]}', {[}{]}, {[}{]}), TautomerTransform(u'special imine r', u'{[}CX4!H0{]}{[}c{]}=,:{[}n{]}', {[}{]}, {[}{]}), TautomerTransform(u`1,3 aromatic heteroatom H shift f', u'{[}\#7!H0{]}{[}\#6R1{]}={[}O,\#7X2{]}', {[}{]}, {[}{]}), TautomerTransform(u`1,3 aromatic heteroatom H shift r', u'{[}O,\#7;!H0{]}{[}\#6R1{]}=,:{[}\#7X2{]}', {[}{]}, {[}{]}), TautomerTransform(u`1,3 heteroatom H shift', u'{[}\#7,S,O,Se,Te;!H0{]}{[}\#7X2,\#6,\#15{]}={[}\#7,\#16,\#8,Se,Te{]}', {[}{]}, {[}{]}), TautomerTransform(u`1,5 aromatic heteroatom H shift', u'{[}n,s,o;!H0{]}:{[}c,n{]}:{[}c{]}:{[}c,n{]}:{[}n,s,o;H0{]}', {[}{]}, {[}{]}), TautomerTransform(u`1,5 aromatic heteroatom H shift f', u'{[}\#7,\#16,\#8,Se,Te;!H0{]}{[}\#6,nX2{]}=,:{[}\#6,nX2{]}{[}\#6,\#7X2{]}=,:{[}\#7X2,S,O,Se,Te{]}', {[}{]}, {[}{]}), TautomerTransform(u`1,5 aromatic heteroatom H shift r', u'{[}\#7,S,O,Se,Te;!H0{]}{[}\#6,\#7X2{]}=,:{[}\#6,nX2{]}{[}\#6,nX2{]}=,:{[}\#7,\#16,\#8,Se,Te{]}', {[}{]}, {[}{]}), TautomerTransform(u`1,7 aromatic heteroatom H shift f', u'{[}\#7,\#8,\#16,Se,Te;!H0{]}{[}\#6,\#7X2{]}=,:{[}\#6,\#7X2{]}{[}\#6,\#7X2{]}=,:{[}\#6{]}{[}\#6,\#7X2{]}=,:{[}\#7X2,S,O,Se,Te,CX3{]}', {[}{]}, {[}{]}), TautomerTransform(u`1,7 aromatic heteroatom H shift r', u'{[}\#7,S,O,Se,Te,CX4;!H0{]}{[}\#6,\#7X2{]}=,:{[}\#6{]}{[}\#6,\#7X2{]}=,:{[}\#6,\#7X2{]}{[}\#6,\#7X2{]}=,:{[}NX2,S,O,Se,Te{]}', {[}{]}, {[}{]}), TautomerTransform(u`1,9 aromatic heteroatom H shift f', u'{[}\#7,O;!H0{]}{[}\#6,\#7X2{]}=,:{[}\#6,\#7X2{]}{[}\#6,\#7X2{]}=,:{[}\#6,\#7X2{]}{[}\#6,\#7X2{]}=,:{[}\#6,\#7X2{]}{[}\#6,\#7X2{]}=,:{[}\#7,O{]}', {[}{]}, {[}{]}), TautomerTransform(u`1,11 aromatic heteroatom H shift f', u'{[}\#7,O;!H0{]}{[}\#6,nX2{]}=,:{[}\#6,nX2{]}{[}\#6,nX2{]}=,:{[}\#6,nX2{]}{[}\#6,nX2{]}=,:{[}\#6,nX2{]}{[}\#6,nX2{]}=,:{[}\#6,nX2{]}{[}\#6,nX2{]}=,:{[}\#7X2,O{]}', {[}{]}, {[}{]}), TautomerTransform(u'furanone f', u'{[}O,S,N;!H0{]}{[}\#6X3r5;\$({[}\#6{]}{[}!\#6{]}){]}=,:{[}\#6X3r5{]}', {[}{]}, {[}{]}), TautomerTransform(u'furanone r', u'{[}\#6r5!H0{]}{[}\#6X3r5;\$({[}\#6{]}{[}!\#6{]}){]}={[}O,S,N{]}', {[}{]}, {[}{]}), TautomerTransform(u'keten/ynol f', u'{[}C!H0{]}={[}C{]}={[}O,S,Se,Te;X1{]}', {[}rdkit.Chem.rdchem.BondType.TRIPLE, rdkit.Chem.rdchem.BondType.SINGLE{]}, {[}{]}), TautomerTransform(u'keten/ynol r', u'{[}O,S,Se,Te;!H0X2{]}{[}C{]}\#{[}C{]}', {[}rdkit.Chem.rdchem.BondType.DOUBLE, rdkit.Chem.rdchem.BondType.DOUBLE{]}, {[}{]}), TautomerTransform(u'ionic nitro/aci-nitro f', u'{[}C!H0{]}{[}N+;\$({[}N{]}{[}O-{]}){]}={[}O{]}', {[}{]}, {[}{]}), TautomerTransform(u'ionic nitro/aci-nitro r', u'{[}O!H0{]}{[}N+;\$({[}N{]}{[}O-{]}){]}={[}C{]}', {[}{]}, {[}{]}), TautomerTransform(u'oxim/nitroso f', u'{[}O!H0{]}{[}N{]}={[}C{]}', {[}{]}, {[}{]}), TautomerTransform(u'oxim/nitroso r', u'{[}C!H0{]}{[}N{]}={[}O{]}', {[}{]}, {[}{]}), TautomerTransform(u'oxim/nitroso via phenol f', u'{[}O!H0{]}{[}N{]}={[}C{]}{[}C{]}={[}C{]}{[}C{]}={[}OH0{]}', {[}{]}, {[}{]}), TautomerTransform(u'oxim/nitroso via phenol r', u'{[}O!H0{]}{[}c{]}:{[}c{]}:{[}c{]}:{[}c{]}{[}N{]}={[}OH0{]}', {[}{]}, {[}{]}), TautomerTransform(u'cyano/iso-cyanic acid f', u'{[}O!H0{]}{[}C{]}\#{[}N{]}', {[}rdkit.Chem.rdchem.BondType.DOUBLE, rdkit.Chem.rdchem.BondType.DOUBLE{]}, {[}{]}), TautomerTransform(u'cyano/iso-cyanic acid r', u'{[}N!H0{]}={[}C{]}={[}O{]}', {[}rdkit.Chem.rdchem.BondType.TRIPLE, rdkit.Chem.rdchem.BondType.SINGLE{]}, {[}{]}), TautomerTransform(u'formamidinesulfinic acid f', u'{[}O,N;!H0{]}{[}C{]}{[}S,Se,Te{]}={[}O{]}', {[}rdkit.Chem.rdchem.BondType.DOUBLE, rdkit.Chem.rdchem.BondType.SINGLE, rdkit.Chem.rdchem.BondType.SINGLE{]}, {[}{]}), TautomerTransform(u'formamidinesulfinic acid r', u'{[}O!H0{]}{[}S,Se,Te{]}{[}C{]}={[}O,N{]}', {[}rdkit.Chem.rdchem.BondType.DOUBLE, rdkit.Chem.rdchem.BondType.SINGLE, rdkit.Chem.rdchem.BondType.SINGLE{]}, {[}{]}), TautomerTransform(u'isocyanide f', u'{[}C-0!H0{]}\#{[}N+0{]}', {[}rdkit.Chem.rdchem.BondType.TRIPLE{]}, {[}-1, 1{]}), TautomerTransform(u'isocyanide r', u'{[}N+!H0{]}\#{[}C-{]}', {[}rdkit.Chem.rdchem.BondType.TRIPLE{]}, {[}-1, 1{]}), TautomerTransform(u'phosphonic acid f', u'{[}OH{]}{[}PH0{]}', {[}rdkit.Chem.rdchem.BondType.DOUBLE{]}, {[}{]}), TautomerTransform(u'phosphonic acid r', u'{[}PH{]}={[}O{]}', {[}rdkit.Chem.rdchem.BondType.SINGLE{]}, {[}{]})), tautomer\_scores=(TautomerScore(u'benzoquinone', u'{[}\#6{]}1({[}\#6{]}={[}\#6{]}{[}\#6{]}({[}\#6{]}={[}\#6{]}1)=,:{[}N,S,O{]})=,:{[}N,S,O{]}', 25), TautomerScore(u'oxim', u'{[}\#6{]}={[}N{]}{[}OH{]}', 4), TautomerScore(u'C=O', u'{[}\#6{]}=,:{[}\#8{]}', 2), TautomerScore(u'N=O', u'{[}\#7{]}=,:{[}\#8{]}', 2), TautomerScore(u'P=O', u'{[}\#15{]}=,:{[}\#8{]}', 2), TautomerScore(u'C=hetero', u'{[}\#6{]}={[}!\#1;!\#6{]}', 1), TautomerScore(u'methyl', u'{[}CX4H3{]}', 1), TautomerScore(u'guanidine terminal=N', u'{[}\#7{]}{[}\#6{]}(={[}NR0{]}){[}\#7H0{]}', 1), TautomerScore(u'guanidine endocyclic=N', u'{[}\#7;R{]}{[}\#6;R{]}({[}N{]})={[}\#7;R{]}', 2), TautomerScore(u'aci-nitro', u'{[}\#6{]}={[}N+{]}({[}O-{]}){[}OH{]}', -4)), max\_restarts=200, max\_tautomers=1000, prefer\_organic=False}}{}
Bases: \href{https://docs.python.org/2/library/functions.html\#object}{\sphinxcode{object}}

The main class for performing standardization of molecules and deriving parent molecules.

The primary usage is via the \sphinxcode{standardize()} method:

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{n}{s} \PYG{o}{=} \PYG{n}{Standardizer}\PYG{p}{(}\PYG{p}{)}
\PYG{n}{mol1} \PYG{o}{=} \PYG{n}{Chem}\PYG{o}{.}\PYG{n}{MolFromSmiles}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{C1=CC=CC=C1}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{mol2} \PYG{o}{=} \PYG{n}{s}\PYG{o}{.}\PYG{n}{standardize}\PYG{p}{(}\PYG{n}{mol1}\PYG{p}{)}
\end{Verbatim}

There are separate methods to derive fragment, charge, tautomer, isotope and stereo parent molecules.
\index{addhs() (PyPretreatMol.StandardizeMol method)}

\begin{fulllineitems}
\phantomsection\label{reference/PyPretreatMol:PyPretreatMol.StandardizeMol.addhs}\pysiglinewithargsret{\sphinxbfcode{addhs}}{\emph{mol}}{}
\end{fulllineitems}

\index{canonicalize\_tautomer (PyPretreatMol.StandardizeMol attribute)}

\begin{fulllineitems}
\phantomsection\label{reference/PyPretreatMol:PyPretreatMol.StandardizeMol.canonicalize_tautomer}\pysigline{\sphinxbfcode{canonicalize\_tautomer}}~\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
A callable \sphinxcode{TautomerCanonicalizer} instance.

\end{description}\end{quote}

\end{fulllineitems}

\index{disconnect\_metals (PyPretreatMol.StandardizeMol attribute)}

\begin{fulllineitems}
\phantomsection\label{reference/PyPretreatMol:PyPretreatMol.StandardizeMol.disconnect_metals}\pysigline{\sphinxbfcode{disconnect\_metals}}~\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
A callable \sphinxcode{MetalDisconnector} instance.

\end{description}\end{quote}

\end{fulllineitems}

\index{largest\_fragment (PyPretreatMol.StandardizeMol attribute)}

\begin{fulllineitems}
\phantomsection\label{reference/PyPretreatMol:PyPretreatMol.StandardizeMol.largest_fragment}\pysigline{\sphinxbfcode{largest\_fragment}}~\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
A callable \sphinxcode{LargestFragmentChooser} instance.

\end{description}\end{quote}

\end{fulllineitems}

\index{normalize (PyPretreatMol.StandardizeMol attribute)}

\begin{fulllineitems}
\phantomsection\label{reference/PyPretreatMol:PyPretreatMol.StandardizeMol.normalize}\pysigline{\sphinxbfcode{normalize}}~\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
A callable \sphinxcode{Normalizer} instance.

\end{description}\end{quote}

\end{fulllineitems}

\index{reionize (PyPretreatMol.StandardizeMol attribute)}

\begin{fulllineitems}
\phantomsection\label{reference/PyPretreatMol:PyPretreatMol.StandardizeMol.reionize}\pysigline{\sphinxbfcode{reionize}}~\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
A callable \sphinxcode{Reionizer} instance.

\end{description}\end{quote}

\end{fulllineitems}

\index{rmhs() (PyPretreatMol.StandardizeMol method)}

\begin{fulllineitems}
\phantomsection\label{reference/PyPretreatMol:PyPretreatMol.StandardizeMol.rmhs}\pysiglinewithargsret{\sphinxbfcode{rmhs}}{\emph{mol}}{}
\end{fulllineitems}

\index{uncharge (PyPretreatMol.StandardizeMol attribute)}

\begin{fulllineitems}
\phantomsection\label{reference/PyPretreatMol:PyPretreatMol.StandardizeMol.uncharge}\pysigline{\sphinxbfcode{uncharge}}~\begin{quote}\begin{description}
\item[{Returns}] \leavevmode
A callable \sphinxcode{Uncharger} instance.

\end{description}\end{quote}

\end{fulllineitems}


\end{fulllineitems}

\index{ValidatorMol() (in module PyPretreatMol)}

\begin{fulllineitems}
\phantomsection\label{reference/PyPretreatMol:PyPretreatMol.ValidatorMol}\pysiglinewithargsret{\sphinxcode{PyPretreatMol.}\sphinxbfcode{ValidatorMol}}{\emph{mol}}{}
Return log messages for a given SMILES string using the default validations.

Note: This is a convenience function for quickly validating a single SMILES string.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\textbf{\texttt{smiles}} (\href{https://docs.python.org/2/library/string.html\#module-string}{\emph{\texttt{string}}}) -- The SMILES for the molecule.

\item[{Returns}] \leavevmode
A list of log messages.

\item[{Return type}] \leavevmode
list of strings.

\end{description}\end{quote}

\end{fulllineitems}

\index{ValidatorSmi() (in module PyPretreatMol)}

\begin{fulllineitems}
\phantomsection\label{reference/PyPretreatMol:PyPretreatMol.ValidatorSmi}\pysiglinewithargsret{\sphinxcode{PyPretreatMol.}\sphinxbfcode{ValidatorSmi}}{\emph{smi}}{}
Return log messages for a given SMILES string using the default validations.

Note: This is a convenience function for quickly validating a single SMILES string.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\textbf{\texttt{smiles}} (\href{https://docs.python.org/2/library/string.html\#module-string}{\emph{\texttt{string}}}) -- The SMILES for the molecule.

\item[{Returns}] \leavevmode
A list of log messages.

\item[{Return type}] \leavevmode
list of strings.

\end{description}\end{quote}

\end{fulllineitems}



\subsection{PyPretreatMolutil module}
\label{reference/PyPretreatMolutil::doc}\label{reference/PyPretreatMolutil:pypretreatmolutil-module}\phantomsection\label{reference/PyPretreatMolutil:module-PyPretreatMolutil}\index{PyPretreatMolutil (module)}
Created on Wed Jun 15 10:13:55 2016

@author: yzj
molvs.tautomer
\textasciitilde{}\textasciitilde{}\textasciitilde{}\textasciitilde{}\textasciitilde{}\textasciitilde{}\textasciitilde{}\textasciitilde{}\textasciitilde{}\textasciitilde{}\textasciitilde{}\textasciitilde{}\textasciitilde{}\textasciitilde{}

This module contains tools for enumerating tautomers and determining a canonical tautomer.
\begin{quote}\begin{description}
\item[{copyright}] \leavevmode
Copyright 2014 by Matt Swain.

\item[{license}] \leavevmode
MIT, see LICENSE file for more details.

\end{description}\end{quote}


\subsection{PyPretreatPro module}
\label{reference/PyPretreatPro:pypretreatpro-module}\label{reference/PyPretreatPro::doc}\label{reference/PyPretreatPro:module-PyPretreatPro}\index{PyPretreatPro (module)}
sequence. You can freely use and distribute it. If you hava any problem, you could

contact with us timely!

Authors: Zhijiang Yao and Dongsheng Cao.

Date: 2016.06.04

Email: \href{mailto:gadsby@163.com}{gadsby@163.com}


\bigskip\hrule{}\bigskip

\index{ProteinCheck() (in module PyPretreatPro)}

\begin{fulllineitems}
\phantomsection\label{reference/PyPretreatPro:PyPretreatPro.ProteinCheck}\pysiglinewithargsret{\sphinxcode{PyPretreatPro.}\sphinxbfcode{ProteinCheck}}{\emph{ProteinSequence}}{}
Check whether the protein sequence is a valid amino acid sequence or not

Usage:

result=ProteinCheck(protein)

Input: protein is a pure protein sequence.

Output: if the check is no problem, result will return the length of protein.

\end{fulllineitems}



\section{PyInteraction}
\label{reference/PyInteraction::doc}\label{reference/PyInteraction:pyinteraction}

\subsection{PyInteraction module}
\label{reference/PyInteraction_module:module-PyInteraction}\label{reference/PyInteraction_module:pyinteraction-module}\label{reference/PyInteraction_module::doc}\index{PyInteraction (module)}
The calculation of interaction descriptors. You can choose three types of

interacation descriptors. You can freely use and distribute it. If you

hava any problem, you could contact with us timely!

Authors: Zhijiang Yao and Dongsheng Cao.

Date: 2016.06.14

Email: \href{mailto:gadsby@163.com}{gadsby@163.com}
\index{CalculateInteraction1() (in module PyInteraction)}

\begin{fulllineitems}
\phantomsection\label{reference/PyInteraction_module:PyInteraction.CalculateInteraction1}\pysiglinewithargsret{\sphinxcode{PyInteraction.}\sphinxbfcode{CalculateInteraction1}}{\emph{dict1=\{\}}, \emph{dict2=\{\}}}{}
Calculate the two interaction features by combining two different

features.

Usage:
\begin{quote}

res=CalculateInteraction(dict1,dict2)

Input: dict1 is a dict form containing features.
\begin{quote}

dict2 is a dict form containing features.
\end{quote}

Output: res is a dict form containing interaction

features.
\end{quote}

\end{fulllineitems}

\index{CalculateInteraction2() (in module PyInteraction)}

\begin{fulllineitems}
\phantomsection\label{reference/PyInteraction_module:PyInteraction.CalculateInteraction2}\pysiglinewithargsret{\sphinxcode{PyInteraction.}\sphinxbfcode{CalculateInteraction2}}{\emph{dict1=\{\}}, \emph{dict2=\{\}}}{}
Calculate the two interaction features by combining two different

features.

Usage:
\begin{quote}

res=CalculateInteraction(dict1,dict2)

Input: dict1 is a dict form containing features.
\begin{quote}

dict2 is a dict form containing features.
\end{quote}

Output: res is a dict form containing interaction

features.
\end{quote}

\end{fulllineitems}

\index{CalculateInteraction3() (in module PyInteraction)}

\begin{fulllineitems}
\phantomsection\label{reference/PyInteraction_module:PyInteraction.CalculateInteraction3}\pysiglinewithargsret{\sphinxcode{PyInteraction.}\sphinxbfcode{CalculateInteraction3}}{\emph{dict1=\{\}}, \emph{dict2=\{\}}}{}
Calculate the two interaction features by

F={[}Fa(i)+Fb(i)),Fa(i)*Fb(i){]} (2n)

It's used in same type of descriptors.

Usage:
\begin{quote}

res=CalculateInteraction(dict1,dict2)

Input: dict1 is a dict form containing features.
\begin{quote}

dict2 is a dict form containing features.
\end{quote}

Output: res is a dict form containing interaction

features.
\end{quote}

\end{fulllineitems}



\section{PyGetMol}
\label{reference/PyGetMol:pygetmol}\label{reference/PyGetMol::doc}

\subsection{GetProtein module}
\label{reference/GetProtein:getprotein-module}\label{reference/GetProtein::doc}\phantomsection\label{reference/GetProtein:module-GetProtein}\index{GetProtein (module)}
Created on Sat Jul 13 11:18:26 2013

This module is used for downloading the PDB file from RCSB PDB web and

extract its amino acid sequence. This module can also download the protein

sequence from the uniprot (\url{http://www.uniprot.org/}) website. You can only

need input a protein ID or prepare a file (ID.txt) related to ID. You can

obtain a .txt (ProteinSequence.txt) file saving protein sequence you need.

Authors: Zhijiang Yao and Dongsheng Cao.

Date: 2016.06.04

Email: \href{mailto:gadsby@163.com}{gadsby@163.com}
\index{GetPDB() (in module GetProtein)}

\begin{fulllineitems}
\phantomsection\label{reference/GetProtein:GetProtein.GetPDB}\pysiglinewithargsret{\sphinxcode{GetProtein.}\sphinxbfcode{GetPDB}}{\emph{pdbidlist={[}{]}}}{}
Download the PDB file from PDB FTP server by providing a list of pdb id.

\end{fulllineitems}

\index{GetProteinSequence() (in module GetProtein)}

\begin{fulllineitems}
\phantomsection\label{reference/GetProtein:GetProtein.GetProteinSequence}\pysiglinewithargsret{\sphinxcode{GetProtein.}\sphinxbfcode{GetProteinSequence}}{\emph{ProteinID}}{}
Get the protein sequence from the uniprot website by ID.

Usage:

result=GetProteinSequence(ProteinID)

Input: ProteinID is a string indicating ID such as ``P48039''.

\end{fulllineitems}

\index{GetProteinSequenceFromTxt() (in module GetProtein)}

\begin{fulllineitems}
\phantomsection\label{reference/GetProtein:GetProtein.GetProteinSequenceFromTxt}\pysiglinewithargsret{\sphinxcode{GetProtein.}\sphinxbfcode{GetProteinSequenceFromTxt}}{\emph{path}, \emph{openfile}, \emph{savefile}}{}
Get the protein sequence from the uniprot website by the file containing ID.

Usage:

result=GetProteinSequenceFromTxt(path,openfile,savefile)

Input: path is a directory path containing the ID file such as ``/home/orient/protein/''

openfile is the ID file such as ``proteinID.txt''

\end{fulllineitems}

\index{GetSeqFromPDB() (in module GetProtein)}

\begin{fulllineitems}
\phantomsection\label{reference/GetProtein:GetProtein.GetSeqFromPDB}\pysiglinewithargsret{\sphinxcode{GetProtein.}\sphinxbfcode{GetSeqFromPDB}}{\emph{pdbfile='`}}{}
Get the amino acids sequences from pdb file.

\end{fulllineitems}

\index{IsFasta() (in module GetProtein)}

\begin{fulllineitems}
\phantomsection\label{reference/GetProtein:GetProtein.IsFasta}\pysiglinewithargsret{\sphinxcode{GetProtein.}\sphinxbfcode{IsFasta}}{\emph{seq}}{}
Judge the Seq object is in FASTA format.
Two situation:
1. No seq name.
2. Seq name is illegal.
3. No sequence.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\textbf{\texttt{seq}} -- Seq object.

\end{description}\end{quote}

\end{fulllineitems}

\index{ReadFasta() (in module GetProtein)}

\begin{fulllineitems}
\phantomsection\label{reference/GetProtein:GetProtein.ReadFasta}\pysiglinewithargsret{\sphinxcode{GetProtein.}\sphinxbfcode{ReadFasta}}{\emph{f}}{}
Read a fasta file.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\textbf{\texttt{f}} -- HANDLE to input. e.g. sys.stdin, or open(\textless{}file\textgreater{})

\end{description}\end{quote}

\end{fulllineitems}

\index{Seq (class in GetProtein)}

\begin{fulllineitems}
\phantomsection\label{reference/GetProtein:GetProtein.Seq}\pysiglinewithargsret{\sphinxstrong{class }\sphinxcode{GetProtein.}\sphinxbfcode{Seq}}{\emph{name}, \emph{seq}, \emph{no}}{}
\end{fulllineitems}

\index{pdbDownload() (in module GetProtein)}

\begin{fulllineitems}
\phantomsection\label{reference/GetProtein:GetProtein.pdbDownload}\pysiglinewithargsret{\sphinxcode{GetProtein.}\sphinxbfcode{pdbDownload}}{\emph{file\_list}, \emph{hostname='ftp.wwpdb.org'}, \emph{directory='/pub/pdb/data/structures/all/pdb/'}, \emph{prefix='pdb'}, \emph{suffix='.ent.gz'}}{}
Download all pdb files in file\_list and unzip them.

\end{fulllineitems}

\index{pdbSeq() (in module GetProtein)}

\begin{fulllineitems}
\phantomsection\label{reference/GetProtein:GetProtein.pdbSeq}\pysiglinewithargsret{\sphinxcode{GetProtein.}\sphinxbfcode{pdbSeq}}{\emph{pdb}, \emph{use\_atoms=False}}{}
Parse the SEQRES entries in a pdb file.  If this fails, use the ATOM
entries.  Return dictionary of sequences keyed to chain and type of
sequence used.

\end{fulllineitems}

\index{unZip() (in module GetProtein)}

\begin{fulllineitems}
\phantomsection\label{reference/GetProtein:GetProtein.unZip}\pysiglinewithargsret{\sphinxcode{GetProtein.}\sphinxbfcode{unZip}}{\emph{some\_file}, \emph{some\_output}}{}
Unzip some\_file using the gzip library and write to some\_output.

\end{fulllineitems}



\subsection{GetDNA module}
\label{reference/GetDNA:getdna-module}\label{reference/GetDNA::doc}\label{reference/GetDNA:module-GetDNA}\index{GetDNA (module)}
This module is used for downloading the DNA sequence from ncbi web. You can only

need input a DNA ID.

Authors: Zhijiang Yao and Dongsheng Cao.

Date: 2016.11.04

Email: \href{mailto:gadsby@163.com}{gadsby@163.com}
\index{GetDNAFromUniGene() (in module GetDNA)}

\begin{fulllineitems}
\phantomsection\label{reference/GetDNA:GetDNA.GetDNAFromUniGene}\pysiglinewithargsret{\sphinxcode{GetDNA.}\sphinxbfcode{GetDNAFromUniGene}}{\emph{SeqID='`}}{}
This module is used for downloading the DNA sequence from ncbi web. You can only

need input a DNA ID.

\end{fulllineitems}

\index{IsFasta() (in module GetDNA)}

\begin{fulllineitems}
\phantomsection\label{reference/GetDNA:GetDNA.IsFasta}\pysiglinewithargsret{\sphinxcode{GetDNA.}\sphinxbfcode{IsFasta}}{\emph{seq}}{}
Judge the Seq object is in FASTA format.
Two situation:
1. No seq name.
2. Seq name is illegal.
3. No sequence.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\textbf{\texttt{seq}} -- Seq object.

\end{description}\end{quote}

\end{fulllineitems}

\index{IsUnderAlphabet() (in module GetDNA)}

\begin{fulllineitems}
\phantomsection\label{reference/GetDNA:GetDNA.IsUnderAlphabet}\pysiglinewithargsret{\sphinxcode{GetDNA.}\sphinxbfcode{IsUnderAlphabet}}{\emph{s}, \emph{alphabet}}{}
Judge the string is within the scope of the alphabet or not.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{s}} -- The string.

\item {} 
\textbf{\texttt{alphabet}} -- alphabet.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{ReadFasta() (in module GetDNA)}

\begin{fulllineitems}
\phantomsection\label{reference/GetDNA:GetDNA.ReadFasta}\pysiglinewithargsret{\sphinxcode{GetDNA.}\sphinxbfcode{ReadFasta}}{\emph{f}}{}
Read a fasta file.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\textbf{\texttt{f}} -- HANDLE to input. e.g. sys.stdin, or open(\textless{}file\textgreater{})

\end{description}\end{quote}

\end{fulllineitems}

\index{Seq (class in GetDNA)}

\begin{fulllineitems}
\phantomsection\label{reference/GetDNA:GetDNA.Seq}\pysiglinewithargsret{\sphinxstrong{class }\sphinxcode{GetDNA.}\sphinxbfcode{Seq}}{\emph{name}, \emph{seq}, \emph{no}}{}
\end{fulllineitems}



\subsection{Getmol module}
\label{reference/Getmol:module-Getmol}\label{reference/Getmol:getmol-module}\label{reference/Getmol::doc}\index{Getmol (module)}
This module is to get different formats of molecules from file and web. If you

have any question please contact me via email.

Authors: Zhijiang Yao and Dongsheng Cao.

Date: 2016.06.04

Email: \href{mailto:gadsby@163.com}{gadsby@163.com}
\index{GetMolFromCAS() (in module Getmol)}

\begin{fulllineitems}
\phantomsection\label{reference/Getmol:Getmol.GetMolFromCAS}\pysiglinewithargsret{\sphinxcode{Getmol.}\sphinxbfcode{GetMolFromCAS}}{\emph{casid='`}}{}
Downloading the molecules from \url{http://www.chemnet.com/cas/} by CAS ID (casid).
if you want to use this function, you must be install pybel.

\end{fulllineitems}

\index{GetMolFromDrugbank() (in module Getmol)}

\begin{fulllineitems}
\phantomsection\label{reference/Getmol:Getmol.GetMolFromDrugbank}\pysiglinewithargsret{\sphinxcode{Getmol.}\sphinxbfcode{GetMolFromDrugbank}}{\emph{dbid='`}}{}
Downloading the molecules from \url{http://www.drugbank.ca/} by dbid (dbid).

\end{fulllineitems}

\index{GetMolFromEBI() (in module Getmol)}

\begin{fulllineitems}
\phantomsection\label{reference/Getmol:Getmol.GetMolFromEBI}\pysiglinewithargsret{\sphinxcode{Getmol.}\sphinxbfcode{GetMolFromEBI}}{}{}
\end{fulllineitems}

\index{GetMolFromKegg() (in module Getmol)}

\begin{fulllineitems}
\phantomsection\label{reference/Getmol:Getmol.GetMolFromKegg}\pysiglinewithargsret{\sphinxcode{Getmol.}\sphinxbfcode{GetMolFromKegg}}{\emph{kid='`}}{}
Downloading the molecules from \url{http://www.genome.jp/} by kegg id (kid).

\end{fulllineitems}

\index{GetMolFromNCBI() (in module Getmol)}

\begin{fulllineitems}
\phantomsection\label{reference/Getmol:Getmol.GetMolFromNCBI}\pysiglinewithargsret{\sphinxcode{Getmol.}\sphinxbfcode{GetMolFromNCBI}}{\emph{cid='`}}{}
Downloading the molecules from \url{http://pubchem.ncbi.nlm.nih.gov/} by cid (cid).

\end{fulllineitems}

\index{ReadMolFromInchi() (in module Getmol)}

\begin{fulllineitems}
\phantomsection\label{reference/Getmol:Getmol.ReadMolFromInchi}\pysiglinewithargsret{\sphinxcode{Getmol.}\sphinxbfcode{ReadMolFromInchi}}{\emph{inchi='`}}{}
Read a molecule by Inchi string.

Usage:
\begin{quote}

res=ReadMolFromInchi(inchi)

Input: inchi is a InChi string.

Output: res is a molecule object.
\end{quote}

\end{fulllineitems}

\index{ReadMolFromMOL() (in module Getmol)}

\begin{fulllineitems}
\phantomsection\label{reference/Getmol:Getmol.ReadMolFromMOL}\pysiglinewithargsret{\sphinxcode{Getmol.}\sphinxbfcode{ReadMolFromMOL}}{\emph{filename='`}}{}
Read a  molecule by mol file format.

Usage:
\begin{quote}

res=ReadMolFromMOL(filename)

Input: filename is a file name with path.

Output: res is a  molecular object.
\end{quote}

\end{fulllineitems}

\index{ReadMolFromMol() (in module Getmol)}

\begin{fulllineitems}
\phantomsection\label{reference/Getmol:Getmol.ReadMolFromMol}\pysiglinewithargsret{\sphinxcode{Getmol.}\sphinxbfcode{ReadMolFromMol}}{\emph{filename='`}}{}
Read a molecule with mol file format.

Usage:
\begin{quote}

res=ReadMolFromMol(filename)

Input: filename is a file name.

Output: res is a molecule object.
\end{quote}

\end{fulllineitems}

\index{ReadMolFromSDF() (in module Getmol)}

\begin{fulllineitems}
\phantomsection\label{reference/Getmol:Getmol.ReadMolFromSDF}\pysiglinewithargsret{\sphinxcode{Getmol.}\sphinxbfcode{ReadMolFromSDF}}{\emph{filename='`}}{}
Read a set of molecules by SDF file format.

Note: the output of this function is a set of molecular objects.

You need to use for statement to call each object.

Usage:
\begin{quote}

res=ReadMolFromSDF(filename)

Input: filename is a file name with path.

Output: res is a set of molecular object.
\end{quote}

\end{fulllineitems}

\index{ReadMolFromSmile() (in module Getmol)}

\begin{fulllineitems}
\phantomsection\label{reference/Getmol:Getmol.ReadMolFromSmile}\pysiglinewithargsret{\sphinxcode{Getmol.}\sphinxbfcode{ReadMolFromSmile}}{\emph{smi='`}}{}
Read a molecule by SMILES string.

Usage:
\begin{quote}

res=ReadMolFromSmile(smi)

Input: smi is a SMILES string.

Output: res is a molecule object.
\end{quote}

\end{fulllineitems}



\section{test package}
\label{reference/test::doc}\label{reference/test:test-package}

\subsection{test module}
\label{reference/test2::doc}\label{reference/test2:test-module}\phantomsection\label{reference/test2:module-test}\index{test (module)}

\subsection{test\_PyBioMed module}
\label{reference/test_PyBioMed::doc}\label{reference/test_PyBioMed:test-pybiomed-module}\phantomsection\label{reference/test_PyBioMed:module-test_PyBioMed}\index{test\_PyBioMed (module)}
Created on Mon Oct 24 09:05:01 2016

@author: Gadsby
\index{test\_pybiomed() (in module test\_PyBioMed)}

\begin{fulllineitems}
\phantomsection\label{reference/test_PyBioMed:test_PyBioMed.test_pybiomed}\pysiglinewithargsret{\sphinxcode{test\_PyBioMed.}\sphinxbfcode{test\_pybiomed}}{}{}
\end{fulllineitems}



\subsection{test\_PyDNA module}
\label{reference/test_PyDNA:test-pydna-module}\label{reference/test_PyDNA::doc}\phantomsection\label{reference/test_PyDNA:module-test_PyDNA}\index{test\_PyDNA (module)}
Created on Mon Oct 24 09:16:55 2016

@author: Gadsby
\index{test\_pydna() (in module test\_PyDNA)}

\begin{fulllineitems}
\phantomsection\label{reference/test_PyDNA:test_PyDNA.test_pydna}\pysiglinewithargsret{\sphinxcode{test\_PyDNA.}\sphinxbfcode{test\_pydna}}{}{}
\end{fulllineitems}



\subsection{test\_PyInteration module}
\label{reference/test_PyInteration:test-pyinteration-module}\label{reference/test_PyInteration::doc}\phantomsection\label{reference/test_PyInteration:module-test_PyInteration}\index{test\_PyInteration (module)}
Created on Thu Nov 03 15:08:29 2016

@author: Gadsby
\index{test\_pyinteration() (in module test\_PyInteration)}

\begin{fulllineitems}
\phantomsection\label{reference/test_PyInteration:test_PyInteration.test_pyinteration}\pysiglinewithargsret{\sphinxcode{test\_PyInteration.}\sphinxbfcode{test\_pyinteration}}{}{}
\end{fulllineitems}



\subsection{test\_PyMolecule module}
\label{reference/test_PyMolecule:test-pymolecule-module}\label{reference/test_PyMolecule::doc}\phantomsection\label{reference/test_PyMolecule:module-test_PyMolecule}\index{test\_PyMolecule (module)}
Created on Thu Oct 27 11:13:27 2016

@author: Gadsby
\index{test\_pymolecule() (in module test\_PyMolecule)}

\begin{fulllineitems}
\phantomsection\label{reference/test_PyMolecule:test_PyMolecule.test_pymolecule}\pysiglinewithargsret{\sphinxcode{test\_PyMolecule.}\sphinxbfcode{test\_pymolecule}}{}{}
\end{fulllineitems}



\subsection{test\_PyPretreat module}
\label{reference/test_PyPretreat:test-pypretreat-module}\label{reference/test_PyPretreat::doc}\phantomsection\label{reference/test_PyPretreat:module-test_PyPretreat}\index{test\_PyPretreat (module)}
Created on Mon Oct 31 14:12:12 2016

@author: Gadsby
\index{test\_pypretreat() (in module test\_PyPretreat)}

\begin{fulllineitems}
\phantomsection\label{reference/test_PyPretreat:test_PyPretreat.test_pypretreat}\pysiglinewithargsret{\sphinxcode{test\_PyPretreat.}\sphinxbfcode{test\_pypretreat}}{}{}
\end{fulllineitems}



\subsection{test\_PyProtein module}
\label{reference/test_PyProtein:test-pyprotein-module}\label{reference/test_PyProtein::doc}\phantomsection\label{reference/test_PyProtein:module-test_PyProtein}\index{test\_PyProtein (module)}
Created on Thu Oct 27 09:38:03 2016

@author: Gadsby
\index{test\_pyprotein() (in module test\_PyProtein)}

\begin{fulllineitems}
\phantomsection\label{reference/test_PyProtein:test_PyProtein.test_pyprotein}\pysiglinewithargsret{\sphinxcode{test\_PyProtein.}\sphinxbfcode{test\_pyprotein}}{}{}
\end{fulllineitems}



\subsection{test\_PyGetMol module}
\label{reference/test_PyGetMol::doc}\label{reference/test_PyGetMol:test-pygetmol-module}\phantomsection\label{reference/test_PyGetMol:module-test_PyGetMol}\index{test\_PyGetMol (module)}
Created on Tue Oct 25 15:07:23 2016

@author: Gadsby
\index{test\_pygetmol() (in module test\_PyGetMol)}

\begin{fulllineitems}
\phantomsection\label{reference/test_PyGetMol:test_PyGetMol.test_pygetmol}\pysiglinewithargsret{\sphinxcode{test\_PyGetMol.}\sphinxbfcode{test\_pygetmol}}{}{}
\end{fulllineitems}



\chapter{Testing}
\label{test:testing}\label{test::doc}

\section{Requirements for testing}
\label{test:requirements-for-testing}
PyBioMed requires RDKit and pybel packages.
If you don't already have the packages installed, follow
the directions here
\url{https://openbabel.org/docs/dev/UseTheLibrary/Python\_Pybel.html}

\url{http://www.rdkit.org/}


\section{Testing an installed package}
\label{test:testing-an-installed-package}
If you have a file-based (not a Python egg) installation you can
test the installed package with

\begin{Verbatim}[commandchars=\\\{\}]
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{k+kn}{from} \PYG{n+nn}{PyBioMed}\PYG{n+nn}{.}\PYG{n+nn}{test} \PYG{k}{import} \PYG{n}{test\PYGZus{}PyBioMed}
\PYG{g+gp}{\PYGZgt{}\PYGZgt{}\PYGZgt{} }\PYG{n}{test\PYGZus{}PyBioMed}\PYG{o}{.}\PYG{n}{test\PYGZus{}pybiomed}\PYG{p}{(}\PYG{p}{)}
\end{Verbatim}


\chapter{Download}
\label{download:download}\label{download::doc}

\section{Python Package}
\label{download:python-package}
Source and binary releases: \url{https://pypi.python.org/pypi/pybiomed/}

Github (latest development): \url{https://github.com/gadsbyfly/PyBioMed/}


\section{Documentation}
\label{download:documentation}
\emph{PDF}

\url{https://github.com/gadsbyfly/PyBioMed/blob/master/doc/Descriptor} introduction/PyBioMedDocumentation.pdf


\section{The introduction of descriptors}
\label{download:the-introduction-of-descriptors}
\emph{PDF}


\subsection{Molecular descriptors introduction}
\label{download:molecular-descriptors-introduction}
\url{https://github.com/gadsbyfly/PyBioMed/blob/master/doc/DescriptorIntroduction/PyBioMedChem.pdf}


\subsection{Protein descriptors introduction}
\label{download:protein-descriptors-introduction}
\url{https://github.com/gadsbyfly/PyBioMed/blob/master/doc/DescriptorIntroduction/PyBioMedProtein.pdf}


\subsection{DNA descriptors introduction}
\label{download:dna-descriptors-introduction}
\url{https://github.com/gadsbyfly/PyBioMed/blob/master/doc/DescriptorIntroduction/PyBioMedDNA.pdf}


\subsection{Interaction descriptors introduction}
\label{download:interaction-descriptors-introduction}
\url{https://github.com/gadsbyfly/PyBioMed/blob/master/doc/DescriptorIntroduction/PyBioMedInteraction.pdf}

\emph{HTML in zip file}

\url{https://github.com/gadsbyfly/PyBioMed/blob/master/doc/DescriptorIntroduction/PyBioMedDocumentation.zip}


\renewcommand{\indexname}{Index}
\printindex
\end{document}
